--- a/kitty/simd-string.c
+++ b/kitty/simd-string.c
@@ -195,7 +195,11 @@
     PyObject *module = (PyObject*)x;
     if (PyModule_AddFunctions(module, module_methods) != 0) return false;
 #define A(x, val) { Py_INCREF(Py_##val); if (0 != PyModule_AddObject(module, #x, Py_##val)) return false; }
+#if __has_builtin(__builtin_cpu_supports)
 #define do_check() { has_sse4_2 = __builtin_cpu_supports("sse4.2") != 0; has_avx2 = __builtin_cpu_supports("avx2") != 0; }
+#else
+#define do_check() { has_sse4_2 = false; has_avx2 = false; }
+#endif

 #ifdef __APPLE__
 #ifdef __arm64__
