--- a/meson.build
+++ b/meson.build
@@ -39,7 +39,7 @@
 
 elif get_option('use_system_wlroots').enabled()
 	use_system_wlroots = true
-	wlroots = dependency('wlroots', version: ['>=0.17.0', '<0.18.0'], required: true)
+	wlroots = dependency('wlroots-0.17', version: ['>=0.17.0', '<0.18.0'], required: true)
 
 elif get_option('use_system_wlroots').auto()
 	message( 'SEARCHING FOR WLROOTS' )
--- a/src/meson.build
+++ b/src/meson.build
@@ -127,7 +127,7 @@
     wayland_server,
     pixman,
     # These might be subprojects so we need to pass them as strings
-    'wlroots',
+    'wlroots-0.17',
     'wf-config',
 ]
 
