diff --git a/src/script/runtime/toolchains.meson b/src/script/runtime/toolchains.meson
index 97682251..a5c5eda7 100644
--- a/src/script/runtime/toolchains.meson
+++ b/src/script/runtime/toolchains.meson
@@ -886,7 +886,8 @@ toolchain.register_compiler(
         endif
     endfunc,
     detect: func(out str) -> int
-        return 'clang' in out.to_lower() ? 50 : 0
+        score = build_machine.system() == 'windows' ? 50 : 100
+        return 'clang' in out.to_lower() ? score : 0
     endfunc,
     handlers: {
         'can_compile_llvm_ir': true,
@@ -1066,7 +1067,8 @@ toolchain.register_compiler(
     linker: 'lld-link',
     archiver: 'llvm-lib',
     detect: func(out str) -> int
-        return 'CL.EXE COMPATIBILITY' in out ? 100 : 0
+        score = build_machine.system() == 'windows' ? 100 : 50
+        return 'CL.EXE COMPATIBILITY' in out ? score : 0
     endfunc,
     handlers: {
         'force_language': func(_c compiler, s str) -> list[str]
