diff --git a/gdk/loaders/gdktiff.c b/gdk/loaders/gdktiff.c
index 93f39a7830..1a4e35edea 100644
--- a/gdk/loaders/gdktiff.c
+++ b/gdk/loaders/gdktiff.c
@@ -341,7 +341,7 @@ static GdkTexture *
 load_fallback (TIFF    *tif,
                GError **error)
 {
-  int width, height;
+  uint32_t width, height;
   guchar *data;
   GBytes *bytes;
   GdkTexture *texture;
diff --git a/gtk/gtklistview.c b/gtk/gtklistview.c
index 53e8558dd8..f6309bf8ee 100644
--- a/gtk/gtklistview.c
+++ b/gtk/gtklistview.c
@@ -186,7 +186,9 @@ gtk_list_view_split (GtkListBase *base,
                           new_tile,
                           &(GdkRectangle) {
                             tile->area.x,
-                            tile->area.y + tile->area.height + spacing,
+                            /* ../gtk/gtklistview.c:189:42: runtime error: signed integer overflow: 2111085624 + 72796048 cannot be represented in type 'int'
+                              when starting nautilus */
+                            (guint)tile->area.y + tile->area.height + spacing,
                             tile->area.width,
                             row_height * new_tile->n_items + spacing * (new_tile->n_items - 1)
                           });
