From faf9730be42511131cff3fd8b40632c77388fa36 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sun, 22 Jan 2023 01:21:26 +0100
Subject: [PATCH] disable execinfo for tests

---
 testsuite/reftests/gtk-reftest.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/testsuite/reftests/gtk-reftest.c b/testsuite/reftests/gtk-reftest.c
index a057a2e..053aa6c 100644
--- a/testsuite/reftests/gtk-reftest.c
+++ b/testsuite/reftests/gtk-reftest.c
@@ -24,7 +24,7 @@
 #include "reftest-module.h"
 #include "reftest-snapshot.h"
 
-#ifndef G_OS_WIN32
+#if 0
 #include <execinfo.h>
 #endif
 #include <string.h>
@@ -475,7 +475,7 @@ log_writer (GLogLevelFlags   log_level,
             gsize            n_fields,
             gpointer         user_data)
 {
-#ifndef G_OS_WIN32
+#if 0
   if (log_level & G_LOG_LEVEL_CRITICAL)
     {
       void *buffer[1024];
-- 
2.39.0

