diff --git a/Cargo.lock b/Cargo.lock
index 60b9ce1..c98abb0 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2180,7 +2180,6 @@ dependencies = [
  "shadow-rs",
  "signal-hook",
  "signal-hook-tokio",
- "tikv-jemallocator",
  "tokio",
  "tokio-metrics-collector",
  "tokio-rustls",
@@ -2299,26 +2298,6 @@ dependencies = [
  "cfg-if",
 ]
 
-[[package]]
-name = "tikv-jemalloc-sys"
-version = "0.6.0+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "cd3c60906412afa9c2b5b5a48ca6a5abe5736aec9eb48ad05037a677e52e4e2d"
-dependencies = [
- "cc",
- "libc",
-]
-
-[[package]]
-name = "tikv-jemallocator"
-version = "0.6.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "4cec5ff18518d81584f477e9bfdf957f5bb0979b0bac3af4ca30b5b3ae2d2865"
-dependencies = [
- "libc",
- "tikv-jemalloc-sys",
-]
-
 [[package]]
 name = "time"
 version = "0.3.41"
diff --git a/Cargo.toml b/Cargo.toml
index 4dba33c..f10b1e6 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -100,8 +100,6 @@ toml = "0.8"
 tracing = { version = "0.1", default-features = false, features = ["std"] }
 tracing-subscriber = { version = "0.3", default-features = false, features = ["smallvec", "registry", "parking_lot", "fmt", "ansi", "tracing-log"] }
 
-[target.'cfg(all(target_env = "musl", target_pointer_width = "64"))'.dependencies.tikv-jemallocator]
-version = "0.6"
 
 [target.'cfg(unix)'.dependencies]
 signal-hook = { version = "0.3", features = ["extended-siginfo"] }
diff --git a/src/bin/server.rs b/src/bin/server.rs
index 4cdd82b..f13bce7 100644
--- a/src/bin/server.rs
+++ b/src/bin/server.rs
@@ -8,9 +8,6 @@
 #![deny(rust_2018_idioms)]
 #![deny(dead_code)]
 
-#[cfg(all(target_env = "musl", target_pointer_width = "64"))]
-#[global_allocator]
-static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
 
 use static_web_server::{
     settings::{cli::General, Commands},

