diff --git a/Cargo.lock b/Cargo.lock
index 91d68ea..a7dd658 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2164,7 +2164,6 @@ dependencies = [
  "shadow-rs",
  "signal-hook",
  "signal-hook-tokio",
- "tikv-jemallocator",
  "tokio",
  "tokio-metrics-collector",
  "tokio-rustls",
@@ -2283,26 +2282,6 @@ dependencies = [
  "cfg-if",
 ]
 
-[[package]]
-name = "tikv-jemalloc-sys"
-version = "0.6.1+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "cd8aa5b2ab86a2cefa406d889139c162cbb230092f7d1d7cbc1716405d852a3b"
-dependencies = [
- "cc",
- "libc",
-]
-
-[[package]]
-name = "tikv-jemallocator"
-version = "0.6.1"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "0359b4327f954e0567e69fb191cf1436617748813819c94b8cd4a431422d053a"
-dependencies = [
- "libc",
- "tikv-jemalloc-sys",
-]
-
 [[package]]
 name = "time"
 version = "0.3.44"
diff --git a/Cargo.toml b/Cargo.toml
index ce92cbb..91e6afe 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -100,8 +100,6 @@ toml = "0.9"
 tracing = { version = "0.1", default-features = false, features = ["std"] }
 tracing-subscriber = { version = "0.3", default-features = false, features = ["smallvec", "registry", "parking_lot", "fmt", "ansi", "tracing-log"] }
 
-[target.'cfg(all(target_env = "musl", target_pointer_width = "64"))'.dependencies.tikv-jemallocator]
-version = "0.6"
 
 [target.'cfg(unix)'.dependencies]
 signal-hook = { version = "0.3", features = ["extended-siginfo"] }
diff --git a/src/bin/server.rs b/src/bin/server.rs
index eed5a1f..a859af1 100644
--- a/src/bin/server.rs
+++ b/src/bin/server.rs
@@ -8,10 +8,6 @@
 #![deny(rust_2018_idioms)]
 #![deny(dead_code)]
 
-#[cfg(all(target_env = "musl", target_pointer_width = "64"))]
-#[global_allocator]
-static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
-
 use static_web_server::{
     Result, Settings,
     settings::{Commands, cli::General},
