diff --git a/third_party/rust/target-lexicon-0.9.0/src/targets.rs b/third_party/rust/target-lexicon-0.9.0/src/targets.rs
index 6ae570e..b1d07a0 100644
--- a/third_party/rust/target-lexicon-0.9.0/src/targets.rs
+++ b/third_party/rust/target-lexicon-0.9.0/src/targets.rs
@@ -298,6 +298,7 @@ pub enum Vendor {
     Unknown,
     Amd,
     Apple,
+    Chimera,
     Experimental,
     Fortanix,
     Nvidia,
@@ -709,6 +710,7 @@ impl fmt::Display for Vendor {
             Vendor::Unknown => "unknown",
             Vendor::Amd => "amd",
             Vendor::Apple => "apple",
+            Vendor::Chimera => "chimera",
             Vendor::Experimental => "experimental",
             Vendor::Fortanix => "fortanix",
             Vendor::Nvidia => "nvidia",
@@ -730,6 +732,7 @@ impl FromStr for Vendor {
             "unknown" => Vendor::Unknown,
             "amd" => Vendor::Amd,
             "apple" => Vendor::Apple,
+            "chimera" => Vendor::Chimera,
             "experimental" => Vendor::Experimental,
             "fortanix" => Vendor::Fortanix,
             "nvidia" => Vendor::Nvidia,
diff --git a/third_party/rust/target-lexicon/src/targets.rs b/third_party/rust/target-lexicon/src/targets.rs
index 87c237a..48fd041 100644
--- a/third_party/rust/target-lexicon/src/targets.rs
+++ b/third_party/rust/target-lexicon/src/targets.rs
@@ -397,6 +397,7 @@ pub enum Vendor {
     Unknown,
     Amd,
     Apple,
+    Chimera,
     Espressif,
     Experimental,
     Fortanix,
@@ -984,6 +985,7 @@ impl fmt::Display for Vendor {
             Unknown => "unknown",
             Amd => "amd",
             Apple => "apple",
+            Chimera => "chimera",
             Espressif => "espressif",
             Experimental => "experimental",
             Fortanix => "fortanix",
@@ -1011,6 +1013,7 @@ impl FromStr for Vendor {
             "unknown" => Unknown,
             "amd" => Amd,
             "apple" => Apple,
+            "chimera" => Chimera,
             "espressif" => Espressif,
             "experimental" => Experimental,
             "fortanix" => Fortanix,