--- a/tools/profiler/core/platform-linux-android.cpp	2019-01-29 12:09:40.980448579 +0100
+++ b/tools/profiler/core/platform-linux-android.cpp	2019-01-29 12:11:09.689590967 +0100
@@ -630,8 +630,10 @@ static void PlatformInit(PSLockRef aLock) {}
 
 #if defined(HAVE_NATIVE_UNWIND)
 void Registers::SyncPopulate() {
+#if defined(__GLIBC__)
   if (!getcontext(&mContextSyncStorage)) {
     PopulateRegsFromContext(*this, &mContextSyncStorage);
   }
+#endif
 }
 #endif
