From: Brian May <bam@debian.org>
Date: Wed, 17 Nov 2021 09:46:28 +1100
Subject: Fix autoconf 2.7 issues

---
 cf/check-var.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cf/check-var.m4 b/cf/check-var.m4
index 2fd7bca..71d6f70 100644
--- a/cf/check-var.m4
+++ b/cf/check-var.m4
@@ -20,7 +20,7 @@ AC_MSG_RESULT($ac_foo)
 if test "$ac_foo" = yes; then
 	AC_DEFINE_UNQUOTED(AS_TR_CPP(HAVE_[]$1), 1, 
 		[Define if you have the `]$1[' variable.])
-	m4_ifval([$2], AC_CHECK_DECLS([$1],[],[],[$2]))
+	m4_ifval([$2], [AC_CHECK_DECLS([$1],[],[],[$2])])
 fi
 ])
 
