needs unpackaged python-polib
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 1e51834..b47711e 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -2,7 +2,7 @@ AM_TESTS_ENVIRONMENT = top_srcdir="$(top_srcdir)" top_builddir="$(top_builddir)"
 
 dist_noinst_SCRIPTS = libbytesize_unittest.sh libbytesize_unittest.py lbs_py_override_unittest.py locale_utils.py testenv.sh canary_tests.sh
 
-TESTS = libbytesize_unittest.sh canary_tests.sh
+TESTS = libbytesize_unittest.sh
 
 # Add the translation-canary source files to the tarball
 EXTRA_DIST = $(top_srcdir)/translation-canary/translation_canary/*.py \
