From b014a08228f9486ae864f171cc8f8e758943ce1a Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 10 Jun 2023 21:54:29 +0200
Subject: [PATCH] portable install(1) use

---
 util/install_helper.sh | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/util/install_helper.sh b/util/install_helper.sh
index 76f2b47..414ddf1 100755
--- a/util/install_helper.sh
+++ b/util/install_helper.sh
@@ -24,7 +24,8 @@ else
     DESTDIR="${DESTDIR%/}"
 fi
 
-install -D -m 644 "${MESON_SOURCE_ROOT}/util/fuse.conf" \
+install -d -m 755 "${DESTDIR}${sysconfdir}"
+install -m 644 "${MESON_SOURCE_ROOT}/util/fuse.conf" \
 	"${DESTDIR}${sysconfdir}/fuse.conf"
 
 if $useroot; then
@@ -38,12 +39,14 @@ if $useroot; then
 fi
 
 if [ "${udevrulesdir}" != "" ]; then
-    install -D -m 644 "${MESON_SOURCE_ROOT}/util/udev.rules" \
+    install -d -m 755 "${DESTDIR}${udevrulesdir}"
+    install -m 644 "${MESON_SOURCE_ROOT}/util/udev.rules" \
         "${DESTDIR}${udevrulesdir}/99-fuse3.rules"
 fi
 
 if [ "$initscriptdir" != "" ]; then
-    install -D -m 755 "${MESON_SOURCE_ROOT}/util/init_script" \
+    install -d -m 755 "${DESTDIR}/etc/init.d"
+    install -m 755 "${MESON_SOURCE_ROOT}/util/init_script" \
             "${DESTDIR}${initscriptdir}/fuse3"
 
     if test -x /usr/sbin/update-rc.d && test -z "${DESTDIR}"; then
-- 
2.39.0

