--- a/sbin/update-ca-certificates
+++ b/sbin/update-ca-certificates
@@ -181,7 +181,7 @@ then
   done
   if [ "$verbose" = 0 ]
   then
-    openssl rehash . > /dev/null
+    openssl rehash .
   else
     openssl rehash -v .
   fi
@@ -192,8 +192,6 @@ if [ -f "$TEMPBUNDLE" ]
 then
   chmod 0644 "$TEMPBUNDLE"
   mv -f "$TEMPBUNDLE" "$CERTBUNDLE"
-  # Restore proper SELinux label after moving the file
-  [ -x /sbin/restorecon ] && /sbin/restorecon "$CERTBUNDLE" >/dev/null 2>&1
 fi
 
 echo "$ADDED_CNT added, $REMOVED_CNT removed; done."
