diff --git a/loaders/meson.build b/loaders/meson.build
index 7913b58..a763863 100644
--- a/loaders/meson.build
+++ b/loaders/meson.build
@@ -51,7 +51,7 @@ foreach loader : get_option('loaders')
     console: true,
     env: cargo_env,
     command: [
-      cargo_bin, 'build',
+      cargo_bin, 'auditable', 'build',
       '--target-dir', target_dir / 'loaders',
       [cargo_options, [ '--package', loader ]],
     ],
