diff --git a/loaders/meson.build b/loaders/meson.build
index 7913b58..7375f28 100644
--- a/loaders/meson.build
+++ b/loaders/meson.build
@@ -26,17 +26,8 @@ cargo_options = [
 ]
 
 cargo_env = {
-  # Don't use $HOME/.cargo/ for this
-  'CARGO_HOME': meson.project_build_root() / 'cargo',
   # Gettext translation
   'LOCALEDIR': get_option('prefix') / get_option('localedir'),
-  # Git commit for debugging
-  'GIT_DESCRIBE': run_command(
-      'git',
-      '-C', meson.project_source_root(),
-      'rev-parse', '--short', 'HEAD',
-      check: false
-    ).stdout().strip(),
 }
 
 test_args = []
