diff --git a/Makefile.in b/Makefile.in
index 8c44a7c..f7df7d9 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -74,9 +74,11 @@ intel-undervolt: $(intel_undervolt_objects)
 	$(CC) $(LDFLAGS) -o $@ $^ -lm
 
 install: all
-	install -Dm755 'intel-undervolt' \
+	install -dm 755 $(DESTDIR)$(BINDIR)
+	install -dm 755 $(DESTDIR)$(SYSCONFDIR)
+	install -m755 'intel-undervolt' \
 	"$(DESTDIR)$(BINDIR)/intel-undervolt"
-	install -Dm644 'intel-undervolt.conf' \
+	install -m644 'intel-undervolt.conf' \
 	"$(DESTDIR)$(SYSCONFDIR)/intel-undervolt.conf"
 ifeq ($(ENABLE_SYSTEMD), 1)
 	install -Dm644 'intel-undervolt.service' \
@@ -85,7 +87,8 @@ ifeq ($(ENABLE_SYSTEMD), 1)
 	"$(DESTDIR)$(UNITDIR)/intel-undervolt-loop.service"
 endif
 ifeq ($(ENABLE_ELOGIND), 1)
-	install -Dm755 'intel-undervolt.elogind' \
+	install -dm 755 $(DESTDIR)$(ELOGINDDIR)/system-sleep
+	install -m755 'intel-undervolt.elogind' \
 	"$(DESTDIR)$(ELOGINDDIR)/system-sleep/50-intel-undervolt"
 endif
 ifeq ($(ENABLE_OPENRC), 1)
