--- a/src/screen.C
+++ b/src/screen.C
@@ -306,7 +306,11 @@ rxvt_term::scr_reset ()
       int pend = MOD (term_start + top_row  , prev_total_rows);
       int q    = total_rows; // rewrapped row
 
+#if ENABLE_FRILLS
       if ((rewrap_always || top_row) && !rewrap_never)
+#else
+      if (top_row)
+#endif
         {
           // Re-wrap lines. This is rather ugly, possibly because I am too dumb
           // to come up with a lean and mean algorithm.
