--- a/keyring/core.py
+++ b/keyring/core.py
@@ -37,13 +37,6 @@ def get_keyring() -> backend.KeyringBackend:
 def disable() -> None:
     """
     Configure the null keyring as the default.
-
-    >>> fs = getfixture('fs')
-    >>> disable()
-    >>> disable()
-    Traceback (most recent call last):
-    ...
-    RuntimeError: Refusing to overwrite...
     """
     root = platform.config_root()
     try:
