--- a/data/meson.build	2024-06-17 07:49:52.000000000 -0300
+++ a/data/meson.build	2024-09-12 10:08:39.800069752 -0300
@@ -66,7 +66,7 @@
   if (appstreamcli.found())
     test('validate-appdata',
          appstreamcli,
-         args: ['validate', '--strict', '--pedantic', '--explain', metainfo_file],
+         args: ['validate', '--strict', '--pedantic', '--explain', '--no-net', metainfo_file],
          workdir: meson.current_build_dir()
         )
   endif
