From 17a1ba555d761483037259b79b7ede145e4553a9 Mon Sep 17 00:00:00 2001
From: ttyyls <contact@behri.org>
Date: Sat, 16 Mar 2024 19:47:57 +0000
Subject: [PATCH] use env flags

---
 Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index efdd10297..34983063b 100644
--- a/Makefile
+++ b/Makefile
@@ -167,8 +167,8 @@ CPU = generic
 ARCH =
 
 #### Toolchain options.
-CC = cc
-LD = $(CC)
+cmd_CC = $(CC)
+cmd_LD = $(CC)
 
 #### Debug flags (typically "-g").
 # Those flags only feed CFLAGS so it is not mandatory to use this form.
@@ -289,13 +289,13 @@ ARCH_FLAGS        = $(ARCH_FLAGS.$(ARCH))
 # These CFLAGS contain general optimization options, CPU-specific optimizations
 # and debug flags. They may be overridden by some distributions which prefer to
 # set all of them at once instead of playing with the CPU and DEBUG variables.
-CFLAGS = $(ARCH_FLAGS) $(CPU_CFLAGS) $(DEBUG_CFLAGS) $(SPEC_CFLAGS)
+CFLAGS := $(CFLAGS) $(SPEC_CFLAGS)
 
 #### Common LDFLAGS
 # These LDFLAGS are used as the first "ld" options, regardless of any library
 # path or any other option. They may be changed to add any linker-specific
 # option at the beginning of the ld command line.
-LDFLAGS = $(ARCH_FLAGS) -g
+LDFLAGS := $(LDFLAGS)
 
 #### list of all "USE_*" options. These ones must be updated if new options are
 # added, so that the relevant options are properly added to the CFLAGS and to
@@ -873,7 +873,7 @@ endif
 #### Global link options
 # These options are added at the end of the "ld" command line. Use LDFLAGS to
 # add options at the beginning of the "ld" command line if needed.
-LDOPTS = $(TARGET_LDFLAGS) $(OPTIONS_LDFLAGS) $(ADDLIB)
+LDOPTS = $(CFLAGS) $(TARGET_LDFLAGS) $(OPTIONS_LDFLAGS) $(ADDLIB)
 
 ifeq ($(TARGET),)
 all:
-- 
2.44.0

