diff -ruN a/src/Makefile b/src/Makefile
--- a/src/Makefile	2025-08-23 10:43:56.000000000 +0200
+++ b/src/Makefile	2025-09-08 09:41:23.145267815 +0200
@@ -89,7 +89,7 @@
 IS_APPLE_SILICON        := $(shell [ "$$(sysctl -in hw.optional.arm64 2>/dev/null)" = "1" ] && echo 1 || echo 0)
 endif
 
-IS_AARCH64              := $(shell [ "$$(arch 2>/dev/null)" = "aarch64" ] && echo 1 || echo 0)
+IS_AARCH64              := $(shell [ "$$CBUILD_TARGET_MACHINE" = "aarch64" ] && echo 1 || echo 0)
 IS_ARM                  := $(or $(filter 1,$(IS_APPLE_SILICON)),$(filter 1,$(IS_AARCH64)))
 
 ifneq (,$(filter $(UNAME),FreeBSD NetBSD DragonFly))
