From 01047cfbd5f95ee50204c6df7a825929024ce8c7 Mon Sep 17 00:00:00 2001
From: Jens Reidel <adrian@travitia.xyz>
Date: Thu, 10 Apr 2025 00:42:33 +0200
Subject: [PATCH] Only use hardware CRC on ARM if supported

Signed-off-by: Jens Reidel <adrian@travitia.xyz>
---
 deps/LZMA-SDK/C/7zCrc.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/deps/LZMA-SDK/C/7zCrc.c b/deps/LZMA-SDK/C/7zCrc.c
index c7ec353d6..b1ae5c0fb 100644
--- a/deps/LZMA-SDK/C/7zCrc.c
+++ b/deps/LZMA-SDK/C/7zCrc.c
@@ -83,12 +83,6 @@ UInt32 MY_FAST_CALL CrcUpdateT1(UInt32 v, const void *data, size_t size, const U
     #endif
   #elif (defined(__clang__) && (__clang_major__ >= 3)) \
      || (defined(__GNUC__) && (__GNUC__ > 4))
-      #if !defined(__ARM_FEATURE_CRC32)
-        #define __ARM_FEATURE_CRC32 1
-          #if (!defined(__clang__) || (__clang_major__ > 3)) // fix these numbers
-            #define ATTRIB_CRC __attribute__((__target__("arch=armv8-a+crc")))
-          #endif
-      #endif
       #if defined(__ARM_FEATURE_CRC32)
         #define USE_ARM64_CRC
         #include <arm_acle.h>
-- 
2.49.0

