commit d9fc339805666b25482823cfad46c3c62074a4ea
Author: Wesley Moore <wes@wezm.net>
Date:   Wed Aug 2 21:28:08 2023 +1000

    Fix error when building with Clang

diff --git a/runtime/grammars/sources/rescript/src/scanner.c b/runtime/grammars/sources/rescript/src/scanner.c
index 9bf9fa4..113ca35 100644
--- a/runtime/grammars/sources/rescript/src/scanner.c
+++ b/runtime/grammars/sources/rescript/src/scanner.c
@@ -131,7 +131,7 @@ bool tree_sitter_rescript_external_scanner_scan(
     const bool* valid_symbols
     ) {
   ScannerState* state = (ScannerState*)payload;
-  const in_string = state->in_quotes || state->in_backticks;
+  const bool in_string = state->in_quotes || state->in_backticks;
 
   if (valid_symbols[TEMPLATE_CHARS]) {
     lexer->result_symbol = TEMPLATE_CHARS;
