--- a/src/3rdparty/chromium/content/browser/BUILD.gn
+++ b/src/3rdparty/chromium/content/browser/BUILD.gn
@@ -198,2 +198,3 @@
     "//components/services/storage/public/mojom",
+    "//components/spellcheck:buildflags",
     "//components/sqlite_proto",
--- a/src/3rdparty/chromium/extensions/browser/api/declarative_net_request/BUILD.gn
+++ b/src/3rdparty/chromium/extensions/browser/api/declarative_net_request/BUILD.gn
@@ -22,2 +22,3 @@
     "//content/public/browser",
+    "//components/web_cache/public/mojom",
     "//extensions/common",
--- a/src/core/configure/BUILD.root.gn.in
+++ b/src/core/configure/BUILD.root.gn.in
@@ -235,2 +235,3 @@
   deps = [
+    "//chrome/app:generated_resources",
     "//components/zoom",
--- a/src/3rdparty/chromium/ui/gl/gl_display.h
+++ b/src/3rdparty/chromium/ui/gl/gl_display.h
@@ -13,2 +13,3 @@
 #include "ui/gl/gl_export.h"
+#include "ui/gl/gpu_preference.h"

--- a/src/3rdparty/chromium/components/history/core/browser/BUILD.gn
+++ b/src/3rdparty/chromium/components/history/core/browser/BUILD.gn
@@ -123,6 +123,7 @@
     "//components/prefs",
     "//components/query_parser",
     "//components/search_engines",
+    "//components/signin/public/base:signin_buildflags",
     "//components/signin/public/identity_manager",
     "//components/sync",
     "//components/url_formatter",
--- a/src/3rdparty/chromium/components/webcrypto/BUILD.gn
+++ b/src/3rdparty/chromium/components/webcrypto/BUILD.gn
@@ -60,6 +60,7 @@
   deps = [
     "//base",
     "//crypto",
+    "//third_party/blink/renderer/platform/heap:heap",
     "//third_party/blink/public:blink_headers",
     "//third_party/boringssl",
   ]
--- a/src/3rdparty/chromium/third_party/blink/renderer/platform/heap/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/blink/renderer/platform/heap/BUILD.gn
@@ -84,6 +84,7 @@
     "//third_party/blink/renderer/platform/instrumentation:instrumentation",
     "//third_party/blink/renderer/platform/loader:loader",
     "//third_party/blink/renderer/platform/scheduler:scheduler",
+    "//components/webcrypto:webcrypto",
   ]
 }

 
