--- a/src/3rdparty/chromium/third_party/libvpx/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/libvpx/BUILD.gn
@@ -47,6 +47,8 @@
   }
 } else if (current_cpu == "riscv64") {
   cpu_arch_full = "generic"
+} else if (current_cpu == "ppc64") {
+  cpu_arch_full = "generic"
 } else if (current_cpu == "loong64") {
   cpu_arch_full = "loongarch"
 } else {
@@ -486,7 +488,7 @@
       public_deps = [ ":libvpx_arm64_headers" ]
     }
   } else if (current_cpu == "ppc64") {
-    sources = libvpx_srcs_ppc64
+    sources = libvpx_srcs_generic
   } else if (current_cpu == "riscv64") {
     sources = libvpx_srcs_generic
     public_deps = [ ":libvpx_generic_headers" ]
