--- a/src/3rdparty/chromium/BUILD.gn
+++ b/src/3rdparty/chromium/BUILD.gn
@@ -111,14 +111,6 @@ group("gn_all") {
       "//skia:skia_unittests",
       "//sql:sql_unittests",
       "//testing/libfuzzer/tests",
-      "//third_party/angle/src/tests:angle_end2end_tests",
-      "//third_party/angle/src/tests:angle_unittests",
-      "//third_party/angle/src/tests:angle_white_box_tests",
-      "//third_party/distributed_point_functions/shim:distributed_point_functions_shim_unittests",
-      "//third_party/flatbuffers:flatbuffers_unittests",
-      "//third_party/highway:highway_tests",
-      "//third_party/liburlpattern:liburlpattern_unittests",
-      "//third_party/rapidhash:rapidhash_fuzztests",
       "//tools/binary_size:binary_size_trybot_py",
       "//tools/ipc_fuzzer:ipc_fuzzer_all",
       "//tools/metrics:metrics_metadata",
