diff --git a/pyproject.toml b/pyproject.toml
index e151cd6..e4088c8 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -110,6 +110,4 @@ filterwarnings = [
   "error",
   # https://github.com/dateutil/dateutil/issues/1314
   "ignore:datetime.datetime.utcfromtimestamp:DeprecationWarning:dateutil.tz.tz",
-  # Python <= 3.11
-  "ignore:sys.monitoring isn't available, using default core:coverage.exceptions.CoverageWarning",
 ]
