commit 5f46924fe0ca55e253a8ac88b0e597e9d5d3526c
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Tue Apr 5 04:14:22 2022 +0200

    disable broken tests

diff --git a/tests/meson.build b/tests/meson.build
index 955dbc9..befb707 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -58,14 +58,6 @@ if cairo_dep.found()
       [ 'testmisc', [ 'testmisc.c' ], [ libpangocairo_dep, libpangoft2_dep, glib_dep, harfbuzz_dep ] ],
       [ 'testserialize', [ 'testserialize.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
     ]
-
-    if host_system != 'darwin'
-      tests += [
-        [ 'test-layout', [ 'test-layout.c', 'test-common.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
-        [ 'test-fonts', [ 'test-fonts.c', 'test-common.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
-        [ 'test-no-fonts', [ 'test-no-fonts.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
-      ]
-    endif
   endif
 
   if false and host_system != 'darwin'
