commit 925cfac66b0d94b796fec62226931ab62756d8cc
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Mon Mar 21 04:23:17 2022 +0100

    disable broken tests

diff --git a/tests/meson.build b/tests/meson.build
index a4c545c..bae3884 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -43,7 +43,6 @@ if cairo_dep.found()
     [ 'testcontext', [ 'testcontext.c' ], [ libpangocairo_dep ] ],
     [ 'testiter', [ 'testiter.c' ], [ libpangocairo_dep ] ],
     [ 'test-ellipsize', [ 'test-ellipsize.c' ], [ libpangocairo_dep ] ],
-    [ 'markup-parse', [ 'markup-parse.c' , 'test-common.c' ], [ libpangocairo_dep ] ],
     [ 'test-itemize', [ 'test-itemize.c', 'test-common.c' ], [ libpangocairo_dep ] ],
     [ 'test-shape', [ 'test-shape.c', 'test-common.c' ], [ libpangocairo_dep ] ],
     [ 'test-font', [ 'test-font.c' ], [ libpangocairo_dep ] ],
@@ -58,12 +57,6 @@ if cairo_dep.found()
       [ 'testmisc', [ 'testmisc.c' ], [ libpangocairo_dep, libpangoft2_dep, glib_dep, harfbuzz_dep ] ],
       [ 'testserialize', [ 'testserialize.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
     ]
-
-    if host_system != 'darwin'
-      tests += [
-        [ 'test-layout', [ 'test-layout.c', 'test-common.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
-      ]
-    endif
   endif
 
   if host_system != 'darwin'
