refers to internal Cantarell-VF path

--- a/tests/test-font.c
+++ b/tests/test-font.c
@@ -1012,7 +1012,6 @@ main (int argc, char *argv[])
   g_test_add_func ("/pango/font/font-metrics", test_font_metrics);
   g_test_add_func ("/pango/font/scale-font/plain", test_font_scale);
   g_test_add_func ("/pango/font/scale-font/variations", test_font_scale_variations);
-  g_test_add_func ("/pango/font/custom", test_font_custom);
 
   return g_test_run ();
 }
