--- a/pango/pango-layout.c
+++ b/pango/pango-layout.c
@@ -5599,6 +5599,10 @@ pango_layout_get_empty_extents_and_height_at_index (PangoLayout    *layout,
     }
 }
 
+/* There is a SubOverflow somewhere in this function that crashes
+ * gtksourceview tests sometimes, not quite sure which subtraction it is
+ */
+__attribute__((no_sanitize("undefined")))
 static void
 pango_layout_run_get_extents_and_height (PangoLayoutRun *run,
                                          PangoRectangle *run_ink,
