sigsegvs during construction

--- a/tests/auto/corelib/itemmodels/qrangemodel/tst_qrangemodel.cpp
+++ b/tests/auto/corelib/itemmodels/qrangemodel/tst_qrangemodel.cpp
@@ -710,12 +710,6 @@ void tst_QRangeModel::autoConnectPolicy()
         QCOMPARE(dataChangedSpy.at(0).at(2), QVariant::fromValue(QList{Qt::UserRole + 1}));
     }();
 
-    // build tests
-    { // make sure we don't kill the compiler with recursive templates
-        QList<std::array<Object *, 1000000>> wideList;
-        QRangeModel model(wideList);
-    }
-
     { // work with custom tuple types
         QList<ObjectRow> objectRows;
         QRangeModel model(objectRows);
