From cbe35de131195ce79acb7090b5726fe2c0ead48a Mon Sep 17 00:00:00 2001
From: Subhaditya Nath <sn03.general@gmail.com>
Date: Sun, 28 Apr 2024 20:42:50 +0530
Subject: [PATCH] add trailing newline at the end of --help text

---
 app/help.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/help.go b/app/help.go
index eef7ebd..f95c573 100644
--- a/help.go
+++ b/help.go
@@ -54,7 +54,7 @@ func helpText() string {
 		pterm.Yellow("WEBSITE"),
 	)
 
-	return description + usage + author + version + flags + options + env + docs + website
+	return description + usage + author + version + flags + options + env + docs + website + "\n"
 }
 
 func envHelp() string {
-- 
2.44.0

