diff --git a/vala/valagirparser.c b/vala/valagirparser.c
index 63fe537..5208798 100644
--- a/vala/valagirparser.c
+++ b/vala/valagirparser.c
@@ -3746,6 +3746,8 @@ vala_gir_parser_parse_repository (ValaGirParser* self)
 					_tmp30_ = _tmp29_;
 					if (g_strcmp0 (_tmp30_, "c:include") == 0) {
 						vala_gir_parser_parse_c_include (self);
+					} else if (g_strcmp0 (_tmp30_, "doc:format") == 0) {
+						vala_gir_parser_skip_element (self);
 					} else {
 						ValaMarkupReader* _tmp31_;
 						const gchar* _tmp32_;
