commit bff494a57560d6177e715819e6b3283a32abcfd3
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Fri Mar 17 18:13:58 2023 +0100

    bla

diff --git a/test/Makefile.in b/test/Makefile.in
index 3df8fab..e0c5903 100644
--- a/test/Makefile.in
+++ b/test/Makefile.in
@@ -456,8 +456,6 @@ EXTRA_DIST = \
 	clos1way4.ok \
 	clos1way5.awk \
 	clos1way5.ok \
-	clos1way6.awk \
-	clos1way6.ok \
 	closebad.awk \
 	closebad.ok \
 	close_status.awk \
@@ -1761,7 +1759,6 @@ GAWK_EXT_TESTS = \
 	aadelete1 aadelete2 aarray1 aasort aasorti argtest arraysort \
 	arraysort2 arraytype asortbool backw badargs beginfile1 beginfile2 \
 	binmode1 charasbytes clos1way clos1way2 clos1way3 clos1way4 \
-	clos1way5 clos1way6 colonwarn commas crlf dbugeval dbugeval2 \
 	dbugeval3 dbugeval4 dbugtypedre1 dbugtypedre2 delsub \
 	devfd devfd1 devfd2 dfacheck1 dumpvars \
 	errno exit fieldwdth forcenum fpat1 fpat2 \
@@ -1893,7 +1890,6 @@ NEED_LOCALE_RU = mtchi18n
 
 # List of tests that fail on MinGW
 EXPECTED_FAIL_MINGW = \
-	backbigs1 backsmalls1 clos1way6 close_status dbugeval4\
 	devfd devfd1 devfd2 errno exitval2 fork fork2 fts functab5 \
 	getfile getlnhd ignrcas3 inetdayt inetecht inf-nan-torture \
 	iolint mbfw1 mbprintf1 mbprintf4 mbstr1 mbstr2 \
@@ -1903,7 +1899,6 @@ EXPECTED_FAIL_MINGW = \
 # List of tests that fail on z/OS
 EXPECTED_FAIL_ZOS = \
 	aasort aasorti arraysort asort asorti backbigs1 backsmalls1 \
-	backsmalls2 beginfile1 beginfile2 charasbytes commas clos1way6 concat4 \
 	dfamb1 double1 double2 errno fmttest forcenum getlndir gsubtst5 \
 	ignrcas2 inf-nan-torture iolint lc_num1 mbfw1 mbprintf1 \
 	mbprintf2 mbprintf3 mbprintf4 mbprintf5 mbstr1 mbstr2 mtchi18n \
@@ -4452,7 +4447,6 @@ clos1way5:
 	@-AWKPATH="$(srcdir)" $(AWK) -f $@.awk  >_$@ 2>&1 || echo EXIT CODE: $$? >>_$@
 	@-$(CMP) "$(srcdir)"/$@.ok _$@ && rm -f _$@
 
-clos1way6:
 	@echo $@ $(ZOS_FAIL)
 	@-AWKPATH="$(srcdir)" $(AWK) -f $@.awk  >_$@ 2>&1 || echo EXIT CODE: $$? >>_$@
 	@-$(CMP) "$(srcdir)"/$@.ok _$@ && rm -f _$@
