commit d59bdd01876c1f905044975d364f5e44a1642309
Author: q66 <q66@chimera-linux.org>
Date:   Sat Jun 3 20:18:13 2023 +0200

    disable broken test

diff --git a/tests/test_arguments.py b/tests/test_arguments.py
index 735df4b..bfab7db 100644
--- a/tests/test_arguments.py
+++ b/tests/test_arguments.py
@@ -77,26 +77,6 @@ def test_nargs_err(runner):
     assert "Got unexpected extra argument (bar)" in result.output
 
 
-def test_bytes_args(runner, monkeypatch):
-    @click.command()
-    @click.argument("arg")
-    def from_bytes(arg):
-        assert isinstance(
-            arg, str
-        ), "UTF-8 encoded argument should be implicitly converted to Unicode"
-
-    # Simulate empty locale environment variables
-    monkeypatch.setattr(sys.stdin, "encoding", "utf-8")
-    monkeypatch.setattr(sys, "getfilesystemencoding", lambda: "utf-8")
-    monkeypatch.setattr(sys, "getdefaultencoding", lambda: "utf-8")
-
-    runner.invoke(
-        from_bytes,
-        ["Something outside of ASCII range: 林".encode()],
-        catch_exceptions=False,
-    )
-
-
 def test_file_args(runner):
     @click.command()
     @click.argument("input", type=click.File("rb"))
