commit 12e6a97edddaa36448105ca5f997ef097527cbec
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sun Nov 13 06:08:52 2022 +0100

    drop unneeded glslang deps

diff --git a/src/glsl/meson.build b/src/glsl/meson.build
index 19b4bc6..89a4adb 100644
--- a/src/glsl/meson.build
+++ b/src/glsl/meson.build
@@ -18,11 +18,7 @@ else
 
   glslang_deps = [
     cxx.find_library('glslang',             required: glslang_req),
-    cxx.find_library('MachineIndependent',  required: false),
-    cxx.find_library('OSDependent',         required: glslang_req),
     cxx.find_library('HLSL',                required: glslang_req),
-    cxx.find_library('OGLCompiler',         required: glslang_req),
-    cxx.find_library('GenericCodeGen',      required: false),
     cxx.find_library('SPVRemapper',         required: glslang_req),
     cxx.find_library('SPIRV',               required: glslang_req),
     cxx.find_library('SPIRV-Tools-opt',     required: false),
