https://github.com/ValveSoftware/gamescope/issues/1410
--
diff --git a/src/Backends/WaylandBackend.cpp b/src/Backends/WaylandBackend.cpp
index 6f578a2..a4deba2 100644
--- a/src/Backends/WaylandBackend.cpp
+++ b/src/Backends/WaylandBackend.cpp
@@ -2200,6 +2200,11 @@ namespace gamescope
             {
                 assert( !m_pRelativePointer );
 
+                // lmao
+                if (!pSurface) {
+                    return;
+                }
+
                 m_pLockedPointer = zwp_pointer_constraints_v1_lock_pointer( m_pPointerConstraints, pSurface, m_pPointer, nullptr, ZWP_POINTER_CONSTRAINTS_V1_LIFETIME_PERSISTENT );
                 m_pRelativePointer = zwp_relative_pointer_manager_v1_get_relative_pointer( m_pRelativePointerManager, m_pPointer );
             }
