From e7b94fb27b9777f9cbe4a7f870c474418706f682 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Fri, 3 Jan 2025 14:42:28 +0100
Subject: [PATCH] zfs_strerror: use strerror_l conditionally

---
 include/libzutil.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/include/libzutil.h b/include/libzutil.h
index 3a52741..64e26e4 100644
--- a/include/libzutil.h
+++ b/include/libzutil.h
@@ -275,10 +275,11 @@ _LIBZUTIL_H void update_vdev_config_dev_sysfs_path(nvlist_t *nv,
  */
 static inline char *zfs_strerror(int errnum) {
 #ifdef HAVE_STRERROR_L
-	return (strerror_l(errnum, uselocale(0)));
-#else
-	return (strerror(errnum));
+	locale_t lval = uselocale(0);
+	if (lval != LC_GLOBAL_LOCALE)
+		return (strerror_l(errnum, lval));
 #endif
+	return (strerror(errnum));
 }
 
 #ifdef	__cplusplus
-- 
2.47.1

