--- a/tests/zfs-tests/cmd/statx.c
+++ b/tests/zfs-tests/cmd/statx.c
@@ -28,6 +28,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <sys/syscall.h>
+#include <sys/stat.h>
 #include <unistd.h>
 
 /*
@@ -54,10 +55,6 @@
 #endif /* __NR_statx */
 
 
-int
-statx(int, const char *, int, unsigned int, void *)
-    __attribute__((weak));
-
 static inline int
 _statx(int fd, const char *path, int flags, unsigned int mask, void *stx)
 {
