--- a/target/riscv/kvm.c
+++ b/target/riscv/kvm.c
@@ -729,7 +729,7 @@ static void kvm_riscv_read_multiext_legacy(RISCVCPU *cpu,
             } else {
                 error_report("Unable to read ISA_EXT KVM register %s, "
                              "error code: %s", multi_ext_cfg->name,
-                             strerrorname_np(errno));
+                             strerror(errno));
                 exit(EXIT_FAILURE);
             }
         } else {
@@ -791,7 +791,7 @@ static void kvm_riscv_init_multiext_cfg(RISCVCPU *cpu, KVMScratchCPU *kvmcpu)
          * Error out if we get any other errno.
          */
         error_report("Error when accessing get-reg-list, code: %s",
-                     strerrorname_np(errno));
+                     strerror(errno));
         exit(EXIT_FAILURE);
     }
 
@@ -801,7 +801,7 @@ static void kvm_riscv_init_multiext_cfg(RISCVCPU *cpu, KVMScratchCPU *kvmcpu)
     ret = ioctl(kvmcpu->cpufd, KVM_GET_REG_LIST, reglist);
     if (ret) {
         error_report("Error when reading KVM_GET_REG_LIST, code %s ",
-                     strerrorname_np(errno));
+                     strerror(errno));
         exit(EXIT_FAILURE);
     }
 
@@ -824,7 +824,7 @@ static void kvm_riscv_init_multiext_cfg(RISCVCPU *cpu, KVMScratchCPU *kvmcpu)
         if (ret != 0) {
             error_report("Unable to read ISA_EXT KVM register %s, "
                          "error code: %s", multi_ext_cfg->name,
-                         strerrorname_np(errno));
+                         strerror(errno));
             exit(EXIT_FAILURE);
         }
 
