commit 953d1edbd94c8122c4319976d700d2d7e28cdc21
Author: q66 <q66@chimera-linux.org>
Date:   Thu May 18 17:31:08 2023 +0200

    fix hexagon idef-parser prepare

diff --git a/target/hexagon/idef-parser/prepare b/target/hexagon/idef-parser/prepare
index 72d6fcb..b0f7700 100755
--- a/target/hexagon/idef-parser/prepare
+++ b/target/hexagon/idef-parser/prepare
@@ -21,4 +21,4 @@ set -e
 set -o pipefail
 
 # Run the preprocessor and drop comments
-cpp "$@"
+clang-cpp "$@"
