--- a/seccomp.sh
+++ b/seccomp.sh
@@ -242,7 +242,7 @@
 	__calls="$(sed -n 's/[\t ]*\*[\t ]*#syscalls\(:'"${__p}"'\|\)[\t ]\{1,\}\(.*\)/\2/p' ${IN})"
 	__calls="${__calls} ${EXTRA_SYSCALLS:-}"
 	__calls="$(filter ${__calls})"
-	echo "seccomp profile ${__p} allows: ${__calls}" | tr '\n' ' ' | fmt -t
+	echo "seccomp profile ${__p} allows: ${__calls}" | tr '\n' ' ' | fmt -p
 
 	# Pad here to keep gen_profile() "simple"
 	__count=0
