--- a/seccomp.sh
+++ b/seccomp.sh
@@ -245,7 +245,7 @@
 
 	cols="$(stty -a | sed -n 's/.*columns \([0-9]*\).*/\1/p' || :)" 2>/dev/null
 	case $cols in [0-9]*) col_args="-w ${cols}";; *) col_args="";; esac
-	echo "seccomp profile ${__p} allows: ${__calls}" | tr '\n' ' ' | fmt -t ${col_args}
+	echo "seccomp profile ${__p} allows: ${__calls}" | tr '\n' ' ' | fmt -p ${col_args}
 
 	# Pad here to keep gen_profile() "simple"
 	__count=0
