From 6ecabd82314b0a8366e03d255290672d8bf0339f Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Mon, 19 Jul 2021 22:09:54 +0200
Subject: [PATCH] fix maketab cross build

---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index 9ceaaad..b2c99f8 100644
--- a/makefile
+++ b/makefile
@@ -61,7 +61,7 @@ proctab.c:	maketab
 	./maketab awkgram.tab.h >proctab.c
 
 maketab:	awkgram.tab.h maketab.c
-	$(HOSTCC) $(CFLAGS) maketab.c -o maketab
+	$(BUILD_CC) $(BUILD_CFLAGS) $(BUILD_LDFLAGS) maketab.c -o maketab
 
 bundle:
 	@cp awkgram.tab.h awkgram.tab.h.bak
-- 
2.32.0

