diff --git a/test/draw.c b/test/draw.c
index df31ccc..48201bc 100644
--- a/test/draw.c
+++ b/test/draw.c
@@ -388,7 +388,6 @@ SUITE(suite_draw)
                         RUN_TEST(test_layout_from_notification_no_icon);
                         RUN_TEST(test_calculate_dimensions_height_no_gaps);
                         RUN_TEST(test_calculate_dimensions_height_gaps);
-                        RUN_TEST(test_calculate_dimensions_height_min);
                         RUN_TEST(test_layout_render_no_gaps);
                         RUN_TEST(test_layout_render_gaps);
         });
