literally doesn't compile
--
--- a/src/Misc/Util.h
+++ b/src/Misc/Util.h
@@ -21,18 +21,9 @@
 #ifndef UTIL_H
 #define UTIL_H
 
+#include <algorithm>
+#include <set>
 #include <string>
-
-namespace std {// forward declarations to avoid pervasive includes
-
-  template<typename T>
-  class allocator;
-  template<typename K, typename CMP, class ALLO>
-  class set;
-
-  template<typename IT, typename V>
-  IT find (IT, IT, V const&);
-}
 
 
 namespace util {
