commit 1a53365ce931080111f2c984f8d52b4d87ae9ecc
Author: q66 <q66@chimera-linux.org>
Date:   Mon Jan 24 23:47:44 2022 +0100

    fix heimdal support

diff --git a/cmake/modules/FindKRB5.cmake b/cmake/modules/FindKRB5.cmake
index b0e46ac..e0f6ed1 100644
--- a/cmake/modules/FindKRB5.cmake
+++ b/cmake/modules/FindKRB5.cmake
@@ -54,7 +54,7 @@ unset(optupper)
 set(mit_includes "${WITH_KRB5}/include")
 set(mit_libs "-lkrb5 -lk5crypto -lcom_err -lgssapi_krb5")
 set(heimdal_includes "${WITH_KRB5}/include/heimdal")
-set(heimdal_libs "-lkrb5 -lcrypto -lasn1 -lcom_err -lroken -lgssapi")
+set(heimdal_libs "-lkrb5 -lhcrypto -lasn1 -lcom_err -lroken -lgssapi")
 set(sun_includes "${WITH_KRB5}/include/kerberosv5")
 set(sun_libs "-lkrb5 -lgss")
 
