commit 97962c1d62b16332571b633dcb819264f49d1160
Author: q66 <q66@chimera-linux.org>
Date:   Fri May 12 02:19:36 2023 +0200

    fix backtrace build

diff --git a/app/core/gimpbacktrace-backend.h b/app/core/gimpbacktrace-backend.h
index 4bfb1c9..51bf979 100644
--- a/app/core/gimpbacktrace-backend.h
+++ b/app/core/gimpbacktrace-backend.h
@@ -22,7 +22,7 @@
 #define __GIMP_BACKTRACE_BACKEND_H__
 
 
-#ifdef __gnu_linux__
+#ifdef __GLIBC__
 # define GIMP_BACKTRACE_BACKEND_LINUX
 #elif defined (G_OS_WIN32) && defined (ARCH_X86)
 # define GIMP_BACKTRACE_BACKEND_WINDOWS
