--- a/meson.build
+++ b/meson.build
@@ -464,7 +464,7 @@
 exiv2             = dependency('exiv2',              version: '>='+exiv2_minver)
 gexiv2_minver     = '0.14.0'
 gexiv2_maxver     = '0.15.0'
-gexiv2            = dependency('gexiv2',             version: ['>='+gexiv2_minver, '<'+gexiv2_maxver])
+gexiv2            = dependency('gexiv2-0.16')
 
 gio               = dependency('gio-2.0')
 gio_specific_name = platform_windows ? 'gio-windows-2.0' : 'gio-unix-2.0'
--- a/libgimp/meson.build
+++ b/libgimp/meson.build
@@ -389,7 +389,7 @@ libgimp_deps_table = [
     { 'gir': 'cairo-1.0',         'vapi': 'cairo-1.0',       },
     { 'gir': 'GdkPixbuf-2.0',     'vapi': 'gdk-pixbuf-2.0',  },
     { 'gir': 'Gegl-0.4',          'vapi': 'gegl-0.4',        },
-    { 'gir': 'GExiv2-0.10',       'vapi': 'gexiv2',          },
+    { 'gir': 'GExiv2-0.16',       'vapi': 'gexiv2-0.16',     },
     { 'gir': 'Gio-2.0',           'vapi': 'gio-2.0',         },
     { 'gir': gio_specific_gir,    'vapi': gio_specific_vapi, },
     { 'gir': 'GLib-2.0',          'vapi': 'glib-2.0',        },
