pulling in xserver-xorg-xvfb creates a cycle
--
--- a/meson.build
+++ b/meson.build
@@ -1193,34 +1193,6 @@ test(
     env: test_env,
 )
 if get_option('enable-x11')
-    has_xvfb = find_program('Xvfb', required: false)
-    has_xkbcomp = find_program('xkbcomp', required: false)
-    # We only warn because the build machine may not be the same
-    # as the host/test machine.
-    if not has_xvfb.found()
-        warning('Xvfb program not found, but is required to run X11 tests.')
-    endif
-    if not has_xkbcomp.found()
-        warning('xkbcomp program not found, but is required to run X11 tests.')
-    endif
-    test(
-        'x11',
-        executable('test-x11', 'test/x11.c', dependencies: x11_xvfb_test_dep),
-        env: test_env,
-        is_parallel : false,
-    )
-    test(
-        'x11comp',
-        executable(
-            'test-x11comp',
-            'test/x11comp.c',
-            'test/utils-text.c',
-            'test/utils-text.h',
-            dependencies: x11_xvfb_test_dep
-        ),
-        env: test_env,
-        is_parallel : false,
-    )
 endif
 if get_option('enable-xkbregistry')
     test(
