Enable xz multithreaded compression by default.

	--xtraeme

diff -ruN a/libarchive/archive_write_add_filter_xz.c b/libarchive/archive_write_add_filter_xz.c
--- a/libarchive/archive_write_add_filter_xz.c	2026-07-28 11:30:38.000000000 +0200
+++ b/libarchive/archive_write_add_filter_xz.c	2026-10-03 09:15:51.892186282 +0200
@@ -152,7 +152,7 @@
 	if (xz == NULL)
 		goto memerr;
 	xz->compression_level = LZMA_PRESET_DEFAULT;
-	xz->threads = 1;
+	xz->threads = lzma_cputhreads() > 12 ? 12 : lzma_cputhreads();
 
 	f = __archive_write_allocate_filter(a);
 	if (f == NULL)
