commit 9caa602d854dc538b8524cfb3ecc16d70fc0d016
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Thu Mar 16 18:35:25 2023 +0100

    drop iconv requirement from pkgconfig

diff --git a/configure b/configure
index ab8c099..cad96a7 100755
--- a/configure
+++ b/configure
@@ -16176,7 +16176,6 @@ fi
 
     am_save_LIBS="$LIBS"
     LIBS="${LIBS} ${LIBICONV}"
-    LIBSREQUIRED="$LIBSREQUIRED${LIBSREQUIRED:+ }iconv"
     ac_fn_c_check_func "$LINENO" "locale_charset" "ac_cv_func_locale_charset"
 if test "x$ac_cv_func_locale_charset" = xyes
 then :
