--- a/meson.build
+++ b/meson.build
@@ -124,7 +124,7 @@ dependency('wayland-client', required: gtk_wayland.found())
 selinux = dependency('libselinux', version: '>= 2.0', required: get_option('selinux'))
 cloudproviders = dependency('cloudproviders', version: '>= 0.3.1', required: get_option('cloudproviders'))
 if get_option('extensions')
-  gexiv = dependency('gexiv2', version: '>= 0.14.2')
+  gexiv = dependency('gexiv2-0.16', version: '>= 0.14.2')
   gdkpixbuf = dependency('gdk-pixbuf-2.0', version: '>= 2.30.0')
   gst_tag_dep = dependency('gstreamer-tag-1.0')
   gst_pbutils_dep = dependency('gstreamer-pbutils-1.0')
