There are some tests with deprecated empty arg lists.

--- a/meson.build
+++ b/meson.build
@@ -63,7 +63,6 @@ add_project_arguments(
     '-Werror=pointer-arith',
     '-Werror=sequence-point',
     '-Werror=shadow',
-    '-Werror=strict-prototypes',
     '-Werror=undef',
     '-Werror=uninitialized',
     # Context: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/commit/ce54a2527555e51e4ebf4cce9cbb6259cafa89a4
