--- a/src/python/setup.py
+++ b/src/python/setup.py
@@ -28,21 +28,23 @@
 from Cython.Distutils import build_ext
 
 setup(
-	name = "seccomp",
-	version = os.environ["VERSION_RELEASE"],
-	description = "Python binding for libseccomp",
-	long_description = "Python API for the Linux Kernel's syscall filtering capability, seccomp.",
-	url = "https://github.com/seccomp/libseccomp",
-	maintainer = "Paul Moore",
-	maintainer_email = "paul@paul-moore.com",
-	license = "LGPLv2.1",
-	platforms = "Linux",
-	cmdclass = {'build_ext': build_ext},
-	ext_modules = [
-		Extension("seccomp", ["seccomp.pyx"],
-			# unable to handle libtool libraries directly
-			extra_objects=["../.libs/libseccomp.a"],
-			# fix build warnings, see PEP 3123
-			extra_compile_args=["-fno-strict-aliasing"])
-	]
+    name="seccomp",
+    version=os.environ["VERSION_RELEASE"],
+    description="Python binding for libseccomp",
+    long_description="Python API for the Linux Kernel's syscall filtering capability, seccomp.",
+    url="https://github.com/seccomp/libseccomp",
+    maintainer="Paul Moore",
+    maintainer_email="paul@paul-moore.com",
+    license="LGPLv2.1",
+    platforms="Linux",
+    cmdclass={"build_ext": build_ext},
+    ext_modules=[
+        Extension(
+            "seccomp",
+            ["seccomp.pyx"],
+            libraries=["seccomp"],
+            # fix build warnings, see PEP 3123
+            extra_compile_args=["-fno-strict-aliasing"],
+        )
+    ],
 )
