diff --git a/lxqt-config-input/touchpaddevice.cpp b/lxqt-config-input/touchpaddevice.cpp
index 59079b9..ac23952 100644
--- a/lxqt-config-input/touchpaddevice.cpp
+++ b/lxqt-config-input/touchpaddevice.cpp
@@ -368,7 +368,7 @@ bool TouchpadDevice::setTapToDragEnabled(bool enabled) const
 
 bool TouchpadDevice::setAccelSpeed(float speed) const
 {
-    return set_xi2_property(LIBINPUT_PROP_ACCEL, QList<QVariant>({speed}));
+    return set_xi2_property(LIBINPUT_PROP_ACCEL, QList<QVariant>({QVariant(speed)}));
 }
 
 int TouchpadDevice::scrollMethodsAvailable() const
