diff --git a/src/libsodium/crypto_aead/aes256gcm/aead_aes256gcm.c b/src/libsodium/crypto_aead/aes256gcm/aead_aes256gcm.c
index 2946ba8..3847f2d 100644
--- a/src/libsodium/crypto_aead/aes256gcm/aead_aes256gcm.c
+++ b/src/libsodium/crypto_aead/aes256gcm/aead_aes256gcm.c
@@ -47,8 +47,7 @@ crypto_aead_aes256gcm_keygen(unsigned char k[crypto_aead_aes256gcm_KEYBYTES])
     randombytes_buf(k, crypto_aead_aes256gcm_KEYBYTES);
 }
 
-#if !((defined(HAVE_ARMCRYPTO) && defined(__clang__) && defined(NATIVE_LITTLE_ENDIAN)) || \
-      (defined(HAVE_TMMINTRIN_H) && defined(HAVE_WMMINTRIN_H)))
+#if !(defined(HAVE_TMMINTRIN_H) && defined(HAVE_WMMINTRIN_H))
 
 #ifndef ENOSYS
 #define ENOSYS ENXIO
@@ -154,4 +153,4 @@ crypto_aead_aes256gcm_is_available(void)
     return 0;
 }
 
-#endif
\ No newline at end of file
+#endif
diff --git a/src/libsodium/crypto_aead/aes256gcm/armcrypto/aead_aes256gcm_armcrypto.c b/src/libsodium/crypto_aead/aes256gcm/armcrypto/aead_aes256gcm_armcrypto.c
index 0a5a128..3886327 100644
--- a/src/libsodium/crypto_aead/aes256gcm/armcrypto/aead_aes256gcm_armcrypto.c
+++ b/src/libsodium/crypto_aead/aes256gcm/armcrypto/aead_aes256gcm_armcrypto.c
@@ -13,7 +13,7 @@
 #include "runtime.h"
 #include "utils.h"
 
-#if defined(HAVE_ARMCRYPTO) && defined(__clang__) && defined(NATIVE_LITTLE_ENDIAN)
+#if 0
 
 #if !defined(MSC_VER) || _MSC_VER < 1800
 #define __vectorcall
