commit 98c5ffde7a2400c6724481d83351010d5aba5895
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Tue Apr 5 05:35:56 2022 +0200

    disable tests broken with new adwaita-icon-theme

diff --git a/testsuite/gtk/meson.build b/testsuite/gtk/meson.build
index 206af0d..db6d38f 100644
--- a/testsuite/gtk/meson.build
+++ b/testsuite/gtk/meson.build
@@ -13,7 +13,6 @@ tests = [
   ['builder', [], [], gtk_tests_export_dynamic_ldflag],
   ['builderparser'],
   ['cellarea'],
-  ['check-icon-names'],
   ['check-cursor-names'],
   ['cssprovider'],
   ['defaultvalue'],
diff --git a/testsuite/reftests/meson.build b/testsuite/reftests/meson.build
index 37a855a..61af974 100644
--- a/testsuite/reftests/meson.build
+++ b/testsuite/reftests/meson.build
@@ -148,9 +148,6 @@ test_data = [
   'box-shadow-with-blend-mode.ui',
   'button-wrapping.ui',
   'button-wrapping.ref.ui',
-  'cellrenderer-pixbuf-stock-rtl.css',
-  'cellrenderer-pixbuf-stock-rtl.ui',
-  'cellrenderer-pixbuf-stock-rtl.ref.ui',
   'color-transition.css',
   'color-transition.ref.ui',
   'color-transition.ui',
