--- Makefile
+++ Makefile
@@ -6,7 +6,7 @@
 binprefix = 
 manprefix = /usr
 exec_prefix = $(binprefix)/
-sbindir = $(exec_prefix)sbin
+sbindir = $(exec_prefix)bin
 mandir = $(manprefix)/share/man
 oldmandir = $(manprefix)/man
 
@@ -22,8 +22,7 @@
 
 OBJS = hdparm.o identify.o sgio.o sysfs.o geom.o fallocate.o fibmap.o fwdownload.o dvdspeed.o wdidle3.o apt.o
 
-all:
-	$(MAKE) -j4 hdparm
+all: hdparm
 
 static:
 	LDFLAGS=-static $(MAKE)
@@ -52,9 +51,9 @@
 	if [ -f $(DESTDIR)$(sbindir)/hdparm ]; then rm -f $(DESTDIR)$(sbindir)/hdparm ; fi
 	if [ -f $(DESTDIR)$(mandir)/man8/hdparm.8 ]; then rm -f $(DESTDIR)$(mandir)/man8/hdparm.8 ;\
 	elif [ -f $(DESTDIR)$(oldmandir)/man8/hdparm.8 ]; then rm -f $(DESTDIR)$(oldmandir)/man8/hdparm.8 ; fi
-	$(INSTALL_PROGRAM) -D hdparm $(DESTDIR)$(sbindir)/hdparm
-	if [ -d $(DESTDIR)$(mandir) ]; then $(INSTALL_DATA) -D hdparm.8 $(DESTDIR)$(mandir)/man8/hdparm.8 ;\
-	elif [ -d $(DESTDIR)$(oldmandir) ]; then $(INSTALL_DATA) -D hdparm.8 $(DESTDIR)$(oldmandir)/man8/hdparm.8 ; fi
+	$(INSTALL_PROGRAM) hdparm $(DESTDIR)$(sbindir)/hdparm
+	if [ -d $(DESTDIR)$(mandir) ]; then $(INSTALL_DATA) hdparm.8 $(DESTDIR)$(mandir)/man8/hdparm.8 ;\
+	elif [ -d $(DESTDIR)$(oldmandir) ]; then $(INSTALL_DATA) hdparm.8 $(DESTDIR)$(oldmandir)/man8/hdparm.8 ; fi
 
 clean:
 	-rm -f hdparm $(OBJS) core 2>/dev/null
