diff --git a/src/config.h b/src/config.h
index 0994f95..ef1cab3 100644
--- a/src/config.h
+++ b/src/config.h
@@ -49,8 +49,8 @@
 #define ETHERSFILE "/etc/ethers"
 #define DEFLEASE 3600 /* default DHCPv4 lease time, one hour */
 #define DEFLEASE6 (3600*24) /* default lease time for DHCPv6. One day. */
-#define CHUSER "nobody"
-#define CHGRP "dip"
+#define CHUSER "_dnsmasq"
+#define CHGRP "_dnsmasq"
 #define TFTP_MAX_CONNECTIONS 50 /* max simultaneous connections */
 #define TFTP_MAX_WINDOW 32 /* max window size to negotiate */
 #define TFTP_TRANSFER_TIME 120 /* Abandon TFTP transfers after this long. Two mins. */
