--- a/configure	2024-02-05 23:52:07.000000000 +0100
+++ b/configure	2024-06-09 16:59:35.785197609 +0200
@@ -793,12 +793,6 @@
 
 CFLAGS="${CFLAGS}${CFLAGS_ADD}"
 
-if [ "$CC_FOR_BUILD" = "" ] ; then
-    CC_FOR_BUILD="${CC} ${CFLAGS}"
-else
-    enableFrompb=no
-fi
-
 # Add automatic thread compilation flags, unless suppressed by --disable-auto-flags
 if [ "$addflags" = "yes" ] ; then
     if [ "$threadFlags" != "" ] ; then
