other one don't exist
--
--- a/c/clearcache.c
+++ b/c/clearcache.c
@@ -45,7 +45,7 @@
 #ifdef TARGET_OS_IPHONE
   sys_icache_invalidate((void *)start, (char *)end-(char *)start);
 #else
-  __clear_cache((char *)start, (char *)end);
+  __builtin___clear_cache((char *)start, (char *)end);
 # if defined(__clang__) && defined(__aarch64__) && !defined(__APPLE__)
   /* Seem to need an extra combination of barriers here to make up for
      something in Clang's __clear_cache() */
