diff --git a/libusb/io.c b/libusb/io.c
index ab84ba6..549cca8 100644
--- a/libusb/io.c
+++ b/libusb/io.c
@@ -21,6 +21,13 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#ifndef TIMESPEC_TO_TIMEVAL
+#define TIMESPEC_TO_TIMEVAL(tv, ts) {		\
+	(tv)->tv_sec = (ts)->tv_sec;		\
+	(tv)->tv_usec = (ts)->tv_nsec / 1000;	\
+}
+#endif
+
 #include "libusbi.h"
 
 /**
