From 20cac24b794987f09321201130452f2102e26194 Mon Sep 17 00:00:00 2001
From: Martin Liska <mliska@suse.cz>
Date: Fri, 19 May 2023 21:04:04 +0200
Subject: [PATCH] Fix current LLVM tip branch.

---
 clang_delta/RemoveUnusedStructField.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/clang_delta/RemoveUnusedStructField.cpp b/clang_delta/RemoveUnusedStructField.cpp
index 44b596f1..e9043df5 100644
--- a/clang_delta/RemoveUnusedStructField.cpp
+++ b/clang_delta/RemoveUnusedStructField.cpp
@@ -247,7 +247,11 @@ const Expr *RemoveUnusedStructField::getInitExprFromDesignatedInitExpr(
       }
       else {
         const DesignatedInitExpr::Designator *DS = DIE->getDesignator(0);
+#if LLVM_VERSION_MAJOR >= 17
+        const FieldDecl *CurrFD = DS->getFieldDecl();
+#else
         const FieldDecl *CurrFD = DS->getField();
+#endif
         if ((CurrFD && FD == CurrFD) ||
             (CurrFD == NULL && DS->getFieldName() == FD->getIdentifier())) {
           IsFirstField = (I == 0);
