From c0e2623434a7736e4c4864a822a4a5cf9c8746c1 Mon Sep 17 00:00:00 2001
From: Martin Liska <martin.liska@melowntech.com>
Date: Mon, 17 Jul 2023 06:20:35 +0200
Subject: [PATCH] LLVM 16: add missing header file for toString

---
 clang_delta/Transformation.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clang_delta/Transformation.cpp b/clang_delta/Transformation.cpp
index 8dc9f75d..dcd5248b 100644
--- a/clang_delta/Transformation.cpp
+++ b/clang_delta/Transformation.cpp
@@ -23,6 +23,7 @@
 #include "clang/Basic/SourceManager.h"
 #include "llvm/Support/raw_ostream.h"
 #include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringExtras.h"
 
 using namespace std;
 using namespace clang;
