diff --git a/include/libunwind-ppc32.h b/include/libunwind-ppc32.h
index 303597c..7f7170a 100644
--- a/include/libunwind-ppc32.h
+++ b/include/libunwind-ppc32.h
@@ -69,7 +69,7 @@ extern "C" {
 
 #define UNW_TDEP_CURSOR_LEN 280
 
-#if __WORDSIZE==32
+#if !defined(__powerpc64__)
 typedef uint32_t unw_word_t;
 typedef int32_t unw_sword_t;
 # define UNW_WORD_MAX UINT32_MAX
diff --git a/include/libunwind-ppc64.h b/include/libunwind-ppc64.h
index 1ce48bf..5839ebd 100644
--- a/include/libunwind-ppc64.h
+++ b/include/libunwind-ppc64.h
@@ -69,7 +69,7 @@ extern "C" {
 
 #define UNW_TDEP_CURSOR_LEN 280
 
-#if __WORDSIZE==32
+#if !defined(__powerpc64__)
 typedef uint32_t unw_word_t;
 typedef int32_t unw_sword_t;
 #define UNW_WORD_MAX UINT32_MAX
diff --git a/src/ppc32/Ginit.c b/src/ppc32/Ginit.c
index 9444cbb..13b5aff 100644
--- a/src/ppc32/Ginit.c
+++ b/src/ppc32/Ginit.c
@@ -49,7 +49,7 @@ uc_addr (ucontext_t *uc, int reg)
 
   if ((unsigned) (reg - UNW_PPC32_R0) < 32)
 #if defined(__linux__)
-    addr = &uc->uc_mcontext.uc_regs->gregs[reg - UNW_PPC32_R0];
+    addr = &uc->uc_mcontext.gregs[reg - UNW_PPC32_R0];
 #elif defined(__FreeBSD__)
     addr = &uc->uc_mcontext.mc_gpr[reg - UNW_PPC32_R0];
 #endif
@@ -58,7 +58,7 @@ uc_addr (ucontext_t *uc, int reg)
   if ( ((unsigned) (reg - UNW_PPC32_F0) < 32) &&
        ((unsigned) (reg - UNW_PPC32_F0) >= 0) )
 #if defined(__linux__)
-    addr = &uc->uc_mcontext.uc_regs->fpregs.fpregs[reg - UNW_PPC32_F0];
+    addr = &uc->uc_mcontext.fpregs.fpregs[reg - UNW_PPC32_F0];
  #elif defined(__FreeBSD__)
     addr = &uc->uc_mcontext.mc_fpreg[reg - UNW_PPC32_F0];
 #endif
@@ -85,7 +85,7 @@ uc_addr (ucontext_t *uc, int reg)
           return NULL;
         }
 #if defined(__linux__)
-      addr = &uc->uc_mcontext.uc_regs->gregs[gregs_idx];
+      addr = &uc->uc_mcontext.gregs[gregs_idx];
 #elif defined(__FreeBSD__)
       addr = &uc->uc_mcontext.mc_gpr[gregs_idx];
 #endif
diff --git a/src/ppc32/ucontext_i.h b/src/ppc32/ucontext_i.h
index ee93c69..71a6278 100644
--- a/src/ppc32/ucontext_i.h
+++ b/src/ppc32/ucontext_i.h
@@ -44,8 +44,8 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
 //#define MQ_IDX                36
 #define LINK_IDX        36
 
-#define _UC_MCONTEXT_GPR(x) ( ((void *)&dmy_ctxt.uc_mcontext.uc_regs->gregs[x] - (void *)&dmy_ctxt) )
-#define _UC_MCONTEXT_FPR(x) ( ((void *)&dmy_ctxt.uc_mcontext.uc_regs->fpregs.fpregs[x] - (void *)&dmy_ctxt) )
+#define _UC_MCONTEXT_GPR(x) ( ((void *)&dmy_ctxt.uc_mcontext.gregs[x] - (void *)&dmy_ctxt) )
+#define _UC_MCONTEXT_FPR(x) ( ((void *)&dmy_ctxt.uc_mcontext.fpregs.fpregs[x] - (void *)&dmy_ctxt) )
 
 /* These are dummy structures used only for obtaining the offsets of the
    various structure members. */
