--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -195,10 +195,6 @@
 
 # Support for std::filesystem
 # GCC version <9 and Clang (all versions) require -lstdc++fs
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR ${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS "9")
-  target_link_libraries(runtime "stdc++fs")
-  target_link_libraries(libbpftrace "stdc++fs")
-endif()
 
 if (BUILD_ASAN)
   target_compile_options(bpftrace PUBLIC "-fsanitize=address")
