commit c5f73cc56edc17eee3fadec7e09ff4090b10e9d0
Author: q66 <q66@chimera-linux.org>
Date:   Sun Jan 23 21:31:08 2022 +0100

    use libedit instead of readline

diff --git a/js/moz.configure b/js/moz.configure
index 3c3d0d43..985a42a1 100644
--- a/js/moz.configure
+++ b/js/moz.configure
@@ -919,12 +919,12 @@ option(
 
 has_readline = check_symbol(
     "readline",
-    flags=["-lreadline"],
+    flags=["-ledit"],
     when="--enable-readline",
     onerror=lambda: die("No system readline library found"),
 )
 
-set_config("EDITLINE_LIBS", ["-lreadline"], when=has_readline)
+set_config("EDITLINE_LIBS", ["-ledit"], when=has_readline)
 
 
 @depends("--enable-readline", editline, when=editline)
