commit abc301fbccc67ed4adf90f883cd696f86df9dd41
Author: q66 <q66@chimera-linux.org>
Date:   Fri Dec 31 02:39:17 2021 +0100

    how about you stop trying to be "helpful"?

diff --git a/configure b/configure
index da631a4..db5aba2 100755
--- a/configure
+++ b/configure
@@ -171,14 +171,6 @@ for t in ${all_targets}; do
     [ -f "${source_path}/${t}.mk" ] && enable_feature ${t}
 done
 
-if ! diff --version >/dev/null; then
-  die "diff missing: Try installing diffutils via your package manager."
-fi
-
-if ! perl --version >/dev/null; then
-    die "Perl is required to build"
-fi
-
 if [ "`cd \"${source_path}\" && pwd`" != "`pwd`" ]; then
   # test to see if source_path already configured
   if [ -f "${source_path}/vpx_config.h" ]; then
