diff --git a/data/pam/lightdm b/data/pam/lightdm
index fed8a9b..5a1b8f1 100644
--- a/data/pam/lightdm
+++ b/data/pam/lightdm
@@ -17,4 +17,5 @@ password  required pam_unix.so
 
 # Setup session
 session   required pam_unix.so
-session   optional pam_systemd.so
+-session   optional pam_turnstile.so
+-session   optional pam_elogind.so
diff --git a/data/pam/lightdm-autologin b/data/pam/lightdm-autologin
index 157f469..acd67ab 100644
--- a/data/pam/lightdm-autologin
+++ b/data/pam/lightdm-autologin
@@ -19,4 +19,5 @@ password  required pam_deny.so
 
 # Setup session
 session   required pam_unix.so
-session   optional pam_systemd.so
+-session   optional pam_turnstile.so
+-session   optional pam_elogind.so
diff --git a/data/pam/lightdm-greeter b/data/pam/lightdm-greeter
index 9a6862b..680ce27 100644
--- a/data/pam/lightdm-greeter
+++ b/data/pam/lightdm-greeter
@@ -14,4 +14,5 @@ password  required pam_deny.so
 
 # Setup session
 session   required pam_unix.so
-session   optional pam_systemd.so
+-session   optional pam_turnstile.so
+-session   optional pam_elogind.so
