From de9fbebfde844a653fe2be382339a6db4aaced32 Mon Sep 17 00:00:00 2001
From: Jens Reidel <adrian@travitia.xyz>
Date: Wed, 16 Apr 2025 14:45:53 +0200
Subject: [PATCH] Remove policies removed from CMake

Signed-off-by: Jens Reidel <adrian@travitia.xyz>
---
 CMakeLists.txt            | 4 ----
 lib/wiiuse/CMakeLists.txt | 6 ------
 2 files changed, 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98dd5dc..8fc9362 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,10 +6,6 @@ set(PROJECT_VERSION "1.4")
 
 add_definitions( -DSUPERTUXKART_VERSION="${PROJECT_VERSION}" )
 
-if(NOT (CMAKE_MAJOR_VERSION VERSION_LESS 3))
-  cmake_policy(SET CMP0043 OLD)
-endif()
-
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake")
 include(CMakeDependentOption)
 
diff --git a/lib/wiiuse/CMakeLists.txt b/lib/wiiuse/CMakeLists.txt
index e56a750..7ceaa38 100644
--- a/lib/wiiuse/CMakeLists.txt
+++ b/lib/wiiuse/CMakeLists.txt
@@ -4,12 +4,6 @@
 
 cmake_minimum_required(VERSION 2.8.0)
 
-# Added for STK, silences CMake warning
-# See https://github.com/supertuxkart/stk-code/commit/b0ff15873ee7fa8901672672b47def9039a5534b#diff-a3e272598233e89e4b577a434cc2a89d
-if(NOT (CMAKE_MAJOR_VERSION VERSION_LESS 3))
-  cmake_policy(SET CMP0048 OLD)
-endif()
-
 # Set package properties
 project(WiiUse)
 
-- 
2.49.0

