diff --git a/Makefile b/Makefile
index 43c4227a..9861571e 100644
--- a/Makefile
+++ b/Makefile
@@ -52,7 +52,7 @@ $(BINOUT)/haredoc: $(BINOUT)/hare
 	@mkdir -p $(BINOUT)
 	@printf 'HARE\t%s\n' "$@"
 	@env $(HARE_BUILD_ENV) \
-		$(BINOUT)/hare build $(HARE_DEFINES) -o $(BINOUT)/haredoc ./cmd/haredoc
+		$(HARE) build -a$(ARCH) $(HARE_DEFINES) -o $(BINOUT)/haredoc ./cmd/haredoc
 
 docs/html: $(BINOUT)/haredoc
 	mkdir -p docs/html
