diff --git a/Cargo.toml b/Cargo.toml
index 79cbb8c..fbacc21 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,7 +48,7 @@ gix = { version = "0.69.1", default-features = false, features = [
     "index",
     "status",
 ] }
-gix-features = { version = "0.39.0", features = ["zlib-ng"] }
+gix-features = { version = "0.39.0", features = ["zlib-stock"] }
 globset = "0.4.15"
 human-panic = "2.0.2"
 image.workspace = true
