commit ed58f32d0f0d1ae852a508cc23300400d6fa6e65
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Jun 17 21:55:52 2023 +0200

    fix build with clang 16

diff --git a/XMPFiles/source/PluginHandler/XMPAtoms.h b/XMPFiles/source/PluginHandler/XMPAtoms.h
index 4807103..23780a2 100644
--- a/XMPFiles/source/PluginHandler/XMPAtoms.h
+++ b/XMPFiles/source/PluginHandler/XMPAtoms.h
@@ -83,7 +83,7 @@ enum
 
 #define XMPAtomNull emptyStr_K
 
-struct StringCompare : std::binary_function<const std::string &, const std::string &, bool>
+struct StringCompare
 {
 	bool operator() (const std::string & a, const std::string & b) const
 	{
diff --git a/XMPFilesPlugins/api/source/PluginRegistry.h b/XMPFilesPlugins/api/source/PluginRegistry.h
index 467fea3..b141987 100644
--- a/XMPFilesPlugins/api/source/PluginRegistry.h
+++ b/XMPFilesPlugins/api/source/PluginRegistry.h
@@ -68,7 +68,7 @@ public:
 	
 private:
 
-	struct StringCompare : std::binary_function< const std::string &, const std::string &, bool >
+	struct StringCompare
 	{
 		bool operator()( const std::string & a, const std::string & b ) const
 		{
