diff --git a/autotests/directoryIteratorTest.cpp b/autotests/directoryIteratorTest.cpp
index 3a8f61d..a9aae1e 100644
--- a/autotests/directoryIteratorTest.cpp
+++ b/autotests/directoryIteratorTest.cpp
@@ -1,6 +1,10 @@
 // SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
 // SPDX-FileCopyrightText: 2022 Harald Sitter <sitter@kde.org>
 
+#ifndef S_BLKSIZE
+#define S_BLKSIZE 512
+#endif
+
 #include <QDebug>
 #include <QProcess>
 #include <QTest>
diff --git a/src/posixWalker.cpp b/src/posixWalker.cpp
index 27b7f1c..b0964c1 100644
--- a/src/posixWalker.cpp
+++ b/src/posixWalker.cpp
@@ -1,6 +1,10 @@
 // SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
 // SPDX-FileCopyrightText: 2022 Harald Sitter <sitter@kde.org>
 
+#ifndef S_BLKSIZE
+#define S_BLKSIZE 512
+#endif
+
 #include "posixWalker.h"
 #include <QDebug>
 
