--- a/meson.build
+++ b/meson.build
@@ -307,7 +307,7 @@
 if time_epoch == ''
     time_epoch = run_command(date, '+%s', check: true).stdout().strip()
 endif
-generate_date = run_command(date, '--utc', '--date=@' + time_epoch, '+%Y-%m-%d', check: true).stdout().strip()
+generate_date = run_command(date, '-z', '--date=@' + time_epoch, '+%Y-%m-%d', check: true).stdout().strip()
 
 ## Manpages.
 docconf = configuration_data()
@@ -983,8 +983,8 @@
     'lxc user network config:	@0@'.format(lxc_user_network_conf),
     'lxc user network database:	@0@'.format(lxc_user_network_db)]
 
-alt_time_epoch = run_command('date', '-Is', '-u', '-d',
-                             '@@0@'.format(time_epoch), check: true).stdout().strip()
+alt_time_epoch = run_command('date', '-u', '-r ' + time_epoch,
+                             '+%Y-%m-%dT%H:%M:%S+00:00', check: true).stdout().strip()
 status += [
     'time epoch:			@0@ (@1@)'.format(time_epoch, alt_time_epoch)]
 
