--- a/extern/lzma/CpuArch.c
+++ b/extern/lzma/CpuArch.c
@@ -766,8 +766,6 @@
 
 #ifdef USE_HWCAP
 
-#include <asm/hwcap.h>
-
   #define MY_HWCAP_CHECK_FUNC_2(name1, name2) \
   BoolInt CPU_IsSupported_ ## name1() { return (getauxval(AT_HWCAP)  & (HWCAP_  ## name2)) ? 1 : 0; }
 
--- a/extern/lzma/Threads.c
+++ b/extern/lzma/Threads.c
@@ -265,7 +265,6 @@
       */
 
       // ret2 =
-      pthread_attr_setaffinity_np(&attr, sizeof(*cpuSet), cpuSet);
       // if (ret2) ret = ret2;
       #endif
     }
@@ -275,14 +274,12 @@
     if (!ret)
     {
       p->_created = 1;
-      /*
       if (cpuSet)
       {
         // ret2 =
         pthread_setaffinity_np(p->_tid, sizeof(*cpuSet), cpuSet);
         // if (ret2) ret = ret2;
       }
-      */
     }
   }
   // ret2 =
--- a/extern/lzma/Threads.h
+++ b/extern/lzma/Threads.h
@@ -4,6 +4,10 @@
 #ifndef ZIP7_INC_THREADS_H
 #define ZIP7_INC_THREADS_H
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #ifdef _WIN32
 #include "7zWindows.h"
 
@@ -20,6 +20,7 @@
 #endif
 
 #include <pthread.h>
+#include <sched.h>
 
 #endif
 
