--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -1877,7 +1879,7 @@
       # Blender's requirement of libraries mean the installation path must be used.
       install(
         CODE "\
-set(BLENDER_BIN \"${CMAKE_INSTALL_PREFIX}/${BLENDER_BIN}\")\n\
+set(BLENDER_BIN \"\$ENV{DESTDIR}/${CMAKE_INSTALL_PREFIX}/${BLENDER_BIN}\")\n\
 set(PYTHON_EXECUTABLE \"${PYTHON_EXECUTABLE}\")\n\
 set(MANPAGE_GEN \"${CMAKE_SOURCE_DIR}/doc/manpage/blender.1.py\")\n\
 set(MANPAGE_OUT \"${CMAKE_CURRENT_BINARY_DIR}/blender.1\")\n\
