--- a/source/blender/freestyle/intern/geometry/VecMat.h
+++ b/source/blender/freestyle/intern/geometry/VecMat.h
@@ -13,6 +13,8 @@
 #include <math.h>
 #include <vector>
 
+#include "BLI_sys_types.h"
+
 #ifdef WITH_CXX_GUARDEDALLOC
 #  include "MEM_guardedalloc.h"
 #endif
--- a/source/blender/freestyle/intern/system/Id.h
+++ b/source/blender/freestyle/intern/system/Id.h
@@ -13,6 +13,8 @@
 #  include "MEM_guardedalloc.h"
 #endif
 
+#include "BLI_sys_types.h"
+
 namespace Freestyle {
 
 /** Class used to tag any object by an id.
