Avoid calling out to git for no reason
--- a/lisp/loadup.el
+++ b/lisp/loadup.el
@@ -423,8 +423,8 @@ lost after dumping")))
                                (string-to-number
                                 (substring name (length base) exelen)))
 			     files)))
-      (setq emacs-repository-version (ignore-errors (emacs-repository-get-version))
-            emacs-repository-branch (ignore-errors (emacs-repository-get-branch)))
+      ;; (setq emacs-repository-version (ignore-errors (emacs-repository-get-version))
+      ;;       emacs-repository-branch (ignore-errors (emacs-repository-get-branch)))
       ;; A constant, so we shouldn't change it with `setq'.
       (defconst emacs-build-number
 	(if versions (1+ (apply #'max versions)) 1))))
