diff --git a/Unix/Makefile b/Unix/Makefile
index 5c45d8f..4c41798 100755
--- a/Unix/Makefile	2025-06-25
+++ b/Unix/Makefile	2025-10-11
@@ -95,14 +95,10 @@
 # Rule: test-code - Check that the counter works
 test-code:
 	t/00_C.t
-	t/02_git.t
-	t/01_opts.t
 
 # Rule: test-unix - Check that the Unix version works
 test-unix:
 	t/00_C.t -u
-	t/02_git.t -u
-	t/01_opts.t -u
 
 # Rule: test - Run tests (exclude test-unix)
 test: test-perl test-pod test-code
diff --git a/Unix/pod2man.mk b/Unix/pod2man.mk
index 025b379..91a0332 100644
--- a/Unix/pod2man.mk
+++ b/Unix/pod2man.mk
@@ -40,7 +40,7 @@ PACKAGE		?= package
 # Optional variables to set
 MANSECT		?= 1
 PODCENTER	?= User Commands
-PODDATE		?= $$(date --utc --date="@$${SOURCE_DATE_EPOCH:-$$(date +%s)}" "+%Y-%m-%d")
+PODDATE		?= $$(date -u -j -r "${SOURCE_DATE_EPOCH}" +'%Y-%m-%d')
 detected_OS = $(shell uname)
 ifeq ($(detected_OS),Darwin)
 	# macOS;  "PODDATE ?=" doesn't work
