diff --git a/panel-plugin/meson.build b/panel-plugin/meson.build
index 01e78ea..1b5dcfe 100644
--- a/panel-plugin/meson.build
+++ b/panel-plugin/meson.build
@@ -17,7 +17,6 @@ shared_library(
     'notification-plugin-dialogs.c',
     'notification-plugin-log.c',
   ] + panel_plugin_generated_sources,
-  soversion: '',
   gnu_symbol_visibility: 'hidden',
   include_directories: [
     include_directories('..'),
