From 0617d34f6f48b860e60864adcd024da6997ff00d Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Tue, 14 Dec 2021 02:07:51 +0100
Subject: [PATCH] disable chacha altivec

this fails tests
---
 fat-ppc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fat-ppc.c b/fat-ppc.c
index 3adbb88..a983245 100644
--- a/fat-ppc.c
+++ b/fat-ppc.c
@@ -212,7 +212,7 @@ fat_init (void)
       _nettle_gcm_hash_vec = _nettle_gcm_hash_c;
 #endif /* GCM_TABLE_BITS == 8 */
     }
-  if (features.have_altivec)
+  if (0)
     {
       if (verbose)
 	fprintf (stderr, "libnettle: enabling altivec code.\n");
-- 
2.34.1

