From 92bb73d2d70e2ef1e82abe5481194ec47b1fa640 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 23 Nov 2024 01:48:25 +0100
Subject: [PATCH] test: fix compat with libarchive/bsd tar

---
 test/solver.sh | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/test/solver.sh b/test/solver.sh
index 1ccbb43..573fcb0 100755
--- a/test/solver.sh
+++ b/test/solver.sh
@@ -1,10 +1,18 @@
 #!/bin/sh
 
+case "$(tar --version | head -n1)" in
+    *GNU*) GNUTAR=1 ;;
+esac
+
 update_repo() {
 	local repo="$1"
 	if [ ! -f "$repo.adb" -o "$repo" -nt "$repo.adb" ]; then
 		local tmpname="$repo.new.$$"
-		tar czf "$tmpname" -P --transform "flags=r;s|$repo|APKINDEX|" "$repo"
+		if [ -n "$GNUTAR" ]; then
+			tar czf "$tmpname" -P --transform "flags=r;s|$repo|APKINDEX|" "$repo"
+		else
+			tar czf "$tmpname" -P -s "|$repo|APKINDEX|r" "$repo"
+		fi
 		mv "$tmpname" "$repo.adb"
 	fi
 }
-- 
2.47.0

