From 2446e2be3f9514117da594fc474934d505ae9476 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 23 Nov 2024 14:10:35 +0100
Subject: [PATCH] disable broken mkpkg for hardlinks for now

---
 src/app_mkpkg.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/app_mkpkg.c b/src/app_mkpkg.c
index c5ecf68..e1f8394 100644
--- a/src/app_mkpkg.c
+++ b/src/app_mkpkg.c
@@ -278,6 +278,7 @@ static int mkpkg_process_dirent(void *pctx, int dirfd, const char *entry)
 
 	switch (fi.mode & S_IFMT) {
 	case S_IFREG:
+#if 0
 		key = (struct mkpkg_hardlink_key) {
 			.device = fi.data_device,
 			.inode = fi.data_inode,
@@ -299,6 +300,7 @@ static int mkpkg_process_dirent(void *pctx, int dirfd, const char *entry)
 			memcpy(link->name, entry, len);
 			apk_hash_insert(&ctx->link_by_inode, link);
 		}
+#endif
 		ctx->installed_size += (fi.size + BLOCK_SIZE - 1) & ~(BLOCK_SIZE-1);
 		break;
 	case S_IFBLK:
-- 
2.47.0

