commit abf9cdcba7666803a2a5d6ea6ac6b9aa68c9a3c2
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Thu Mar 31 01:58:27 2022 +0200

    package: respect version fuzziness when creating dep from adb

diff --git a/src/package.c b/src/package.c
index a1c2c66..2a6b9fd 100644
--- a/src/package.c
+++ b/src/package.c
@@ -461,6 +461,7 @@ void apk_dep_from_adb(struct apk_dependency *dep, struct apk_database *db, struc
 	*dep = (struct apk_dependency) {
 		.name = apk_db_get_name(db, adb_ro_blob(d, ADBI_DEP_NAME)),
 		.version = apk_atomize_dup(&db->atoms, adb_ro_blob(d, ADBI_DEP_VERSION)),
+		.fuzzy = !!(mask & APK_VERSION_FUZZY),
 		.conflict = !!(mask & APK_VERSION_CONFLICT),
 		.result_mask = (mask & ~APK_VERSION_CONFLICT) ?: APK_VERSION_EQUAL,
 	};
