bugfix for https://gitlab.alpinelinux.org/alpine/apk-tools/-/issues/11020#note_433198 for older commit.
--
diff --git a/src/package.c b/src/package.c
index 68d2e60..3f61878 100644
--- a/src/package.c
+++ b/src/package.c
@@ -545,7 +545,7 @@
 	uid = adb_ro_blob(pkginfo, ADBI_PI_UNIQUE_ID);
 	if (uid.len >= APK_CHECKSUM_SHA1) {
 		pkg->csum.type = APK_CHECKSUM_SHA1;
-		memcpy(pkg->csum.data, uid.ptr, uid.len);
+		memcpy(pkg->csum.data, uid.ptr, APK_CHECKSUM_SHA1);
 	}
 
 	pkg->name = apk_db_get_name(db, adb_ro_blob(pkginfo, ADBI_PI_NAME));
