diff --git a/shumate/vector/shumate-vector-render-scope.c b/shumate/vector/shumate-vector-render-scope.c
index ecacb66..65eec81 100644
--- a/shumate/vector/shumate-vector-render-scope.c
+++ b/shumate/vector/shumate-vector-render-scope.c
@@ -471,7 +471,7 @@ shumate_vector_render_scope_index_layer (ShumateVectorRenderScope *self)
 
   for (int key = 0; key < layer->n_keys; key ++)
     {
-      int val;
+      gpointer valp;
       const char *field_name;
       GHashTableIter field_iter;
       ShumateVectorIndexBitset *bitset;
@@ -483,12 +483,12 @@ shumate_vector_render_scope_index_layer (ShumateVectorRenderScope *self)
       field_name = layer->keys[key];
 
       g_hash_table_iter_init (&field_iter, field->indexes);
-      while (g_hash_table_iter_next (&field_iter, (gpointer *)&val, (gpointer *)&bitset))
+      while (g_hash_table_iter_next (&field_iter, &valp, (gpointer *)&bitset))
         {
           g_auto(ShumateVectorValue) value = SHUMATE_VECTOR_VALUE_INIT;
           VectorTile__Tile__Value *v;
 
-          v = layer->values[val];
+          v = layer->values[GPOINTER_TO_INT(valp)];
           convert_vector_value (v, &value);
 
           shumate_vector_index_add_bitset (self->index, self->source_layer_idx, field_name, &value, bitset);
