diff --git a/shumate/shumate-vector-sprite-sheet.c b/shumate/shumate-vector-sprite-sheet.c
index e60d6fd..388e716 100644
--- a/shumate/shumate-vector-sprite-sheet.c
+++ b/shumate/shumate-vector-sprite-sheet.c
@@ -280,7 +280,7 @@ search_sprites (GPtrArray *sprites,
                 gboolean   higher,
                 gboolean   lower)
 {
-  int best_scale = higher ? G_MAXDOUBLE : G_MINDOUBLE;
+  double best_scale = higher ? G_MAXDOUBLE : G_MINDOUBLE;
   ShumateVectorSprite *best_sprite = NULL;
 
   for (int i = 0; i < sprites->len; i++)
