From eb416d52ac7ec898089f3040e8028814dec8f7ea Mon Sep 17 00:00:00 2001
From: ItsHarper <git-f4cbe@harperandrews.org>
Date: Sun, 10 May 2026 13:17:06 -0400
Subject: unbundle sqlite

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index fa2a8b128a..260244988d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -402,7 +402,7 @@ jupyter-protocol = "1.1.0"
 jupyter-websocket-client = "1.0.0"
 kdl = "6.5.0"
 libc = "0.2"
-libsqlite3-sys = { version = "0.37", features = ["bundled"] }
+libsqlite3-sys = { version = "0.37" }
 linkify = "0.11"
 log = { version = "0.4.16", features = ["kv_unstable_serde", "serde"] }
 lsp-types = { git = "https://github.com/zed-industries/lsp-types", rev = "b71ab4eeb27d9758be8092020a46fe33fbca4e33" }
