--- a/pyproject.toml
+++ b/pyproject.toml
@@ -39,7 +39,6 @@
   "pygls>=1.1.0",
   "lsprotocol>=2023.0.0",
   "ruff>=0.0.274",
-  "typing_extensions",
 ]
 
 [project.optional-dependencies]
--- a/tests/test_server.py
+++ b/tests/test_server.py
@@ -8,7 +8,7 @@
 from threading import Event
 
 from packaging.version import Version
-from typing_extensions import Self
+from typing import Self
 
 from tests.client import defaults, session, utils
 
--- a/ruff_lsp/settings.py
+++ b/ruff_lsp/settings.py
@@ -2,7 +2,7 @@
 
 import enum
 
-from typing_extensions import Literal, TypedDict
+from typing import Literal, TypedDict
 
 
 @enum.unique
--- a/ruff_lsp/server.py
+++ b/ruff_lsp/server.py
@@ -76,7 +76,7 @@
 from packaging.specifiers import SpecifierSet, Version
 from pygls import server, uris, workspace
 from pygls.workspace.position_codec import PositionCodec
-from typing_extensions import Literal, Self, TypedDict, assert_never
+from typing import Literal, Self, TypedDict, assert_never
 
 from ruff_lsp import __version__, utils
 from ruff_lsp.settings import (
