matches the autotools scripts

diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-01-03 01:03:10.000000000 +0100
+++ b/CMakeLists.txt	2024-08-27 17:28:22.363327734 +0200
@@ -358,7 +358,7 @@
 # to resolve generator expression in OUTPUT_NAME
 file(
   GENERATE
-  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lept_$<CONFIG>.pc
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lept.pc
   INPUT ${CMAKE_CURRENT_BINARY_DIR}/lept.pc.in)
 
 configure_file(
@@ -368,7 +368,7 @@
   ${CMAKE_CURRENT_SOURCE_DIR}/cmake/templates/LeptonicaConfig.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/LeptonicaConfig.cmake @ONLY)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lept_$<CONFIG>.pc
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lept.pc
         DESTINATION lib/pkgconfig)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/LeptonicaConfig.cmake
               ${CMAKE_CURRENT_BINARY_DIR}/LeptonicaConfig-version.cmake
