--- a/Makefile
+++ b/Makefile
@@ -117,8 +117,10 @@
 	install -d $(DESTDIR)$(MAN1DIR)/;
 	install sslscan.1 $(DESTDIR)$(MAN1DIR)/sslscan.1;
 else
-	install -D sslscan $(DESTDIR)$(BINDIR)/sslscan;
-	install -D sslscan.1 $(DESTDIR)$(MAN1DIR)/sslscan.1;
+	install -d $(DESTDIR)$(BINDIR);
+	install -d $(DESTDIR)$(MAN1DIR);
+	install sslscan $(DESTDIR)$(BINDIR)/sslscan;
+	install sslscan.1 $(DESTDIR)$(MAN1DIR)/sslscan.1;
 endif
 
 uninstall:
