commit 578d9c84e93786e53f75eceb723d47c5b869e520
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Thu Jun 1 23:02:13 2023 +0200

    drop groff requirement

diff --git a/src/Makefile.am b/src/Makefile.am
index 48fce70..53545b4 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -14,16 +14,13 @@
 AM_CPPFLAGS = -I$(top_srcdir) $(ISYSTEM)$(top_builddir)/libgnu $(ISYSTEM)$(top_srcdir)/libgnu -I$(top_srcdir)/lib $(GLIB_CFLAGS) $(WARN_CFLAGS) -DG_LOG_DOMAIN='"libenchant"'
 
 dist_man_MANS = enchant-@ENCHANT_MAJOR_VERSION@.1 enchant-lsmod-@ENCHANT_MAJOR_VERSION@.1
-nodist_doc_DATA = enchant-@ENCHANT_MAJOR_VERSION@.html enchant-lsmod-@ENCHANT_MAJOR_VERSION@.html
+nodist_doc_DATA =
 
 edit = sed \
 	-e 's|@PKGDATADIR[@]|$(pkgdatadir)|g' \
 	-e 's|@SYSCONFDIR[@]|$(sysconfdir)|g'
 
 DISTCLEANFILES = $(dist_man_MANS) $(nodist_doc_DATA)
-
-.1.html:
-	groff -mandoc -Thtml $< > $@
 
 enchant-@ENCHANT_MAJOR_VERSION@.1: $(builddir)/enchant.1 Makefile.am $(top_builddir)/config.status
 	cp $(abs_builddir)/enchant.1 $@
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -47,7 +47,7 @@
 libenchant_data_DATA = enchant.ordering
 
 dist_man_MANS = enchant.5
-nodist_doc_DATA = enchant.html
+nodist_doc_DATA =
 
 edit = sed \
 	-e 's|@PKGDATADIR[@]|$(pkgdatadir)|g' \
@@ -59,8 +59,5 @@
 	rm -f $@ $@.tmp
 	$(edit) $(abs_builddir)/enchant.5.in >$@.tmp
 	mv $@.tmp $@
-
-enchant.html: enchant.5
-	groff -mandoc -Thtml $< > $@
 
 EXTRA_DIST = $(libenchant_data_DATA) $(VAPIS)
