commit 578d9c84e93786e53f75eceb723d47c5b869e520
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Thu Jun 1 23:02:13 2023 +0200

    drop groff requirement

diff --git a/src/Makefile.am b/src/Makefile.am
index 48fce70..53545b4 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -24,19 +24,13 @@ noinst_HEADERS = debug.h
 pkgdata_DATA = enchant.ordering
 
 dist_man_MANS = enchant-@ENCHANT_MAJOR_VERSION@.1 enchant-lsmod-@ENCHANT_MAJOR_VERSION@.1 enchant.5
-nodist_doc_DATA = enchant-@ENCHANT_MAJOR_VERSION@.html enchant-lsmod-@ENCHANT_MAJOR_VERSION@.html enchant.html
+nodist_doc_DATA =
 
 edit = sed \
 	-e 's|DATADIR|$(datadir)|g'
 
 DISTCLEANFILES = $(dist_man_MANS) $(nodist_doc_DATA)
 
-.1.html:
-	groff -mandoc -Thtml $< > $@
-
-enchant.html: enchant.5
-	groff -mandoc -Thtml $< > $@
-
 enchant-@ENCHANT_MAJOR_VERSION@.1: $(builddir)/enchant.1 Makefile.am $(top_builddir)/config.status
 	rm -f $@ $@.tmp
 	$(edit) $(abs_builddir)/enchant.1 >$@.tmp
