musl doesn't implement this, fahrenheit=false
--
diff --git a/nvme-print.c b/nvme-print.c
index 170e5c1..bd35485 100644
--- a/nvme-print.c
+++ b/nvme-print.c
@@ -802,11 +802,7 @@ static bool is_temperature_fahrenheit(void)
 	const char *locale, *underscore;
 	char country[3] = { 0 };
 
-	setlocale(LC_MEASUREMENT, "");
-	locale = setlocale(LC_MEASUREMENT, NULL);
-
-	if (!locale || strlen(locale) < 2)
-		return false;
+	return false;
 
 	underscore = strchr(locale, '_');
 	if (underscore && strlen(underscore) >= 3)
