From 45a6dea45579db2b3a31db4fb186f2fa6e2e84e5 Mon Sep 17 00:00:00 2001
From: Gnarwhal <git.aspect893@passmail.net>
Date: Tue, 4 Feb 2025 04:50:51 +0000
Subject: [PATCH 1/2] Zero index sh args

---
 t/geturl_response.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/geturl_response.pl b/t/geturl_response.pl
index beb1a92..0e77d16 100644
--- a/t/geturl_response.pl
+++ b/t/geturl_response.pl
@@ -5,7 +5,7 @@ eval { require 'ddclient'; } or BAIL_OUT($@);
 # Fake curl.  Use the printf utility, which can process escapes.  This allows Perl to drive the fake
 # curl with plain ASCII and get arbitrary bytes back, avoiding problems caused by any encoding that
 # might be done by Perl (e.g., "use open ':encoding(UTF-8)';").
-my @fakecurl = ('sh', '-c', 'printf %b "$1"', '--');
+my @fakecurl = ('sh', '-c', 'printf %b "$0"');
 
 my @test_cases = (
     {
-- 
2.48.1

