diff --color -ruN ddclient-3.11.2/t/get_ip_from_if.pl ddclient-3.11.2-patched/t/get_ip_from_if.pl
--- ddclient-3.11.2/t/get_ip_from_if.pl	2023-11-23 12:06:21.000000000 +0000
+++ ddclient-3.11.2-patched/t/get_ip_from_if.pl	2024-07-10 03:20:56.907890172 +0000
@@ -39,25 +39,4 @@
     }
 };
 
-subtest "Get default interface and IP for test system" => sub {
-    my $interface = ddclient::get_default_interface(4);
-    if ($interface) {
-        isnt($interface, "lo", "Check for loopback 'lo'");
-        isnt($interface, "lo0", "Check for loopback 'lo0'");
-        my $ip1 = ddclient::get_ip_from_interface("default", 4);
-        my $ip2 = ddclient::get_ip_from_interface($interface, 4);
-        is($ip1, $ip2, "Check IPv4 from default interface");
-        ok(ddclient::is_ipv4($ip1), "Valid IPv4 from get_ip_from_interface($interface)");
-    }
-    $interface = ddclient::get_default_interface(6);
-    if ($interface) {
-        isnt($interface, "lo", "Check for loopback 'lo'");
-        isnt($interface, "lo0", "Check for loopback 'lo0'");
-        my $ip1 = ddclient::get_ip_from_interface("default", 6);
-        my $ip2 = ddclient::get_ip_from_interface($interface, 6);
-        is($ip1, $ip2, "Check IPv6 from default interface");
-        ok(ddclient::is_ipv6($ip1), "Valid IPv6 from get_ip_from_interface($interface)");
-    }
-};
-
 done_testing();
