--- a/meson.build
+++ b/meson.build
@@ -121,7 +121,7 @@ gi_docgen_dep = dependency('gi-docgen', version: '>= 2021.7',
                            native: true,
                            required: get_option('gtk_doc'))
 gjs_dep = dependency('gjs-1.0', version: gjs_req, required: get_option('gjs'))
-mozjs_dep = dependency('mozjs-128', required: get_option('gjs'))
+mozjs_dep = dependency('mozjs-140', required: get_option('gjs'))
 
 # Check if user specified a Python installation path, if so, use it
 python_path = get_option('python3_path')
