From 5606f315416d019576fc849aad92ea7fcbd7bcae Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Sat, 4 Jan 2025 01:40:19 +0100
Subject: [PATCH] media: remove use of custom AVFMT_FLAG_NOH264PARSE

---
 media/filters/ffmpeg_glue.cc | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/media/filters/ffmpeg_glue.cc b/media/filters/ffmpeg_glue.cc
index 26bb9e8b92614..05143a0130b22 100644
--- a/media/filters/ffmpeg_glue.cc
+++ b/media/filters/ffmpeg_glue.cc
@@ -109,10 +109,6 @@ FFmpegGlue::FFmpegGlue(FFmpegURLProtocol* protocol) {
   // Enable fast, but inaccurate seeks for MP3.
   format_context_->flags |= AVFMT_FLAG_FAST_SEEK;
 
-  // We don't allow H.264 parsing during demuxing since we have our own parser
-  // and the ffmpeg one increases memory usage unnecessarily.
-  format_context_->flags |= AVFMT_FLAG_NOH264PARSE;
-
   // Ensures format parsing errors will bail out. From an audit on 11/2017, all
   // instances were real failures. Solves bugs like http://crbug.com/710791.
   format_context_->error_recognition |= AV_EF_EXPLODE;
