we don't have qt5
--
--- a/ui/qt/BUILD.gn
+++ b/ui/qt/BUILD.gn
@@ -102,12 +102,6 @@
     }
   }
 }
-qt_shim("qt5_shim") {
-  qt_version = "5"
-  if (!use_sysroot) {
-    moc_qt_path = "$moc_qt5_path"
-  }
-}
 if (use_qt6) {
   qt_shim("qt6_shim") {
     qt_version = "6"
@@ -123,7 +117,6 @@
   defines = [ "IS_QT_IMPL" ]
 
   # qt_shim is in data_deps since we want to load it manually.
-  data_deps = [ ":qt5_shim" ]
   if (use_qt6) {
-    data_deps += [ ":qt6_shim" ]
+    data_deps = [ ":qt6_shim" ]
   }
