--- a/BUILD.gn
+++ b/BUILD.gn
@@ -111,13 +111,6 @@
       "//skia:skia_unittests",
       "//sql:sql_unittests",
       "//testing/libfuzzer/tests",
-      "//third_party/angle/src/tests:angle_end2end_tests",
-      "//third_party/angle/src/tests:angle_unittests",
-      "//third_party/angle/src/tests:angle_white_box_tests",
-      "//third_party/flatbuffers:flatbuffers_unittests",
-      "//third_party/highway:highway_tests",
-      "//third_party/liburlpattern:liburlpattern_unittests",
-      "//third_party/rapidhash:rapidhash_fuzztests",
       "//tools/binary_size:binary_size_trybot_py",
       "//tools/metrics:metrics_metadata",
       "//ui/base:ui_base_unittests",
