--- a/configure.ac
+++ b/configure.ac
@@ -1226,11 +1226,6 @@
 		]])], [
 			enable_sandbox=found
 
-			AS_CASE(["$CC $CFLAGS"], [*-fsanitize=*],
-				[AC_MSG_ERROR([
-    CC or CFLAGS contain '-fsanitize=' which is incompatible with the Landlock
-    sandboxing. Use --disable-sandbox when using '-fsanitize'.])])
-
 			AC_DEFINE([HAVE_LINUX_LANDLOCK], [1],
 				[Define to 1 if Linux Landlock is supported.
 				See configure.ac for details.])
