--- a/libgit2-glib/ggit-types.h
+++ b/libgit2-glib/ggit-types.h
@@ -989,8 +989,7 @@ typedef enum
 
 typedef enum
 {
-	GGIT_CHECKOUT_NONE                    = 0,
-	GGIT_CHECKOUT_SAFE                    = (1u << 0),
+	GGIT_CHECKOUT_SAFE                    = 0,
 	GGIT_CHECKOUT_FORCE                   = (1u << 1),
 	GGIT_CHECKOUT_RECREATE_MISSING        = (1u << 2),
 	GGIT_CHECKOUT_ALLOW_CONFLICTS         = (1u << 4),
@@ -1010,7 +1009,8 @@ typedef enum
 	GGIT_CHECKOUT_DONT_REMOVE_EXISTING    = (1u << 22),
 	GGIT_CHECKOUT_DONT_WRITE_INDEX        = (1u << 23),
 	GGIT_CHECKOUT_UPDATE_SUBMODULES       = (1u << 16),
-	GGIT_CHECKOUT_UPDATE_SUBMODULES_IF_CHANGED = (1u << 17)
+	GGIT_CHECKOUT_UPDATE_SUBMODULES_IF_CHANGED = (1u << 17),
+	GGIT_CHECKOUT_NONE = (1u << 30),
 } GgitCheckoutStrategy;
 
 typedef enum {
