cbuild handles enabling cet on x86_64 on its own anyway
--- a/cmake/compilers/Clang.cmake	2025-07-06 09:53:32.436997198 +0100
+++ b/cmake/compilers/Clang.cmake	2025-07-06 10:04:11.749570151 +0100
@@ -88,12 +88,6 @@
         check_cxx_source_compiles("int main(int, char*[]) { return 0; }" CF_PROTECTION_FULL_SUPPORTED)
         unset(CMAKE_TRY_COMPILE_TARGET_TYPE)
         unset(CMAKE_REQUIRED_FLAGS)
-
-        if (CF_PROTECTION_FULL_SUPPORTED)
-            set(TBB_LIB_COMPILE_FLAGS ${TBB_LIB_COMPILE_FLAGS} -fcf-protection=full)
-        else()
-            message(WARNING "Compiler does not support -fcf-protection=full.")
-        endif()
     endif()
 endif()
 
