commit 42d255c920662fd6562023569814eef79995a930
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Apr 2 20:13:56 2022 +0200

    install portability

diff --git a/main/Makefile b/main/Makefile
index 1da18a8..93a930b 100644
--- a/main/Makefile
+++ b/main/Makefile
@@ -152,8 +152,8 @@ install:
 	done
 	$(INSTALL) -m 755 $(BINDIR)/flite_time $(DESTDIR)$(INSTALLBINDIR)
 #       The libraries: static and shared (if built)
-	cp -pd $(flite_LIBS_deps) $(DESTDIR)$(INSTALLLIBDIR)
+	cp -a $(flite_LIBS_deps) $(DESTDIR)$(INSTALLLIBDIR)
 ifdef SHFLAGS
-	cp -pd $(SHAREDLIBS) $(VERSIONSHAREDLIBS) $(DESTDIR)$(INSTALLLIBDIR)
+	cp -a $(SHAREDLIBS) $(VERSIONSHAREDLIBS) $(DESTDIR)$(INSTALLLIBDIR)
 endif
 
