--- a/tests/tst_plugin.pro
+++ b/tests/tst_plugin.pro
@@ -28,6 +28,6 @@ DATA_PATH = $${TOP_SRC_DIR}/tests/data/
 DEFINES += \
     TEST_DATA_DIR=\\\"$$DATA_PATH\\\"
 
-check.commands = "LD_LIBRARY_PATH=mock:${LD_LIBRARY_PATH} xvfb-run -a dbus-test-runner -m 120 -t ./$${TARGET}"
+check.commands = "LD_LIBRARY_PATH=mock:${LD_LIBRARY_PATH} wlheadless-run -- dbus-test-runner -m 120 -t ./$${TARGET}"
 check.depends = $${TARGET}
 QMAKE_EXTRA_TARGETS += check
