diff --git a/cmake/today.cmake b/cmake/today.cmake
index 659e5ec..2c8133d 100644
--- a/cmake/today.cmake
+++ b/cmake/today.cmake
@@ -4,7 +4,7 @@
 #
 MACRO (TODAY RESULT)
     if (DEFINED ENV{SOURCE_DATE_EPOCH} AND NOT WIN32)
-        EXECUTE_PROCESS(COMMAND "date" "-u" "-d" "@$ENV{SOURCE_DATE_EPOCH}" "+%Y-%m-%d"
+        EXECUTE_PROCESS(COMMAND "date" "-u" "-r" "$ENV{SOURCE_DATE_EPOCH}" "+%Y-%m-%d"
                         OUTPUT_VARIABLE ${RESULT} OUTPUT_STRIP_TRAILING_WHITESPACE)
     elseif(CMAKE_VERSION VERSION_LESS "2.8.11")
 	if (WIN32)
