nope

--- a/cmake/CommonConfigOptions.cmake
+++ b/cmake/CommonConfigOptions.cmake
@@ -12,24 +12,6 @@ if(WITH_VERBOSE_WINPR_ASSERT)
   add_compile_definitions(WITH_VERBOSE_WINPR_ASSERT)
 endif()
 
-# known issue on android, thus disabled until we support newer CMake
-# https://github.com/android/ndk/issues/1444
-if(NOT ANDROID)
-  if(POLICY CMP0069)
-    cmake_policy(SET CMP0069 NEW)
-  endif()
-  if(POLICY CMP0138)
-    cmake_policy(SET CMP0138 NEW)
-  endif()
-  include(CheckIPOSupported)
-  check_ipo_supported(RESULT supported OUTPUT error)
-  if(NOT supported)
-    message(WARNING "LTO not supported, got ${error}")
-  endif()
-
-  option(CMAKE_INTERPROCEDURAL_OPTIMIZATION "build with link time optimization" ${supported})
-endif()
-
 set(SUPPORTED_BUILD_TYPES "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 
 # Default to release build type
