diff -ruN a/configure b/configure
--- a/configure	2024-07-21 21:03:39.000000000 +0200
+++ b/configure	2024-08-16 21:56:12.430992265 +0200
@@ -1132,10 +1132,10 @@
 	rm -f Makefile $DISTCLEAN_OTHER
 
 install-bin:
-	./install-sh -c -s \$(TARGET) \$(PREFIX)/bin/\$(TARGET)
+	./install-sh -c -s \$(TARGET) \$(DESTDIR)\$(PREFIX)/bin/\$(TARGET)
 install: install-bin
 uninstall:
-	rm \$(PREFIX)/bin/\$(TARGET)
+	rm \$(DESTDIR)\$(PREFIX)/bin/\$(TARGET)
 _EOF
 
 if [ "$SYS" = "MinGW" ]
