--- a/Makefile.in
+++ b/Makefile.in
@@ -677,7 +677,7 @@ SUBST_INSTALL_HEADER = \
 	  $(SED) -e "s/\<EXPORT\>/extern/" < "$$1" > "include/$$file.t" || exit 1; \
 	  $(MKDIR_P) "$$2" || exit 1; \
 	  $(INSTALL_HEADER) "include/$$file.t" "$$2/$$file" || exit 1; \
-	  $(RM) "include/$$file.t"; \
+	  rm -f "include/$$file.t"; \
 	}; \
 	subst_install_header
 
@@ -1866,11 +1866,11 @@ update-po:
 	$(MAKE) -C po $@
 
 install-attributes_h: include/attributes.h
-	$(SUBST_INSTALL_HEADER) "$<" "$(DESTDIR)$(pkgincludedir)"
+	$(SUBST_INSTALL_HEADER) "$(srcdir)/include/attributes.h" "$(DESTDIR)$(pkgincludedir)"
 install-error_context_h: include/error_context.h
-	$(SUBST_INSTALL_HEADER) "$<" "$(DESTDIR)$(pkgincludedir)"
+	$(SUBST_INSTALL_HEADER) "$(srcdir)/include/error_context.h" "$(DESTDIR)$(pkgincludedir)"
 install-libattr_h: include/libattr.h
-	$(SUBST_INSTALL_HEADER) "$<" "$(DESTDIR)$(pkgincludedir)"
+	$(SUBST_INSTALL_HEADER) "$(srcdir)/include/libattr.h" "$(DESTDIR)$(pkgincludedir)"
 install-data-local: install-attributes_h install-error_context_h install-libattr_h
 uninstall-local:
 	-rm -f \
