diff --git a/Makefile b/Makefile
index 612a35f..372fc50 100644
--- a/Makefile
+++ b/Makefile
@@ -52,20 +52,20 @@ R= $V.3
 all:	$(PLAT)
 
 $(PLATS) help test clean:
-	@cd src && $(MAKE) $@ V=$(V) R=$(R)
+	@cd src && $(MAKE) $@ V=$(V) R=$(R) && cd ..
 
 install: dummy
-	cd src && $(MKDIR) $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_MAN) $(INSTALL_LMOD) $(INSTALL_CMOD)
-	cd src && $(INSTALL_EXEC) $(TO_BIN) $(INSTALL_BIN)
-	cd src && $(INSTALL_DATA) $(TO_INC) $(INSTALL_INC)
-	cd src && $(INSTALL_DATA) $(TO_LIB) $(INSTALL_LIB)
-	cd doc && $(INSTALL_DATA) $(TO_MAN) $(INSTALL_MAN)
+	cd src && $(MKDIR) $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_MAN) $(INSTALL_LMOD) $(INSTALL_CMOD) && cd ..
+	cd src && $(INSTALL_EXEC) $(TO_BIN) $(INSTALL_BIN) && cd ..
+	cd src && $(INSTALL_DATA) $(TO_INC) $(INSTALL_INC) && cd ..
+	cd src && $(INSTALL_DATA) $(TO_LIB) $(INSTALL_LIB) && cd ..
+	cd doc && $(INSTALL_DATA) $(TO_MAN) $(INSTALL_MAN) && cd ..
 
 uninstall:
-	cd src && cd $(INSTALL_BIN) && $(RM) $(TO_BIN)
-	cd src && cd $(INSTALL_INC) && $(RM) $(TO_INC)
-	cd src && cd $(INSTALL_LIB) && $(RM) $(TO_LIB)
-	cd doc && cd $(INSTALL_MAN) && $(RM) $(TO_MAN)
+	cd src && cd $(INSTALL_BIN) && $(RM) $(TO_BIN) && cd ..
+	cd src && cd $(INSTALL_INC) && $(RM) $(TO_INC) && cd ..
+	cd src && cd $(INSTALL_LIB) && $(RM) $(TO_LIB) && cd ..
+	cd doc && cd $(INSTALL_MAN) && $(RM) $(TO_MAN) && cd ..
 
 local:
 	$(MAKE) install INSTALL_TOP=../install
