diff --git a/src/Makefile b/src/Makefile
index 0e0804e..ab00468 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -32,16 +32,16 @@ CMCFLAGS=
 
 PLATS= guess aix bsd c89 freebsd generic ios linux linux-readline macosx mingw posix solaris
 
-LUA_A=	liblua.a
-LUA_SO= liblua.so
+LUA_A=	liblua5.4.a
+LUA_SO= liblua5.4.so
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o lundump.o lvm.o lzio.o
 LIB_O=	lauxlib.o lbaselib.o lcorolib.o ldblib.o liolib.o lmathlib.o loadlib.o loslib.o lstrlib.o ltablib.o lutf8lib.o linit.o
 BASE_O= $(CORE_O) $(LIB_O) $(MYOBJS)
 
-LUA_T=	lua
+LUA_T=	lua5.4
 LUA_O=	lua.o
 
-LUAC_T=	luac
+LUAC_T=	luac5.4
 LUAC_O=	luac.o
 
 ALL_O= $(BASE_O) $(LUA_O) $(LUAC_O)
