commit ebeefee7d599ff96fec660c96c61f722acea6f96
Author: q66 <q66@chimera-linux.org>
Date:   Thu Aug 31 21:18:54 2023 +0200

    use correct gimptool

diff --git a/src/Makefile b/src/Makefile
index 5433c87..2d8c7b0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -87,7 +87,7 @@ INCLUDE ?= include
 PIC = -fPIC
 EXE =
 WGET = wget --no-check-certificate --quiet -O
-PLUGIN = $(shell gimptool-2.0 --gimpplugindir)/plug-ins
+PLUGIN = $(shell gimptool-2.99 --gimpplugindir)/plug-ins
 SHARE = $(PREFIX)/share
 BASH_COMPLETIONSDIR = $(shell pkg-config --variable completionsdir bash-completion || echo $(SHARE)/bash-completion/completions)
 VERSION = $(shell grep -F 'gmic_version ' gmic.h | tail -c4 | head -c3)
