diff -ruN a/src/Makefile b/src/Makefile
--- a/src/Makefile	2025-01-27 18:30:40.000000000 +0100
+++ b/src/Makefile	2025-01-28 08:24:50.830328629 +0100
@@ -404,7 +404,7 @@
 	$(CXX) -dynamiclib -o libgmic.$(SOVERSION).dylib -install_name $(USR)/lib/libgmic.$(SOVERSION).dylib -compatibility_version $(SOVERSION) -current_version $(SVERSION) $(MANDATORY_CFLAGS) libgmic.o $(LIBS)
 else
 ifdef IS_CLANG
-	$(CXX) -shared -std=c++11 -pedantic -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
+	$(CXX) -shared -std=c++11 -pedantic -Wl,-soname,libgmic.so.$(VERSION1) -o libgmic.so.$(VERSION) libgmic.o $(LDFLAGS) $(LIBS)
 else
 	$(CXX) -shared -std=c++11 -pedantic -Wl,-soname,libgmic.so.$(VERSION1) -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
 endif
@@ -450,7 +450,7 @@
 _cli: gmic$(EXE)
 
 gmic$(EXE): gmic_cli.cpp gmic.o
-	$(CXX) -o gmic$(EXE) gmic_cli.cpp $(CFLAGS) $(LIBS)
+	$(CXX) -o gmic$(EXE) gmic_cli.cpp $(CFLAGS) $(LDFLAGS) $(LIBS)
 
 gmic.o: gmic.cpp gmic.h gmic_stdlib_community.h CImg.h
 	$(CXX) -o gmic.o -c gmic.cpp -Dgmic_core $(CFLAGS)
