diff -ruN a/src/Makefile b/src/Makefile
--- a/src/Makefile	2024-12-31 12:11:35.000000000 +0100
+++ b/src/Makefile	2025-01-06 10:58:56.171909885 +0100
@@ -401,7 +401,7 @@
 libgmic.so: libgmic.o
 	ar rcs libgmic.a libgmic.o
 ifdef IS_CLANG
-	$(CXX) -shared -std=c++11 -pedantic -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
+	$(CXX) -shared -std=c++11 -pedantic -Wl,-soname,libgmic.so.$(VERSION1) -o libgmic.so.$(VERSION) libgmic.o $(LDFLAGS) $(LIBS)
 else
 	$(CXX) -shared -std=c++11 -pedantic -Wl,-soname,libgmic.so.$(VERSION1) -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
 endif
@@ -442,7 +442,7 @@
 _cli: gmic$(EXE)
 
 gmic$(EXE): gmic_cli.cpp gmic.o
-	$(CXX) -o gmic$(EXE) gmic_cli.cpp $(CFLAGS) $(LIBS)
+	$(CXX) -o gmic$(EXE) gmic_cli.cpp $(CFLAGS) $(LDFLAGS) $(LIBS)
 
 gmic.o: gmic.cpp gmic.h gmic_stdlib_community.h CImg.h
 	$(CXX) -o gmic.o -c gmic.cpp -Dgmic_core $(CFLAGS)
