Disable some panels while we are missing dependencies.

diff --git a/panels/meson.build b/panels/meson.build
index f603db9..0af5083 100644
--- a/panels/meson.build
+++ b/panels/meson.build
@@ -19,7 +19,7 @@ panels = [
   'notifications',
   'online-accounts',
   'power',
-  'printers',
+ #'printers',
   'region',
   'removable-media',
   'search',
diff --git a/tests/meson.build b/tests/meson.build
index d4fe361..b879160 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -6,5 +6,5 @@ endif
 
 subdir('interactive-panels')
 
-subdir('printers')
+#subdir('printers')
 subdir('info')
diff --git a/panels/meson.build b/panels/meson.build
index 0af5083..6e75b3b 100644
--- a/panels/meson.build
+++ b/panels/meson.build
@@ -27,7 +27,7 @@ panels = [
   'sound',
   'universal-access',
   'usage',
-  'user-accounts',
+ #'user-accounts',
   'wwan',
 ]
 
diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index 67f3f1b..44d9b9a 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -51,7 +51,7 @@ extern GType cc_wifi_panel_get_type (void);
 extern GType cc_notifications_panel_get_type (void);
 extern GType cc_goa_panel_get_type (void);
 extern GType cc_power_panel_get_type (void);
-extern GType cc_printers_panel_get_type (void);
+//extern GType cc_printers_panel_get_type (void);
 extern GType cc_region_panel_get_type (void);
 extern GType cc_removable_media_panel_get_type (void);
 extern GType cc_search_panel_get_type (void);
@@ -61,7 +61,7 @@ extern GType cc_sound_panel_get_type (void);
 extern GType cc_bolt_panel_get_type (void);
 #endif /* BUILD_THUNDERBOLT */
 extern GType cc_ua_panel_get_type (void);
-extern GType cc_user_panel_get_type (void);
+//extern GType cc_user_panel_get_type (void);
 #ifdef BUILD_WACOM
 extern GType cc_wacom_panel_get_type (void);
 #endif /* BUILD_WACOM */
@@ -122,7 +122,7 @@ static CcPanelLoaderVtable default_panels[] =
   PANEL_TYPE("notifications",    cc_notifications_panel_get_type,        NULL),
   PANEL_TYPE("online-accounts",  cc_goa_panel_get_type,                  NULL),
   PANEL_TYPE("power",            cc_power_panel_get_type,                NULL),
-  PANEL_TYPE("printers",         cc_printers_panel_get_type,             NULL),
+  //PANEL_TYPE("printers",         cc_printers_panel_get_type,             NULL),
   PANEL_TYPE("region",           cc_region_panel_get_type,               NULL),
   PANEL_TYPE("removable-media",  cc_removable_media_panel_get_type,      NULL),
   PANEL_TYPE("search",           cc_search_panel_get_type,               NULL),
@@ -133,7 +133,7 @@ static CcPanelLoaderVtable default_panels[] =
 #endif
   PANEL_TYPE("universal-access", cc_ua_panel_get_type,                   NULL),
   PANEL_TYPE("usage",            cc_usage_panel_get_type,                NULL),
-  PANEL_TYPE("user-accounts",    cc_user_panel_get_type,                 NULL),
+  //PANEL_TYPE("user-accounts",    cc_user_panel_get_type,                 NULL),
 #ifdef BUILD_WACOM
   PANEL_TYPE("wacom",            cc_wacom_panel_get_type,                cc_wacom_panel_static_init_func),
 #endif
