commit bedcb0d9f334ca20cbfaba4d1233f2703fe65c0d
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sun Mar 19 18:29:01 2023 +0100

    meson: fix pkgconf version

diff --git a/meson.build b/meson.build
index 123768a..9f364c7 100644
--- a/meson.build
+++ b/meson.build
@@ -1,6 +1,6 @@
 project(
   'gnome-control-center', 'c',
-        version : '44.rc',
+        version : '44.0',
         license : 'GPL2+',
   meson_version : '>= 0.57.0'
 )
