commit c2780b95435214682d6028b47d02f8708c030122
Author: q66 <q66@chimera-linux.org>
Date:   Fri Jul 7 16:57:32 2023 +0200

    drop ks_ldap_help_variables

diff --git a/dirmngr/server.c b/dirmngr/server.c
index 51a149c..f1e2eac 100644
--- a/dirmngr/server.c
+++ b/dirmngr/server.c
@@ -2776,7 +2776,6 @@ cmd_ad_query (assuan_context_t ctx, char *line)
 
   if (opt_help)
     {
-      ks_ldap_help_variables (ctrl);
       err = 0;
       goto leave;
     }
