commit 5d3b06b94efc3f2e6293394eb7453ddaf1d31901
Author: q66 <q66@chimera-linux.org>
Date:   Thu Oct 19 17:56:22 2023 +0200

    fix build

diff --git a/src/liborcus/gnumeric_value_format_parser.cpp b/src/liborcus/gnumeric_value_format_parser.cpp
index 822c1ba..20ce2d5 100644
--- a/src/liborcus/gnumeric_value_format_parser.cpp
+++ b/src/liborcus/gnumeric_value_format_parser.cpp
@@ -41,7 +41,7 @@ void gnumeric_value_format_parser::segment()
                 if (pos != 2)
                     throw parse_error("value format segment is not formatted properly", get_pos());
 
-                std::string_view s{p0, std::distance(p0, m_cur)};
+                std::string_view s{p0, size_t(std::distance(p0, m_cur))};
                 if (s.empty())
                     throw parse_error("segment value is empty", get_pos());
 
@@ -51,7 +51,7 @@ void gnumeric_value_format_parser::segment()
             }
             case '=':
             {
-                std::string_view s{p0, std::distance(p0, m_cur)};
+                std::string_view s{p0, size_t(std::distance(p0, m_cur))};
                 seg.type = to_gnumeric_value_format_type(s);
                 if (seg.type == gnumeric_value_format_type::unknown)
                 {
@@ -65,7 +65,7 @@ void gnumeric_value_format_parser::segment()
             }
             case ':':
             {
-                std::string_view s{p0, std::distance(p0, m_cur)};
+                std::string_view s{p0, size_t(std::distance(p0, m_cur))};
 
                 switch (pos)
                 {
