From 27db72556d9a00bb24c9aad9a7ae6447bbc50a71 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sat, 27 Nov 2021 01:03:28 +0100
Subject: [PATCH 01/19] llvm: always set a larger stack size explicitly

---
 llvm/lib/Support/Threading.cpp | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/llvm/lib/Support/Threading.cpp b/llvm/lib/Support/Threading.cpp
index 04a1a9e19..776badaad 100644
--- a/llvm/lib/Support/Threading.cpp
+++ b/llvm/lib/Support/Threading.cpp
@@ -82,16 +82,7 @@ unsigned llvm::ThreadPoolStrategy::compute_thread_count() const {
 // keyword.
 #include "llvm/Support/thread.h"
 
-#if defined(__APPLE__)
-  // Darwin's default stack size for threads except the main one is only 512KB,
-  // which is not enough for some/many normal LLVM compilations. This implements
-  // the same interface as std::thread but requests the same stack size as the
-  // main thread (8MB) before creation.
 const llvm::Optional<unsigned> llvm::thread::DefaultStackSize = 8 * 1024 * 1024;
-#else
-const llvm::Optional<unsigned> llvm::thread::DefaultStackSize = None;
-#endif
-
 
 #endif
 
-- 
2.37.3

