From 6f1be080b311c23d8905539111d649060de6c8d9 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 4 Nov 2023 08:50:28 +0100
Subject: [PATCH 02/30] llvm: musl workarounds

---
 llvm/include/llvm/Analysis/TargetLibraryInfo.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/llvm/include/llvm/Analysis/TargetLibraryInfo.h b/llvm/include/llvm/Analysis/TargetLibraryInfo.h
index db5e80ccd..5043c1a1a 100644
--- a/llvm/include/llvm/Analysis/TargetLibraryInfo.h
+++ b/llvm/include/llvm/Analysis/TargetLibraryInfo.h
@@ -17,6 +17,15 @@
 #include "llvm/TargetParser/Triple.h"
 #include <optional>
 
+#undef fopen64
+#undef fseeko64
+#undef fstat64
+#undef fstatvfs64
+#undef ftello64
+#undef lstat64
+#undef stat64
+#undef tmpfile64
+
 namespace llvm {
 
 template <typename T> class ArrayRef;
-- 
2.47.0

