From 8634ee49c9357354bc61d35cc6e41ec7671a0499 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 4 Nov 2023 08:55:34 +0100
Subject: [PATCH 06/27] compiler-rt: default to libc++ for sanitizers

---
 compiler-rt/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler-rt/CMakeLists.txt b/compiler-rt/CMakeLists.txt
index bbb4e8d7c..819a5a5e5 100644
--- a/compiler-rt/CMakeLists.txt
+++ b/compiler-rt/CMakeLists.txt
@@ -203,7 +203,7 @@ macro(handle_default_cxx_lib var)
       set(${var}_LIBNAME "libc++")
       set(${var}_INTREE 1)
     else()
-      set(${var}_LIBNAME "libstdc++")
+      set(${var}_LIBNAME "libc++")
       set(${var}_SYSTEM 1)
     endif()
   else()
-- 
2.44.0

