From ee096ffac895b0c637c1a004f9922726c98f9fdd Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sun, 14 Apr 2024 14:46:54 +0200
Subject: [PATCH 15/27] clang: drop incorrect warning about vector equality
 results

---
 clang/lib/Sema/SemaExpr.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/clang/lib/Sema/SemaExpr.cpp b/clang/lib/Sema/SemaExpr.cpp
index 4cce0abc2..581483e5f 100644
--- a/clang/lib/Sema/SemaExpr.cpp
+++ b/clang/lib/Sema/SemaExpr.cpp
@@ -13641,8 +13641,6 @@ QualType Sema::CheckVectorCompareOperands(ExprResult &LHS, ExprResult &RHS,
       if (vType->castAs<VectorType>()->getVectorKind() ==
           VectorKind::AltiVecVector)
         return Context.getLogicalOperationType();
-      else
-        Diag(Loc, diag::warn_deprecated_altivec_src_compat);
       break;
     case LangOptions::AltivecSrcCompatKind::GCC:
       // For GCC we always return the vector type.
-- 
2.44.0

