From 2d9deaaa1b866a77277e412460b866b1d2cf8df9 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Tue, 2 May 2023 16:19:42 +0200
Subject: [PATCH 20/27] clang: enable fortify by default on optimizing musl
 targets

---
 clang/lib/Frontend/InitPreprocessor.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/clang/lib/Frontend/InitPreprocessor.cpp b/clang/lib/Frontend/InitPreprocessor.cpp
index 208c6a8db..6f9ae584c 100644
--- a/clang/lib/Frontend/InitPreprocessor.cpp
+++ b/clang/lib/Frontend/InitPreprocessor.cpp
@@ -1301,6 +1301,11 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
     Builder.defineMacro("__GLIBCXX_BITSIZE_INT_N_0", "128");
   }
 
+  // Enable fortify by default on musl when optimizing, but not with asan
+  if (TI.getTriple().isMusl() && LangOpts.Optimize &&
+      !LangOpts.Sanitize.has(SanitizerKind::Address))
+    Builder.defineMacro("_FORTIFY_SOURCE", "2");
+
   // Get other target #defines.
   TI.getTargetDefines(LangOpts, Builder);
 }
-- 
2.39.0

