From ea82d4de395f920a4c8d13ed9bc152f95c986ded Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Tue, 2 May 2023 16:23:22 +0200
Subject: [PATCH 21/29] clang: use strong stack protector by default

---
 clang/lib/Driver/ToolChains/Linux.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/clang/lib/Driver/ToolChains/Linux.h b/clang/lib/Driver/ToolChains/Linux.h
index 2eb2d0578..80a0b035c 100644
--- a/clang/lib/Driver/ToolChains/Linux.h
+++ b/clang/lib/Driver/ToolChains/Linux.h
@@ -63,6 +63,11 @@ public:
 
   const char *getDefaultLinker() const override;
 
+  LangOptions::StackProtectorMode
+  GetDefaultStackProtectorLevel(bool KernelOrKext) const override {
+    return LangOptions::SSPStrong;
+  }
+
 protected:
   Tool *buildAssembler() const override;
   Tool *buildLinker() const override;
-- 
2.49.0

