From 7240e7712c6a5ab7961e3bd09d8622a8a555f044 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Tue, 2 May 2023 16:19:58 +0200
Subject: [PATCH 21/27] clang: use --as-needed by default

---
 clang/lib/Driver/ToolChains/Gnu.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/clang/lib/Driver/ToolChains/Gnu.cpp b/clang/lib/Driver/ToolChains/Gnu.cpp
index 4f2340316..97d831ba9 100644
--- a/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -566,6 +566,13 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   bool NeedsSanitizerDeps = addSanitizerRuntimes(ToolChain, Args, CmdArgs);
   bool NeedsXRayDeps = addXRayRuntime(ToolChain, Args, CmdArgs);
   addLinkerCompressDebugSectionsOption(ToolChain, Args, CmdArgs);
+
+  // Use --as-needed by default for all explicit linker inputs on Linux
+  // We don't reset it afterwards because explicit argument does not
+  // get reset either (and that is permitted, so it should be fine)
+  if (Triple.isOSLinux())
+    CmdArgs.push_back("--as-needed");
+
   AddLinkerInputs(ToolChain, Inputs, Args, CmdArgs, JA);
 
   addHIPRuntimeLibArgs(ToolChain, Args, CmdArgs);
-- 
2.39.0

