--- a/openmp/libomptarget/deviceRTLs/amdgcn/CMakeLists.txt
+++ b/openmp/libomptarget/deviceRTLs/amdgcn/CMakeLists.txt
@@ -30,7 +30,11 @@ else()
   return()
 endif()
 
-if (LLVM_DIR)
+if (CMAKE_CROSSCOMPILING)
+  set(CLANG_TOOL /usr/bin/clang)
+  set(LINK_TOOL /usr/bin/llvm-link)
+  set(OPT_TOOL /usr/bin/opt)
+elseif (LLVM_DIR)
   # Builds that use pre-installed LLVM have LLVM_DIR set.
   find_program(CLANG_TOOL clang PATHS ${LLVM_TOOLS_BINARY_DIR} NO_DEFAULT_PATH)
   find_program(LINK_TOOL llvm-link PATHS ${LLVM_TOOLS_BINARY_DIR}
