--- a/lldb/CMakeLists.txt
+++ b/lldb/CMakeLists.txt
@@ -31,21 +31,13 @@ if (WIN32)
 endif()
 
 if (LLDB_ENABLE_PYTHON)
-  if (NOT CMAKE_CROSSCOMPILING)
     execute_process(
-      COMMAND ${Python3_EXECUTABLE}
+      COMMAND python3
           -c "import distutils.sysconfig; print(distutils.sysconfig.get_python_lib(True, False, ''))"
       OUTPUT_VARIABLE LLDB_PYTHON_DEFAULT_RELATIVE_PATH
       OUTPUT_STRIP_TRAILING_WHITESPACE)
 
     file(TO_CMAKE_PATH ${LLDB_PYTHON_DEFAULT_RELATIVE_PATH} LLDB_PYTHON_DEFAULT_RELATIVE_PATH)
-  else ()
-    if ("${LLDB_PYTHON_RELATIVE_PATH}" STREQUAL "")
-      message(FATAL_ERROR
-        "Crosscompiling LLDB with Python requires manually setting
-        LLDB_PYTHON_RELATIVE_PATH.")
-    endif ()
-  endif ()
 
   set(LLDB_PYTHON_RELATIVE_PATH ${LLDB_PYTHON_DEFAULT_RELATIVE_PATH}
     CACHE STRING "Path where Python modules are installed, relative to install prefix")
