From 83b30ba32578117595b3e40f6b3543df74660b14 Mon Sep 17 00:00:00 2001
From: Davin McCall <davmac@davmac.org>
Date: Sat, 26 Mar 2022 21:36:10 +1000
Subject: [PATCH] Examples should be a top-level section

---
 doc/manpages/dinit-service.5.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/manpages/dinit-service.5.m4 b/doc/manpages/dinit-service.5.m4
index 0f956f3..1b1e8b6 100644
--- a/doc/manpages/dinit-service.5.m4
+++ b/doc/manpages/dinit-service.5.m4
@@ -527,7 +527,7 @@ To avoid substitution, a single `\fB$\fR' can be escaped with a second, as in `\
 Variables for substitution come from the \fBdinit\fR environment at the time the service is loaded.
 In particular, variables set via \fBenv\-file\fR are not visible to the substitution function.
 .\"
-.SS EXAMPLES
+.SH EXAMPLES
 .LP
 Here is an example service description for the \fBmysql\fR database server.
 It has a dependency on the \fBrcboot\fR service (not shown) which is
