commit 6f2701f479a9f0ce1487ed48d2c15b0b5b783c1f
Author: q66 <q66@chimera-linux.org>
Date:   Thu Feb 9 00:09:17 2023 +0000

    fix assertion error on shutdown

diff --git a/src/control.cc b/src/control.cc
index 8b1d3a9..ac8801b 100644
--- a/src/control.cc
+++ b/src/control.cc
@@ -1207,13 +1207,13 @@ bool control_conn_t::data_ready() noexcept
 
 bool control_conn_t::send_data() noexcept
 {
-    if (outbuf.empty() && bad_conn_close) {
+    if (outbuf.empty()) {
         if (oom_close) {
             // Send oom response
             char oomBuf[] = { DINIT_RP_OOM };
             bp_sys::write(iob.get_watched_fd(), oomBuf, 1);
         }
-        return true;
+        return bad_conn_close;
     }
     
     vector<char> & pkt = outbuf.front();
