From 5be171d1cbeba810480488d35d8adb6bb0f66fd5 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Thu, 25 Nov 2021 02:14:27 +0100
Subject: [PATCH] change HOST flags to cbuild setup

---
 build/tools/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/build/tools/Makefile b/build/tools/Makefile
index d501550..d8df7c7 100644
--- a/build/tools/Makefile
+++ b/build/tools/Makefile
@@ -1,6 +1,6 @@
-HOSTCXX ?= $(CXX)
-HOSTCXXFLAGS ?= $(CXXFLAGS)
-HOSTLDFLAGS ?= $(LDFLAGS)
+HOSTCXX ?= $(BUILD_CXX)
+HOSTCXXOPTS ?= $(BUILD_CXXFLAGS)
+HOSTLDFLAGS ?= $(BUILD_LDFLAGS)
 
 mconfig-gen: mconfig-gen.cc
 	$(HOSTCXX) $(HOSTCXXOPTS) -o mconfig-gen mconfig-gen.cc $(HOSTLDFLAGS)
-- 
2.34.0

