From 15f572483da67c3a18edaf044745f48f9f144106 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Tue, 9 Nov 2021 06:16:45 +0100
Subject: [PATCH] use symlinks instead of hardlinks

---
 doc/manpages/Makefile | 4 ++--
 src/Makefile          | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/doc/manpages/Makefile b/doc/manpages/Makefile
index 3fb0d7d..2df9f20 100644
--- a/doc/manpages/Makefile
+++ b/doc/manpages/Makefile
@@ -12,8 +12,8 @@ install: generate
 	install -m644 dinit-service.5 "$(DESTDIR)$(MANDIR)/man5"
 ifeq ($(BUILD_SHUTDOWN),yes)
 	install -m644 shutdown.8 "$(DESTDIR)$(MANDIR)/man8"
-	ln -f "$(DESTDIR)$(MANDIR)/man8/shutdown.8" "$(DESTDIR)$(MANDIR)/man8/halt.8" 
-	ln -f "$(DESTDIR)$(MANDIR)/man8/shutdown.8" "$(DESTDIR)$(MANDIR)/man8/reboot.8" 
+	ln -sf "shutdown.8" "$(DESTDIR)$(MANDIR)/man8/halt.8"
+	ln -sf "shutdown.8" "$(DESTDIR)$(MANDIR)/man8/reboot.8"
 endif
 
 clean:
diff --git a/src/Makefile b/src/Makefile
index bb8caa0..917977f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -61,8 +61,8 @@ install: all
 	install -d $(DESTDIR)$(SBINDIR)
 	install $(STRIPOPTS) dinit dinitctl dinitcheck $(SHUTDOWN) $(DESTDIR)$(SBINDIR)
 ifeq ($(BUILD_SHUTDOWN),yes)
-	ln -f $(DESTDIR)$(SBINDIR)/$(SHUTDOWN) $(DESTDIR)$(SBINDIR)/$(SHUTDOWNPREFIX)halt
-	ln -f $(DESTDIR)$(SBINDIR)/$(SHUTDOWN) $(DESTDIR)$(SBINDIR)/$(SHUTDOWNPREFIX)reboot
+	ln -sf $(SHUTDOWN) $(DESTDIR)$(SBINDIR)/$(SHUTDOWNPREFIX)halt
+	ln -sf $(SHUTDOWN) $(DESTDIR)$(SBINDIR)/$(SHUTDOWNPREFIX)reboot
 endif
 
 clean:
-- 
2.33.1

