--- a/Makefile
+++ b/Makefile
@@ -275,30 +275,38 @@ sha1.o : sha1.c sha1.h md5.h
 install : install-bin install-man install-udev
 
 install-static : mdadm.static install-man
-	$(INSTALL) -D $(STRIP) -m 755 mdadm.static $(DESTDIR)$(BINDIR)/mdadm
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 mdadm.static $(DESTDIR)$(BINDIR)/mdadm
 
 install-tcc : mdadm.tcc install-man
-	$(INSTALL) -D $(STRIP) -m 755 mdadm.tcc $(DESTDIR)$(BINDIR)/mdadm
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 mdadm.tcc $(DESTDIR)$(BINDIR)/mdadm
 
 install-uclibc : mdadm.uclibc install-man
-	$(INSTALL) -D $(STRIP) -m 755 mdadm.uclibc $(DESTDIR)$(BINDIR)/mdadm
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 mdadm.uclibc $(DESTDIR)$(BINDIR)/mdadm
 
 install-klibc : mdadm.klibc install-man
-	$(INSTALL) -D $(STRIP) -m 755 mdadm.klibc $(DESTDIR)$(BINDIR)/mdadm
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 mdadm.klibc $(DESTDIR)$(BINDIR)/mdadm
 
 install-man: mdadm.8 md.4 mdadm.conf.5 mdmon.8
-	$(INSTALL) -D -m 644 mdadm.8 $(DESTDIR)$(MAN8DIR)/mdadm.8
-	$(INSTALL) -D -m 644 mdmon.8 $(DESTDIR)$(MAN8DIR)/mdmon.8
-	$(INSTALL) -D -m 644 md.4 $(DESTDIR)$(MAN4DIR)/md.4
-	$(INSTALL) -D -m 644 mdadm.conf.5 $(DESTDIR)$(MAN5DIR)/mdadm.conf.5
+	$(INSTALL) -d $(DESTDIR)$(MAN8DIR)
+	$(INSTALL) -d $(DESTDIR)$(MAN5DIR)
+	$(INSTALL) -d $(DESTDIR)$(MAN4DIR)
+	$(INSTALL) -m 644 mdadm.8 $(DESTDIR)$(MAN8DIR)/mdadm.8
+	$(INSTALL) -m 644 mdmon.8 $(DESTDIR)$(MAN8DIR)/mdmon.8
+	$(INSTALL) -m 644 md.4 $(DESTDIR)$(MAN4DIR)/md.4
+	$(INSTALL) -m 644 mdadm.conf.5 $(DESTDIR)$(MAN5DIR)/mdadm.conf.5
 
 install-udev: udev-md-raid-arrays.rules udev-md-raid-assembly.rules udev-md-raid-creating.rules \
 		udev-md-clustered-confirm-device.rules 
+	$(INSTALL) -d $(DESTDIR)$(UDEVDIR)/rules.d
 	@for file in 01-md-raid-creating.rules 63-md-raid-arrays.rules 64-md-raid-assembly.rules \
 		69-md-clustered-confirm-device.rules ; \
 	do sed -e 's,BINDIR,$(BINDIR),g' udev-$${file#??-} > .install.tmp.1 && \
-	   $(ECHO) $(INSTALL) -D -m 644 udev-$${file#??-} $(DESTDIR)$(UDEVDIR)/rules.d/$$file ; \
-	   $(INSTALL) -D -m 644 .install.tmp.1 $(DESTDIR)$(UDEVDIR)/rules.d/$$file ; \
+	   $(ECHO) $(INSTALL) -m 644 udev-$${file#??-} $(DESTDIR)$(UDEVDIR)/rules.d/$$file ; \
+	   $(INSTALL) -m 644 .install.tmp.1 $(DESTDIR)$(UDEVDIR)/rules.d/$$file ; \
 	   rm -f .install.tmp.1; \
 	done
 
@@ -323,8 +331,9 @@ install-systemd: systemd/mdmon@.service
 	if [ -f /etc/SuSE-release -o -n "$(SUSE)" ] ;then $(INSTALL) -D -m 755 systemd/SUSE-mdadm_env.sh $(DESTDIR)$(LIB_DIR)/mdadm_env.sh ;fi
 
 install-bin: mdadm mdmon
-	$(INSTALL) -D $(STRIP) -m 755 mdadm $(DESTDIR)$(BINDIR)/mdadm
-	$(INSTALL) -D $(STRIP) -m 755 mdmon $(DESTDIR)$(BINDIR)/mdmon
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 mdadm $(DESTDIR)$(BINDIR)/mdadm
+	$(INSTALL) -m 755 mdmon $(DESTDIR)$(BINDIR)/mdmon
 
 uninstall:
 	rm -f $(DESTDIR)$(MAN8DIR)/mdadm.8 $(DESTDIR)$(MAN8DIR)/mdmon.8 $(DESTDIR)$(MAN4DIR)/md.4 $(DESTDIR)$(MAN5DIR)/mdadm.conf.5 $(DESTDIR)$(BINDIR)/mdadm
