--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -408,7 +408,7 @@ libnetfilter_conntrack_la_LIBADD = conntrack/libnfconntrack.la \
 				   expect/libnfexpect.la \
 				   ${LIBNFNETLINK_LIBS} ${LIBMNL_LIBS}
 
-libnetfilter_conntrack_la_LDFLAGS = -Wc,-nostartfiles -lnfnetlink \
+libnetfilter_conntrack_la_LDFLAGS = -lnfnetlink \
 				    -version-info $(LIBVERSION)
 
 libnetfilter_conntrack_la_SOURCES = main.c callback.c
