--- a/core/backends/bluetooth/multiplexchannel.cpp
+++ b/core/backends/bluetooth/multiplexchannel.cpp
@@ -52,7 +52,7 @@ qint64 MultiplexChannel::bytesToWrite() const
 qint64 MultiplexChannel::readData(char *data, qint64 maxlen)
 {
     if (maxlen <= state->read_buffer.size() || state->read_buffer.size() > 0) {
-        const auto num_to_read = std::min(maxlen, state->read_buffer.size());
+        const auto num_to_read = std::min(maxlen, qint64(state->read_buffer.size()));
         std::memcpy(data, state->read_buffer.data(), num_to_read);
         state->read_buffer.remove(0, num_to_read);
         Q_EMIT state->readAvailable();
