--- a/configure
+++ b/configure
@@ -15683,15 +15683,15 @@ fi
 fi
 
 
-    ac_fn_c_check_header_mongrel "$LINENO" "readline/readline.h" "ac_cv_header_readline_readline_h" "$ac_includes_default"
+    ac_fn_c_check_header_mongrel "$LINENO" "editline/readline.h" "ac_cv_header_readline_readline_h" "$ac_includes_default"
 if test "x$ac_cv_header_readline_readline_h" = xyes; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for readline in -lreadline" >&5
-$as_echo_n "checking for readline in -lreadline... " >&6; }
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for readline in -ledit" >&5
+$as_echo_n "checking for readline in -ledit... " >&6; }
 if ${ac_cv_lib_readline_readline+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lreadline $tcap $LIBS"
+LIBS="-ledit $tcap $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -15723,7 +15723,7 @@ fi
 $as_echo "$ac_cv_lib_readline_readline" >&6; }
 if test "x$ac_cv_lib_readline_readline" = xyes; then :
 
-	   RDL_LIBS="-lreadline $RDL_LIBS $tcap"
+	   RDL_LIBS="-ledit $RDL_LIBS $tcap"
 
 $as_echo "#define HAVE_LIBREADLINE /**/" >>confdefs.h
 
--- a/xmlcatalog.c
+++ b/xmlcatalog.c
@@ -17,7 +17,7 @@
 #endif
 
 #ifdef HAVE_LIBREADLINE
-#include <readline/readline.h>
+#include <editline/readline.h>
 #ifdef HAVE_LIBHISTORY
 #include <readline/history.h>
 #endif
--- a/xmllint.c
+++ b/xmllint.c
@@ -46,7 +46,7 @@
 #include <stdlib.h>
 #endif
 #ifdef HAVE_LIBREADLINE
-#include <readline/readline.h>
+#include <editline/readline.h>
 #ifdef HAVE_LIBHISTORY
 #include <readline/history.h>
 #endif
