From 239e25f613cb0d63bb26ebae433fcc4a9e0cee16 Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Mon, 5 Aug 2024 15:14:21 +0200
Subject: [PATCH] parser: Report at least one fatal error

---
 parserInternals.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/parserInternals.c b/parserInternals.c
index 1ab0b11c6..5724af1cb 100644
--- a/parserInternals.c
+++ b/parserInternals.c
@@ -256,7 +256,9 @@ xmlCtxtVErr(xmlParserCtxtPtr ctxt, xmlNodePtr node, xmlErrorDomain domain,
             goto done;
         ctxt->nbWarnings += 1;
     } else {
-        if (ctxt->nbErrors >= XML_MAX_ERRORS)
+        /* Report at least one fatal error. */
+        if ((ctxt->nbErrors >= XML_MAX_ERRORS) &&
+            ((level < XML_ERR_FATAL) || (ctxt->wellFormed == 0)))
             goto done;
         ctxt->nbErrors += 1;
     }
-- 
GitLab

