From ed8b4264f65b1ced1e3b13967dd1cf90102cfa40 Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Thu, 18 Jul 2024 20:56:31 +0200
Subject: [PATCH] xinclude: Set error handler when parsing text

---
 xinclude.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xinclude.c b/xinclude.c
index faf22883b..6c1e3ace6 100644
--- a/xinclude.c
+++ b/xinclude.c
@@ -1653,6 +1653,8 @@ xmlXIncludeLoadTxt(xmlXIncludeCtxtPtr ctxt, xmlXIncludeRefPtr ref) {
         xmlXIncludeErrMemory(ctxt);
         goto error;
     }
+    if (ctxt->errorHandler != NULL)
+        xmlCtxtSetErrorHandler(pctxt, ctxt->errorHandler, ctxt->errorCtxt);
     inputStream = xmlLoadExternalEntity((const char*)url, NULL, pctxt);
     if (inputStream == NULL) {
         /*
-- 
GitLab

