commit f9220227a69de4fd8cc25c4ec872ef7a647eb1e7
Author: q66 <q66@chimera-linux.org>
Date:   Sat Aug 5 14:36:55 2023 +0200

    portable realpath invocation

diff --git a/copy-firmware.sh b/copy-firmware.sh
index 7086019..498e873 100755
--- a/copy-firmware.sh
+++ b/copy-firmware.sh
@@ -96,7 +96,7 @@ grep -E '^Link:' WHENCE | sed -e 's/^Link: *//g;s/-> //g' | while read f d; do
     else
         directory="$destdir/$(dirname "$f")"
         install -d "$directory"
-        target="$(cd "$directory" && realpath -m -s "$d")"
+        target="$(cd "$directory" && realpath "$d")"
         if test -d "$target"; then
             $verbose "creating link $f -> $d"
             ln -s "$d" "$destdir/$f"
