commit 666bb19fbba4d30d0320123ab81cd391eeaa3d33
Author: q66 <q66@chimera-linux.org>
Date:   Fri Jun 30 15:48:10 2023 +0200

    portability fixes

diff --git a/copy-firmware.sh b/copy-firmware.sh
index 58eb7e3..0dd3927 100755
--- a/copy-firmware.sh
+++ b/copy-firmware.sh
@@ -75,7 +75,7 @@ grep -E '^Link:' WHENCE | sed -e 's/^Link: *//g;s/-> //g' | while read f d; do
         test -f "$destdir/$f$compext" && continue
         $verbose "copying link $f$compext"
         install -d "$destdir/$(dirname "$f")"
-        cp -d "$f$compext" "$destdir/$f$compext"
+        cp -R "$f$compext" "$destdir/$f$compext"
 
         if test "x$d" != "x"; then
             target="$(readlink "$f")"
