--- a/copy-firmware.sh
+++ b/copy-firmware.sh
@@ -92,7 +92,7 @@ rdfind -makesymlinks true -makeresultsfile false "$destdir" >/dev/null
 find "$destdir" -type l | while read -r l; do
 	target="$(realpath "$l")"
 	$verbose "Correcting path for $l"
-	ln -fs "$(realpath --relative-to="$(dirname "$(realpath -s "$l")")" "$target")" "$l"
+	ln -fs "$(python3 -c "import os, sys; print(os.path.relpath(sys.argv[1], os.path.dirname(os.path.normpath(sys.argv[2]))))" "$target" "$l")" "$l"
 done
 
 # shellcheck disable=SC2162 # file/folder name can include escaped symbols
