commit 79e669213d614d3d460d46259fbb608493230b20
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Mon Jun 5 23:43:56 2023 +0200

    fix deprecated declarations

diff --git a/src/nss/crypto.c b/src/nss/crypto.c
index 9171313..429d209 100644
--- a/src/nss/crypto.c
+++ b/src/nss/crypto.c
@@ -412,7 +412,7 @@ xmlSecNssKeysMngrInit(xmlSecKeysMngrPtr mngr) {
  * Returns: internal key slot and initializes it if needed.
  */
 PK11SlotInfo *
-xmlSecNssGetInternalKeySlot()
+xmlSecNssGetInternalKeySlot(void)
 {
     PK11SlotInfo *slot = NULL;
     SECStatus rv;
