diff --git a/meson.build b/meson.build
index 6f79707..98e6470 100644
--- a/meson.build
+++ b/meson.build
@@ -194,21 +194,21 @@ conf.set('GPERF_LEN_TYPE', gperf_len_type,
 nobody_user = get_option('nobody-user')
 
 if not meson.is_cross_build()
-        getent_result = run_command('getent', 'passwd', '65534')
+        getent_result = run_command('getent', 'passwd', '99')
         if getent_result.returncode() == 0
                 name = getent_result.stdout().split(':')[0]
                 if name != nobody_user
                         warning('\n' +
-                                'The local user with the UID 65534 does not match the configured user name "@0@" of the nobody user (its name is @1@).\n'.format(nobody_user, name) +
+                                'The local user with the UID 99 does not match the configured user name "@0@" of the nobody user (its name is @1@).\n'.format(nobody_user, name) +
                                 'Your build will result in an user table setup that is incompatible with the local system.')
                 endif
         endif
         id_result = run_command('id', '-u', nobody_user)
         if id_result.returncode() == 0
                 id = id_result.stdout().to_int()
-                if id != 65534
+                if id != 99
                         warning('\n' +
-                                'The local user with the configured user name "@0@" of the nobody user does not have UID 65534 (it has @1@).\n'.format(nobody_user, id) +
+                                'The local user with the configured user name "@0@" of the nobody user does not have UID 99 (it has @1@).\n'.format(nobody_user, id) +
                                 'Your build will result in an user table setup that is incompatible with the local system.')
                 endif
         endif
diff --git a/meson_options.txt b/meson_options.txt
index 87adfc2..3d7ad1e 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -5,7 +5,7 @@ option('debug-extra', type : 'array', choices : ['hashmap'], value : [],
        description : 'enable extra debugging')
 
 option('nobody-user', type : 'string',
-       description : 'The name of the nobody user (the one with UID 65534)',
+       description : 'The name of the nobody user (the one with UID 99)',
        value : 'nobody')
 
 option('system-bus-address', type : 'string',
diff --git a/src/basic/user-util.h b/src/basic/user-util.h
index d58f568..7e5977b 100644
--- a/src/basic/user-util.h
+++ b/src/basic/user-util.h
@@ -17,7 +17,7 @@ char* uid_to_name(uid_t uid);
 
 #define UID_INVALID ((uid_t) -1)
 
-#define UID_NOBODY ((uid_t) 65534U)
+#define UID_NOBODY ((uid_t) 99U)
 
 bool synthesize_nobody(void);
 
