--- a/beets/util/__init__.py
+++ b/beets/util/__init__.py
@@ -46,7 +46,7 @@
     Union,
 )
 
-from typing_extensions import TypeAlias
+from typing import TypeAlias
 from unidecode import unidecode
 
 from beets.util import hidden
--- a/setup.py
+++ b/setup.py
@@ -91,7 +91,6 @@
         "munkres>=1.0.0",
         "musicbrainzngs>=0.4",
         "pyyaml",
-        "typing_extensions",
         "unidecode>=1.3.6",
     ]
     + (
