diff --git a/yazi-fm/Cargo.toml b/yazi-fm/Cargo.toml
index 9d7879b..9fa99b8 100644
--- a/yazi-fm/Cargo.toml
+++ b/yazi-fm/Cargo.toml
@@ -52,9 +52,6 @@ signal-hook-tokio = { version = "0.3.1", features = [ "futures-v0_3" ] }
 [target.'cfg(target_os = "macos")'.dependencies]
 crossterm = { workspace = true, features = [ "use-dev-tty", "libc" ] }
 
-[target.'cfg(all(not(target_os = "macos"), not(target_os = "windows")))'.dependencies]
-tikv-jemallocator = "0.6.0"
-
 [[bin]]
 name = "yazi"
 path = "src/main.rs"
diff --git a/yazi-fm/src/main.rs b/yazi-fm/src/main.rs
index 96d8bff..df510ce 100644
--- a/yazi-fm/src/main.rs
+++ b/yazi-fm/src/main.rs
@@ -1,10 +1,6 @@
 #![allow(clippy::module_inception)]
 #![allow(clippy::unit_arg)]
 
-#[cfg(all(not(target_os = "macos"), not(target_os = "windows")))]
-#[global_allocator]
-static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
-
 yazi_macro::mod_pub!(app completion confirm help input lives manager notify pick spot tasks which);
 
 yazi_macro::mod_flat!(context executor logs panic root router signals term);
