--- a/test/test-list.h
+++ b/test/test-list.h
@@ -128,7 +128,6 @@ TEST_DECLARE   (tcp_bind_or_listen_error_after_close)
 TEST_DECLARE   (tcp_listen_without_bind)
 TEST_DECLARE   (tcp_connect_error_fault)
 TEST_DECLARE   (tcp_connect6_error_fault)
-TEST_DECLARE   (tcp_connect6_link_local)
 TEST_DECLARE   (tcp_connect_timeout)
 TEST_DECLARE   (tcp_local_connect_timeout)
 TEST_DECLARE   (tcp6_local_connect_timeout)
@@ -726,7 +725,6 @@ TASK_LIST_START
   TEST_ENTRY  (tcp_listen_without_bind)
   TEST_ENTRY  (tcp_connect_error_fault)
   TEST_ENTRY  (tcp_connect6_error_fault)
-  TEST_ENTRY  (tcp_connect6_link_local)
   TEST_ENTRY  (tcp_connect_timeout)
   TEST_ENTRY  (tcp_local_connect_timeout)
   TEST_ENTRY  (tcp6_local_connect_timeout)
