diff -urN a/test/test-list.h b/test/test-list.h
--- a/test/test-list.h	2026-03-06 20:05:38.000000000 +0000
+++ b/test/test-list.h	2026-08-26 21:59:00.827204836 +0100
@@ -379,7 +379,6 @@
 #ifdef _WIN32
 TEST_FS_DECLARE   (fs_unlink_archive_readonly)
 #endif
-TEST_FS_DECLARE   (fs_chown)
 TEST_FS_DECLARE   (fs_link)
 TEST_FS_DECLARE   (fs_readlink)
 TEST_FS_DECLARE   (fs_realpath)
@@ -1107,7 +1106,6 @@
 #ifdef _WIN32
   TEST_FS_ENTRY  (fs_unlink_archive_readonly)
 #endif
-  TEST_FS_ENTRY  (fs_chown)
   TEST_FS_ENTRY  (fs_link)
   TEST_FS_ENTRY  (fs_utime)
   TEST_FS_ENTRY  (fs_utime_round)
