diff --git a/meson.build b/meson.build
index 6d1f986..3d334d3 100644
--- a/meson.build
+++ b/meson.build
@@ -377,7 +377,7 @@ endif
 
 lib_cryptsetup = dependency(
   'libcryptsetup',
-  required : get_option('cryptsetup'))
+  required : false)
 conf.set('HAVE_CRYPTSETUP', lib_cryptsetup.found() ? 1 : false)
 
 cryptsetup_dlopen = not get_option('cryptsetup').disabled() and get_option('cryptsetup-dlopen').enabled()
