commit 6a0cc1aa2cb1669581203391aa291d4899315a3c
Author: q66 <q66@chimera-linux.org>
Date:   Fri Jun 30 04:07:41 2023 +0200

    work around missing ln -r

diff --git a/libmount/meson.build b/libmount/meson.build
index c52e2be..1c6b38e 100644
--- a/libmount/meson.build
+++ b/libmount/meson.build
@@ -143,10 +143,6 @@
       dependencies : lib__mount_deps + blkid_static_dep,
       c_args : ['-DTEST_PROGRAM'],
     )
-    # the test-setup expects the helpers in the toplevel build-directory
-    link = meson.project_build_root() / test_name
-    run_command('ln', '-srf', exe.full_path(), link,
-      check : true)
   endforeach
 endif
 
--- a/liblastlog2/meson.build
+++ b/liblastlog2/meson.build
@@ -54,10 +54,6 @@
       link_with : [lib_common],
       dependencies : [libdl, lastlog2_dep],
     )
-    # the test-setup expects the helpers in the toplevel build-directory
-    link = meson.project_build_root() / test_name
-    run_command('ln', '-srf', exe.full_path(), link,
-      check : true)
   endforeach
 
 endif  
