commit ed7ef1c54aea6d8ccf88fa961bf7d16d44d00de3
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Thu Jun 1 03:05:32 2023 +0200

    portable handling of symlinks

diff --git a/meson.build b/meson.build
index 107e54a..c657aca 100644
--- a/meson.build
+++ b/meson.build
@@ -813,7 +813,6 @@ bison_gen = generator(
 
 
 
-meson_make_symlink = meson.current_source_dir() + '/tools/meson-make-symlink.sh'
 meson_make_manpage_stub = meson.current_source_dir() + '/tools/meson-make-manpage-stub.sh'
 
 configure_file(
@@ -941,9 +940,7 @@ exe = executable(
   build_by_default : opt)
 if opt and not is_disabler(exe)
   exes += exe
-  meson.add_install_script(meson_make_symlink,
-                           'last',
-                           join_paths(usrbin_exec_dir, 'lastb'))
+  install_symlink('lastb', pointing_to: 'last', install_dir: usrbin_exec_dir)
   manadocs += ['login-utils/last.1.adoc']
   manlinks += {'lastb.1': 'last.1'}
   bashcompletions += ['last']
@@ -1048,9 +1045,7 @@ exe = executable(
   build_by_default : opt)
 if opt and not is_disabler(exe)
   exes += exe
-  meson.add_install_script(meson_make_symlink,
-                           'vipw',
-                           join_paths(usrbin_exec_dir, 'vigr'))
+  install_symlink('vigr', pointing_to: 'vipw', install_dir: usrbin_exec_dir)
   manadocs += ['login-utils/vipw.8.adoc']
 endif
 
@@ -1500,26 +1495,6 @@ exes += exe
 manadocs += ['sys-utils/setarch.8.adoc']
 bashcompletions += ['setarch']
 
-setarch_links = ['uname26', 'linux32', 'linux64']
-setarch_links_arch = {
-  's390x' :   ['s390', 's390x'],
-  'x86' :     ['i386'],
-  'x86_64' :  ['i386', 'x86_64'],
-  'ppc64' :   ['ppc', 'ppc64', 'ppc32'],
-  'space64' : ['sparc', 'sparc64', 'sparc32', 'sparc32bash'],
-  'mips64' :  ['mips', 'mips64', 'mips32'],
-  'ia64' :    ['i386', 'ia64'],
-  'hppa' :    ['parisc', 'parisc64', 'parisc32'],
-}
-setarch_links += setarch_links_arch.get(host_machine.cpu_family(), [])
-
-foreach link: setarch_links
-  meson.add_install_script(meson_make_symlink,
-                           'setarch',
-                           join_paths(usrbin_exec_dir, link))
-  manlinks += {link + '.8': 'setarch.8'}
-endforeach
-
 opt = not get_option('build-eject').disabled()
 exe = executable(
   'eject',
@@ -3417,8 +3392,6 @@ if bash_completion.found()
     )
   endforeach
   foreach link_name, target : bashcompletionslinks
-    meson.add_install_script(meson_make_symlink,
-      target,
-      join_paths(bash_completion.get_variable(pkgconfig : 'completionsdir'), link_name))
+    install_symlink(link_name, pointing_to: target, install_dir: bash_completion.get_variable(pkgconfig : 'completionsdir'))
   endforeach
 endif
