diff --git a/./configure.ac b/./configure.ac.new
index cb4e43fa..ad172f9e 100644
--- a/./configure.ac
+++ b/./configure.ac.new
@@ -461,7 +461,7 @@ AM_CONDITIONAL(BUILD_PINENTRY_GTK_2, test "$pinentry_gtk_2" = "yes")
 if test "$pinentry_gnome_3" != "no"; then
 	PKG_CHECK_MODULES(
 		GNOME3,
-		[gcr-3,gcr-base-3],
+		[gcr-4],
 		[
 			pinentry_gnome_3=yes
 			AC_DEFINE(GCR_API_SUBJECT_TO_CHANGE, 1, [Nod nod])
diff --git a/gnome3/pinentry-gnome3.c b/gnome3/pinentry-gnome3.c.new
index 0b8d8d18..d0f8873e 100644
--- a/gnome3/pinentry-gnome3.c
+++ b/gnome3/pinentry-gnome3.c.new
@@ -23,7 +23,7 @@
 # include "config.h"
 #endif
 
-#include <gcr/gcr-base.h>
+#include <gcr/gcr.h>
 
 #include <string.h>
 #include <stdlib.h>
