diff --color -ruN gdm-settings-4.4/gdms/settings.py gdm-settings-4.4-patched/gdms/settings.py
--- gdm-settings-4.4/gdms/settings.py	2024-05-08 19:34:15.000000000 +0000
+++ gdm-settings-4.4-patched/gdms/settings.py	2024-07-11 02:56:48.859743696 +0000
@@ -59,7 +59,32 @@
 
 _commands = CommandList()
 
+def _ensure_path(path):
+    (head, tail) = os.path.split(path)
+    if not head == "":
+        _commands.add(f"mkdir -p '{head}'")
 
+def _cmd_t(cmd, flags, src, dest):
+    if not os.path.isdir(dest):
+        if os.path.exists(dest):
+            _commands.add(f"rm '{dest}'")
+        _commands.add(f"mkdir '{dest}'")
+    _commands.add(f"{cmd} {flags} '{src}' '{dest}'")
+
+# A portable alternative to the -t flag for install
+def _cp_t(flags, src, dest):
+    _cmd_t('cp', flags, src, dest)
+
+# A portable alternative to the -t flag for install
+def _install_t(flags, src, dest):
+    _cmd_t('install', flags, src, dest)
+
+# A portable alternative to the -T flag for install
+def _install_T(flags, src, dest):
+    if os.path.isdir(dest):
+        _commands.add(f"rm -r '{dest}'")
+    _commands.add(f"install {flags} '{src}' '{dest}'")
+
 def init():
     '''Initialize the settings module'''
     os.makedirs(env.TEMP_DIR, exist_ok=True)
@@ -335,7 +360,7 @@
 
     _commands.add(f"rm -rf {gresource.ThemesDir}/default-pure")
     _commands.add(f"mkdir -p {gresource.ThemesDir}")
-    _commands.add(f"cp -r {env.TEMP_DIR}/default-pure -t {gresource.ThemesDir}")
+    _cp_t("-r", f"{env.TEMP_DIR}/default-pure", gresource.ThemesDir)
 
 
 def _gresource_apply():
@@ -550,14 +575,16 @@
 
         logo_temp = os.path.join(env.TEMP_DIR, 'logo.temp')
         shutil.copy(logo_file, logo_temp)
-        _commands.add(f"install -m644 '{logo_temp}' -T '{logo}'")
+        _install_T("-m644", logo_temp, logo)
 
     overriding_files = get_overriding_files()
     if overriding_files:
         _commands.add('rm', *overriding_files)
 
-    _commands.add(f"install -Dm644 '{temp_conf_path}' -t '{gdm_conf_dir}'")
-    _commands.add(f"install -Dm644 '{temp_profile_path}' -T '{gdm_profile_path}'")
+    _ensure_path(gdm_conf_dir)
+    _install_t("-m644", temp_conf_path, gdm_conf_dir)
+    _ensure_path(gdm_profile_path)
+    _install_T("-m644", temp_profile_path, gdm_profile_path)
     _commands.add("dconf update")
 
 
@@ -611,7 +638,8 @@
                      '&>/dev/null',
                    ])
 
-    _commands.add(['install', '-Dm644',
+    _ensure_path(f"~{gresource.GdmUsername}/.config/monitors.xml")
+    _commands.add(['install', '-m644',
                      '-o', gresource.GdmUsername,
                      temp_monitors_xml,
                      f'~{gresource.GdmUsername}/.config/monitors.xml',
