--- a/share/mk/dist/check/_.mk
+++ b/share/mk/dist/check/_.mk
@@ -11,7 +11,7 @@
 include $(MAKEFILEDIR)/configure/version.mk
 
 
-_DISTCHECKDIR      := $(shell $(REALPATH) -m $(builddir)/distcheck)
+_DISTCHECKDIR      := $(shell $(REALPATH) $(builddir)/distcheck)
 _DISTCHECKSRCDIR   := $(_DISTCHECKDIR)/$(DISTNAME)
 _DISTCHECKBUILDDIR := $(_DISTCHECKDIR)/$(DISTNAME)_builddir
 _DISTCHECKDESTDIR  := $(_DISTCHECKDIR)/$(DISTNAME)_destdir
