commit 9942955d09ff9c1e47b023bd50cb28f995927dfd
Author: q66 <q66@chimera-linux.org>
Date:   Mon Mar 21 01:06:13 2022 +0100

    drop bad version requirement

diff --git a/avahi-libevent.pc.in b/avahi-libevent.pc.in
index a1dca01..c178e15 100644
--- a/avahi-libevent.pc.in
+++ b/avahi-libevent.pc.in
@@ -6,6 +6,6 @@ includedir=${prefix}/include
 Name: avahi-libevent
 Description: Avahi Multicast DNS Responder (libevent Support)
 Version: @PACKAGE_VERSION@
-Requires: libevent-2.1.5
+Requires: libevent
 Libs: -L${libdir} -lavahi-libevent
 Cflags: -D_REENTRANT -I${includedir}
