diff --git a/test/signal/test-signalhandler.cpp b/test/signal/test-signalhandler.cpp
index a3e77b08e..f8c3503a2 100644
--- a/test/signal/test-signalhandler.cpp
+++ b/test/signal/test-signalhandler.cpp
@@ -52,17 +52,13 @@
     ++*static_cast<int*>(nullptr); // NOLINT(clang-analyzer-core.NullDereference)
 }
 
-#if !defined(__APPLE__)
 /*static*/ int my_fpe() // NOLINT(misc-use-internal-linkage)
 {
-    if (feenableexcept(FE_ALL_EXCEPT) == -1)
-        return 2;
     if (std::feraiseexcept(FE_ALL_EXCEPT) != 0)
         return 3;
     return 1 % -1;
 }
 #endif
-#endif
 
 int main(int argc, const char * const argv[])
 {
