From d8657fa119ee3807a274dc7357ae1e2d045d5e50 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Tue, 2 Aug 2022 23:47:48 +0200
Subject: [PATCH] do not use bash and drop chown

---
 install-sh | 25 +++----------------------
 1 file changed, 3 insertions(+), 22 deletions(-)

diff --git a/install-sh b/install-sh
index 86fc9e3..295ce67 100755
--- a/install-sh
+++ b/install-sh
@@ -1,4 +1,4 @@
-#! /bin/bash
+#!/bin/sh
 # SPDX-License-Identifier: GPL-2.0
 # Copyright (c) 2000-2001 Silicon Graphics, Inc.  All Rights Reserved.
 #
@@ -23,6 +23,7 @@
 #       set           not set  |    no         yes
 #       set           set      |    yes        yes
 #
+
 _usage() {
     echo "Usage: $prog [-o owner] [-g group] [-m mode] -d directory"
     echo "or     $prog [-D] [-o owner] [-g group] [-m mode] file directory/file"
@@ -40,27 +41,7 @@ _usage() {
 
 _chown ()
 {
-    _st=255
-    if [ $# -eq 3 ] ; then
-	chown $1:$2 $3
-	_st=$?
-	if [ $_st -ne 0 ] ; then
-	    if [ $REAL_UID != '0' ] ; then
-		if [ ! -f $DIST_ROOT/.chown.quiet ] ; then
-		    echo '==============================================='
-		    echo Ownership of files under ${DIST_ROOT:-/}
-		    echo cannot be changed
-		    echo '==============================================='
-		    if [ -n "$DIST_ROOT" ] ; then
-			touch $DIST_ROOT/.chown.quiet
-		    fi
-		fi
-	       _st=0
-	    fi
-	fi
-    fi
-
-    return $_st
+    return 0
 }
 
 
-- 
2.37.1

