--- a/image-5.29.2/registries.conf
+++ b/image-5.29.2/registries.conf
@@ -18,7 +18,7 @@
 # of these registries, it should be added at the end of the list.
 #
 # # An array of host[:port] registries to try when pulling an unqualified image, in order.
-# unqualified-search-registries = ["example.com"]
+unqualified-search-registries = ["docker.io"]
 #
 # [[registry]]
 # # The "prefix" field is used to choose the relevant [[registry]] TOML table;
--- a/storage-1.52.0/docs/Makefile
+++ b/storage-1.52.0/docs/Makefile
@@ -1,4 +1,4 @@
-GOMD2MAN = ../tests/tools/build/go-md2man
+GOMD2MAN = go-md2man
 PREFIX ?= ${DESTDIR}/usr
 MANINSTALLDIR=${PREFIX}/share/man
 MANPAGES_MD = $(wildcard docs/*.5.md)
--- a/common-0.57.4/pkg/config/containers.conf
+++ b/common-0.57.4/pkg/config/containers.conf
@@ -417,7 +417,7 @@ default_sysctls = [
 # Cgroup management implementation used for the runtime.
 # Valid options "systemd" or "cgroupfs"
 #
-#cgroup_manager = "systemd"
+cgroup_manager = "cgroupfs"
 
 # Environment variables to pass into conmon
 #
