diff --git a/osdep/language-posix.c b/osdep/language-posix.c
index f5d1a3a72c..8fd68c6420 100644
--- a/osdep/language-posix.c
+++ b/osdep/language-posix.c
@@ -33,6 +33,7 @@ char **mp_get_user_langs(void)
 
     size_t nb = 0;
     char **ret = NULL;
+    bool has_c = false;
 
     // Prefer anything we get from LANGUAGE first
     for (const char *langList = getenv("LANGUAGE"); langList && *langList;) {
@@ -49,11 +50,21 @@ char **mp_get_user_langs(void)
         const char *envval = getenv(list[i]);
         if (envval && *envval) {
             size_t len = strcspn(envval, ".@");
+            if (!strncmp("C", envval, len)) {
+                has_c = true;
+                continue;
+            }
+
             MP_TARRAY_GROW(NULL, ret, nb);
             ret[nb++] = talloc_strndup(ret, envval, len);
         }
     }
 
+    if (has_c && !nb) {
+        MP_TARRAY_GROW(NULL, ret, nb);
+        ret[nb++] = talloc_strdup(ret, "en");
+    }
+
     // Null-terminate the list
     MP_TARRAY_APPEND(NULL, ret, nb, NULL);
 
