diff --git a/Cargo.lock b/Cargo.lock
index bfe9e646c..541b4c21a 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -270,7 +270,6 @@ dependencies = [
  "streaming-iterator",
  "strsim 0.10.0",
  "strum",
- "tikv-jemallocator",
  "tree-sitter",
  "tree-sitter-bash",
  "tree-sitter-c",
@@ -965,26 +964,6 @@ version = "0.4.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "3369f5ac52d5eb6ab48c6b4ffdc8efbcad6b89c765749064ba298f2c68a16a76"
 
-[[package]]
-name = "tikv-jemalloc-sys"
-version = "0.6.0+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "cd3c60906412afa9c2b5b5a48ca6a5abe5736aec9eb48ad05037a677e52e4e2d"
-dependencies = [
- "cc",
- "libc",
-]
-
-[[package]]
-name = "tikv-jemallocator"
-version = "0.6.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "4cec5ff18518d81584f477e9bfdf957f5bb0979b0bac3af4ca30b5b3ae2d2865"
-dependencies = [
- "libc",
- "tikv-jemalloc-sys",
-]
-
 [[package]]
 name = "tree-sitter"
 version = "0.24.5"
diff --git a/Cargo.toml b/Cargo.toml
index a5ec1378b..c280c4d7c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -110,9 +110,6 @@ tree-sitter-xml = "0.7.0"
 tree-sitter-yaml = "0.7.0"
 tree-sitter-zig = "1.1.2"
 
-[target.'cfg(not(any(target_env = "msvc", target_os = "illumos")))'.dependencies]
-tikv-jemallocator = "0.6"
-
 [dev-dependencies]
 # assert_cmd 2.0.10 requires predicates 3.
 # TODO: update.
diff --git a/src/main.rs b/src/main.rs
index c5818a187..27bfcc6f1 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -75,29 +75,6 @@ use crate::parse::guess_language::language_globs;
 use crate::parse::guess_language::{guess, language_name, Language, LanguageOverride};
 use crate::parse::syntax;
 
-/// The global allocator used by difftastic.
-///
-/// Diffing allocates a large amount of memory, and both Jemalloc and
-/// MiMalloc perform better than the system allocator.
-///
-/// Some versions of MiMalloc (specifically libmimalloc-sys greater
-/// than 0.1.24) handle very large, mostly unused allocations
-/// badly. This makes large line-oriented diffs very slow, as
-/// discussed in #297.
-///
-/// MiMalloc is generally faster than Jemalloc, but older versions of
-/// MiMalloc don't compile on GCC 15+, so use Jemalloc for now. See
-/// #805.
-///
-/// For reference, Jemalloc uses 10-20% more time (although up to 33%
-/// more instructions) when testing on sample files.
-#[cfg(not(any(target_env = "msvc", target_os = "illumos")))]
-use tikv_jemallocator::Jemalloc;
-
-#[cfg(not(any(target_env = "msvc", target_os = "illumos")))]
-#[global_allocator]
-static GLOBAL: Jemalloc = Jemalloc;
-
 use std::path::Path;
 use std::{env, thread};
 
