diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 27e4680..ab4c404 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -418,7 +418,7 @@ if (USE_ICU_REGEXP)
 endif ()
 
 if (NOT WIN32)
-  add_definitions ("-Wall -Werror")
+  add_definitions ("-Wall")
 endif ()
 
 include_directories ("src")
diff --git a/tools/cpp/CMakeLists.txt b/tools/cpp/CMakeLists.txt
index 314d91c..1ded960 100644
--- a/tools/cpp/CMakeLists.txt
+++ b/tools/cpp/CMakeLists.txt
@@ -25,7 +25,7 @@ set (
 )
 
 if (NOT WIN32)
-  add_definitions ("-Wall -Werror")
+  add_definitions ("-Wall")
 endif ()
 
 include_directories ("src")
