From b8c7ef336ef423c97508b5590f299657ecbe9d50 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Tue, 2 May 2023 16:08:06 +0200
Subject: [PATCH 08/27] compiler-rt: default to libc++ for sanitizers

---
 compiler-rt/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler-rt/CMakeLists.txt b/compiler-rt/CMakeLists.txt
index 8a13508fc..3cd47c8e4 100644
--- a/compiler-rt/CMakeLists.txt
+++ b/compiler-rt/CMakeLists.txt
@@ -224,7 +224,7 @@ macro(handle_default_cxx_lib var)
       set(${var}_LIBNAME "libc++")
       set(${var}_INTREE 1)
     else()
-      set(${var}_LIBNAME "libstdc++")
+      set(${var}_LIBNAME "libc++")
       set(${var}_SYSTEM 1)
     endif()
   else()
-- 
2.39.0

