commit 44f6b4be8052a5cfd8302a87683a83355cd622ba
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Tue Mar 7 05:04:19 2023 +0100

    fix sed gnuism breaking exported symbols from networkmanager

diff --git a/tools/create-exports-NetworkManager.sh b/tools/create-exports-NetworkManager.sh
index 21ba369..f1535ae 100755
--- a/tools/create-exports-NetworkManager.sh
+++ b/tools/create-exports-NetworkManager.sh
@@ -71,7 +71,7 @@ get_symbols_missing() {
 }
 
 pretty() {
-    sed 's/.*/\t\0;/'
+    sed 's/\(.*\)/\t\1;/'
 }
 
 do_build() {
