commit 3c2a25e8f2b0a9ad61b3d6e12602d7442cb2f024
Author: q66 <q66@chimera-linux.org>
Date:   Sun Jan 23 06:23:11 2022 +0100

    drop some gcc-related lto junk

diff --git a/meson.build b/meson.build
index 1a7dadf..0f9a6d0 100644
--- a/meson.build
+++ b/meson.build
@@ -173,7 +173,7 @@
 endif
 
 enable_lto = get_option('b_lto')
-if enable_lto
+if false
   # meson already adds '-flto'
   lto_flag = '-flto-partition=none'
   assert(cc.has_argument(lto_flag), '-flto-partition=none not supported. Disable link-time optimization with -Db_lto=false.')
