commit 5cb1f50717e777b21463462c5bebf1d681265d75
Author: q66 <q66@chimera-linux.org>
Date:   Tue Dec 5 01:21:54 2023 +0100

    less crappy big endian check

diff --git a/configure b/configure
index 3cd3bdf..9d91611 100755
--- a/configure
+++ b/configure
@@ -5917,11 +5917,7 @@ done
 
 check_cc pragma_deprecated "" '_Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")'
 
-# The global variable ensures the bits appear unchanged in the object file.
-test_cc <<EOF || die "endian test failed"
-unsigned int endian = 'B' << 24 | 'I' << 16 | 'G' << 8 | 'E';
-EOF
-od -t x1 $TMPO | grep -q '42 *49 *47 *45' && enable bigendian
+test_cpp_condition stdlib.h "defined(__BIG_ENDIAN__)" && enable bigendian
 
 check_cc const_nan math.h "struct { double d; } static const bar[] = { { NAN } }"
 
