commit 0a96729183a001e55fc3ba4d7a5cfa48cdd6ceb9
Author: q66 <q66@chimera-linux.org>
Date:   Fri Mar 17 01:37:03 2023 +0000

    disable riscv asm using unratified exts

diff --git a/libavcodec/bswapdsp.c b/libavcodec/bswapdsp.c
index f0ea2b5..ef74c92 100644
--- a/libavcodec/bswapdsp.c
+++ b/libavcodec/bswapdsp.c
@@ -51,7 +51,7 @@ av_cold void ff_bswapdsp_init(BswapDSPContext *c)
     c->bswap_buf   = bswap_buf;
     c->bswap16_buf = bswap16_buf;
 
-#if ARCH_RISCV
+#if 0
     ff_bswapdsp_init_riscv(c);
 #elif ARCH_X86
     ff_bswapdsp_init_x86(c);
diff --git a/libavcodec/riscv/Makefile b/libavcodec/riscv/Makefile
index 965942f..85a3b04 100644
--- a/libavcodec/riscv/Makefile
+++ b/libavcodec/riscv/Makefile
@@ -5,8 +5,6 @@ RVV-OBJS-$(CONFIG_ALAC_DECODER) += riscv/alacdsp_rvv.o
 OBJS-$(CONFIG_AUDIODSP) += riscv/audiodsp_init.o \
                            riscv/audiodsp_rvf.o
 RVV-OBJS-$(CONFIG_AUDIODSP) += riscv/audiodsp_rvv.o
-OBJS-$(CONFIG_BSWAPDSP) += riscv/bswapdsp_init.o \
-                           riscv/bswapdsp_rvb.o
 RVV-OBJS-$(CONFIG_BSWAPDSP) += riscv/bswapdsp_rvv.o
 OBJS-$(CONFIG_FMTCONVERT) += riscv/fmtconvert_init.o
 RVV-OBJS-$(CONFIG_FMTCONVERT) += riscv/fmtconvert_rvv.o
