--- a/src/pdf/XPDFRenderer.cpp
+++ b/src/pdf/XPDFRenderer.cpp
@@ -179,7 +179,7 @@ QString XPDFRenderer::title() const
             if (title.isString())
             {
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 72
-                return QString(title.getString()->c_str());
+                return QString(title.getString().c_str());
 #else
                 return QString(title.getString()->getCString());
 #endif
