commit 1257394a7583e59d0cf9534aea0adcc739a75f51
Author: q66 <q66@chimera-linux.org>
Date:   Sat Dec 2 12:16:36 2023 +0100

    fix plugin install script

diff --git a/scripts/install_plugins.sh b/scripts/install_plugins.sh
index 591514a..c54f8d0 100755
--- a/scripts/install_plugins.sh
+++ b/scripts/install_plugins.sh
@@ -6,8 +6,8 @@ if [ -e $HARDCODED_PLUGIN_DEST_2 ]; then rm -r $HARDCODED_PLUGIN_DEST_2; fi;
 
 
 mkdir -p $DEST_DIR/usr/lib/lv2
-cp -r bin/lv2/g++/valve_deck_speakers.lv2 $DEST_DIR/usr/lib/lv2/valve_deck_speakers.lv2
-cp -r bin/lv2/g++/valve_binaural.lv2 $DEST_DIR/usr/lib/lv2/valve_binaural.lv2
-cp -r bin/lv2/g++/valve_deck_microphone.lv2 $DEST_DIR/usr/lib/lv2/valve_deck_microphone.dsp
-cp -r bin/svg $DEST_DIR/usr/lib/lv2/svg
+cp -R bin/lv2/clang++/valve_deck_speakers.lv2 $DEST_DIR/usr/lib/lv2/valve_deck_speakers.lv2
+cp -R bin/lv2/clang++/valve_binaural.lv2 $DEST_DIR/usr/lib/lv2/valve_binaural.lv2
+cp -R bin/lv2/clang++/valve_deck_microphone.lv2 $DEST_DIR/usr/lib/lv2/valve_deck_microphone.dsp
+cp -R bin/svg $DEST_DIR/usr/lib/lv2/svg
 cp version.txt $DEST_DIR/usr/lib/lv2/valve_audio_dsp_version.txt
