--- a/src/bin/system/e_system_power.c
+++ b/src/bin/system/e_system_power.c
@@ -32,6 +32,9 @@ _cb_power_hibernate(void *data EINA_UNUSED, const char *params EINA_UNUSED)
 static void
 _power_halt_init(void)
 {
+#if 1
+   _cmd_halt = strdup("loginctl poweroff");
+#else
 #if defined (__FreeBSD__) || defined (__OpenBSD__)
    _cmd_halt = strdup("shutdown -p now");
 #else
@@ -43,11 +46,15 @@ _power_halt_init(void)
    // linux systemd: PATH/systemctl poweroff
    //           bsd: /sbin/shutdown -p no
    // *            : /sbin/shutdown -h now
+#endif
 }
 
 static void
 _power_reboot_init(void)
 {
+#if 1
+   _cmd_reboot = strdup("loginctl reboot");
+#else
 #if defined (__FreeBSD__) || defined (__OpenBSD__)
    _cmd_reboot = strdup("shutdown -r now");
 #else
@@ -58,11 +65,15 @@ _power_reboot_init(void)
 #endif
    // linux systemd: PATH/systemctl reboot
    //             *: /sbin/shutdown -r now
+#endif
 }
 
 static void
 _power_suspend_init(void)
 {
+#if 1
+   _cmd_suspend = strdup("loginctl suspend");
+#else
 #if defined (__FreeBSD__) || defined (__OpenBSD__)
    if (ecore_file_app_installed("zzz"))
      _cmd_suspend = strdup("zzz");
@@ -85,11 +96,15 @@ _power_suspend_init(void)
    //    /etc/acpi/sleep.sh force
    //    PATH/pm-suspend
    //    /etc/acpi/pm-suspend
+#endif
 }
 
 static void
 _power_hibernate_init(void)
 {
+#if 1
+   _cmd_hibernate = strdup("loginctl hibernate");
+#else
 #if defined (__FreeBSD__)
    if (ecore_file_app_installed("acpiconf"))
      _cmd_hibernate = strdup("acpiconf -s4");
@@ -112,6 +127,7 @@ _power_hibernate_init(void)
    //    /etc/acpi/hibernate.sh force
    //    PATH/pm-hibernate
    //    /etc/acpi/pm-hibernate
+#endif
 }
 
 static void
