diff --git a/configure.ac b/configure.ac
index 2f1c525..6911283 100644
--- a/configure.ac
+++ b/configure.ac
@@ -235,10 +235,7 @@ AS_IF([test "x$enable_coverage" = "xyes"], [
 ])
 AM_CONDITIONAL([ENABLE_COVERAGE], [test "x$enable_coverage" = "xyes"])
 
-m4_ifdef([GTK_DOC_CHECK], [
-GTK_DOC_CHECK([1.14],[--flavour no-tmpl-flat])
-], [
-AM_CONDITIONAL([ENABLE_GTK_DOC], false)])
+AM_CONDITIONAL([ENABLE_GTK_DOC], false)
 
 PKG_INSTALLDIR
 PKG_NOARCH_INSTALLDIR
diff --git a/libkmod/docs/Makefile.am b/libkmod/docs/Makefile.am
index c4f3d69..5c686a4 100644
--- a/libkmod/docs/Makefile.am
+++ b/libkmod/docs/Makefile.am
@@ -25,10 +25,4 @@ IGNORE_HFILES = libkmod-internal.h \
 
 content_files = version.xml
 
-EXTRA_DIST =
-
-if ENABLE_GTK_DOC
-include $(top_srcdir)/libkmod/docs/gtk-doc.make
-else
-EXTRA_DIST += libkmod-docs.xml libkmod-sections.txt
-endif
+EXTRA_DIST = libkmod-docs.xml libkmod-sections.txt
