--- a/player.c	2022-07-05 22:20:28.000000000 +0200
+++ b/player.c	2024-01-13 02:42:55.219834610 +0100
@@ -829,7 +829,7 @@
 		ip = ip_new(ti->filename);
 		_producer_status_update(PS_STOPPED);
 		/* PS_STOPPED, CS_PLAYING */
-		if (player_cont && (player_cont_album == 1 || strcmp(player_info_priv.ti->album,ti->album) == 0)) {
+		if (player_cont && (player_cont_album == 1 || (player_info_priv.ti->album && ti->album && strcmp(player_info_priv.ti->album,ti->album) == 0))) {
 			_producer_play();
 			if (producer_status == PS_UNLOADED) {
 				_consumer_stop();
