commit e5e5a1ee1a659c60664d8099ed21be2574d73da1
Author: q66 <q66@chimera-linux.org>
Date:   Thu Apr 6 23:07:41 2023 +0000

    bsd date fixups

diff --git a/make/InitSupport.gmk b/make/InitSupport.gmk
index d2291c5..fc3211c 100644
--- a/make/InitSupport.gmk
+++ b/make/InitSupport.gmk
@@ -319,7 +319,7 @@ else # $(HAS_SPEC)=true
                                                        +"%Y-%m-%dT%H:%M:%SZ" 2> /dev/null)
     else
       export SOURCE_DATE_ISO_8601 := $$(shell $$(DATE) -u \
-                                                       -j -f "%s" "$$(SOURCE_DATE_EPOCH)" \
+                                                       -j -r "$$(SOURCE_DATE_EPOCH)" \
                                                        +"%Y-%m-%dT%H:%M:%SZ" 2> /dev/null)
     endif
   endef
diff --git a/make/autoconf/jdk-options.m4 b/make/autoconf/jdk-options.m4
index dea69dc..fab5f16 100644
--- a/make/autoconf/jdk-options.m4
+++ b/make/autoconf/jdk-options.m4
@@ -220,7 +220,7 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_JDK_OPTIONS],
     if test "x$IS_GNU_DATE" = xyes; then
       COPYRIGHT_YEAR=`date --date=@$SOURCE_DATE_EPOCH +%Y`
     else
-      COPYRIGHT_YEAR=`date -j -f %s $SOURCE_DATE_EPOCH +%Y`
+      COPYRIGHT_YEAR=`date -j -r $SOURCE_DATE_EPOCH +%Y`
     fi
   else
     COPYRIGHT_YEAR=`$DATE +'%Y'`
diff --git a/make/autoconf/util.m4 b/make/autoconf/util.m4
index 0e1f033..7d93bc8 100644
--- a/make/autoconf/util.m4
+++ b/make/autoconf/util.m4
@@ -246,10 +246,10 @@ AC_DEFUN([UTIL_GET_EPOCH_TIMESTAMP],
     timestamp=$($DATE --utc --date=$2 +"%s" 2> /dev/null)
   else
     # BSD date
-    timestamp=$($DATE -u -j -f "%F %T" "$2" "+%s" 2> /dev/null)
+    timestamp=$($DATE -u -j -f "%Y-%m-%d %T" "$2" "+%s" 2> /dev/null)
     if test "x$timestamp" = x; then
       # Perhaps the time was missing
-      timestamp=$($DATE -u -j -f "%F %T" "$2 00:00:00" "+%s" 2> /dev/null)
+      timestamp=$($DATE -u -j -f "%Y-%m-%d %T" "$2 00:00:00" "+%s" 2> /dev/null)
       # If this did not work, we give up and return the empty string
     fi
   fi
