--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@ DESTDIR ?=
 PREFIX ?= /usr/local
 BINDIR ?= bin
 
-MANDIR ?= $(PREFIX)/share/man
+MANDIR ?= $(DESTDIR)$(PREFIX)/share/man
 
 # implemented backends: wayland x11 gdi
 backends ?= wayland x11
@@ -25,8 +25,8 @@ with     ?= libconfig
 
 IS_CLANG = $(shell $(CC) -v 2>&1 | grep -q clang && echo true)
 ifeq ($(IS_CLANG),true)
-	CFLAGS += -Wno-gnu-zero-variadic-macro-arguments
-	CFLAGS += -Wno-empty-translation-unit
+	override CFLAGS += -Wno-gnu-zero-variadic-macro-arguments
+	override CFLAGS += -Wno-empty-translation-unit
 endif
 
 # Echo function
@@ -43,38 +43,38 @@ endif
 <<addon-srcs>> = src/config.c
 
 ifeq ($(filter x11,$(<<backends>>)),x11)
-	PKGS += x11 xfixes xinerama xrandr xext x11
-	CFLAGS += -DX11
+	override PKGS += x11 xfixes xinerama xrandr xext x11
+	override CFLAGS += -DX11
 endif
 ifeq ($(filter wayland,$(<<backends>>)),wayland)
-	PKGS += wayland-client
-	CFLAGS += -DWAYLAND
-	LDFLAGS += -lrt
+	override PKGS += wayland-client
+	override CFLAGS += -DWAYLAND
+	override LDFLAGS += -lrt
 endif
 ifneq ($(filter wayland x11,$(<<backends>>)),)
-	PKGS += cairo pango pangocairo
-	CFLAGS += -DCOLOR_HELP -DCAIRO
+	override PKGS += cairo pango pangocairo
+	override CFLAGS += -DCOLOR_HELP -DCAIRO
 endif
 ifeq ($(filter gdi,$(<<backends>>)),gdi)
 # Current toolchain architecture variable from MSYS2 project
 	ifeq ($(MSYSTEM_CARCH),i686)
-		CFLAGS += -m32
+		override CFLAGS += -m32
 	endif
-	CFLAGS += -DGDI
-	LDFLAGS += -lgdi32
+	override CFLAGS += -DGDI
+	override LDFLAGS += -lgdi32
 endif
 # rebuild on optional deps change
 ifneq ($(filter libconfig,$(<<with>>)),)
 	ifneq ($(shell pkg-config --exists libconfig && echo exists),)
-		PKGS += libconfig
-		CFLAGS += -DLIBCONFIG
+		override PKGS += libconfig
+		override CFLAGS += -DLIBCONFIG
 		<<addons>> += libconfig
 	endif
 endif
 
 ifneq ($(PKGS),)
-	CFLAGS += $(shell pkg-config --cflags $(PKGS))
-	LDFLAGS += $(shell pkg-config --libs $(PKGS))
+	override CFLAGS += $(shell pkg-config --cflags $(PKGS))
+	override LDFLAGS += $(shell pkg-config --libs $(PKGS))
 endif
 
 <<sources>> := \
