--- a/Makefile
+++ b/Makefile
@@ -65,7 +65,7 @@ ifeq ($(filter gdi,$(<<backends>>)),gdi)
 endif
 # rebuild on optional deps change
 ifneq ($(filter libconfig,$(<<with>>)),)
-	ifneq ($(shell pkg-config --exists libconfig && echo exists),)
+	ifneq ($(shell $(PKG_CONFIG) --exists libconfig && echo exists),)
 		override PKGS += libconfig
 		override CFLAGS += -DLIBCONFIG
 		<<addons>> += libconfig
@@ -73,8 +73,8 @@ ifneq ($(filter libconfig,$(<<with>>)),)
 endif
 
 ifneq ($(PKGS),)
-	override CFLAGS += $(shell pkg-config --cflags $(PKGS))
-	override LDFLAGS += $(shell pkg-config --libs $(PKGS))
+	override CFLAGS += $(shell $(PKG_CONFIG) --cflags $(PKGS))
+	override LDFLAGS += $(shell $(PKG_CONFIG) --libs $(PKGS))
 endif
 
 <<sources>> := \
--- a/src/wayland/xdg-shell.cgen
+++ b/src/wayland/xdg-shell.cgen
@@ -1,5 +1,5 @@
 #!/bin/sh
-PROTOCOLS=$(pkg-config --variable=pkgdatadir wayland-protocols)
+PROTOCOLS=$(${PKG_CONFIG:-pkg-config} --variable=pkgdatadir wayland-protocols)
 
 wayland-scanner private-code \
 	"$PROTOCOLS/stable/xdg-shell/xdg-shell.xml" \
--- a/src/x11/XEventTypes.cgen
+++ b/src/x11/XEventTypes.cgen
@@ -4,7 +4,7 @@ cat << EOF
 const char *XEventName(int type) {
 	switch(type) {
 EOF
-DIR="$(pkg-config x11 --variable=includedir)/X11"
+DIR="$(${PKG_CONFIG:-pkg-config} x11 --variable=includedir)/X11"
 if [ -f "$DIR/X.h" ]; then
 	gawk '
 		/Event names\./,/must be bigger than any event/ {
