diff --git a/Makefile b/Makefile
index adc7c84..521df71 100644
--- a/Makefile
+++ b/Makefile
@@ -17,11 +17,11 @@ backends ?= wayland x11
 
 IS_CLANG = $(shell $(CC) -v 2>&1 | grep -q clang && echo true)
 ifeq ($(IS_CLANG),true)
-	CFLAGS += -Wno-gnu-zero-variadic-macro-arguments
-	CFLAGS += -Wno-empty-translation-unit
+	override CFLAGS += -Wno-gnu-zero-variadic-macro-arguments
+	override CFLAGS += -Wno-empty-translation-unit
 endif
 
-CFLAGS += -Isrc
+override CFLAGS += -Isrc
 
 # Echo function
 << := @echo
@@ -32,35 +32,35 @@ endif
 # Mess with backends
 <<backends>> = $(sort $(filter x11 wayland gdi,$(backends)))
 ifeq ($(filter x11,$(<<backends>>)),x11)
-	PKGS += x11 xfixes xinerama xrandr
-	CFLAGS += -DX11
+	override PKGS += x11 xfixes xinerama xrandr
+	override CFLAGS += -DX11
 endif
 ifeq ($(filter wayland,$(<<backends>>)),wayland)
-	PKGS += wayland-client
-	CFLAGS += -DWAYLAND
-	LDFLAGS += -lrt
+	override PKGS += wayland-client
+	override CFLAGS += -DWAYLAND
+	override LDFLAGS += -lrt
 endif
 ifneq ($(filter wayland x11,$(<<backends>>)),)
-	PKGS += cairo
-	CFLAGS += -DCOLOR_HELP -DCAIRO
+	override PKGS += cairo
+	override CFLAGS += -DCOLOR_HELP -DCAIRO
 endif
 ifeq ($(filter gdi,$(<<backends>>)),gdi)
 # Current toolchain architecture variable from MSYS2 project
 	ifeq ($(MSYSTEM_CARCH),i686)
-		CFLAGS += -m32
+		override CFLAGS += -m32
 	endif
-	CFLAGS += -DGDI
-	LDFLAGS += -lgdi32
+	override CFLAGS += -DGDI
+	override LDFLAGS += -lgdi32
 endif
 
 ifeq ($(shell pkg-config --exists libconfig && echo exists),exists)
-	PKGS += libconfig
-	CFLAGS += -DLIBCONFIG
+	override PKGS += libconfig
+	override CFLAGS += -DLIBCONFIG
 endif
 
 ifneq ($(PKGS),)
-	CFLAGS += $(shell pkg-config --cflags $(PKGS))
-	LDFLAGS += $(shell pkg-config --libs $(PKGS))
+	override CFLAGS += $(shell pkg-config --cflags $(PKGS))
+	override LDFLAGS += $(shell pkg-config --libs $(PKGS))
 endif
 
 <<sources>> := \
