--- a/meson.build
+++ b/meson.build
@@ -1846,9 +1846,7 @@ if conf.get('BPF_FRAMEWORK') == 1
                 endif
                 target_triplet_cmd = run_command(cc.cmd_array(), '-print-multiarch', check: false)
         else
-                # clang does not support -print-multiarch (D133170) and its -dump-machine
-                # does not match multiarch. Query gcc instead.
-                target_triplet_cmd = run_command('gcc', '-print-multiarch', check: false)
+                target_triplet_cmd = run_command('echo', check: false)
         endif
         if target_triplet_cmd.returncode() == 0
                 sysroot = meson.get_external_property('sys_root', '/')
