diff --git a/Makefile.am b/Makefile.am
index 8ac71a8..4cd8f1d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -663,10 +663,3 @@ endif
 
 
 check: prepare-check
-
-if !HAVE_PANDOC
-# If pandoc is not enabled, we want to complain that you need pandoc for make dist,
-# so hook the target and complain.
-	@(>&2 echo "You do not have pandoc, a requirement for the distribution of manpages")
-	@exit 1
-endif
