commit 4c1966da815bb8c72855280be30484ac6d1b109b
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Tue Apr 5 04:49:23 2022 +0200

    use elogind for systemd support

diff --git a/configure b/configure
index ba5a7ae..8d0179f 100755
--- a/configure
+++ b/configure
@@ -22630,7 +22630,7 @@ printf "%s\n" "#define WITH_VALGRIND 1" >>confdefs.h
 fi
 
 #### Set up final flags
-LIBDBUS_LIBS="$THREAD_LIBS $NETWORK_libs $SYSTEMD_LIBS"
+LIBDBUS_LIBS="$THREAD_LIBS $NETWORK_libs $SYSTEMD_LIBS -lelogind"
 
 
 ### X11 detection
diff --git a/dbus/dbus-spawn-unix.c b/dbus/dbus-spawn-unix.c
index f3577e7..3f8b290 100644
--- a/dbus/dbus-spawn-unix.c
+++ b/dbus/dbus-spawn-unix.c
@@ -34,6 +34,8 @@
 #include "dbus-test.h"
 #include "dbus-protocol.h"
 
+#define HAVE_SYSTEMD 1
+
 #include <unistd.h>
 #include <fcntl.h>
 #include <signal.h>
diff --git a/dbus/dbus-sysdeps-unix.c b/dbus/dbus-sysdeps-unix.c
index 8743589..fd1b178 100644
--- a/dbus/dbus-sysdeps-unix.c
+++ b/dbus/dbus-sysdeps-unix.c
@@ -59,6 +59,8 @@
 #include <grp.h>
 #include <arpa/inet.h>
 
+#define HAVE_SYSTEMD 1
+
 #ifdef HAVE_ERRNO_H
 #include <errno.h>
 #endif
diff --git a/dbus/dbus-sysdeps-util-unix.c b/dbus/dbus-sysdeps-util-unix.c
index 314ce64..f6b8137 100644
--- a/dbus/dbus-sysdeps-util-unix.c
+++ b/dbus/dbus-sysdeps-util-unix.c
@@ -52,6 +52,8 @@
 #include <dirent.h>
 #include <sys/un.h>
 
+#define HAVE_SYSTEMD 1
+
 #ifdef HAVE_SYS_PRCTL_H
 #include <sys/prctl.h>
 #endif
diff --git a/dbus/dbus-userdb-util.c b/dbus/dbus-userdb-util.c
index 1ca21eb..6df2891 100644
--- a/dbus/dbus-userdb-util.c
+++ b/dbus/dbus-userdb-util.c
@@ -30,6 +30,8 @@
 #include <dbus/dbus-test-tap.h>
 #include <string.h>
 
+#define HAVE_SYSTEMD 1
+
 /* It isn't obvious from its name, but this file is part of the Unix
  * system-dependent part of libdbus. */
 #if defined(DBUS_WIN) || !defined(DBUS_UNIX)
