--- a/configure
+++ b/configure
@@ -22407,7 +22407,7 @@ $as_echo "#define WITH_VALGRIND 1" >>confdefs.h
 fi
 
 #### Set up final flags
-LIBDBUS_LIBS="$THREAD_LIBS $NETWORK_libs $SYSTEMD_LIBS"
+LIBDBUS_LIBS="$THREAD_LIBS $NETWORK_libs $SYSTEMD_LIBS -lelogind"
 
 
 ### X11 detection
--- a/dbus/dbus-spawn.c
+++ b/dbus/dbus-spawn.c
@@ -24,6 +24,8 @@
 
 #include <config.h>
 
+#define HAVE_SYSTEMD 1
+
 #include "dbus-spawn.h"
 #include "dbus-sysdeps-unix.h"
 #include "dbus-internals.h"
--- a/dbus/dbus-sysdeps-unix.c
+++ b/dbus/dbus-sysdeps-unix.c
@@ -24,6 +24,8 @@
 
 #include <config.h>
 
+#define HAVE_SYSTEMD 1
+
 #include "dbus-internals.h"
 #include "dbus-sysdeps.h"
 #include "dbus-sysdeps-unix.h"
--- a/dbus/dbus-sysdeps-util-unix.c
+++ b/dbus/dbus-sysdeps-util-unix.c
@@ -34,6 +34,8 @@
 #include "dbus-userdb.h"
 #include "dbus-test.h"
 
+#define HAVE_SYSTEMD 1
+
 #include <sys/types.h>
 #include <stdlib.h>
 #include <string.h>
--- a/dbus/dbus-userdb-util.c
+++ b/dbus/dbus-userdb-util.c
@@ -29,6 +29,8 @@
 #include "dbus-protocol.h"
 #include <string.h>
 
+#define HAVE_SYSTEMD 1
+
 #ifdef HAVE_SYSTEMD
 #include <systemd/sd-login.h>
 #endif
