From b120ec8101ff44eae1c381a3141c7c6325b96d27 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Tue, 2 Nov 2021 23:52:18 +0100
Subject: [PATCH] enable elogind support

---
 configure               | 2 +-
 dbus/dbus-userdb-util.c | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 38db1dd..8e2e9e6 100755
--- a/configure
+++ b/configure
@@ -22407,7 +22407,7 @@ $as_echo "#define WITH_VALGRIND 1" >>confdefs.h
 fi
 
 #### Set up final flags
-LIBDBUS_LIBS="$THREAD_LIBS $NETWORK_libs $SYSTEMD_LIBS"
+LIBDBUS_LIBS="$THREAD_LIBS $NETWORK_libs $SYSTEMD_LIBS -lelogind"
 
 
 ### X11 detection
diff --git a/dbus/dbus-userdb-util.c b/dbus/dbus-userdb-util.c
index 170d233..8238caf 100644
--- a/dbus/dbus-userdb-util.c
+++ b/dbus/dbus-userdb-util.c
@@ -29,6 +29,8 @@
 #include "dbus-protocol.h"
 #include <string.h>
 
+#define HAVE_SYSTEMD 1
+
 #ifdef HAVE_SYSTEMD
 #include <systemd/sd-login.h>
 #endif
-- 
2.33.0

