--- a/bus/activation-helper-bin.c
+++ b/bus/activation-helper-bin.c
@@ -33,6 +33,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <err.h>
+#include <grp.h>
+#include <pwd.h>
 
 static int
 convert_error_to_exit_code (DBusError *error)
@@ -82,6 +86,28 @@ main (int argc, char **argv)
   /* default is all okay */
   retval = 0;
 
+  /* get the group structure */
+  errno = ENOENT;
+  struct group *gr = getgrnam("dbus");
+  if (!gr) err(1, "could not get dbus group");
+
+  /* get the calling user name */
+  errno = ENOENT;
+  struct passwd *pwd = getpwuid(getuid());
+  if (!pwd) err(1, "could not get passwd");
+
+  /* check if it's in there */
+  int found = !strcmp(pwd->pw_name, "dbus");
+  for (char **grps = gr->gr_mem; *grps; ++grps)
+    {
+      if (!strcmp(*grps, pwd->pw_name))
+        {
+          found = 1;
+          break;
+        }
+    }
+  if (!found) errx(1, "user '%s' not allowed", pwd->pw_name);
+
   /* have we used a help option or not specified the correct arguments? */
   if (argc != 2 ||
       strcmp (argv[1], "--help") == 0 ||
