--- a/meson.build
+++ b/meson.build
@@ -9,7 +9,7 @@ with_lynx = get_option('lynx')
 cpp = meson.get_compiler('cpp')
 
 gtkmm_dep = dependency('gtkmm-4.0', version : '>= 4.0', required : true)
-sigcpp_dep = dependency('sigc++-2.0', required : true)
+sigcpp_dep = dependency('sigc++-3.0', required : true)
 canberragtk_dep = dependency('libcanberra', version : '>= 0.16', required : true)
 
 libpulse_dep = dependency('libpulse', version : '>= 5.0', required : true)
