diff --git a/libtaskmanager/autotests/waylandtasksmodeltest.cpp b/libtaskmanager/autotests/waylandtasksmodeltest.cpp
index ee5050abc7..942e23cea7 100644
--- a/libtaskmanager/autotests/waylandtasksmodeltest.cpp
+++ b/libtaskmanager/autotests/waylandtasksmodeltest.cpp
@@ -6,6 +6,8 @@
 
 #define QT_FORCE_ASSERTS 1
 
+#include <unistd.h>
+
 #include <QProcess>
 #include <qpa/qplatformnativeinterface.h>
 
