--- a/src/lib/evas/canvas/evas_object_textblock.c
+++ b/src/lib/evas/canvas/evas_object_textblock.c
@@ -976,7 +976,7 @@ struct _Escape_Value
    size_t value_len;
 };
 
-#define ESCAPE_VALUE(e,v) {e,v,strlen(e),strlen(v)}
+#define ESCAPE_VALUE(e,v) {e,v,sizeof(e) - 1,sizeof(v) - 1}
 
 /**
  * @internal
--- a/src/lib/evas/common/evas_text_utils.c
+++ b/src/lib/evas/common/evas_text_utils.c
@@ -19,7 +19,7 @@ struct _Color_Name_Value
    unsigned char b;
 };
 
-#define COLOR_NAME(name,r,g,b) {name,(unsigned char)strlen(name),r,g,b}
+#define COLOR_NAME(name,r,g,b) {name,(unsigned char)sizeof(name) - 1,r,g,b}
 
 /**
  * @internal
