From 5f5a0158fdacf8495fce69bbbda023a735ad9611 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Thu, 24 Jun 2021 04:36:16 +0200
Subject: [PATCH] fix build with clang

---
 src/chartype.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/chartype.h b/src/chartype.h
index bfa3d54..5c98fca 100644
--- a/chartype.h
+++ b/chartype.h
@@ -38,7 +38,8 @@
 	!(defined(__APPLE__) && defined(__MACH__)) && \
 	!defined(__OpenBSD__) && \
 	!defined(__FreeBSD__) && \
-	!defined(__DragonFly__)
+	!defined(__DragonFly__) && \
+	!defined(__clang__)
 #ifndef __STDC_ISO_10646__
 /* In many places it is assumed that the first 127 code points are ASCII
  * compatible, so ensure wchar_t indeed does ISO 10646 and not some other
-- 
2.32.0

