revert unstable api crap

--- a/src/main.c
+++ b/src/main.c
@@ -477,7 +477,6 @@ static void init_tox_options(const Run_Options *run_opts, Init_Queue *init_q, st
     tox_options_set_tcp_port(tox_opts, run_opts->tcp_port);
     tox_options_set_local_discovery_enabled(tox_opts, !run_opts->disable_local_discovery);
     tox_options_set_experimental_groups_persistence(tox_opts, true);
-    tox_options_set_experimental_disable_dns(tox_opts, false);
 
     if (run_opts->logging) {
         tox_options_set_log_callback(tox_opts, cb_toxcore_logger);
@@ -495,9 +494,7 @@ static void init_tox_options(const Run_Options *run_opts, Init_Queue *init_q, st
         init_queue_add(init_q, "TCP relaying enabled on port %d", tox_options_get_tcp_port(tox_opts));
     }
 
-    const bool proxy_set = tox_options_get_proxy_type(tox_opts) != TOX_PROXY_TYPE_NONE;
-
-    if (proxy_set) {
+    if (tox_options_get_proxy_type(tox_opts) != TOX_PROXY_TYPE_NONE) {
         tox_options_set_proxy_port(tox_opts, run_opts->proxy_port);
         tox_options_set_proxy_host(tox_opts, run_opts->proxy_address);
         const char *ps = tox_options_get_proxy_type(tox_opts) == TOX_PROXY_TYPE_SOCKS5 ? "SOCKS5" : "HTTP";
@@ -508,12 +505,8 @@ static void init_tox_options(const Run_Options *run_opts, Init_Queue *init_q, st
     }
 
     if (!tox_options_get_udp_enabled(tox_opts)) {
-        if (proxy_set) {
-            tox_options_set_experimental_disable_dns(tox_opts, true);
-        }
-
         init_queue_add(init_q, "UDP disabled");
-    } else if (proxy_set) {
+    } else if (tox_options_get_proxy_type(tox_opts) != TOX_PROXY_TYPE_NONE) {
         const char *msg = "WARNING: Using a proxy without disabling UDP may leak your real IP address.";
         init_queue_add(init_q, "%s", msg);
         msg = "Use the -t option to disable UDP.";
@@ -844,7 +837,6 @@ _Noreturn static void *thread_av(void *data)
         sleep_thread(sleep_duration);
     }
 }
-
 #endif /* AUDIO */
 
 static void print_usage(void)
