--- a/typogrify/packages/titlecase/__init__.py
+++ b/typogrify/packages/titlecase/__init__.py
@@ -15,7 +15,7 @@ import re
 __all__ = ['titlecase']
 __version__ = '0.5.1'
 
-SMALL = 'a|an|and|as|at|but|by|en|for|if|in|of|on|or|the|to|v\.?|via|vs\.?'
+SMALL = r'a|an|and|as|at|but|by|en|for|if|in|of|on|or|the|to|v\.?|via|vs\.?'
 PUNCT = r"""!"#$%&'‘()*+,\-./:;?@[\\\]_`{|}~"""
 
 SMALL_WORDS = re.compile(r'^(%s)$' % SMALL, re.I)
