--- a/configure.ac
+++ b/configure.ac
@@ -22,9 +22,9 @@
 # automake 1.12 seems to require this, but automake 1.11 doesn't recognize it
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 
-# Protobuf 3.6+ requires C++11.
+# Current protobuf needs C++17
 AS_IF([pkg-config --atleast-version 3.6.0 protobuf],
-  [AX_CXX_COMPILE_STDCXX([11])])
+  [AX_CXX_COMPILE_STDCXX([17])])
 
 WARNING_CXXFLAGS=""
 PICKY_CXXFLAGS=""
