commit 24d5520e67b29aa54275caa963f904aa201ed3c9
Author: q66 <q66@chimera-linux.org>
Date:   Wed Aug 30 03:09:18 2023 +0200

    patch out build-time xserver dependency

diff --git a/meson.build b/meson.build
index a358354..ecd4ffd 100644
--- a/meson.build
+++ b/meson.build
@@ -74,8 +74,7 @@ x_deps = declare_dependency(
   ],
 )
 # Xserver 1.17 & later default to -nolisten and require -listen for remote access
-xserver_deps = dependency('xorg-server', version : '>=1.17', required : false)
-xserver_has_listen = xserver_deps.found()
+xserver_has_listen = true
 find_x_server_script = find_program('build-aux/find-x-server.sh', native: true)
 find_x_server_out = run_command(find_x_server_script).stdout().strip()
 if find_x_server_out != ''
