commit f8a34c75b770a2e5b4f5cae7122082130f7e14bd
Author: q66 <q66@chimera-linux.org>
Date:   Sat Nov 12 19:17:46 2022 +0100

    fix up pam for chimera

diff --git a/data/pam-arch/gdm-launch-environment.pam b/data/pam-arch/gdm-launch-environment.pam
index 9d05e516d..ea6c6a05f 100644
--- a/data/pam-arch/gdm-launch-environment.pam
+++ b/data/pam-arch/gdm-launch-environment.pam
@@ -1,18 +1,19 @@
 #%PAM-1.0
 
-auth       required                    pam_succeed_if.so    audit quiet_success user in gdm:gnome-initial-setup
+auth       required                    pam_succeed_if.so    audit quiet_success user = _gdm
 auth       optional                    pam_permit.so
 auth       required                    pam_env.so
 
-account    required                    pam_succeed_if.so    audit quiet_success user in gdm:gnome-initial-setup
+account    required                    pam_succeed_if.so    audit quiet_success user = _gdm
 account    optional                    pam_permit.so
 
 password   required                    pam_deny.so
 
 session    optional                    pam_loginuid.so
 session    optional                    pam_keyinit.so       force revoke
-session    required                    pam_succeed_if.so    audit quiet_success user in gdm:gnome-initial-setup
+session    required                    pam_succeed_if.so    audit quiet_success user = _gdm
 session    optional                    pam_permit.so
--session   optional                    pam_systemd.so
+-session   optional                    pam_turnstile.so
+-session   optional                    pam_elogind.so
 -session   optional                    pam_elogind.so
 session    required                    pam_env.so
