commit 6a36f4816fde0822391193398f8ebead99f29120
Author: q66 <q66@chimera-linux.org>
Date:   Thu May 18 00:26:34 2023 +0200

    always libadd gslcblas
    
    Otherwise we fail tests.

diff --git a/configure b/configure
index b593652..e45d3ef 100755
--- a/configure.ac
+++ b/configure.ac
@@ -171,15 +171,11 @@
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 
-case $host in
-  *-*-cygwin* | *-*-mingw* )
   if test "$enable_shared" = yes; then
     GSLCBLAS_LDFLAGS="$GSLCBLAS_LDFLAGS -no-undefined"
     GSL_LDFLAGS="$GSL_LDFLAGS -no-undefined"
     GSL_LIBADD="cblas/libgslcblas.la"
   fi
-  ;;
-esac
 
 AC_SUBST(GSLCBLAS_LDFLAGS)
 AC_SUBST(GSL_LDFLAGS)
