https://github.com/schismtracker/schismtracker/issues/374

--- a/Makefile.am	2023-10-29 03:15:48.000000000 +0100
+++ b/Makefile.am	2024-01-20 09:43:14.531064935 +0100
@@ -189,23 +189,6 @@
 lib_win32=-lwinmm
 endif
 
-if HAVE_WINDRES
-
-## use today's date if we didn't get a commit date from git
-if HAVE_GIT
-wrcflags_version = -DWRC_VERSION=0,`echo '$(PACKAGE_VERSION)' | sed 's/\(....\)\(..\)\(..\).*/\1,\2,\3/'`
-else
-wrcflags_version = -DWRC_VERSION=0,`date +%Y%m%d | sed 's/\(....\)\(..\)\(..\).*/\1,\2,\3/'`
-endif
-
-## --use-temp-file is needed to work around stupid bugs
-WRCFLAGS = --use-temp-file -I. -I$(srcdir) $(cflags_version) $(wrcflags_version)
-.rc.$(OBJEXT):
-	$(WINDRES) $(WRCFLAGS) -i $< -o $@
-files_windres=sys/win32/schismres.rc
-sys/win32/schismres.$(OBJEXT): icons/schismres.ico config.h Makefile.am
-endif HAVE_WINDRES
-
 if USE_WII
 files_wii=sys/wii/isfs.c sys/wii/osdefs.c
 cflags_wii=-I$(srcdir)/sys/wii
