https://github.com/chimera-linux/cports/pull/1273#discussion_r1466266497
--- a/schism/version.c	2024-05-29 18:27:13.000000000 +0200
+++ b/schism/version.c	2024-05-30 14:04:34.232059657 +0200
@@ -49,11 +49,7 @@
 
 */
 static const char* top_banner_normal =
-#if !defined(EMPTY_VERSION)
-	"Schism Tracker " VERSION
-#else
-	"Schism Tracker built " __DATE__ " " __TIME__
-#endif
+	"Schism Tracker " VERSION2
 	;
 
 /*
