--- a/Makefile
+++ b/Makefile
@@ -53,17 +53,17 @@ R= 5.1.5
 all:	$(PLAT)
 
 $(PLATS) clean:
-	cd src && $(MAKE) $@ V=$(V) R=$(R)
+	cd src && $(MAKE) $@ V=$(V) R=$(R) && cd ..
 
 test:	dummy
 	src/lua test/hello.lua
 
 install: dummy
-	cd src && $(MKDIR) $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_MAN) $(INSTALL_LMOD) $(INSTALL_CMOD)
-	cd src && $(INSTALL_EXEC) $(TO_BIN) $(INSTALL_BIN)
-	cd src && $(INSTALL_DATA) $(TO_INC) $(INSTALL_INC)
-	cd src && $(INSTALL_DATA) $(TO_LIB) $(INSTALL_LIB)
-	cd doc && $(INSTALL_DATA) $(TO_MAN) $(INSTALL_MAN)
+	cd src && $(MKDIR) $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_MAN) $(INSTALL_LMOD) $(INSTALL_CMOD) && cd ..
+	cd src && $(INSTALL_EXEC) $(TO_BIN) $(INSTALL_BIN) && cd ..
+	cd src && $(INSTALL_DATA) $(TO_INC) $(INSTALL_INC) && cd ..
+	cd src && $(INSTALL_DATA) $(TO_LIB) $(INSTALL_LIB) && cd ..
+	cd doc && $(INSTALL_DATA) $(TO_MAN) $(INSTALL_MAN) && cd ..
 
 ranlib:
 	cd src && cd $(INSTALL_LIB) && $(RANLIB) $(TO_LIB)
