only LANG is read by musl
---
diff --git a/src/gettext.cpp b/src/gettext.cpp
index bb99e39bc..bdc046145 100644
--- a/src/gettext.cpp
+++ b/src/gettext.cpp
@@ -171,7 +171,7 @@ void init_gettext(const char *path, const std::string &configured_language,
 		// Set LANGUAGE which overrides all others, see
 		// <https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html>
 #ifndef _MSC_VER
-		setenv("LANGUAGE", configured_language.c_str(), 1);
+		setenv("LANG", configured_language.c_str(), 1);
 
 		// Reload locale with changed environment
 		setlocale(LC_ALL, "");
