--- a/src/provider/core/provider.cpp
+++ b/src/provider/core/provider.cpp
@@ -384,7 +384,7 @@
         const auto targetTime = lastEncouragementTime.addDays(encouragementInterval);
         timeToEncouragement = std::max(timeToEncouragement, (int)QDateTime::currentDateTime().secsTo(targetTime));
     }
-    encouragementTimer.start(timeToEncouragement * 1000);
+    encouragementTimer.start((unsigned)timeToEncouragement * 1000);
 }
 
 void ProviderPrivate::emitShowEncouragementMessage()
