From 4eb5f467dfb9d93223699171c4d43ee9a26e90fb Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Wed, 9 Oct 2024 13:19:36 +0200
Subject: [PATCH 3/3] implement graphical.monitor service for dinit

---
 backend/dinit | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/backend/dinit b/backend/dinit
index 713a253..7bacaac 100644
--- a/backend/dinit
+++ b/backend/dinit
@@ -51,6 +51,15 @@ case "$1" in
     stop)
         exec kill -s TERM "$2"
         ;;
+    graphical-notify)
+        if [ -z "$DINIT_CS_FD" ]; then
+            # must have a control socket
+            echo "dinit: control socket not given" >&2
+            exit 69
+        fi
+        # this is not invoked by turnstile, but by the monitor service
+        exec dinitctl trigger graphical.target
+        ;;
     *)
         exit 32
         ;;
@@ -112,6 +121,7 @@ type = internal
 depends-on = system
 waits-for.d = ${boot_dir}
 depends-on = login.target
+depends-ms = graphical.monitor
 depends-ms = graphical.target
 EOF
 
@@ -121,6 +131,14 @@ type = internal
 waits-for.d = ${system_boot_dir}
 EOF
 
+# monitor service to watch for environment changes
+cat << EOF > "${DINIT_DIR}/graphical.monitor"
+type = process
+depends-on = login.target
+options = pass-cs-fd
+command = /usr/bin/dinit-monitor -E -c "$0 graphical-notify" WAYLAND_DISPLAY DISPLAY
+EOF
+
 # this is needed for login to proceed
 cat << EOF > "${DINIT_DIR}/login.target"
 type = internal
@@ -129,6 +147,7 @@ EOF
 # this is not necessary to have started for login to proceed
 cat << EOF > "${DINIT_DIR}/graphical.target"
 type = triggered
+depends-on = graphical.monitor
 depends-on = login.target
 EOF
 
-- 
2.47.0

