diff --git a/src/install/dracut-install.c b/src/install/dracut-install.c
index efa6430..8c15c01 100644
--- a/src/install/dracut-install.c
+++ b/src/install/dracut-install.c
@@ -441,9 +441,9 @@ normal_copy:
 
         if (pid == 0) {
                 if (preservation)
-                        execlp("cp", "cp", "--reflink=auto", "--preserve=xattr", "-fLp", src, dst, NULL);
+                        execlp("cp", "cp", "-afL", src, dst, NULL);
                 else
-                        execlp("cp", "cp", "--reflink=auto", "-fLp", src, dst, NULL);
+                        execlp("cp", "cp", "-pfL", src, dst, NULL);
                 _exit(errno == ENOENT ? 127 : 126);
         }
 
@@ -456,9 +456,9 @@ normal_copy:
         ret = WIFSIGNALED(ret) ? 128 + WTERMSIG(ret) : WEXITSTATUS(ret);
         if (ret != 0) {
                 if (preservation)
-                        log_error("ERROR: 'cp --reflink=auto --preserve=xattr -fLp %s %s' failed with %d", src, dst, ret);
+                        log_error("ERROR: 'cp -afL %s %s' failed with %d", src, dst, ret);
                 else
-                        log_error("ERROR: 'cp --reflink=auto -fLp %s %s' failed with %d", src, dst, ret);
+                        log_error("ERROR: 'cp -pfL %s %s' failed with %d", src, dst, ret);
         }
         log_debug("cp ret = %d", ret);
         return ret;
