remove sponsored tips by default

diff -ruN a/ui/recent.c b/ui/recent.c
--- a/ui/recent.c	2026-07-15 19:43:00.000000000 +0200
+++ b/ui/recent.c	2026-07-24 11:32:04.792432412 +0200
@@ -1723,7 +1723,7 @@
     recent.gui_welcome_page_sidebar_learn_visible = true;
     recent.gui_welcome_page_sidebar_tips_visible = true;
     recent.gui_welcome_page_sidebar_tips_events = true;
-    recent.gui_welcome_page_sidebar_tips_sponsorship = true;
+    recent.gui_welcome_page_sidebar_tips_sponsorship = false;
     recent.gui_welcome_page_sidebar_tips_tips = true;
     recent.gui_welcome_page_sidebar_tips_auto_advance = true;
     recent.gui_welcome_page_sidebar_tips_interval = 8;
