From 2843bcc65d9b3fa461c4348ea7dc33e0d7a55cc6 Mon Sep 17 00:00:00 2001
From: psykose <alice@ayaya.dev>
Date: Thu, 24 Aug 2023 14:46:32 +0000
Subject: [PATCH] src/util: include sstream for stringstream

---
 src/util.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/util.cpp b/src/util.cpp
index 3e7200aeb..f648eff8b 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -20,6 +20,7 @@
 #include "util.hpp"
 
 #include <regex>
+#include <sstream>
 #include <utility>
 
 namespace util {
