doesn't exist in experimental, 16 has this under c++20 fine
--
--- a/include/util.hpp
+++ b/include/util.hpp
@@ -35,10 +35,6 @@
 #include <thread>
 #include <vector>
 
-#ifdef __clang__
-#include <experimental/source_location>
-#endif
-
 namespace util {
 
 // Minimum dB level reported here has to be used in gsettings and spinbuttons
@@ -48,11 +44,7 @@
 constexpr float minimum_linear_level = 0.00001F;
 constexpr double minimum_linear_d_level = 0.00001;
 
-#ifdef __clang__
-using source_location = std::experimental::source_location;
-#else
 using source_location = std::source_location;
-#endif
 
 void debug(const std::string& s, source_location location = source_location::current());
 void error(const std::string& s, source_location location = source_location::current());
