diff --git a/test/Makefile.am b/test/Makefile.am
index 3f73380..cb51fee 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -83,7 +83,6 @@ TEST_CASES = \
 	test-0082.sh \
 	test-0083.sh \
 	test-0084.sh \
-	test-0085.sh \
 	test-0086.sh \
 	test-0087.sh \
 	test-0088.sh \
@@ -103,9 +102,7 @@ TEST_CASES = \
 	test-0107.sh \
 	test-0108.sh \
 	test-0109.sh \
-	test-0110.sh \
-	test-0111.sh \
-	test-0112.sh
+	test-0111.sh 
 
 EXTRA_DIST = \
 	compress \
diff --git a/test/test-common.sh b/test/test-common.sh
index 21cbd97..45a5730 100644
--- a/test/test-common.sh
+++ b/test/test-common.sh
@@ -52,16 +52,7 @@ fi
 
 # check for date(1) support of operating on a time given from the command
 # line instead of the current time. Necessary for test 0085.
-if date --date @42 +%Y%m%d%H%M > /dev/null 2>&1; then
-  DATE_DATEARG='date --date'
-elif gdate --date @42 +%Y%m%d%H%M > /dev/null 2>&1; then
-  DATE_DATEARG='gdate --date'
-else
-  echo "no date command supporting argument --date found:"
-  date --date @42 +%Y%m%d%H%M
-  gdate --date @42 +%Y%m%d%H%M
-  exit 1
-fi
+
 
 TESTDIR="$(basename "$0" .sh)"
 mkdir -p "$TESTDIR"
