commit 97a08043d55d03fd2310afed46a241c3c1cc4ee6
Author: q66 <q66@chimera-linux.org>
Date:   Sun Jan 23 02:48:21 2022 +0100

    disable network-using tests

diff --git a/tests/meson.build b/tests/meson.build
index b3087a7..e523d69 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -8,10 +8,6 @@ test_suites = {
     'oauth2',
     'params',
   ],
-  'rest-extras': [
-    'flickr',
-    'lastfm',
-  ],
 }
 
 test_deps = [
