--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,7 @@ function(CREATE_TARGET target)
         env ${VARS_FOR_CARGO}
           ${Rust_CARGO}
             build --bin ${target}
-            $<$<CONFIG:Release>:--release>
+            $<$<CONFIG:None>:--release>
             $<$<CONFIG:RelWithDebInfo>:--profile=release-with-debug>
             --target ${Rust_CARGO_TARGET}
             --no-default-features
