--- a/cmake/Rust.cmake
+++ b/cmake/Rust.cmake
@@ -22,7 +22,7 @@ else()
     set(rust_target_dir "${FISH_RUST_BUILD_DIR}/${Rust_CARGO_HOST_TARGET}")
 endif()
 
-set(rust_profile $<IF:$<CONFIG:Debug>,debug,$<IF:$<CONFIG:RelWithDebInfo>,release-with-debug,release>>)
+set(rust_profile $<IF:$<CONFIG:None>,debug,$<IF:$<CONFIG:RelWithDebInfo>,release-with-debug,release>>)
 set(rust_debugflags "$<$<CONFIG:Debug>:-g>$<$<CONFIG:RelWithDebInfo>:-g>")
 
 
