--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,18 +57,18 @@ if(NOT "${CMAKE_BUILD_TYPE}" IN_LIST build_types)
 endif()

 add_custom_target(
   fish ALL
   COMMAND
     "${CMAKE_COMMAND}" -E
       env ${VARS_FOR_CARGO}
         ${Rust_CARGO}
-          build --bin fish
-          $<$<CONFIG:Release>:--release>
+          auditable build --bin fish
+          $<$<CONFIG:None>:--release>
           $<$<CONFIG:RelWithDebInfo>:--profile=release-with-debug>
           --target ${Rust_CARGO_TARGET}
           --no-default-features
           --features=${FISH_CARGO_FEATURES}
           ${CARGO_FLAGS}
     &&
     "${CMAKE_COMMAND}" -E
       copy "${rust_target_dir}/${rust_profile}/fish" "${CMAKE_CURRENT_BINARY_DIR}"
