--- a/Lib/test/test_ctypes/test_dlerror.py
+++ b/Lib/test/test_ctypes/test_dlerror.py
@@ -58,10 +58,13 @@ class TestNullDlsym(unittest.TestCase):
         import subprocess
         import tempfile
 
-        retcode = subprocess.call(["gcc", "--version"],
-                                  stdout=subprocess.DEVNULL,
-                                  stderr=subprocess.DEVNULL)
-        if retcode != 0:
+        try:
+            retcode = subprocess.call(["gcc", "--version"],
+                                      stdout=subprocess.DEVNULL,
+                                      stderr=subprocess.DEVNULL)
+            if retcode != 0:
+                self.skipTest("gcc is missing")
+        except FileNotFoundError:
             self.skipTest("gcc is missing")
 
         pipe_r, pipe_w = os.pipe()
