This only works on glibc.

--- a/Modules/_testcapi/run.c
+++ b/Modules/_testcapi/run.c
@@ -75,7 +75,7 @@ run_fileexflags(PyObject *mod, PyObject *pos_args)
 
     result = PyRun_FileExFlags(fp, filename, start, globals, locals, closeit, pflags);
 
-#if defined(__linux__) || defined(MS_WINDOWS) || defined(__APPLE__)
+#if defined(__GLIBC__) || defined(MS_WINDOWS) || defined(__APPLE__)
     /* The behavior of fileno() after fclose() is undefined, but it is
      * the only practical way to check whether the file was closed.
      * Only test this on the known platforms. */
