--- a/meson.build
+++ b/meson.build
@@ -35,7 +35,7 @@
   )
 endif
 
-wlroots        = dependency('wlroots', version: '>= 0.16.0', fallback: ['wlroots', 'wlroots'])
+wlroots        = dependency('wlroots-0.16', version: '>= 0.16.0', fallback: ['wlroots', 'wlroots'])
 wayland_protos = dependency('wayland-protocols', version: '>=1.14')
 wayland_server = dependency('wayland-server')
 xkbcommon      = dependency('xkbcommon')
