diff --git a/external/libcmis/UnpackedTarball_libcmis.mk b/external/libcmis/UnpackedTarball_libcmis.mk
index e47c7f8..51163cf 100644
--- a/external/libcmis/UnpackedTarball_libcmis.mk
+++ b/external/libcmis/UnpackedTarball_libcmis.mk
@@ -26,6 +26,7 @@ $(eval $(call gb_UnpackedTarball_add_patches,libcmis,\
 	external/libcmis/0001-Take-into-account-m_CurlInitProtocolsFunction-in-cop.patch \
 	external/libcmis/initprotocols.patch.1 \
 	external/libcmis/0001-Fix-boost-1.86-breakage.patch \
+	external/libcmis/libcmis-libxml2.patch.1 \
 ))
 
 # vim: set noet sw=4 ts=4:
diff --git a/external/skia/UnpackedTarball_skia.mk b/external/skia/UnpackedTarball_skia.mk
index cf2cd6c..37a32fd 100644
--- a/external/skia/UnpackedTarball_skia.mk
+++ b/external/skia/UnpackedTarball_skia.mk
@@ -23,7 +23,6 @@ skia_patches := \
     windows-do-not-modify-logfont.patch.0 \
     windows-force-unicode-api.patch.0 \
     fix-without-gl.patch.1 \
-    windows-typeface-directwrite.patch.1 \
     windows-raster-surface-no-copies.patch.1 \
     fix-windows-dwrite.patch.1 \
     swap-buffers-rect.patch.1 \
