diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
index a6205d0..bfa29cd 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
@@ -1249,11 +1249,11 @@ void PDFOutDev::setSkipImages( bool bSkipImages )
 
 #if POPPLER_CHECK_VERSION(21, 3, 0)
 poppler_bool PDFOutDev::tilingPatternFill(GfxState *state, Gfx *, Catalog *,
-                                          GfxTilingPattern *tPat, const double *mat,
+                                          GfxTilingPattern *tPat, const std::array<double, 6> &mat,
                                           int x0, int y0, int x1, int y1,
                                           double xStep, double yStep)
 {
-    const double *pBbox = tPat->getBBox();
+    auto pBbox = tPat->getBBox();
     const int nPaintType = tPat->getPaintType();
     Dict *pResDict = tPat->getResDict();
     Object *aStr = tPat->getContentStream();
diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
index eb0e3e2..af96d9e 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
@@ -287,7 +287,7 @@ namespace pdfi
 #if POPPLER_CHECK_VERSION(21, 3, 0)
         poppler_bool useTilingPatternFill() override { return true; };
         poppler_bool tilingPatternFill(GfxState *state, Gfx *gfx, Catalog *cat,
-                                       GfxTilingPattern *tPat, const double *mat,
+                                       GfxTilingPattern *tPat, const std::array<double, 6> &mat,
                                        int x0, int y0, int x1, int y1,
                                        double xStep, double yStep) override;
 #endif
