--- a/vendor/nix/src/dir.rs	2024-07-23 04:37:21.274757256 -0700
+++ b/vendor/nix/src/dir.rs	2024-07-23 04:19:48.158917652 -0700
@@ -7,7 +7,7 @@
 use crate::sys;
 
 #[cfg(target_os = "linux")]
-use libc::{dirent64 as dirent, readdir64_r as readdir_r};
+use libc::{dirent, readdir_r};
 
 #[cfg(not(target_os = "linux"))]
 use libc::{dirent, readdir_r};


--- a/vendor/getrandom/src/util_libc.rs	2024-07-23 05:27:23.264594653 -0700
+++ b/vendor/getrandom/src/util_libc.rs	2024-07-23 04:19:29.511782646 -0700
@@ -98,7 +98,7 @@
 
 cfg_if! {
     if #[cfg(any(target_os = "linux", target_os = "emscripten"))] {
-        use libc::open64 as open;
+        use libc::open as open;
     } else {
         use libc::open;
     }
