commit bf033aa4a4dda2943eddf762b6c5d2f96a75d56e
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Apr 9 02:24:54 2022 +0200

    disable playbin-complex test
    
    It times out.

diff --git a/tests/check/meson.build b/tests/check/meson.build
index f135c1f..b519130 100644
--- a/tests/check/meson.build
+++ b/tests/check/meson.build
@@ -71,7 +71,6 @@ if host_machine.system() != 'windows'
     [ 'elements/multifdsink.c', not core_conf.has('HAVE_SYS_SOCKET_H') or not core_conf.has('HAVE_UNISTD_H') ],
     # FIXME: multisocketsink test on windows/msvc
     [ 'elements/multisocketsink.c', not core_conf.has('HAVE_SYS_SOCKET_H') or not core_conf.has('HAVE_UNISTD_H') ],
-    [ 'elements/playbin-complex.c', not ogg_dep.found() ],
     [ 'elements/textoverlay.c', not pango_dep.found() ],
     [ 'elements/vorbisdec.c', not vorbis_dep.found(), [ vorbis_dep, vorbisenc_dep ] ],
     [ 'elements/vorbistag.c', not vorbisenc_dep.found(), [ vorbis_dep, vorbisenc_dep ] ],
