--- a/fnc.bld.mk	2023-11-26 12:31:23.000000000 +0100
+++ b/fnc.bld.mk	2024-06-24 22:37:02.278588656 +0200
@@ -10,32 +10,6 @@
 HASH !=		cut -f 1 manifest.uuid
 DATE !=		sed '2q;d' manifest | cut -d ' ' -f 2 | tr T ' '
 
-# FLAGS NEEDED TO BUILD SQLITE3
-SQLITE_CFLAGS =	${CFLAGS} -Wall -Werror -Wno-sign-compare -pedantic -std=c99 \
-		-DNDEBUG=1 \
-		-DSQLITE_DQS=0 \
-		-DSQLITE_DEFAULT_MEMSTATUS=0 \
-		-DSQLITE_DEFAULT_WAL_SYNCHRONOUS=1 \
-		-DSQLITE_LIKE_DOESNT_MATCH_BLOBS \
-		-DSQLITE_OMIT_DECLTYPE \
-		-DSQLITE_OMIT_PROGRESS_CALLBACK \
-		-DSQLITE_OMIT_SHARED_CACHE \
-		-DSQLITE_OMIT_LOAD_EXTENSION \
-		-DSQLITE_MAX_EXPR_DEPTH=0 \
-		-DSQLITE_USE_ALLOCA \
-		-DSQLITE_ENABLE_LOCKING_STYLE=0 \
-		-DSQLITE_DEFAULT_FILE_FORMAT=4 \
-		-DSQLITE_ENABLE_EXPLAIN_COMMENTS \
-		-DSQLITE_ENABLE_FTS4 \
-		-DSQLITE_ENABLE_DBSTAT_VTAB \
-		-DSQLITE_ENABLE_JSON1 \
-		-DSQLITE_ENABLE_FTS5 \
-		-DSQLITE_ENABLE_STMTVTAB \
-		-DSQLITE_HAVE_ZLIB \
-		-DSQLITE_INTROSPECTION_PRAGMAS \
-		-DSQLITE_ENABLE_DBPAGE_VTAB \
-		-DSQLITE_TRUSTED_SCHEMA=0
-
 # FLAGS NEEDED TO BUILD LIBFOSSIL
 FOSSIL_CFLAGS =	${CFLAGS} -Wall -Werror -Wsign-compare -pedantic -std=c99
 
@@ -47,15 +21,12 @@
 		-D_XOPEN_SOURCE_EXTENDED -DVERSION=${VERSION} -DHASH=${HASH} \
 		-DDATE="${DATE}"
 
-FNC_LDFLAGS =	${LDFLAGS} -lm -lutil -lz -lpthread -fPIC
+FNC_LDFLAGS =	${LDFLAGS} -lm -lutil -lz -lpthread -fPIC -lsqlite3
 
 all: bin
 
-bin: lib/sqlite3.o lib/libfossil.o src/fnc.o src/fnc
+bin: lib/libfossil.o src/fnc.o src/fnc
 
-lib/sqlite3.o: lib/sqlite3.c lib/sqlite3.h
-	${CC} ${SQLITE_CFLAGS} -c $< -o $@
-
 lib/libfossil.o: lib/libfossil.c lib/libfossil.h
 	${CC} ${FOSSIL_CFLAGS} -c $< -o $@
 
@@ -66,8 +37,8 @@
 	include/fnc.h include/opt.h fnc.bld.mk
 	${CC} ${FNC_CFLAGS} -c $< -o $@
 
-src/fnc: src/fnc.o src/diff.o lib/libfossil.o lib/sqlite3.o fnc.bld.mk
-	${CC} -o $@ src/fnc.o src/diff.o lib/libfossil.o lib/sqlite3.o \
+src/fnc: src/fnc.o src/diff.o lib/libfossil.o fnc.bld.mk
+	${CC} -o $@ src/fnc.o src/diff.o lib/libfossil.o \
 	${FNC_LDFLAGS}
 
 ccdb: clean
