diff --git a/Makefile b/Makefile
index c9e0f0e..27324d2 100644
--- a/Makefile
+++ b/Makefile
@@ -42,9 +42,10 @@ ZSH_COMPLETION = completion/zsh/_ec_probe completion/zsh/_nbfc completion/zsh/_n
 all: deprecation_warning $(CORE) $(DOC) $(SYSTEMD) $(OPEN_RC) $(SYSTEMV) $(BASH_COMPLETION) $(FISH_COMPLETION) $(ZSH_COMPLETION)
 
 install-core: $(CORE)
-	install -Dm 755 src/nbfc_service  $(DESTDIR)$(bindir)/nbfc_service
-	install -Dm 755 src/ec_probe      $(DESTDIR)$(bindir)/ec_probe
-	install -Dm 755 src/nbfc          $(DESTDIR)$(bindir)/nbfc
+	install -d -m 755 $(DESTDIR)$(bindir)
+	install -m 755 src/nbfc_service  $(DESTDIR)$(bindir)/nbfc_service
+	install -m 755 src/ec_probe      $(DESTDIR)$(bindir)/ec_probe
+	install -m 755 src/nbfc          $(DESTDIR)$(bindir)/nbfc
 
 REPLACE_VARS = sed \
 	-e 's|@BINDIR@|$(bindir)|g'           \
@@ -121,7 +122,8 @@ install-configs:
 
 install-systemd: etc/systemd/system/nbfc_service.service
 	# /usr/local/lib/systemd/system
-	install -Dm 644 etc/systemd/system/nbfc_service.service $(DESTDIR)$(sysddir)/nbfc_service.service
+	install -d -m 755 $(DESTDIR)$(sysddir)
+	install -m 644 etc/systemd/system/nbfc_service.service $(DESTDIR)$(sysddir)/nbfc_service.service
 
 install-openrc: etc/init.d/nbfc_service.openrc
 	# /usr/local/etc/init.d
@@ -132,24 +134,29 @@ install-systemv: etc/init.d/nbfc_service.systemv
 	install -Dm 755 etc/init.d/nbfc_service.systemv	 $(DESTDIR)$(systemvdir)/nbfc_service
 
 install-docs:
-	install -Dm 644 doc/ec_probe.1           $(DESTDIR)$(man1dir)/ec_probe.1
-	install -Dm 644 doc/nbfc.1               $(DESTDIR)$(man1dir)/nbfc.1
-	install -Dm 644 doc/nbfc_service.1       $(DESTDIR)$(man1dir)/nbfc_service.1
-	install -Dm 644 doc/nbfc_service.json.5  $(DESTDIR)$(man5dir)/nbfc_service.json.5
+	install -d -m 755 $(DESTDIR)$(man1dir)
+	install -d -m 755 $(DESTDIR)$(man5dir)
+	install -m 644 doc/ec_probe.1           $(DESTDIR)$(man1dir)/ec_probe.1
+	install -m 644 doc/nbfc.1               $(DESTDIR)$(man1dir)/nbfc.1
+	install -m 644 doc/nbfc_service.1       $(DESTDIR)$(man1dir)/nbfc_service.1
+	install -m 644 doc/nbfc_service.json.5  $(DESTDIR)$(man5dir)/nbfc_service.json.5
 
 install-completion:
+	install -d -m 755 $(DESTDIR)$(datadir)/zsh/site-functions
+	install -d -m 755 $(DESTDIR)$(datadir)/bash-completion/completions
+	install -d -m 755 $(DESTDIR)$(datadir)/fish/completions
 	# ZSH
-	install -Dm 644 completion/zsh/_nbfc               $(DESTDIR)$(datadir)/zsh/site-functions/_nbfc
-	install -Dm 644 completion/zsh/_nbfc_service       $(DESTDIR)$(datadir)/zsh/site-functions/_nbfc_service
-	install -Dm 644 completion/zsh/_ec_probe           $(DESTDIR)$(datadir)/zsh/site-functions/_ec_probe
+	install -m 644 completion/zsh/_nbfc               $(DESTDIR)$(datadir)/zsh/site-functions/_nbfc
+	install -m 644 completion/zsh/_nbfc_service       $(DESTDIR)$(datadir)/zsh/site-functions/_nbfc_service
+	install -m 644 completion/zsh/_ec_probe           $(DESTDIR)$(datadir)/zsh/site-functions/_ec_probe
 	# BASH	
-	install -Dm 644 completion/bash/nbfc               $(DESTDIR)$(datadir)/bash-completion/completions/nbfc
-	install -Dm 644 completion/bash/nbfc_service       $(DESTDIR)$(datadir)/bash-completion/completions/nbfc_service
-	install -Dm 644 completion/bash/ec_probe           $(DESTDIR)$(datadir)/bash-completion/completions/ec_probe
+	install -m 644 completion/bash/nbfc               $(DESTDIR)$(datadir)/bash-completion/completions/nbfc
+	install -m 644 completion/bash/nbfc_service       $(DESTDIR)$(datadir)/bash-completion/completions/nbfc_service
+	install -m 644 completion/bash/ec_probe           $(DESTDIR)$(datadir)/bash-completion/completions/ec_probe
 	# FISH	
-	install -Dm 644 completion/fish/nbfc.fish          $(DESTDIR)$(datadir)/fish/completions/nbfc.fish
-	install -Dm 644 completion/fish/nbfc_service.fish  $(DESTDIR)$(datadir)/fish/completions/nbfc_service.fish
-	install -Dm 644 completion/fish/ec_probe.fish      $(DESTDIR)$(datadir)/fish/completions/ec_probe.fish
+	install -m 644 completion/fish/nbfc.fish          $(DESTDIR)$(datadir)/fish/completions/nbfc.fish
+	install -m 644 completion/fish/nbfc_service.fish  $(DESTDIR)$(datadir)/fish/completions/nbfc_service.fish
+	install -m 644 completion/fish/ec_probe.fish      $(DESTDIR)$(datadir)/fish/completions/ec_probe.fish
 
 install: install-core install-configs install-$(INIT_SYSTEM) install-docs install-completion
 
