From 99a62b114b0f32b3c4a46f18dbfe22fe33405dab Mon Sep 17 00:00:00 2001
From: midipix <writeonce@midipix.org>
Date: Jul 19 2024 01:34:48 +0000
Subject: slibtool.m4: slibtool_set_flavor(): define shared and static suffix variables.


---

diff --git a/m4/slibtool.m4 b/m4/slibtool.m4
index bf1c4e1..e2e6ad5 100644
--- a/m4/slibtool.m4
+++ b/m4/slibtool.m4
@@ -260,6 +260,17 @@ slibtool_set_flavor()
 	enable_win32_dll=${slibtool_enable_win32_dll}
 	enable_fast_install=${slibtool_enable_fast_install}
 	pic_mode=${slibtool_pic_mode}
+
+	# suffix variables
+	if [[ -n "${host}" ]]; then
+		shrext_cmds="$($_slibtool -print-shared-ext --host=${host})"
+		libext="$($_slibtool -print-static-ext --host=${host})"
+		libext="${libext#[.]}"
+	else
+		shrext_cmds="$($_slibtool -print-shared-ext)"
+		libext="$($_slibtool -print-static-ext)"
+		libext="${libext#[.]}"
+	fi
 }
 ])
 

