--- a/src/arbits/slbt_archive_merge.c
+++ b/src/arbits/slbt_archive_merge.c
@@ -603,7 +603,7 @@ int slbt_ar_merge_archives(
 						if (meminfo->ar_file_header.ar_header_attr & AR_HEADER_ATTR_NAME_REF) {
 							nwritten = sprintf(
 								&base[omembers],"/"PPRII64,
-								namestr - namebase);
+								(int64_t)(namestr - namebase));
 
 							if (nwritten < 0)
 								SLBT_SYSTEM_ERROR(dctx,0);
