From 2559c857971fa6e2097042305ab65da90c82e63d Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Tue, 22 Apr 2025 15:56:28 +0200
Subject: [PATCH 2/8] remove remaining dpkg references

---
 mkinitramfs      |  2 +-
 update-initramfs | 11 ++---------
 2 files changed, 3 insertions(+), 10 deletions(-)

diff --git a/mkinitramfs b/mkinitramfs
index 774ef68..5c850a1 100755
--- a/mkinitramfs
+++ b/mkinitramfs
@@ -109,7 +109,7 @@ maybe_add_conf() {
 	if [ -e "$1" ] && \
 	   basename "$1" \
 	   | grep '^[[:alnum:]][[:alnum:]\._-]*$' \
-	   | grep -qv '\.dpkg-.*$'; then
+	   | grep -qv '\.chimera-.*$'; then
 		if [ -d "$1" ]; then
 			echo "W: $1 is a directory instead of file" >&2
 		else
diff --git a/update-initramfs b/update-initramfs
index 25a3051..0fb08ea 100755
--- a/update-initramfs
+++ b/update-initramfs
@@ -17,13 +17,6 @@ backup_initramfs=no
 
 [ -r ${CONF} ] && . ${CONF}
 
-if [ -n "${DPKG_MAINTSCRIPT_PACKAGE-}" ] && [ $# = 1 ] && [ "$1" = -u ]; then
-	if dpkg-trigger --no-await update-initramfs; then
-		echo "update-initramfs: deferring update (trigger activated)"
-		exit 0
-	fi
-fi
-
 usage()
 {
 	cat << EOF
@@ -85,7 +78,7 @@ set_initramfs()
 backup_initramfs()
 {
 	[ ! -r "${initramfs}" ] && return 0
-	initramfs_bak="${initramfs}.dpkg-bak"
+	initramfs_bak="${initramfs}.chimera-bak"
 	[ -r "${initramfs_bak}" ] && rm -f "${initramfs_bak}"
 	ln -f "${initramfs}" "${initramfs_bak}" 2>/dev/null \
 		|| cp -a "${initramfs}" "${initramfs_bak}"
@@ -95,7 +88,7 @@ backup_initramfs()
 # keep booted initramfs
 backup_booted_initramfs()
 {
-	initramfs_bak="${initramfs}.dpkg-bak"
+	initramfs_bak="${initramfs}.chimera-bak"
 
 	# first time run thus no backup
 	[ ! -r "${initramfs_bak}" ] && return 0
-- 
2.50.1

