From c94327f8c4420e244b3e5a22170b971ad53dc7e3 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Mon, 3 Jun 2024 16:45:18 +0200
Subject: [PATCH 5/8] general portability fixes for bsdutils and musl

---
 hook-functions | 53 +++++++++++++++++++++++---------------------------
 mkinitramfs    | 21 ++++----------------
 unmkinitramfs  | 12 ++++++------
 3 files changed, 34 insertions(+), 52 deletions(-)

diff --git a/hook-functions b/hook-functions
index d30ff25..e221f95 100644
--- a/hook-functions
+++ b/hook-functions
@@ -166,6 +166,19 @@ _sanitize_target() {
 	printf "%s" "${target}"
 }
 
+_relpath() {
+    local pos="$(dirname $1 | sed -e 's,//*,/,g')" ref="${2%%/}" down=''
+
+    while :; do
+        test "$pos" = '/' && break
+        case "$ref" in $pos/*) break;; esac
+        down="../$down"
+        pos=${pos%/*}
+    done
+
+    echo "$down${ref##$pos/}"
+}
+
 # $1 = file type (for logging)
 # $2 = file to copy to initramfs
 # $3 (optional) Name for the file on the initramfs
@@ -173,7 +186,7 @@ _sanitize_target() {
 # If the target exists, we leave it and return 1.
 # On any other error, we return >1.
 copy_file() {
-	local type src target link_target
+	local type src target link_target relp
 
 	type="${1}"
 	src="${2}"
@@ -209,7 +222,8 @@ copy_file() {
 
 			# Create a relative link so it always points
 			# to the right place
-			ln -rs "${DESTDIR}/${link_target}" "${DESTDIR}/${target}"
+			relp=$(_relpath "${target}" "${link_target}")
+			ln -sf "${relp}" "${DESTDIR}/${target}"
 		fi
 
 		# Copy the link target if it doesn't already exist
@@ -236,30 +250,10 @@ copy_exec() {
 	copy_file binary "${src}" "${target}" || return $(($? - 1))
 
 	# Copy the dependant libraries
-	for x in $(env --unset=LD_PRELOAD ldd "${src}" 2>/dev/null | sed -e '
-		/^[^\t]/d;
-		/\//!d;
-		/linux-gate/d;
-		/=>/ {s/.*=>[[:blank:]]*\([^[:blank:]]*\).*/\1/};
-		s/[[:blank:]]*\([^[:blank:]]*\) (.*)/\1/' 2>/dev/null); do
-
-		# Try to use non-optimised libraries where possible.
-		# We assume that all HWCAP libraries will be in tls,
-		# sse2, vfp or neon.
-		nonoptlib=$(echo "${x}" | sed -e 's#/lib/\([^/]*/\)\?\(tls\|i686\|sse2\|neon\|vfp\).*/\(lib.*\)#/lib/\1\3#')
-		nonoptlib=$(echo "${nonoptlib}" | sed -e 's#-linux-gnu/\(tls\|i686\|sse2\|neon\|vfp\).*/\(lib.*\)#-linux-gnu/\2#')
-
-		if [ -e "${nonoptlib}" ]; then
-			x="${nonoptlib}"
-		fi
-
-		# Handle common dlopen() dependency (Debian bug #950254)
-		case "${x}" in
-		*/libpthread.so.*)
-			copy_libgcc "${x%/*}" || return
-			;;
-		esac
-
+	for x in $(env -u LD_PRELOAD ldd "${src}" 2>/dev/null | awk '{print $3}'); do
+		# XXX: when running ldd on a .solib, the libc.so dependency is resolved
+		# as `ldd` and not the ld-musl loader, so this loop would copy `ldd` and fail
+		if [ "$x" = "ldd" ]; then continue; fi
 		copy_file binary "${x}" || {
 			ret=$?
 			[ ${ret} = 1 ] || return $((ret - 1))
@@ -307,7 +301,8 @@ copy_modules_dir()
 	done
 
 	# shellcheck disable=SC2044
-	for kmod in $(find "${MODULESDIR}/${dir}" "$@" -name '*.ko*' -printf '%f\n'); do
+	for kmod in $(find "${MODULESDIR}/${dir}" "$@" -name '*.ko*'); do
+		kmod=$(basename "$kmod")
 		modules="$modules ${kmod%%.*}"
 	done
 	# shellcheck disable=SC2086
@@ -403,8 +398,8 @@ block_dev_mod_add()
 	dev_node="$1"
 
 	# Look up device number and convert to decimal as it appears in sysfs
-	dev_num="$(stat -L -c %t:%T "$dev_node")"
-	dev_num="$((0x${dev_num%:*})):$((0x${dev_num#*:}))"
+	dev_num="$(stat -L -f %Z "$dev_node")"
+	dev_num="$(echo $dev_num | tr ',' ':')"
 
 	# Look up device in sysfs
 	dev_sys_path="/sys/dev/block/$dev_num"
diff --git a/mkinitramfs b/mkinitramfs
index 2c60ee0..459b421 100755
--- a/mkinitramfs
+++ b/mkinitramfs
@@ -380,16 +380,16 @@ add_builtin_firmware
 cp -p /usr/share/initramfs-tools/init "${DESTDIR}/init"
 
 # add existant boot scripts
-for b in $(cd /usr/share/initramfs-tools/scripts/ && find . \
-	-regextype posix-extended -regex '.*/[[:alnum:]\._-]+$' -type f); do
+for b in $(cd /usr/share/initramfs-tools/scripts/ && find -E . \
+	-regex '.*/[[:alnum:]\._-]+$' -type f); do
 	[ -d "${DESTDIR}/scripts/$(dirname "${b}")" ] \
 		|| mkdir -p "${DESTDIR}/scripts/$(dirname "${b}")"
 	cp -p "/usr/share/initramfs-tools/scripts/${b}" \
 		"${DESTDIR}/scripts/$(dirname "${b}")/"
 done
 # Prune dot-files/directories and limit depth to exclude VCS files
-for b in $(cd "${CONFDIR}/scripts" && find . -maxdepth 2 -name '.?*' -prune -o \
-	-regextype posix-extended -regex '.*/[[:alnum:]\._-]+$' -type f -print); do
+for b in $(cd "${CONFDIR}/scripts" && find -E . -maxdepth 2 -name '.?*' -prune -o \
+	-regex '.*/[[:alnum:]\._-]+$' -type f -print); do
 	[ -d "${DESTDIR}/scripts/$(dirname "${b}")" ] \
 		|| mkdir -p "${DESTDIR}/scripts/$(dirname "${b}")"
 	cp -p "${CONFDIR}/scripts/${b}" "${DESTDIR}/scripts/$(dirname "${b}")/"
@@ -461,19 +461,6 @@ done
 depmod -a -b "${DESTDIR}" "${version}"
 rm -f "${DESTDIR}/lib/modules/${version}"/modules.*map
 
-# make sure that library search path is up to date
-cp -pPr /etc/ld.so.conf* "$DESTDIR"/etc/
-if ! ldconfig -r "$DESTDIR" ; then
-	[ "$(id -u)" != "0" ] \
-	&& echo "ldconfig might need uid=0 (root) for chroot()" >&2
-fi
-# The auxiliary cache is not reproducible and is always invalid at boot
-# (see #845034)
-if [ -d "${DESTDIR}"/var/cache/ldconfig ]; then
-	rm -f "${DESTDIR}"/var/cache/ldconfig/aux-cache
-	rmdir --ignore-fail-on-non-empty "${DESTDIR}"/var/cache/ldconfig
-fi
-
 # Apply DSDT to initramfs
 if [ -e "${CONFDIR}/DSDT.aml" ]; then
 	copy_file DSDT "${CONFDIR}/DSDT.aml"
diff --git a/unmkinitramfs b/unmkinitramfs
index a296029..b3b101d 100755
--- a/unmkinitramfs
+++ b/unmkinitramfs
@@ -110,13 +110,13 @@ splitinitramfs()
 		else
 			subdir=early$count
 		fi
-		dd < "$initramfs" skip=$start count=$((end - start)) iflag=skip_bytes,count_bytes 2> /dev/null |
+		dd < "$initramfs" skip=$((start / 512)) count=$(((end - start) / 512)) 2> /dev/null |
 		(
 			if [ -n "$dir" ]; then
 				mkdir -p -- "$dir/$subdir"
 				cd -- "$dir/$subdir"
 			fi
-			cpio -i "$@"
+			cpio -di "$@"
 		)
 		start=$end
 	done
@@ -125,18 +125,18 @@ splitinitramfs()
 		# Extract to main subdirectory
 		subarchive=$(mktemp "${TMPDIR:-/var/tmp}/unmkinitramfs_XXXXXX")
 		trap 'rm -f "$subarchive"' EXIT
-		dd < "$initramfs" skip="$end" iflag=skip_bytes 2> /dev/null \
+		dd < "$initramfs" skip=$((end / 512)) 2> /dev/null \
 			> "$subarchive"
-		xcpio "$subarchive" "${dir:+$dir/main}" -i "$@"
+		xcpio "$subarchive" "${dir:+$dir/main}" -di "$@"
 	else
 		# Don't use subdirectories (for backward compatibility)
-		xcpio "$initramfs" "$dir" -i "$@"
+		xcpio "$initramfs" "$dir" -di "$@"
 	fi
 }
 
 OPTIONS=$(getopt -o hv --long help,list,verbose -n "$0" -- "$@") || usage_error
 
-cpio_opts="--preserve-modification-time --no-absolute-filenames --quiet"
+cpio_opts="--preserve-modification-time --quiet"
 expected_args=2
 eval set -- "$OPTIONS"
 
-- 
2.45.2

