From dc2fe64a86ce0ab3f793034858eb6b90e9010d01 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Wed, 20 Jul 2022 04:06:22 +0200
Subject: [PATCH 7/8] use fstype from klibc-kinit-standalone

---
 hook-functions | 2 +-
 hooks/fsck     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hook-functions b/hook-functions
index e221f95..1cf0e41 100644
--- a/hook-functions
+++ b/hook-functions
@@ -520,7 +520,7 @@ dep_add_modules_mount()
 	fi
 
 	# do not trust mount, check superblock
-	eval "$(/usr/lib/klibc/bin/fstype "${dev_node}")"
+	eval "$(/usr/libexec/kinit-standalone/fstype "${dev_node}")"
 
 	# check that fstype fs recognition
 	if [ "${FSTYPE}" = "unknown" ]; then
diff --git a/hooks/fsck b/hooks/fsck
index 012c97e..26a50d5 100755
--- a/hooks/fsck
+++ b/hooks/fsck
@@ -53,7 +53,7 @@ get_fsck_type_fstab () {
 		if [ "${MNT_DIR}" = "/" ] || [ "${MNT_TYPE}" = "auto" ]; then
 			MNT_FSNAME="$(resolve_device "${MNT_FSNAME}")"
 			# shellcheck disable=SC2317
-			fstype() { "/usr/lib/klibc/bin/fstype" "$@"; }
+			fstype() { "/usr/libexec/kinit-standalone/fstype" "$@"; }
 			if ! get_fstype "${MNT_FSNAME}"; then
 				echo "W: Couldn't identify type of $2 file system for fsck hook" >&2
 			fi
-- 
2.45.2

