From c0a2df51953d983d059ae25652af3317035db399 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Wed, 20 Jul 2022 04:06:22 +0200
Subject: [PATCH 7/8] use fstype from klibc-kinit-standalone

---
 hook-functions | 2 +-
 hooks/fsck     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hook-functions b/hook-functions
index 5a10ed8..1639100 100644
--- a/hook-functions
+++ b/hook-functions
@@ -480,7 +480,7 @@ dep_add_modules_mount()
 	fi
 
 	# do not trust mount, check superblock
-	eval "$(/usr/lib/klibc/bin/fstype "${dev_node}")"
+	eval "$(/usr/libexec/kinit-standalone/fstype "${dev_node}")"
 
 	# check that fstype fs recognition
 	if [ "${FSTYPE}" = "unknown" ]; then
diff --git a/hooks/fsck b/hooks/fsck
index 01e47cc..4e80070 100755
--- a/hooks/fsck
+++ b/hooks/fsck
@@ -52,7 +52,7 @@ get_fsck_type_fstab () {
 		# therefore never used at boot time
 		if [ "${MNT_DIR}" = "/" ] || [ "${MNT_TYPE}" = "auto" ]; then
 			MNT_FSNAME="$(resolve_device "${MNT_FSNAME}")"
-			fstype() { "/usr/lib/klibc/bin/fstype" "$@"; }
+			fstype() { "/usr/libexec/kinit-standalone/fstype" "$@"; }
 			if ! get_fstype "${MNT_FSNAME}"; then
 				echo "W: Couldn't identify type of $2 file system for fsck hook" >&2
 			fi
-- 
2.43.0

