From 077c2a8dc90189c3ffcdf1bc8451de3837fc2b82 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Wed, 28 Aug 2024 14:02:04 +0200
Subject: [PATCH 8/9] force ugetopt

---
 lsinitramfs      | 2 +-
 mkinitramfs      | 2 +-
 unmkinitramfs    | 2 +-
 update-initramfs | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/lsinitramfs b/lsinitramfs
index 8d2a967..0e75019 100755
--- a/lsinitramfs
+++ b/lsinitramfs
@@ -24,7 +24,7 @@ usage_error()
 
 umi_opts="--list"
 
-OPTIONS=$(getopt -o hl --long help,long -n "$0" -- "$@") || usage_error
+OPTIONS=$(ugetopt -o hl --long help,long -n "$0" -- "$@") || usage_error
 
 eval set -- "$OPTIONS"
 
diff --git a/mkinitramfs b/mkinitramfs
index c15833c..6e2df7e 100755
--- a/mkinitramfs
+++ b/mkinitramfs
@@ -34,7 +34,7 @@ usage_error()
 	exit 2
 }
 
-OPTIONS=$(getopt -o c:d:hl:m:ko:r:v --long help -n "$0" -- "$@") || usage_error
+OPTIONS=$(ugetopt -o c:d:hl:m:ko:r:v --long help -n "$0" -- "$@") || usage_error
 
 eval set -- "$OPTIONS"
 
diff --git a/unmkinitramfs b/unmkinitramfs
index b3b101d..edcc5c0 100755
--- a/unmkinitramfs
+++ b/unmkinitramfs
@@ -134,7 +134,7 @@ splitinitramfs()
 	fi
 }
 
-OPTIONS=$(getopt -o hv --long help,list,verbose -n "$0" -- "$@") || usage_error
+OPTIONS=$(ugetopt -o hv --long help,list,verbose -n "$0" -- "$@") || usage_error
 
 cpio_opts="--preserve-modification-time --quiet"
 expected_args=2
diff --git a/update-initramfs b/update-initramfs
index f45169e..ae98df4 100755
--- a/update-initramfs
+++ b/update-initramfs
@@ -291,7 +291,7 @@ verbose=0
 
 ##
 
-OPTIONS=$(getopt -o "k:cudvtb:h?" --long help -n "$0" -- "$@") || usage_error
+OPTIONS=$(ugetopt -o "k:cudvtb:h?" --long help -n "$0" -- "$@") || usage_error
 
 eval set -- "$OPTIONS"
 
-- 
2.46.0

