From 209a3224fdd9d443a2e8f860890c1f5174fa77d7 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Wed, 28 Aug 2024 14:02:29 +0200
Subject: [PATCH 9/9] decompress firmware in initramfs

---
 mkinitramfs | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/mkinitramfs b/mkinitramfs
index 6e2df7e..a712a84 100755
--- a/mkinitramfs
+++ b/mkinitramfs
@@ -463,6 +463,39 @@ find "${DESTDIR}/${MODULESDIR}" -name '*.ko.*' | while read -r ko; do
 	esac
 done
 
+# decompress firmware for size
+if [ -d "${DESTDIR}/lib/firmware" ]; then
+    find "${DESTDIR}/lib/firmware" -name '*.zst' -o -name '*.xz' | while read -r fw; do
+        case "$fw" in
+        *.zst)
+            if ! command -v zstd >/dev/null 2>&1; then
+                break
+            fi
+            ;;
+        *.xz)
+            if ! command -v xz >/dev/null 2>&1; then
+                break
+            fi
+            ;;
+        esac
+        if [ -L "$fw" ]; then
+            # point the symlink to uncompressed file
+            fwdest=$(readlink "$fw")
+            rm -f "$fw"
+            case "$fw" in
+            *.zst) ln -sf "${fwdest%.zst}" "${fw%.zst}" ;;
+            *.xz) ln -sf "${fwdest%.xz}" "${fw%.xz}" ;;
+            esac
+        else
+            # uncompress
+            case "$fw" in
+            *.zst) zstd -q -d --rm "$fw" ;;
+            *.xz) xz -d "$fw" ;;
+            esac
+        fi
+    done
+fi
+
 # generate module deps
 depmod -a -b "${DESTDIR}" "${version}"
 rm -f "${DESTDIR}/lib/modules/${version}"/modules.*map
-- 
2.46.0

