--- a/loaders/meson.build
+++ b/loaders/meson.build
@@ -53,12 +53,11 @@
 
 cargo_env = {
   # Don't use $HOME/.cargo/ for this
-  'CARGO_HOME': cargo_home,
   # Gettext translation
   'LOCALEDIR': get_option('prefix') / get_option('localedir'),
   # Git commit for debugging
   'GIT_DESCRIBE': run_command(
-    'git',
+    'true',
     '-C', meson.project_source_root(),
     'rev-parse',
     '--short', 'HEAD',
@@ -124,6 +124,7 @@
       find_program('meson'),
       args: [
         'install',
+        '--no-rebuild',
         '-C', meson.project_build_root(),
         '--destdir', destdir,
       ],
--- a/libglycin/meson.build
+++ b/libglycin/meson.build
@@ -21,7 +21,6 @@
 pkg_uninstalled_path = {'PKG_CONFIG_PATH': meson.project_build_root() / 'meson-uninstalled'}
 
 cargo_env = {
-    'CARGO_HOME': cargo_home,
 } + pkg_uninstalled_path
 
 packages = [
