--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@ PACKAGE_NAME := ifupdown-ng
 PACKAGE_VERSION := 0.12.1
 PACKAGE_BUGREPORT := https://github.com/ifupdown-ng/ifupdown-ng/issues/new
 
-SBINDIR := /sbin
+SBINDIR := /usr/bin
 
 INTERFACES_FILE := /etc/network/interfaces
 STATE_FILE := /run/ifstate
@@ -144,20 +144,23 @@ check: ${LIBIFUPDOWN_LIB} ${CMDS}
 	kyua test || (kyua report --verbose && exit 1)
 
 install: all
-	install -D -m755 ${MULTICALL} ${DESTDIR}${SBINDIR}/${MULTICALL}
+	install -d ${DESTDIR}/usr/share/examples/ifupdown-ng
+	install -d ${DESTDIR}${SBINDIR}
+	install -d ${DESTDIR}${EXECUTOR_PATH}
+	install -m755 ${MULTICALL} ${DESTDIR}${SBINDIR}/${MULTICALL}
 	for i in ${CMDS}; do \
 		ln -s ${SBINDIR}/${MULTICALL} ${DESTDIR}${SBINDIR}/$$i; \
 	done
 	for i in ${EXECUTOR_SCRIPTS}; do \
-		install -D -m755 executor-scripts/${LAYOUT}/$$i ${DESTDIR}${EXECUTOR_PATH}/$$i; \
+		install -m755 executor-scripts/${LAYOUT}/$$i ${DESTDIR}${EXECUTOR_PATH}/$$i; \
 	done
 	for i in ${EXECUTOR_SCRIPTS_STUB}; do \
-		install -D -m755 executor-scripts/stub/$$i ${DESTDIR}${EXECUTOR_PATH}/$$i; \
+		install -m755 executor-scripts/stub/$$i ${DESTDIR}${EXECUTOR_PATH}/$$i; \
 	done
 	for i in ${EXECUTOR_SCRIPTS_NATIVE}; do \
-		install -D -m755 executor-scripts/${LAYOUT}-native/$$i ${DESTDIR}${EXECUTOR_PATH}/$$i; \
+		install -m755 executor-scripts/${LAYOUT}-native/$$i ${DESTDIR}${EXECUTOR_PATH}/$$i; \
 	done
-	install -D -m644 dist/ifupdown-ng.conf.example ${DESTDIR}${CONFIG_FILE}.example
+	install -m644 dist/ifupdown-ng.conf.example ${DESTDIR}/usr/share/examples/ifupdown-ng/ifupdown-ng.conf
 
 .scd.1 .scd.2 .scd.3 .scd.4 .scd.5 .scd.6 .scd.7 .scd.8:
 	${SCDOC} < $< > $@
@@ -193,17 +196,20 @@ MANPAGES = ${MANPAGES_5} ${MANPAGES_7} ${MANPAGES_8}
 docs: ${MANPAGES}
 
 install_docs: docs
+	install -d ${DESTDIR}/usr/share/man/man5
+	install -d ${DESTDIR}/usr/share/man/man7
+	install -d ${DESTDIR}/usr/share/man/man8
 	for i in ${MANPAGES_5}; do \
 		target=$$(basename $$i); \
-		install -D -m644 $$i ${DESTDIR}/usr/share/man/man5/$$target; \
+		install -m644 $$i ${DESTDIR}/usr/share/man/man5/$$target; \
 	done
 	for i in ${MANPAGES_7}; do \
 		target=$$(basename $$i); \
-		install -D -m644 $$i ${DESTDIR}/usr/share/man/man7/$$target; \
+		install -m644 $$i ${DESTDIR}/usr/share/man/man7/$$target; \
 	done
 	for i in ${MANPAGES_8}; do \
 		target=$$(basename $$i); \
-		install -D -m644 $$i ${DESTDIR}/usr/share/man/man8/$$target; \
+		install -m644 $$i ${DESTDIR}/usr/share/man/man8/$$target; \
 	done
 
 .SUFFIXES: .scd .1 .2 .3 .4 .5 .6 .7 .8
