commit 06f447d54c721241bea8b5c51a19e79ab292d748
Author: q66 <q66@chimera-linux.org>
Date:   Thu Jan 9 17:19:31 2025 +0100

    bla

diff --git a/Cargo.lock b/Cargo.lock
index 710480f..3129b5c 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2545,7 +2545,6 @@ dependencies = [
  "tempfile",
  "test-case",
  "thiserror 2.0.9",
- "tikv-jemallocator",
  "toml",
  "tracing",
  "walkdir",
@@ -2571,7 +2570,6 @@ dependencies = [
  "rustc-hash 2.1.0",
  "serde",
  "serde_json",
- "tikv-jemallocator",
  "ureq",
  "url",
 ]
@@ -3643,26 +3641,6 @@ dependencies = [
  "once_cell",
 ]
 
-[[package]]
-name = "tikv-jemalloc-sys"
-version = "0.6.0+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "cd3c60906412afa9c2b5b5a48ca6a5abe5736aec9eb48ad05037a677e52e4e2d"
-dependencies = [
- "cc",
- "libc",
-]
-
-[[package]]
-name = "tikv-jemallocator"
-version = "0.6.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "4cec5ff18518d81584f477e9bfdf957f5bb0979b0bac3af4ca30b5b3ae2d2865"
-dependencies = [
- "libc",
- "tikv-jemalloc-sys",
-]
-
 [[package]]
 name = "tinystr"
 version = "0.7.6"
diff --git a/crates/ruff/Cargo.toml b/crates/ruff/Cargo.toml
index 5f437b3..3450bbc 100644
--- a/crates/ruff/Cargo.toml
+++ b/crates/ruff/Cargo.toml
@@ -79,8 +79,5 @@ ignored = ["chrono"]
 [target.'cfg(target_os = "windows")'.dependencies]
 mimalloc = { workspace = true }
 
-[target.'cfg(all(not(target_os = "windows"), not(target_os = "openbsd"), not(target_os = "aix"), any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "powerpc64")))'.dependencies]
-tikv-jemallocator = { workspace = true }
-
 [lints]
 workspace = true
diff --git a/crates/ruff/src/main.rs b/crates/ruff/src/main.rs
index 2271cca..34c3e9d 100644
--- a/crates/ruff/src/main.rs
+++ b/crates/ruff/src/main.rs
@@ -11,19 +11,6 @@ use ruff::{run, ExitStatus};
 #[global_allocator]
 static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
 
-#[cfg(all(
-    not(target_os = "windows"),
-    not(target_os = "openbsd"),
-    not(target_os = "aix"),
-    any(
-        target_arch = "x86_64",
-        target_arch = "aarch64",
-        target_arch = "powerpc64"
-    )
-))]
-#[global_allocator]
-static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
-
 pub fn main() -> ExitCode {
     // Enabled ANSI colors on Windows 10.
     #[cfg(windows)]
diff --git a/crates/ruff_benchmark/Cargo.toml b/crates/ruff_benchmark/Cargo.toml
index c6c8510..9688650 100644
--- a/crates/ruff_benchmark/Cargo.toml
+++ b/crates/ruff_benchmark/Cargo.toml
@@ -64,6 +64,3 @@ codspeed = ["codspeed-criterion-compat"]
 
 [target.'cfg(target_os = "windows")'.dev-dependencies]
 mimalloc = { workspace = true }
-
-[target.'cfg(all(not(target_os = "windows"), not(target_os = "openbsd"), any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "powerpc64")))'.dev-dependencies]
-tikv-jemallocator = { workspace = true }
