--- a/linux.cmake
+++ b/linux.cmake
@@ -18,8 +18,6 @@
 endif()
 
 # More warnings and warning-as-error
-set(COMPILE_OPTIONS
-  -Wall -Werror)
 include(CheckCCompilerFlag)
 check_c_compiler_flag(-Wshadow WARNING_SHADOW)
 if(WARNING_SHADOW)
@@ -55,8 +55,6 @@
 # Security options
 target_compile_options(
   ${PROJECT_NAME}
-  PRIVATE -fstack-protector-strong -D_FORTIFY_SOURCE=2 -O2 -Wformat
-          -Wformat-security
   PRIVATE $<$<C_COMPILER_ID:GNU>:-fno-strict-overflow>
   PRIVATE $<$<C_COMPILER_ID:GNU>:-fno-delete-null-pointer-checks>
   PRIVATE -fwrapv
