From 4dc88803f84ea11caaac78a39d2084ac6f1313d3 Mon Sep 17 00:00:00 2001
From: Christian Hergert <christian@sourceandstack.com>
Date: Tue, 23 Jun 2026 17:32:49 +0200
Subject: [PATCH] main: make sure libdex is initialized early

---
 src/main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/main.c b/src/main.c
index 8c8f19cde..7b65a1105 100644
--- a/src/main.c
+++ b/src/main.c
@@ -36,6 +36,8 @@
 # include <sysprof-capture.h>
 #endif
 
+#include <libdex.h>
+
 #include <libide-core.h>
 #include <libide-code.h>
 #include <libide-gui.h>
@@ -200,6 +202,8 @@ main (gint   argc,
   messages_debug = g_strdup (getenv ("G_MESSAGES_DEBUG"));
   unsetenv ("G_MESSAGES_DEBUG");
 
+  dex_init ();
+
   /* Setup our gdb fork()/exec() helper if we're in a terminal */
   if (is_running_in_shell ())
     bug_buddy_init ();
-- 
GitLab

