https://gitlab.alpinelinux.org/alpine/aports/-/issues/11634
--
--- a/taskcluster/docker/firefox-snap/firefox.desktop
+++ b/taskcluster/docker/firefox-snap/firefox.desktop
@@ -150,7 +150,7 @@
 Keywords[vi]=Internet;WWW;Browser;Web;Explorer;Trình duyệt;Trang web
 Keywords[zh_CN]=Internet;WWW;Browser;Web;Explorer;网页;浏览;上网;火狐;Firefox;ff;互联网;网站;
 Keywords[zh_TW]=Internet;WWW;Browser;Web;Explorer;網際網路;網路;瀏覽器;上網;網頁;火狐
-Exec=firefox %u
+Exec=/usr/lib/firefox/firefox %u
 Terminal=false
 X-MultipleArgs=false
 Type=Application
@@ -198,7 +198,7 @@
 Name[vi]=Mở cửa sổ mới
 Name[zh_CN]=新建窗口
 Name[zh_TW]=開啟新視窗
-Exec=firefox -new-window
+Exec=/usr/lib/firefox/firefox -new-window
 
 [Desktop Action NewPrivateWindow]
 Name=Open a New Private Window
@@ -217,4 +217,4 @@
 Name[tr]=Yeni bir pencere aç
 Name[uk]=Відкрити нове вікно у потайливому режимі
 Name[zh_TW]=開啟新隱私瀏覽視窗
-Exec=firefox -private-window
+Exec=/usr/lib/firefox/firefox -private-window
