commit 2071d683905185e24f13268380876a6543f829b4
Author: q66 <q66@chimera-linux.org>
Date:   Sat Oct 28 06:47:20 2023 +0200

    allow mremap
    
    This prevents hitting a sandbox violation on start.

diff --git a/security/sandbox/linux/SandboxFilter.cpp b/security/sandbox/linux/SandboxFilter.cpp
index 7d04c0d..a8af47a 100644
--- a/security/sandbox/linux/SandboxFilter.cpp
+++ b/security/sandbox/linux/SandboxFilter.cpp
@@ -1006,6 +1006,7 @@ class SandboxPolicyCommon : public SandboxPolicyBase {
         // Memory mapping
       CASES_FOR_mmap:
       case __NR_munmap:
+      case __NR_mremap:
         return Allow();
 
         // Shared memory
