diff --git a/Makefile.toolchain b/Makefile.toolchain
index ddb6e82ffa..26c8f975ae 100644
--- a/Makefile.toolchain
+++ b/Makefile.toolchain
@@ -58,7 +58,7 @@ PROTOC?=protoc
 GCOV=$(CROSS_COMPILE)gcov
 HOSTGCOV=$(CURDIR)/util/llvm-gcov.sh
 
-C_WARN = -Wstrict-prototypes -Wdeclaration-after-statement -Wno-pointer-sign
+C_WARN = -Wdeclaration-after-statement -Wno-pointer-sign
 COMMON_WARN = -Wall -Wundef -Werror-implicit-function-declaration \
 	-Wno-trigraphs -Wno-format-security -Wno-address-of-packed-member \
 	-fno-common -fno-strict-aliasing -fno-strict-overflow
