diff --git a/status.c b/status.c
index 0bd58f1..2aaac73 100644
--- a/status.c
+++ b/status.c
@@ -56,11 +56,7 @@ start_log_filter(int safe) {
 	argv[2] = "|";
 	argv[3] = "-f";
 	argv[4] = awk_script;
-#if defined(_LINUX_PORT)
-	argv[5] = "-S";
-#else
 	argv[5] = "-safe";
-#endif
 	argv[6] = NULL;
 	if (safe == 2)
 		argv[5] = NULL;
diff --git a/system_test.sh b/system_test.sh
index 45da6e6..6f6b1cc 100755
--- a/system_test.sh
+++ b/system_test.sh
@@ -58,13 +58,6 @@ for util in $utils; do
 	}
 done
 
-if [ $(uname) == 'Linux' ]; then
-	date | awk -S '{}' 2> /dev/null || {
-		echo "ERROR: gawk required on Linux"
-		exit 1
-	}
-fi
-
 # fast tests
 
 try "no arguments"
