--- a/src/multimedia/pulseaudio/qpulseaudiosink.cpp
+++ b/src/multimedia/pulseaudio/qpulseaudiosink.cpp
@@ -496,8 +496,8 @@ qint64 QPulseAudioSink::write(const char *data, qint64 len)
     // or even affect the system volume if flat volumes are enabled
 
     QAudioHelperInternal::applyVolume(volume(), m_format,
-                                      QSpan{ reinterpret_cast<const std::byte *>(data), len },
-                                      QSpan{ reinterpret_cast<std::byte *>(dest), len });
+                                      QSpan{ reinterpret_cast<const std::byte *>(data), qsizetype(len) },
+                                      QSpan{ reinterpret_cast<std::byte *>(dest), qsizetype(len) });
 
     if ((pa_stream_write(m_stream.get(), dest, len, nullptr, 0, PA_SEEK_RELATIVE)) < 0) {
         engineLock.unlock();
