diff --git a/src/daemon/Display.cpp b/src/daemon/Display.cpp
index 4014e0d..971598e 100644
--- a/src/daemon/Display.cpp
+++ b/src/daemon/Display.cpp
@@ -36,6 +36,7 @@
 #include <QTimer>
 #include <QLocalSocket>
 
+#include <cstdlib>
 #include <pwd.h>
 #include <unistd.h>
 #include <sys/time.h>
@@ -77,7 +78,39 @@ namespace SDDM {
     }
 
     int fetchAvailableVt() {
-        if (!isTtyInUse(QStringLiteral("tty" STRINGIFY(SDDM_INITIAL_VT)))) {
+        /* chimera active ttys */
+        QFile ttys{QString::fromUtf8("/run/agetty-active")};
+        if (ttys.open(QIODevice::ReadOnly)) {
+            char buf[4096], *bufp = buf, *ttystr = nullptr;
+            memset(buf, 0, sizeof(buf));
+            auto sz = ttys.read(buf, sizeof(buf) - 1);
+            if (sz > 0) {
+                unsigned long minvt = 1;
+                while (*bufp) {
+                    if (strncmp(bufp, "tty", 3)) {
+                        break;
+                    }
+                    char *err = nullptr;
+                    auto gettyn = strtoul(bufp + 3, &err, 10);
+                    if (!err || (*err && *err != '\n')) {
+                        break;
+                    }
+                    if (gettyn >= minvt) {
+                        ttystr = bufp;
+                        minvt = gettyn + 1;
+                    }
+                    if (!*err) {
+                        break;
+                    }
+                    *err = '\0';
+                    bufp = err + 1;
+                }
+                if (minvt < 64 && ttystr && !isTtyInUse(QString::fromUtf8(ttystr))) {
+                    return int(minvt);
+                }
+            }
+        }
+        if (!isTtyInUse(QStringLiteral("tty%1").arg(SDDM_INITIAL_VT))) {
             return SDDM_INITIAL_VT;
         }
         const auto vt = VirtualTerminal::currentVt();
