--- a/src/meson.build
+++ b/src/meson.build
@@ -27,15 +27,12 @@ if gtk4_x11.found()
   cargo_options += ['--features', 'x11']
 endif
 
-cargo_env = {'CARGO_HOME': build_root / 'cargo-home'}
-
 cargo_build = custom_target(
   'cargo-build',
   build_by_default: true,
   build_always_stale: true,
   output: 'src',
   console: true,
-  env: cargo_env,
   command: [
     cargo,
     'build',
@@ -43,21 +40,6 @@ cargo_build = custom_target(
   ],
 )
 
-custom_target(
-  'cp-binary',
-  depends: cargo_build,
-  build_by_default: true,
-  build_always_stale: true,
-  install: true,
-  install_dir: get_option('bindir'),
-  output: meson.project_name(),
-  command: [
-    'cp',
-    'src' / rust_target / meson.project_name(),
-    '@OUTPUT@',
-  ],
-)
-
 custom_target(
   'cargo-doc',
   build_by_default: false,
@@ -82,4 +64,4 @@ test(
   ],
   is_parallel: false,
   timeout: 1800,
-)
\ No newline at end of file
+)
