commit 433226ecf11220926c691ed30063e2d580085f4c
Author: q66 <q66@chimera-linux.org>
Date:   Thu Jan 12 01:30:10 2023 +0100

    disable test that triggers ubsan in test code

diff --git a/tests/libtest/lib518.c b/tests/libtest/lib518.c
index 22f727a..624b714 100644
--- a/tests/libtest/lib518.c
+++ b/tests/libtest/lib518.c
@@ -48,7 +48,7 @@
 #define DEV_NULL "/dev/null"
 #endif
 
-#if defined(HAVE_GETRLIMIT) && defined(HAVE_SETRLIMIT)
+#if 0
 
 static int *fd = NULL;
 static struct rlimit num_open;
