--- a/1.0.0.rc16-3/dmraid/lib/Makefile.in
+++ b/1.0.0.rc16-3/dmraid/lib/Makefile.in
@@ -87,7 +87,8 @@ $(LIB_EVENTS_SHARED): $(OBJECTS2)
 .PHONY: $(addprefix install_,$(TARGETS)) $(addprefix remove_,$(TARGETS))
 
 install_$(LIB_EVENTS_SHARED): $(LIB_EVENTS_SHARED)
-	$(INSTALL_PROGRAM) -D $< $(DESTDIR)$(libdir)/device-mapper/$(<F)
+	$(INSTALL_DIR) $(DESTDIR)$(libdir)/device-mapper
+	$(INSTALL_PROGRAM) $< $(DESTDIR)$(libdir)/device-mapper/$(<F)
 	$(LN_S) -f device-mapper/$(<F) $(DESTDIR)$(libdir)/$(<F)
 
 remove_$(LIB_EVENTS_SHARED):
@@ -95,7 +96,8 @@ remove_$(LIB_EVENTS_SHARED):
 	$(RM) $(DESTDIR)$(libdir)/$(notdir $(LIB_EVENTS_SHARED))
 
 install_$(LIB_SHARED): $(LIB_SHARED)
-	$(INSTALL_PROGRAM) -D $< $(DESTDIR)$(libdir)/$(<F).$(DMRAID_LIB_VERSION)
+	$(INSTALL_DIR) $(DESTDIR)$(libdir)
+	$(INSTALL_PROGRAM) $< $(DESTDIR)$(libdir)/$(<F).$(DMRAID_LIB_VERSION)
 	$(INSTALL_DIR) $(DESTDIR)$(usrlibdir)
 	$(LN_S) -f $(<F).$(DMRAID_LIB_VERSION) $(DESTDIR)$(libdir)/$(<F).$(DMRAID_LIB_MAJOR)
 	$(LN_S) -f $(USRLIB_RELPATH)$(<F).$(DMRAID_LIB_VERSION) $(DESTDIR)$(usrlibdir)/$(<F)
@@ -106,7 +108,8 @@ remove_$(LIB_SHARED):
 	$(RM) $(DESTDIR)$(libdir)/$(notdir $(LIB_SHARED)).$(DMRAID_LIB_VERSION)
 
 install_$(LIB_STATIC): $(LIB_STATIC)
-	$(INSTALL_DATA) -D $< $(DESTDIR)$(usrlibdir)/$(LIB_STATIC)
+	$(INSTALL_DIR) $(DESTDIR)$(usrlibdir)
+	$(INSTALL_DATA) $< $(DESTDIR)$(usrlibdir)/$(LIB_STATIC)
 
 remove_$(LIB_STATIC):
 	$(RM) $(DESTDIR)$(usrlibdir)/$(notdir $(LIB_STATIC))
