commit 6a1ed7742d066b7dc04dd4a3ec9afbc15b4e8283
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Mon Jan 17 03:51:38 2022 +0100

    drop some tests-breaking nonsense

diff --git a/src/tests/runtest b/src/tests/runtest
index a6e4194..e5a5baf 100755
--- a/src/tests/runtest
+++ b/src/tests/runtest
@@ -37,12 +37,7 @@ ibus-keypress
 test-stress
 "
 IBUS_SCHEMA_FILE='org.freedesktop.ibus.gschema.xml'
-GTK_QUERY_MODULE=gtk-query-immodules-3.0-32
-MACHINE=`uname -m`
-
-if test x"$MACHINE" = xx86_64 ; then
-    GTK_QUERY_MODULE=gtk-query-immodules-3.0-64
-fi
+GTK_QUERY_MODULE=gtk-query-immodules-3.0
 
 retval=0
 
