commit 2486cc8
Author: Renato Botelho do Couto <renato@netgate.com>
Date:   Mon Jun 24 16:56:35 2024

    Fix install

diff --git a/hostapd/Makefile b/hostapd/Makefile
index e37c13b..a05b997 100644
--- a/hostapd/Makefile
+++ b/hostapd/Makefile
@@ -1272,10 +1272,17 @@ ifdef CONFIG_NO_TKIP
 CFLAGS += -DCONFIG_NO_TKIP
 endif
 
+MANDIR = $(DESTDIR)/usr/share/man
+
 $(DESTDIR)$(BINDIR)/%: %
-	install -D $(<) $(@)
+	mkdir -p $(dir $@)
+	install $(<) $(@)
 
 install: $(addprefix $(DESTDIR)$(BINDIR)/,$(ALL))
+	mkdir -p $(DESTDIR)/etc $(MANDIR)/man1 $(MANDIR)/man8
+	install -m 0644 hostapd_cli.1 $(MANDIR)/man1
+	install -m 0644 hostapd.8 $(MANDIR)/man8
+	install -m 0644 hostapd.conf $(DESTDIR)/etc
 
 _OBJS_VAR := OBJS
 include ../src/objs.mk
