commit cec3ddc52e93e33e2351c8fe45f841384f25348e
Author: q66 <q66@chimera-linux.org>
Date:   Fri Dec 22 04:06:26 2023 +0100

    fix a bunch of basename fallout junk

diff --git a/src/basic/conf-files.c b/src/basic/conf-files.c
index 484749d7c..64d62fa88 100644
--- a/src/basic/conf-files.c
+++ b/src/basic/conf-files.c
@@ -21,6 +21,11 @@
 #include "strv.h"
 //#include "terminal-util.h"
 
+#if !defined(__GLIBC__)
+#include <string.h>
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 static int files_add(
                 Hashmap *h,
                 Set *masked,
diff --git a/src/login/logind-inhibit.c b/src/login/logind-inhibit.c
index c1564149d..b8d377740 100644
--- a/src/login/logind-inhibit.c
+++ b/src/login/logind-inhibit.c
@@ -27,6 +27,11 @@
 #include "user-util.h"
 #include "util.h"
 
+#if !defined(__GLIBC__)
+#include <string.h>
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 static void inhibitor_remove_fifo(Inhibitor *i);
 
 int inhibitor_new(Inhibitor **ret, Manager *m, const char* id) {
diff --git a/src/login/logind-seat.c b/src/login/logind-seat.c
index 7b1540276..d1b09571e 100644
--- a/src/login/logind-seat.c
+++ b/src/login/logind-seat.c
@@ -25,6 +25,11 @@
 #include "tmpfile-util.h"
 #include "util.h"
 
+#if !defined(__GLIBC__)
+#include <string.h>
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 int seat_new(Seat** ret, Manager *m, const char *id) {
         _cleanup_(seat_freep) Seat *s = NULL;
         int r;
diff --git a/src/login/logind-session.c b/src/login/logind-session.c
index 8122a69e3..8935b3399 100644
--- a/src/login/logind-session.c
+++ b/src/login/logind-session.c
@@ -43,6 +43,11 @@
 #include "cgroup-setup.h"
 #include "extract-word.h"
 
+#if !defined(__GLIBC__)
+#include <string.h>
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 #define RELEASE_USEC (20*USEC_PER_SEC)
 
 static void session_remove_fifo(Session *s);
diff --git a/src/shared/btrfs-util.c b/src/shared/btrfs-util.c
index bbe2ca7b9..3f431b4c2 100644
--- a/src/shared/btrfs-util.c
+++ b/src/shared/btrfs-util.c
@@ -34,6 +34,11 @@
 //#include "time-util.h"
 //#include "util.h"
 
+#if !defined(__GLIBC__)
+#include <string.h>
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 /* WARNING: Be careful with file system ioctls! When we get an fd, we
  * need to make sure it either refers to only a regular file or
  * directory, or that it is located on btrfs, before invoking any
diff --git a/src/shared/exec-util.c b/src/shared/exec-util.c
index 5a233f66c..fdbcb9a2e 100644
--- a/src/shared/exec-util.c
+++ b/src/shared/exec-util.c
@@ -32,6 +32,11 @@
 //#include "tmpfile-util.h"
 //#include "util.h"
 
+#if !defined(__GLIBC__)
+#include <string.h>
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 /* Put this test here for a lack of better place */
 assert_cc(EAGAIN == EWOULDBLOCK);
 
diff --git a/src/test/test-mountpoint-util.c b/src/test/test-mountpoint-util.c
index b2d1eab68..a6be234b5 100644
--- a/src/test/test-mountpoint-util.c
+++ b/src/test/test-mountpoint-util.c
@@ -19,6 +19,11 @@
 #include "virt.h"
 #include "tmpfile-util.h"
 
+#if !defined(__GLIBC__)
+#include <string.h>
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 static void test_mount_propagation_flags_one(const char *name, int ret, unsigned long expected) {
         unsigned long flags;
 
diff --git a/src/test/test-path-util.c b/src/test/test-path-util.c
index cad532792..0a077da8d 100644
--- a/src/test/test-path-util.c
+++ b/src/test/test-path-util.c
@@ -18,6 +18,11 @@
 #include "tmpfile-util.h"
 #include "util.h"
 
+#if !defined(__GLIBC__)
+#include <string.h>
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 TEST(print_paths) {
         log_info("DEFAULT_PATH=%s", DEFAULT_PATH);
         log_info("DEFAULT_USER_PATH=%s", DEFAULT_USER_PATH);
