From 235bac56a45a14692c6b6eae7187b1b70a1d82cc Mon Sep 17 00:00:00 2001
From: Guilhem Baccialone <guilhem.baccialone@zaclys.net>
Date: Mon, 27 Jan 2025 00:09:29 +0100
Subject: [PATCH 1/1] keep using last blueprint-compiler version available

---
 refine/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/refine/meson.build b/refine/meson.build
index e1e77c1..a946fce 100644
--- a/refine/meson.build
+++ b/refine/meson.build
@@ -22,7 +22,7 @@ blueprints = custom_target('blueprints',
   output: '.',
   command: [
     find_program('blueprint-compiler',
-      version: '0.14.0',
+      version: '>=0.14.0',
       version_argument: '--version',
     ), 'batch-compile', '@OUTPUT@', '@CURRENT_SOURCE_DIR@', '@INPUT@'
   ],
-- 
2.48.1

