--- a/examples/smb.conf.default
+++ b/examples/smb.conf.default
@@ -52,7 +52,7 @@
 
 # this tells Samba to use a separate log file for each machine
 # that connects
-   log file = /usr/local/samba/var/log.%m
+   log file = /var/log/samba/%m.log
 
 # Put a capping on the size of the log files (in Kb).
    max log size = 50
@@ -136,7 +136,7 @@
 # specifically define each individual printer
 [printers]
    comment = All Printers
-   path = /usr/spool/samba
+   path = /var/spool/samba
    browsable = no
 # Change 'guest ok' from 'no' to 'yes' to allow the 'guest account' user to print
    guest ok = no
