commit 57afdc2d80bff39cf72a261a6d077472c8fdac44
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Wed May 31 23:59:14 2023 +0200

    fix implicit function decls

diff --git a/ctdb/common/pkt_read.c b/ctdb/common/pkt_read.c
index 212ace5..630b327 100644
--- a/ctdb/common/pkt_read.c
+++ b/ctdb/common/pkt_read.c
@@ -26,6 +26,7 @@
 
 #include <talloc.h>
 #include <tevent.h>
+#include <unistd.h>
 
 #include "lib/util/tevent_unix.h"
 
diff --git a/ctdb/common/pkt_write.c b/ctdb/common/pkt_write.c
index b1c1730..86d82af 100644
--- a/ctdb/common/pkt_write.c
+++ b/ctdb/common/pkt_write.c
@@ -22,6 +22,7 @@
 
 #include <talloc.h>
 #include <tevent.h>
+#include <unistd.h>
 
 #include "lib/util/tevent_unix.h"
 
diff --git a/ctdb/common/system_socket.c b/ctdb/common/system_socket.c
index 06dc558..5bd5a6d 100644
--- a/ctdb/common/system_socket.c
+++ b/ctdb/common/system_socket.c
@@ -64,6 +64,8 @@
 #include "common/logging.h"
 #include "common/system_socket.h"
 
+#include <unistd.h>
+
 /*
   uint16 checksum for n bytes
  */
diff --git a/ctdb/protocol/protocol_types.c b/ctdb/protocol/protocol_types.c
index 0eb1923..fff92cb 100644
--- a/ctdb/protocol/protocol_types.c
+++ b/ctdb/protocol/protocol_types.c
@@ -22,6 +22,7 @@
 
 #include <talloc.h>
 #include <tdb.h>
+#include <unistd.h>
 
 #include "protocol.h"
 #include "protocol_private.h"
diff --git a/ctdb/server/ctdb_fork.c b/ctdb/server/ctdb_fork.c
index 1065423..ebc7a18 100644
--- a/ctdb/server/ctdb_fork.c
+++ b/ctdb/server/ctdb_fork.c
@@ -23,6 +23,7 @@
 
 #include <talloc.h>
 #include <tevent.h>
+#include <unistd.h>
 
 #include "lib/util/debug.h"
 #include "lib/util/time.h"
diff --git a/lib/cmdline/closefrom_except.c b/lib/cmdline/closefrom_except.c
index fe4e0cc..58aa32f 100644
--- a/lib/cmdline/closefrom_except.c
+++ b/lib/cmdline/closefrom_except.c
@@ -16,6 +16,7 @@
 #include "replace.h"
 #include "closefrom_except.h"
 #include <popt.h>
+#include <unistd.h>
 
 int closefrom_except(int lower, int *fds, size_t num_fds)
 {
diff --git a/lib/printer_driver/printer_driver.c b/lib/printer_driver/printer_driver.c
index 35260c2..637c956 100644
--- a/lib/printer_driver/printer_driver.c
+++ b/lib/printer_driver/printer_driver.c
@@ -22,6 +22,7 @@
 #include "rpc_client/init_spoolss.h"
 #include "libgpo/gpo_ini.h"
 #include "printer_driver.h"
+#include <unistd.h>
 
 #define ADD_TO_ARRAY(mem_ctx, type, elem, array, num) \
 do { \
diff --git a/librpc/rpc/dcesrv_core.c b/librpc/rpc/dcesrv_core.c
index d0e68d5..178c56e 100644
--- a/librpc/rpc/dcesrv_core.c
+++ b/librpc/rpc/dcesrv_core.c
@@ -36,6 +36,7 @@
 #include "system/network.h"
 #include "lib/util/idtree_random.h"
 #include "nsswitch/winbind_client.h"
+#include <unistd.h>
 
 /**
  * @file
diff --git a/nsswitch/wbinfo.c b/nsswitch/wbinfo.c
index de54373..0b7ca8c 100644
--- a/nsswitch/wbinfo.c
+++ b/nsswitch/wbinfo.c
@@ -30,6 +30,7 @@
 #include "lib/afs/afs_settoken.h"
 #include "lib/util/smb_strtox.h"
 #include "lib/util/string_wrappers.h"
+#include <unistd.h>
 
 #ifdef DBGC_CLASS
 #undef DBGC_CLASS
