--- a/examples/smb.conf.default
+++ b/examples/smb.conf.default
@@ -52,7 +52,7 @@
 
 # this tells Samba to use a separate log file for each machine
 # that connects
-   log file = /usr/local/samba/var/log.%m
+   log file = /var/log/samba/%m.log
 
 # Put a capping on the size of the log files (in Kb).
    max log size = 50
@@ -136,7 +136,7 @@
 # specifically define each individual printer
 [printers]
    comment = All Printers
-   path = /usr/spool/samba
+   path = /var/spool/samba
    browseable = no
 # Set public = yes to allow user 'guest account' to print
    guest ok = no
--- a/third_party/waf/waflib/Tools/python.py
+++ b/third_party/waf/waflib/Tools/python.py
@@ -328,7 +328,7 @@ def check_python_headers(conf, features='pyembed pyext'):
 	x = 'MACOSX_DEPLOYMENT_TARGET'
 	if dct[x]:
 		env[x] = conf.environ[x] = dct[x]
-	env.pyext_PATTERN = '%s' + dct['SO'] # not a mistake
+	env.pyext_PATTERN = '%s' + '.so' # not a mistake
 
 
 	# Try to get pythonX.Y-config
