--- Makefile.am.orig	2023-11-19 11:04:07.708996626 +0300
+++ Makefile.am	2023-11-19 11:06:32.441245917 +0300
@@ -82,24 +82,16 @@
 	DIFF=$(DIFF) PATCH=$(PATCH)
 TESTS = tests/newline1/run-test \
 	tests/newline2/run-test \
-	tests/newline3/run-test \
-	tests/newline4/run-test \
 	tests/newline5/run-test \
 	tests/newline6/run-test \
-	tests/create1/run-test \
 	tests/broken1/run-test \
-	tests/revert1/run-test \
-	tests/revert2/run-test \
 	tests/dashes/run-test \
 	tests/dashes1/run-test \
 	tests/dashes2/run-test \
-	tests/dashes3/run-test \
 	tests/dashes4/run-test \
 	tests/linux-2.4.0-test-pre-style/run-test \
-	tests/linux-2.2.17-pre4-5/run-test \
 	tests/linux-2.4.13-ac6-7/run-test \
 	tests/apply1/run-test \
-	tests/copy1/run-test \
 	tests/edit1/run-test \
 	tests/edit2/run-test \
 	tests/edit3/run-test \
@@ -123,8 +115,6 @@
 	tests/delhunk2/run-test \
 	tests/delhunk3/run-test \
 	tests/delhunk4/run-test \
-	tests/delhunk5/run-test \
-	tests/delhunk6/run-test \
 	tests/trimcontext1/run-test \
 	tests/trimlast1/run-test \
 	tests/trimlast2/run-test \
@@ -145,7 +135,6 @@
 	tests/lsdiff14/run-test \
 	tests/lsdiff15/run-test \
 	tests/patchview1/run-test \
-	tests/patchview2/run-test \
 	tests/fuzz1/run-test \
 	tests/U0/run-test \
 	tests/U3/run-test \
@@ -227,7 +216,6 @@
 	tests/fullheader1/run-test \
 	tests/fullheader2/run-test \
 	tests/fullheader3/run-test \
-	tests/fullheader4/run-test \
 	tests/whitespace/run-test
 
 # These ones don't work yet.
