diff --git a/test/gtest/CMakeLists.txt b/test/gtest/CMakeLists.txt
index 5e2714df4..b36b86e00 100644
--- a/test/gtest/CMakeLists.txt
+++ b/test/gtest/CMakeLists.txt
@@ -1,10 +1,3 @@
-if(BUILD_FUZZTEST)
-  include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/third_party/fuzztest)
-else()
-  include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/third_party/googletest/googletest/include)
-  include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/third_party/googletest/googlemock/include)
-endif()
-
 set(unittest_SOURCES
   arena.cpp
   cast-check.cpp
diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index fde5276..1de1dc6 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -1,16 +1,4 @@
-if(BUILD_FUZZTEST)
-  add_subdirectory(fuzztest)
-elseif(BUILD_TESTS)
-  # fuzztest includes gtest, but if we're not building fuzztest, build gtest ourselves.
-  add_library(gtest STATIC googletest/googletest/src/gtest-all.cc)
-  add_library(gtest_main STATIC googletest/googletest/src/gtest_main.cc)
-  target_compile_options(gtest PRIVATE "-fno-rtti")
-  target_compile_options(gtest_main PRIVATE "-fno-rtti")
-  include_directories(
-    googletest/googletest
-    googletest/googletest/include
-  )
-endif()
+find_package(GTest)
 
 if(BUILD_LLVM_DWARF)
   add_subdirectory(llvm-project)
