diff --git a/helpers/wl-kbptr-sway-active-win b/helpers/wl-kbptr-sway-active-win
index 75a3f3d..bddd2fe 100755
--- a/helpers/wl-kbptr-sway-active-win
+++ b/helpers/wl-kbptr-sway-active-win
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 active_area=$(swaymsg -t get_tree | jq --raw-output \
 	'.. | (.nodes? + .floating_nodes? // empty)[] | select (.focused) | .rect | "\(.width)x\(.height)+\(.x)+\(.y)"')
