commit d1f6c0a8e4efff8e2b4910b929c2c2b744b603d4
Author: q66 <q66@chimera-linux.org>
Date:   Tue Nov 26 20:50:48 2024 +0100

    replace unsafe use of atol

diff --git a/gettext-tools/src/x-c.c b/gettext-tools/src/x-c.c
index 4137f8e..2401712 100644
--- a/gettext-tools/src/x-c.c
+++ b/gettext-tools/src/x-c.c
@@ -1684,7 +1684,7 @@ phase5_get (token_ty *tp)
             break;
           }
         tp->type = token_type_number;
-        tp->number = atol (sb_xdupfree_c (&buffer));
+        tp->number = strtol (sb_xdupfree_c (&buffer), NULL, 10);
       }
       return;
 
