--- a/check.t
+++ b/check.t
@@ -176,23 +176,6 @@ stdin:
 expected-stdout:
 	;
 ---
-name: selftest-tty-absent
-description:
-	Check that a controlling tty is not present as regress:no-ctty was used
-	(if this test fails for you DO NOT PASS regress:no-ctty and fix every
-	other test that fails: why u use it if u haz ctty?)
-category: regress:no-ctty
-env-setup: !ENV=./envf!
-file-setup: file 644 "envf"
-	PS1=X
-arguments: !-i!
-stdin:
-	echo ok
-expected-stdout:
-	ok
-expected-stderr-pattern:
-	/ksh: warning: won't have full job control\nXX/
----
 name: selftest-tty-present
 description:
 	Check that a controlling tty is present as regress:no-ctty was not used
