diff -ru a/meson.build b/meson.build
--- a/meson.build	2024-06-11 22:27:18.000000000 +0800
+++ b/meson.build	2024-09-17 15:44:16.689878646 +0800
@@ -595,7 +595,7 @@
 gjs_private_gir = gnome.generate_gir(libgjs,
     includes: ['GObject-2.0', 'Gio-2.0'], sources: libgjs_private_sources,
     namespace: 'CjsPrivate', nsversion: '1.0', identifier_prefix: 'Gjs',
-    symbol_prefix: 'gjs_', extra_args: '--warn-error', install: true,
+    symbol_prefix: 'gjs_', install: true,
     install_dir_gir: false, install_dir_typelib: pkglibdir / 'girepository-1.0')
 gjs_private_typelib = gjs_private_gir[1]
 
