commit 5b896470d833238785556925915065596fa1dd2e
Author: q66 <q66@chimera-linux.org>
Date:   Mon Mar 25 03:11:28 2024 +0100

    disable systemd session stuff

diff --git a/plugins/sharing/gsd-sharing-manager.c b/plugins/sharing/gsd-sharing-manager.c
index 7713569..7cdc863 100644
--- a/plugins/sharing/gsd-sharing-manager.c
+++ b/plugins/sharing/gsd-sharing-manager.c
@@ -20,7 +20,6 @@
 #include "config.h"
 
 #include <locale.h>
-#include <systemd/sd-login.h>
 #include <glib.h>
 #include <gio/gio.h>
 #include <gio/gdesktopappinfo.h>
@@ -1066,6 +1065,7 @@ manage_configurable_services (GsdSharingManager *manager)
 static void
 manage_assigned_services (GsdSharingManager *manager)
 {
+#if 0
         size_t i;
         int ret;
         g_autofree char *session_id = NULL;
@@ -1126,6 +1126,7 @@ manage_assigned_services (GsdSharingManager *manager)
 
                 g_hash_table_insert (manager->assigned_services, (gpointer) service->system_bus_name, info);
         }
+#endif
 }
 
 static void
@@ -1144,12 +1144,7 @@ gsd_sharing_manager_init (GsdSharingManager *manager)
         manager->carrier_type = g_strdup ("");
         manager->sharing_status = GSD_SHARING_STATUS_OFFLINE;
 
-        ret = sd_pid_get_user_unit (getpid (), &systemd_unit);
-
-        if (ret < 0)
-                manager->is_systemd_managed = FALSE;
-        else
-                manager->is_systemd_managed = TRUE;
+        manager->is_systemd_managed = FALSE;
 
         manage_configurable_services (manager);
         manage_assigned_services (manager);
