commit c3939493b3f80807bfa3d57d00b83a43e9bbe72c
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sun Mar 19 18:17:03 2023 +0100

    fix pkgconfig version for apk compat

diff --git a/meson.build b/meson.build
index d6b8d5a..6fdc76d 100644
--- a/meson.build
+++ b/meson.build
@@ -1,6 +1,6 @@
 project(
   'gnome-settings-daemon', 'c',
-  version: '44.rc',
+  version: '44.0',
   license: [ 'GPL2+', 'LGPLv2+' ],
   meson_version: '>= 0.57.0'
 )
