commit a671444c922dbb829831021f9129a1c5d44ad170
Author: q66 <q66@chimera-linux.org>
Date:   Tue May 16 01:55:33 2023 +0200

    migrate regexp for qt6

diff --git a/apps/launcher/graphicspage.cpp b/apps/launcher/graphicspage.cpp
index ebb031e..c5df958 100644
--- a/apps/launcher/graphicspage.cpp
+++ b/apps/launcher/graphicspage.cpp
@@ -200,10 +200,11 @@ void Launcher::GraphicsPage::saveSettings()
     int cWidth = 0;
     int cHeight = 0;
     if (standardRadioButton->isChecked()) {
-        QRegExp resolutionRe(QString("(\\d+) x (\\d+).*"));
-        if (resolutionRe.exactMatch(resolutionComboBox->currentText().simplified())) {
-            cWidth = resolutionRe.cap(1).toInt();
-            cHeight = resolutionRe.cap(2).toInt();
+        QRegularExpression resolutionRe(QRegularExpression::anchoredPattern(QString("(\\d+) x (\\d+).*")));
+        QRegularExpressionMatch match = resolutionRe.match(resolutionComboBox->currentText().simplified());
+        if (match.hasMatch()) {
+            cWidth = match.captured(1).toInt();
+            cHeight = match.captured(2).toInt();
         }
     } else {
         cWidth = customWidthSpinBox->value();
diff --git a/apps/launcher/utils/textinputdialog.cpp b/apps/launcher/utils/textinputdialog.cpp
index 70b8275..c022bba 100644
--- a/apps/launcher/utils/textinputdialog.cpp
+++ b/apps/launcher/utils/textinputdialog.cpp
@@ -20,7 +20,8 @@ Launcher::TextInputDialog::TextInputDialog(const QString& title, const QString &
     label->setText(text);
 
     // Line edit
-    QValidator *validator = new QRegExpValidator(QRegExp("^[a-zA-Z0-9_]*$"), this); // Alpha-numeric + underscore
+    QValidator* validator
+        = new QRegularExpressionValidator(QRegularExpression("^[a-zA-Z0-9_]*$"), this); // Alpha-numeric + underscore
     mLineEdit = new LineEdit(this);
     mLineEdit->setValidator(validator);
     mLineEdit->setCompleter(nullptr);
diff --git a/apps/wizard/inisettings.cpp b/apps/wizard/inisettings.cpp
index e9cec12..f423f07 100644
--- a/apps/wizard/inisettings.cpp
+++ b/apps/wizard/inisettings.cpp
@@ -4,7 +4,7 @@
 #include <QFile>
 #include <QStringList>
 #include <QString>
-#include <QRegExp>
+#include <QRegularExpression>
 #include <QDebug>
 
 Wizard::IniSettings::IniSettings()
@@ -35,12 +35,12 @@ bool Wizard::IniSettings::readFile(QTextStream &stream)
     // Look for a square bracket, "'\\["
     // that has one or more "not nothing" in it, "([^]]+)"
     // and is closed with a square bracket, "\\]"
-    QRegExp sectionRe(QLatin1String("^\\[([^]]+)\\]"));
+    QRegularExpression sectionRe(QRegularExpression::anchoredPattern("^\\[([^]]+)\\]"));
 
     // Find any character(s) that is/are not equal sign(s), "[^=]+"
     // followed by an optional whitespace, an equal sign, and another optional whitespace, "\\s*=\\s*"
     // and one or more periods, "(.+)"
-    QRegExp keyRe(QLatin1String("^([^=]+)\\s*=\\s*(.+)$"));
+    QRegularExpression keyRe(QLatin1String("^([^=]+)\\s*=\\s*(.+)$"));
 
     QString currentSection;
 
@@ -51,14 +51,18 @@ bool Wizard::IniSettings::readFile(QTextStream &stream)
         if (line.isEmpty() || line.startsWith(QLatin1Char(';')))
             continue;
 
-        if (sectionRe.exactMatch(line))
+        QRegularExpressionMatch sectionMatch = sectionRe.match(line);
+        if (sectionMatch.hasMatch())
         {
-            currentSection = sectionRe.cap(1);
+            currentSection = sectionMatch.captured(1);
+            continue;
         }
-        else if (keyRe.indexIn(line) != -1)
+
+        QRegularExpressionMatch match = keyRe.match(line);
+        if (match.hasMatch())
         {
-            QString key = keyRe.cap(1).trimmed();
-            QString value = keyRe.cap(2).trimmed();
+            QString key = match.captured(1).trimmed();
+            QString value = match.captured(2).trimmed();
 
             // Append the section, but only if there is one
             if (!currentSection.isEmpty())
@@ -76,12 +80,12 @@ bool Wizard::IniSettings::writeFile(const QString &path, QTextStream &stream)
     // Look for a square bracket, "'\\["
     // that has one or more "not nothing" in it, "([^]]+)"
     // and is closed with a square bracket, "\\]"
-    QRegExp sectionRe(QLatin1String("^\\[([^]]+)\\]"));
+    QRegularExpression sectionRe(QRegularExpression::anchoredPattern("^\\[([^]]+)\\]"));
 
     // Find any character(s) that is/are not equal sign(s), "[^=]+"
     // followed by an optional whitespace, an equal sign, and another optional whitespace, "\\s*=\\s*"
     // and one or more periods, "(.+)"
-    QRegExp keyRe(QLatin1String("^([^=]+)\\s*=\\s*(.+)$"));
+    QRegularExpression keyRe(QLatin1String("^([^=]+)\\s*=\\s*(.+)$"));
 
     const QStringList keys(mSettings.keys());
 
@@ -97,11 +101,16 @@ bool Wizard::IniSettings::writeFile(const QString &path, QTextStream &stream)
             continue;
         }
 
-        if (sectionRe.exactMatch(line)) {
+        QRegularExpressionMatch sectionMatch = sectionRe.match(line);
+        if (sectionMatch.hasMatch()) {
             buffer.append(line + QLatin1String("\n"));
-            currentSection = sectionRe.cap(1);
-        } else  if (keyRe.indexIn(line) != -1) {
-            QString key(keyRe.cap(1).trimmed());
+            currentSection = sectionMatch.captured(1);
+            continue;
+        }
+
+        QRegularExpressionMatch match = keyRe.match(line);
+        if (match.hasMatch()) {
+            QString key(match.captured(1).trimmed());
             QString lookupKey(key);
 
             // Append the section, but only if there is one
diff --git a/components/config/gamesettings.cpp b/components/config/gamesettings.cpp
index 8717a68..6276e27 100644
--- a/components/config/gamesettings.cpp
+++ b/components/config/gamesettings.cpp
@@ -3,7 +3,7 @@
 
 #include <QTextCodec>
 #include <QDir>
-#include <QRegExp>
+#include <QRegularExpression>
 
 #include <components/files/configurationmanager.hpp>
 
@@ -83,7 +83,7 @@ bool Config::GameSettings::readUserFile(QTextStream &stream)
 bool Config::GameSettings::readFile(QTextStream &stream, QMultiMap<QString, QString> &settings)
 {
     QMultiMap<QString, QString> cache;
-    QRegExp keyRe("^([^=]+)\\s*=\\s*(.+)$");
+    QRegularExpression keyRe("^([^=]+)\\s*=\\s*(.+)$");
 
     while (!stream.atEnd()) {
         QString line = stream.readLine();
@@ -91,10 +91,11 @@ bool Config::GameSettings::readFile(QTextStream &stream, QMultiMap<QString, QStr
         if (line.isEmpty() || line.startsWith("#"))
             continue;
 
-        if (keyRe.indexIn(line) != -1) {
+        QRegularExpressionMatch match = keyRe.match(line);
+        if (match.hasMatch()) {
 
-            QString key = keyRe.cap(1).trimmed();
-            QString value = keyRe.cap(2).trimmed();
+            QString key = match.captured(1).trimmed();
+            QString value = match.captured(2).trimmed();
 
             // Don't remove composing entries
             if (key != QLatin1String("data")
@@ -196,13 +197,13 @@ bool Config::GameSettings::writeFile(QTextStream &stream)
 
 bool Config::GameSettings::isOrderedLine(const QString& line)
 {
-    return line.contains(QRegExp("^\\s*fallback-archive\\s*="))
-           || line.contains(QRegExp("^\\s*fallback\\s*="))
-           || line.contains(QRegExp("^\\s*data\\s*="))
-           || line.contains(QRegExp("^\\s*data-local\\s*="))
-           || line.contains(QRegExp("^\\s*resources\\s*="))
-           || line.contains(QRegExp("^\\s*groundcover\\s*="))
-           || line.contains(QRegExp("^\\s*content\\s*="));
+    return line.contains(QRegularExpression("^\\s*fallback-archive\\s*="))
+        || line.contains(QRegularExpression("^\\s*fallback\\s*="))
+        || line.contains(QRegularExpression("^\\s*data\\s*="))
+        || line.contains(QRegularExpression("^\\s*data-local\\s*="))
+        || line.contains(QRegularExpression("^\\s*resources\\s*="))
+        || line.contains(QRegularExpression("^\\s*groundcover\\s*="))
+        || line.contains(QRegularExpression("^\\s*content\\s*="));
 }
 
 // Policy:
@@ -261,7 +262,7 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
     //        +----------------------------------------------------------+
     //
     //
-    QRegExp settingRegex("^([^=]+)\\s*=\\s*([^,]+)(.*)$");
+    QRegularExpression settingRegex("^([^=]+)\\s*=\\s*([^,]+)(.*)$");
     std::vector<QString> comments;
     auto commentStart = fileCopy.end();
     std::map<QString, std::vector<QString> > commentsMap;
@@ -272,9 +273,10 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
             // save in a separate map of comments keyed by "ordered" line
             if (!comments.empty())
             {
-                if (settingRegex.indexIn(*iter) != -1)
+                QRegularExpressionMatch match = settingRegex.match(*iter);
+                if (match.hasMatch())
                 {
-                    commentsMap[settingRegex.cap(1)+"="+settingRegex.cap(2)] = comments;
+                    commentsMap[match.captured(1) + "=" + match.captured(2)] = comments;
                     comments.clear();
                     commentStart = fileCopy.end();
                 }
@@ -283,14 +285,14 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
 
             *iter = QString(); // "ordered" lines to be removed later
         }
-        else if ((*iter).isEmpty() || (*iter).contains(QRegExp("^\\s*#")))
+        else if ((*iter).isEmpty() || (*iter).contains(QRegularExpression("^\\s*#")))
         {
             // comment line, save in temp buffer
             if (comments.empty())
                 commentStart = iter;
 
             // special removed content processing
-            if ((*iter).contains(QRegExp("^##content\\s*=")))
+            if ((*iter).contains(QRegularExpression("^##content\\s*=")))
             {
                 if (!comments.empty())
                 {
@@ -306,11 +308,11 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
         }
         else
         {
-            int index = settingRegex.indexIn(*iter);
+            QRegularExpressionMatch match = settingRegex.match(*iter);
 
             // blank or non-"ordered" line, write saved comments
-            if (!comments.empty() && index != -1 && settingRegex.captureCount() >= 2 &&
-                mUserSettings.find(settingRegex.cap(1)) != mUserSettings.end())
+            if (!comments.empty() && match.hasMatch() && settingRegex.captureCount() >= 2
+                && mUserSettings.find(match.captured(1)) != mUserSettings.end())
             {
                 if (commentStart == fileCopy.end())
                     throw std::runtime_error("Config::GameSettings: failed to parse settings - iterator is past of end of settings file");
@@ -327,10 +329,10 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
             // keep blank lines and non-"ordered" lines other than comments
 
             // look for a key in the line
-            if (index == -1 || settingRegex.captureCount() < 2)
+            if (!match.hasMatch() || settingRegex.captureCount() < 2)
             {
                 // no key or first part of value found in line, replace with a null string which
-                // will be remved later
+                // will be removed later
                 *iter = QString();
                 comments.clear();
                 commentStart = fileCopy.end();
@@ -339,15 +341,16 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
 
             // look for a matching key in user settings
             *iter = QString(); // assume no match
-            QString key = settingRegex.cap(1);
-            QString keyVal = settingRegex.cap(1)+"="+settingRegex.cap(2);
+            QString key = match.captured(1);
+            QString keyVal = match.captured(1) + "=" + match.captured(2);
             QMultiMap<QString, QString>::const_iterator i = mUserSettings.find(key);
             while (i != mUserSettings.end() && i.key() == key)
             {
                 QString settingLine = i.key() + "=" + i.value();
-                if (settingRegex.indexIn(settingLine) != -1)
+                QRegularExpressionMatch keyMatch = settingRegex.match(settingLine);
+                if (keyMatch.hasMatch())
                 {
-                    if ((settingRegex.cap(1)+"="+settingRegex.cap(2)) == keyVal)
+                    if ((keyMatch.captured(1) + "=" + keyMatch.captured(2)) == keyVal)
                     {
                         *iter = settingLine;
                         break;
@@ -366,7 +369,7 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
 
         // Below is based on readFile() code, if that changes corresponding change may be
         // required (for example duplicates may be inserted if the rules don't match)
-        if (/*(*iter).isEmpty() ||*/ iter.contains(QRegExp("^\\s*#")))
+        if (/*(*iter).isEmpty() ||*/ iter.contains(QRegularExpression("^\\s*#")))
         {
             stream << iter << "\n";
             continue;
@@ -406,13 +409,14 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
         else
             settingLine = it.key() + "=" + it.value();
 
-        if (settingRegex.indexIn(settingLine) != -1)
+        QRegularExpressionMatch match = settingRegex.match(settingLine);
+        if (match.hasMatch())
         {
-            auto i = commentsMap.find(settingRegex.cap(1)+"="+settingRegex.cap(2));
+            auto i = commentsMap.find(match.captured(1) + "=" + match.captured(2));
 
             // check if previous removed content item with comments
             if (i == commentsMap.end())
-                i = commentsMap.find("##"+settingRegex.cap(1)+"="+settingRegex.cap(2));
+                i = commentsMap.find("##" + match.captured(1) + "=" + match.captured(2));
 
             if (i != commentsMap.end())
             {
@@ -433,7 +437,7 @@ bool Config::GameSettings::writeFileWithComments(QFile &file)
         auto i = commentsMap.begin();
         for (; i != commentsMap.end(); ++i)
         {
-            if (i->first.contains(QRegExp("^\\s*content\\s*=")))
+            if (i->first.contains(QRegularExpression("^\\s*content\\s*=")))
             {
                 std::vector<QString> cLines = i->second;
                 for (const auto & cLine : cLines)
diff --git a/components/config/launchersettings.cpp b/components/config/launchersettings.cpp
index 025bc43..3314adf 100644
--- a/components/config/launchersettings.cpp
+++ b/components/config/launchersettings.cpp
@@ -1,12 +1,11 @@
 #include "launchersettings.hpp"
 
+#include <QDebug>
 #include <QTextStream>
 #include <QString>
-#include <QRegExp>
+#include <QRegularExpression>
 #include <QMultiMap>
 
-#include <QDebug>
-
 const char Config::LauncherSettings::sCurrentContentListKey[] = "Profiles/currentprofile";
 const char Config::LauncherSettings::sLauncherConfigFileName[] = "launcher.cfg";
 const char Config::LauncherSettings::sContentListsSectionPrefix[] = "Profiles/";
@@ -17,16 +16,16 @@ QStringList Config::LauncherSettings::subKeys(const QString &key)
     QMultiMap<QString, QString> settings = SettingsBase::getSettings();
     QStringList keys = settings.uniqueKeys();
 
-    QRegExp keyRe("(.+)/");
+    QRegularExpression keyRe("(.+)/");
 
     QStringList result;
 
     for (const QString &currentKey : keys)
     {
-
-        if (keyRe.indexIn(currentKey) != -1)
+        QRegularExpressionMatch match = keyRe.match(currentKey);
+        if (match.hasMatch())
         {
-            QString prefixedKey = keyRe.cap(1);
+            QString prefixedKey = match.captured(1);
 
             if(prefixedKey.startsWith(key))
             {
@@ -45,7 +44,7 @@ QStringList Config::LauncherSettings::subKeys(const QString &key)
 bool Config::LauncherSettings::writeFile(QTextStream &stream)
 {
     QString sectionPrefix;
-    QRegExp sectionRe("([^/]+)/(.+)$");
+    QRegularExpression sectionRe(QRegularExpression::anchoredPattern("([^/]+)/(.+)$"));
     QMultiMap<QString, QString> settings = SettingsBase::getSettings();
 
     QMapIterator<QString, QString> i(settings);
@@ -57,9 +56,10 @@ bool Config::LauncherSettings::writeFile(QTextStream &stream)
         QString prefix;
         QString key;
 
-        if (sectionRe.exactMatch(i.key())) {
-             prefix = sectionRe.cap(1);
-             key = sectionRe.cap(2);
+        QRegularExpressionMatch match = sectionRe.match(i.key());
+        if (match.hasMatch()) {
+            prefix = match.captured(1);
+            key = match.captured(2);
         }
 
         // Get rid of legacy settings
diff --git a/components/config/settingsbase.hpp b/components/config/settingsbase.hpp
index 86fa962..1c36b31 100644
--- a/components/config/settingsbase.hpp
+++ b/components/config/settingsbase.hpp
@@ -4,7 +4,7 @@
 #include <QTextStream>
 #include <QStringList>
 #include <QString>
-#include <QRegExp>
+#include <QRegularExpression>
 #include <QMultiMap>
 
 namespace Config
@@ -54,8 +54,8 @@ namespace Config
 
             QString sectionPrefix;
 
-            QRegExp sectionRe("^\\[([^]]+)\\]");
-            QRegExp keyRe("^([^=]+)\\s*=\\s*(.+)$");
+            QRegularExpression sectionRe(QRegularExpression::anchoredPattern("^\\[([^]]+)\\]"));
+            QRegularExpression keyRe("^([^=]+)\\s*=\\s*(.+)$");
 
             while (!stream.atEnd()) {
                 QString line = stream.readLine();
@@ -63,16 +63,18 @@ namespace Config
                 if (line.isEmpty() || line.startsWith("#"))
                     continue;
 
-                if (sectionRe.exactMatch(line)) {
-                    sectionPrefix = sectionRe.cap(1);
+                QRegularExpressionMatch sectionMatch = sectionRe.match(line);
+                if (sectionMatch.hasMatch()) {
+                    sectionPrefix = sectionMatch.captured(1);
                     sectionPrefix.append("/");
                     continue;
                 }
 
-                if (keyRe.indexIn(line) != -1) {
+                QRegularExpressionMatch match = keyRe.match(line);
+                if (match.hasMatch()) {
 
-                    QString key = keyRe.cap(1).trimmed();
-                    QString value = keyRe.cap(2).trimmed();
+                    QString key = match.captured(1).trimmed();
+                    QString value = match.captured(2).trimmed();
 
                     if (!sectionPrefix.isEmpty())
                         key.prepend(sectionPrefix);
diff --git a/components/contentselector/model/esmfile.cpp b/components/contentselector/model/esmfile.cpp
index 46a7c96..3991412 100644
--- a/components/contentselector/model/esmfile.cpp
+++ b/components/contentselector/model/esmfile.cpp
@@ -2,6 +2,7 @@
 
 #include <QMimeData>
 #include <QDataStream>
+#include <QIODevice>
 
 int ContentSelectorModel::EsmFile::sPropertyCount = 7;
 QString ContentSelectorModel::EsmFile::sToolTip = QString("<b>Author:</b> %1<br/> \
diff --git a/components/contentselector/view/combobox.cpp b/components/contentselector/view/combobox.cpp
index 742e236..01bd547 100644
--- a/components/contentselector/view/combobox.cpp
+++ b/components/contentselector/view/combobox.cpp
@@ -6,7 +6,8 @@
 ContentSelectorView::ComboBox::ComboBox(QWidget *parent) :
     QComboBox(parent)
 {
-    mValidator = new QRegExpValidator(QRegExp("^[a-zA-Z0-9_]*$"), this); // Alpha-numeric + underscore
+    mValidator
+        = new QRegularExpressionValidator(QRegularExpression("^[a-zA-Z0-9_]*$"), this); // Alpha-numeric + underscore
     setValidator(mValidator);
     setEditable(true);
     setCompleter(nullptr);
diff --git a/components/contentselector/view/combobox.hpp b/components/contentselector/view/combobox.hpp
index 9af3c83..c25f5be 100644
--- a/components/contentselector/view/combobox.hpp
+++ b/components/contentselector/view/combobox.hpp
@@ -5,7 +5,7 @@
 #include <QStylePainter>
 
 class QString;
-class QRegExpValidator;
+class QRegularExpressionValidator;
 
 namespace ContentSelectorView
 {
@@ -23,7 +23,7 @@ namespace ContentSelectorView
 
     protected:
         void paintEvent(QPaintEvent *) override;
-        QRegExpValidator *mValidator;
+        QRegularExpressionValidator* mValidator;
     };
 }
 
diff --git a/components/contentselector/view/contentselector.cpp b/components/contentselector/view/contentselector.cpp
index 6bb8e6e..9e0b485 100644
--- a/components/contentselector/view/contentselector.cpp
+++ b/components/contentselector/view/contentselector.cpp
@@ -63,7 +63,7 @@ void ContentSelectorView::ContentSelector::buildAddonView()
     ui.addonView->setVisible (true);
 
     mAddonProxyModel = new AddOnProxyModel(this);
-    mAddonProxyModel->setFilterRegExp(searchFilter()->text());
+    mAddonProxyModel->setFilterRegularExpression(searchFilter()->text());
     mAddonProxyModel->setFilterCaseSensitivity(Qt::CaseInsensitive);
     mAddonProxyModel->setDynamicSortFilter (true);
     mAddonProxyModel->setSourceModel (mContentModel);
