commit 98ac4b1967ed10d5b5f0b9a23d3f1ccf3c539df2
Author: q66 <q66@chimera-linux.org>
Date:   Fri Sep 13 14:04:01 2024 +0200

    install(1): use getopt_long in gnumode

--- a/src.freebsd/coreutils/xinstall/xinstall.c
+++ b/src.freebsd/coreutils/xinstall/xinstall.c
@@ -71,6 +71,7 @@ static char sccsid[] = "@(#)xinstall.c	8.1 (Berkeley) 7/21/93";
 #include <sysexits.h>
 #include <unistd.h>
 #include <vis.h>
+#include <getopt.h>
 
 /*
  * Memory strategy threshold, in pages: if physmem is larger then this, use a
@@ -169,6 +170,15 @@ static int	parseid(const char *, id_t *);
 static int	strip(const char *, int, const char *, char **);
 static void	usage(void);
 
+static int
+do_getopt(int argc, char **argv)
+{
+	if (!gnumode)
+		return getopt(argc, argv, "B:bCcD:dg:h:l:M:m:o:pSsT:Uv");
+	/* use getopt_long to get argument interposition */
+	return getopt_long(argc, argv, "B:bCcDdg:l:m:o:pSst:Tv", NULL, NULL);
+}
+
 int
 main(int argc, char *argv[])
 {
@@ -179,20 +189,14 @@ main(int argc, char *argv[])
 	u_int iflags;
 	char *p;
 	const char *to_name;
-	const char *getopt_str;
 	int notarg = 0;
 
 	gnumode = !!strcmp(getprogname(), "binstall");
-	if (gnumode)
-		getopt_str = "B:bCcDdg:l:m:o:pSst:Tv";
-	else
-		getopt_str = "B:bCcD:dg:h:l:M:m:o:pSsT:Uv";
-
 	fset = 0;
 	iflags = 0;
 	set = NULL;
 	group = owner = NULL;
-	while ((ch = getopt(argc, argv, getopt_str)) != -1)
+	while ((ch = do_getopt(argc, argv)) != -1)
 		switch((char)ch) {
 		case 'B':
 			suffix = optarg;
@@ -1539,13 +1543,13 @@ static void
 usage(void)
 {
 	(void)fprintf(stderr,
-"usage: install [-bCcDpSsTv] [-f flags] [-g group] [-m mode] [-o owner]\n"
+"usage: install [-bCcDpSsTv] [-g group] [-m mode] [-o owner]\n"
 "               [-B suffix] [-l linkflags] [-N dbdir]\n"
 "               file1 file2\n"
-"       install [-bCcDpSsv] [-f flags] [-g group] [-m mode] [-o owner]\n"
+"       install [-bCcDpSsv] [-g group] [-m mode] [-o owner]\n"
 "               [-B suffix] [-l linkflags] [-N dbdir]\n"
 "               file1 ... fileN directory\n"
-"       install [-bCcDpSsv] [-f flags] [-g group] [-m mode] [-o owner]\n"
+"       install [-bCcDpSsv] [-g group] [-m mode] [-o owner]\n"
 "               [-B suffix] [-l linkflags] [-N dbdir] -t directory\n"
 "               file1 ... fileN\n"
 "       install -d [-v] [-g group] [-m mode] [-N dbdir] [-o owner]\n"
