commit 43ce6d964d69e941be9ae0d18dd84cd6a4633f0b
Author: q66 <q66@chimera-linux.org>
Date:   Mon Jul 15 14:26:42 2024 +0200

    jot(1): fix random flag

diff --git a/src.freebsd/jot/jot.c b/src.freebsd/jot/jot.c
index 6f333da..976bd25 100644
--- a/src.freebsd/jot/jot.c
+++ b/src.freebsd/jot/jot.c
@@ -323,9 +323,10 @@ main(int argc, char **argv)
 			if (use_random)
 				y = random() / divisor;
 			else {
-				if (getrandom(&y, sizeof(y), 0) < 0)
+				uint32_t yv;
+				if (getrandom(&yv, sizeof(yv), 0) < 0)
 					err(1, "getrandom");
-				y /= divisor;
+				y = yv / divisor;
 			}
 			if (putdata(y * x + begin, !(reps - i)))
 				errx(1, "range error in conversion");
