commit e08073264132f69c4217e7c0060d3c2ce36a1af0
Author: q66 <q66@chimera-linux.org>
Date:   Mon Sep 30 00:16:56 2024 +0200

    realpath: handle empty paths

diff --git a/src.custom/realpath/realpath.cc b/src.custom/realpath/realpath.cc
index 4c84123..65be494 100644
--- a/src.custom/realpath/realpath.cc
+++ b/src.custom/realpath/realpath.cc
@@ -102,7 +102,14 @@ static bool do_realpath(fs::path sp, bool newl) {
     fs::path np;
     std::error_code ec{};
     /* then do the actual resolution */
-    if (strip && sp.is_relative()) {
+    if (sp.empty()) {
+        /* empty paths should issue ENOENT regardless of strip, like gnu */
+        errno = ENOENT;
+        if (!quiet) {
+            warn("''");
+        }
+        return false;
+    } if (strip && sp.is_relative()) {
         /* no symlinks are expanded + relative input */
         np = (fs::current_path(ec) / sp).lexically_normal();
     } else if (strip) {
