commit d22f7a31358689026d97d50785ccaebe1bdcbef1
Author: q66 <q66@chimera-linux.org>
Date:   Fri Aug 30 17:07:32 2024 +0200

    locate updatedb: create /var/db if needed

diff --git a/src.freebsd/findutils/locate/updatedb.sh b/src.freebsd/findutils/locate/updatedb.sh
index cb2d68f..d77491d 100644
--- a/src.freebsd/findutils/locate/updatedb.sh
+++ b/src.freebsd/findutils/locate/updatedb.sh
@@ -10,8 +10,10 @@ echo "Rebuilding locate database..."
 . /etc/locate.rc
 : ${FCODES:="/var/db/locate.database"}
 locdb="$FCODES"
+locbase=$(dirname "$locdb")
 
-touch "$locdb" && rc=0 || rc=3
+mkdir -p "$locbase" && rc=0 || rc=3
+touch "$locdb" || rc=3
 chown nobody "$locdb" || rc=3
 chmod 644 "$locdb" || rc=3
 
