diff --git a/beets/_version.py b/beets/_version.py
index 4dea560..77199ae 100644
--- a/beets/_version.py
+++ b/beets/_version.py
@@ -3,5 +3,5 @@
 # Placeholders are replaced during substitution.
 # Run `git update-index --assume-unchanged beets/_version.py`
 # to ignore local changes to this file.
-__version__ = "0.0.0"
-__version_tuple__ = (0, 0, 0)
+__version__ = "2.5.0"
+__version_tuple__ = (2, 5, 0)
diff --git a/pyproject.toml b/pyproject.toml
index 62b5ac2..df8a8ef 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -157,17 +157,9 @@ web = ["flask", "flask-cors"]
 beet = "beets.ui:main"
 
 
-[tool.poetry-dynamic-versioning]
-enable = true
-vcs = "git"
-format = "{base}.dev{distance}+{commit}"
-
-[tool.poetry-dynamic-versioning.files."beets/_version.py"]
-persistent-substitution = true
-
 [build-system]
-requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning>=1.0.0,<2.0.0"]
-build-backend = "poetry_dynamic_versioning.backend"
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
 
 [tool.pipx-install]
 poethepoet = ">=0.26"
