From cddc7e9ea835929fa3031336c8454a6abe5514e8 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <dkolesa@igalia.com>
Date: Wed, 7 Jul 2021 15:45:58 +0200
Subject: [PATCH] enforce compiler-rt libcc

---
 configure | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/configure b/configure
index a5231a0..ca2efcd 100755
--- a/configure
+++ b/configure
@@ -592,11 +592,7 @@ tryldflag LDFLAGS_AUTO -Wl,--exclude-libs=ALL
 tryldflag LDFLAGS_AUTO -Wl,--dynamic-list="$srcdir/dynamic.list"
 
 # Find compiler runtime library
-test -z "$LIBCC" && tryldflag LIBCC -lgcc && tryldflag LIBCC -lgcc_eh
-test -z "$LIBCC" && tryldflag LIBCC -lcompiler_rt
-test -z "$LIBCC" && try_libcc=`$CC -print-libgcc-file-name 2>/dev/null` \
-                 && tryldflag LIBCC "$try_libcc"
-test -z "$LIBCC" && try_libcc=`$CC -print-file-name=libpcc.a 2>/dev/null` \
+test -z "$LIBCC" && try_libcc=`$CC $LIBCC_LDFLAGS -print-libgcc-file-name 2>/dev/null` \
                  && tryldflag LIBCC "$try_libcc"
 printf "using compiler runtime libraries: %s\n" "$LIBCC"
 
-- 
2.30.2

