diff --git a/Makefile b/Makefile
index fd1844d..e2c0f44 100644
--- a/Makefile
+++ b/Makefile
@@ -116,8 +116,7 @@ MANFILES1 = \
 
 MANFILES8 = \
 	tlp.8 \
-	tlp-stat.8 \
-	tlp.service.8
+	tlp-stat.8
 
 MANFILESRDW8 = \
 	tlp-rdw.8
@@ -298,9 +297,9 @@ install-man-pd:
 	install -d -m 755 $(_MAN)/man8
 	cd man-pd && install -m 644 $(MANFILESPD8) $(_MAN)/man8/
 
-install: install-tlp install-rdw install-pd
+install: install-tlp install-rdw
 
-install-man: install-man-tlp install-man-rdw install-man-pd
+install-man: install-man-tlp install-man-rdw
 
 uninstall-tlp:
 	# Package tlp
