diff --color=auto -urN a/Makefile b/Makefile
--- a/Makefile	2026-05-03 15:29:57.000000000 +0100
+++ b/Makefile	2026-06-23 22:28:44.954840993 +0100
@@ -114,8 +114,7 @@
 
 MANFILES8 = \
 	tlp.8 \
-	tlp-stat.8 \
-	tlp.service.8
+	tlp-stat.8
 
 MANFILESRDW8 = \
 	tlp-rdw.8
@@ -290,9 +289,9 @@
 	install -d -m 755 $(_MAN)/man8
 	cd man-pd && install -m 644 $(MANFILESPD8) $(_MAN)/man8/
 
-install: install-tlp install-rdw install-pd
+install: install-tlp install-rdw
 
-install-man: install-man-tlp install-man-rdw install-man-pd
+install-man: install-man-tlp install-man-rdw
 
 uninstall-tlp:
 	# Package tlp
