diff --git a/Makefile b/Makefile
index f3dd8f2..cc278a9 100644
--- a/Makefile
+++ b/Makefile
@@ -37,9 +37,6 @@ install: etckeeper.version
 	$(INSTALL_DATA) bash_completion $(DESTDIR)$(bashcompletiondir)/etckeeper
 	mkdir -p $(DESTDIR)$(zshcompletiondir)
 	$(INSTALL_DATA) zsh_completion $(DESTDIR)$(zshcompletiondir)/_etckeeper
-	mkdir -p $(DESTDIR)$(systemddir)
-	$(INSTALL_DATA) systemd/etckeeper.service $(DESTDIR)$(systemddir)/etckeeper.service
-	$(INSTALL_DATA) systemd/etckeeper.timer $(DESTDIR)$(systemddir)/etckeeper.timer
 ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),apt)
 	mkdir -p $(DESTDIR)$(etcdir)/apt/apt.conf.d
 	$(INSTALL_DATA) apt.conf $(DESTDIR)$(etcdir)/apt/apt.conf.d/05etckeeper
