commit 6d0a97d0455b025e55da00610136287643bca194
Author: q66 <q66@chimera-linux.org>
Date:   Sat Jun 10 19:58:37 2023 +0200

    use meson

diff --git a/meson.build b/meson.build
new file mode 100644
index 0000000..1f1e493
--- /dev/null
+++ b/meson.build
@@ -0,0 +1,30 @@
+project(
+  'xdg-user-dirs-gtk', 'c',
+        version : '0.11',
+        license : 'GPL2+',
+  meson_version : '>= 0.50.0'
+)
+
+localedir = join_paths (get_option ('prefix'), get_option ('localedir'))
+sysconfdir = join_paths (get_option ('prefix'), get_option ('sysconfdir'))
+
+executable ('xdg-user-dirs-gtk-update',
+            [ 'parse.c',
+              'update.c' ],
+            dependencies: dependency('gtk+-3.0'),
+            c_args: [ '-DGETTEXT_PACKAGE="xdg-user-dirs-gtk"',
+                      '-DGLIBLOCALEDIR="@0@"'.format (localedir),
+                      '-DXDG_USER_DIRS_UPDATE=""' ],
+            install: true )
+
+i18n = import('i18n')
+
+i18n.merge_file (
+                 input: 'user-dirs-update-gtk.desktop.in',
+                 output: 'user-dirs-update-gtk.desktop',
+                 install: true,
+                 install_dir: join_paths (sysconfdir, 'xdg', 'autostart'),
+                 po_dir: 'po',
+                 type: 'desktop')
+
+subdir('po')
diff --git a/po/meson.build b/po/meson.build
new file mode 100644
index 0000000..93f475a
--- /dev/null
+++ b/po/meson.build
@@ -0,0 +1,3 @@
+i18n.gettext ('xdg-user-dirs-gtk', preset: 'glib')
+
+podir = meson.current_source_dir ()
diff --git a/update.c b/update.c
index 3b21ab1..7d25786 100644
--- a/update.c
+++ b/update.c
@@ -1,5 +1,3 @@
-#include "config.h"
-
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
@@ -106,7 +104,7 @@ update_locale (XdgDirEntry *old_entries)
     return;
   close (fd);
   
-  cmdline = g_strdup_printf (XDG_USER_DIRS_UPDATE " --force --dummy-output %s", filename);
+  cmdline = g_strdup_printf ("xdg-user-dirs-update --force --dummy-output %s", filename);
   if (!g_spawn_command_line_sync  (cmdline, &std_out, &std_err, &exit_status, NULL))
     {
       g_free (std_out);
diff --git a/user-dirs-update-gtk.desktop.in b/user-dirs-update-gtk.desktop.in
index 78d5f7c..38c45dd 100644
--- a/user-dirs-update-gtk.desktop.in
+++ b/user-dirs-update-gtk.desktop.in
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=xdg-user-dirs-gtk-update
-_Name=User folders update
-_Comment=Update common folders names to match current locale
+Name=User folders update
+Comment=Update common folders names to match current locale
 Terminal=false
 OnlyShowIn=GNOME;LXDE;Unity;
 Type=Application
