--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -117,7 +117,7 @@ checked_find_package (pugixml REQUIRED
 # LLVM library setup
 checked_find_package (LLVM REQUIRED
                       VERSION_MIN 9.0
-                      VERSION_MAX 19.9
+                      VERSION_MAX 20.9
                       PRINT LLVM_SYSTEM_LIBRARIES CLANG_LIBRARIES)
 # ensure include directory is added (in case of non-standard locations
 include_directories (BEFORE SYSTEM "${LLVM_INCLUDES}")
--- a/src/liboslexec/llvm_util.cpp
+++ b/src/liboslexec/llvm_util.cpp
@@ -3204,7 +3204,7 @@ LLVM_Util::loop_after_block() const
 llvm::Type*
 LLVM_Util::type_union(cspan<llvm::Type*> types)
 {
-    llvm::DataLayout target(module());
+    const llvm::DataLayout &target = module()->getDataLayout();
     size_t max_size  = 0;
     size_t max_align = 1;
     for (auto t : types) {
