commit 7f3f3b8e77a5414d824d7a9bed738f6af5f4ab0c
Author: q66 <q66@chimera-linux.org>
Date:   Sat Oct 14 16:37:14 2023 +0200

    add missing include

diff --git a/src/rendervulkan.hpp b/src/rendervulkan.hpp
index 4d040f0..3b3bbac 100644
--- a/src/rendervulkan.hpp
+++ b/src/rendervulkan.hpp
@@ -9,6 +9,7 @@
 #include <unordered_map>
 #include <array>
 #include <bitset>
+#include <mutex>
 
 #include "main.hpp"
 
