From 3d08e4a95b915201d30db03a35e6ef5c099d2c1d Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 8 Jul 2023 16:15:15 +0200
Subject: [PATCH] portability fixes

---
 Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 401d2ac..a755bba 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -693,7 +693,7 @@ ell/shared: Makefile
 	$(AM_V_at)$(MKDIR_P) ell
 	$(AM_V_GEN)for f in $(ell_shared) ; do \
 		if [ ! -f $$f ] ; then \
-			$(LN_S) -t ell -f $(abs_srcdir)/../ell/$$f ; \
+			$(LN_S) -f $(abs_srcdir)/../ell/$$f ell ; \
 		fi \
 	done > $@
 
@@ -701,7 +701,7 @@ ell/internal: Makefile
 	$(AM_V_at)$(MKDIR_P) ell
 	$(AM_V_GEN)for f in $(ell_headers) $(ell_sources) ; do \
 		if [ ! -f $$f ] ; then \
-			$(LN_S) -t ell -f $(abs_srcdir)/../ell/$$f ; \
+			$(LN_S) -f $(abs_srcdir)/../ell/$$f ell ; \
 		fi \
 	done > $@
 
-- 
2.41.0

