From 9cb229646b8839b5286ec08734b6c98473b892ba Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Sat, 13 Nov 2021 23:51:36 +0100
Subject: [PATCH] portability fixes

---
 Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 6958f2d..b5a7795 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -3714,7 +3714,7 @@ ell/shared: Makefile
 	$(AM_V_at)$(MKDIR_P) ell
 	$(AM_V_GEN)for f in $(ell_shared) ; do \
 		if [ ! -f $$f ] ; then \
-			$(LN_S) -t ell -f $(abs_srcdir)/../ell/$$f ; \
+			$(LN_S) -f $(abs_srcdir)/../ell/$$f ell ; \
 		fi \
 	done > $@
 
@@ -3722,7 +3722,7 @@ ell/internal: Makefile
 	$(AM_V_at)$(MKDIR_P) ell
 	$(AM_V_GEN)for f in $(ell_headers) $(ell_sources) ; do \
 		if [ ! -f $$f ] ; then \
-			$(LN_S) -t ell -f $(abs_srcdir)/../ell/$$f ; \
+			$(LN_S) -f $(abs_srcdir)/../ell/$$f ell ; \
 		fi \
 	done > $@
 
-- 
2.33.0

