commit 9d8cbacc76c08bbd5da506028cf9613808050fe4
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sun Apr 10 00:44:22 2022 +0200

    tests portability

diff --git a/tests/r_corrupt_fs/script b/tests/r_corrupt_fs/script
index 08af91e..14f8e3f 100644
--- a/tests/r_corrupt_fs/script
+++ b/tests/r_corrupt_fs/script
@@ -17,11 +17,7 @@ $MKE2FS -q -F -t ext4 -o Linux -b 1024 $TMPFILE 32M >> $OUT.new 2>&1
 echo debugfs -w -R \"set_bg 1 free_blocks_count 65536\" /tmp/foo.img >> $OUT.new
 $DEBUGFS -w -R "set_bg 1 free_blocks_count 65536" $TMPFILE > /dev/null 2>&1
 
-if type timeout > /dev/null 2>&1 ; then
-   TIMEOUT="timeout -v 30s"
-else
-   TIMEOUT=
-fi
+TIMEOUT=
 
 echo resize2fs -P /tmp/foo.img >> $OUT.new
 $TIMEOUT $RESIZE2FS -P $TMPFILE  >> $OUT.new 2>&1
