diff -ruN a/setup.py b/setup.py
--- a/setup.py	2022-07-31 15:40:32.000000000 +0200
+++ b/setup.py	2024-10-07 12:08:37.338347135 +0200
@@ -5,7 +5,6 @@
 
 from _setup_libssh2 import build_ssh2
 
-import versioneer
 from setuptools import setup, find_packages
 
 cpython = platform.python_implementation() == 'CPython'
@@ -87,13 +86,13 @@
         'msvc*.dll', 'vcruntime*.dll',
     ])
 
-cmdclass = versioneer.get_cmdclass()
+cmdclass = {}
 if USING_CYTHON:
     cmdclass['build_ext'] = build_ext
 
 setup(
     name='ssh2-python',
-    version=versioneer.get_version(),
+    version=os.getenv('SSH2_PYTHON_VERSION'),
     cmdclass=cmdclass,
     url='https://github.com/ParallelSSH/ssh2-python',
     license='LGPLv2',
