From 4c20d2535ad171bc722fda79838fdd9fad8fb45e Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Wed, 18 Jun 2025 17:53:15 +0300
Subject: [PATCH] Don't break announcement URL on C locale (with e.g. musl
 libc)

---
 src/release.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/release.cpp b/src/release.cpp
index 6b2b2381..47ebbf32 100644
--- a/src/release.cpp
+++ b/src/release.cpp
@@ -147,7 +147,9 @@ void Release::getPreview()
     if (announcementUrl.startsWith(kdeSite)) {
         // It's ours, we know how to localise it
         QString languageCode = KLocalizedString::languages().first().toLower().replace("_", "-");
-        announcementUrl.insert(kdeSite.length(), languageCode + "/");
+        if (languageCode != "c") {
+            announcementUrl.insert(kdeSite.length(), languageCode + "/");
+        }
     }
 
     m_previewNetworkAccessManager->get(QNetworkRequest(QUrl(announcementUrl)));
-- 
GitLab
