Use a more ubiquitous default.

--- a/src/libOpenImageIO/imagebufalgo_draw.cpp	2024-05-02 08:23:34.634754057 +0200
+++ b/src/libOpenImageIO/imagebufalgo_draw.cpp	2024-05-02 08:25:37.892241977 +0200
@@ -882,7 +882,7 @@
 static FT_Library ft_library = NULL;
 static bool ft_broken        = false;
 
-static const char* default_font_name[] = { "DroidSans", "cour", "Courier New",
+static const char* default_font_name[] = { "DejaVu Sans", "cour", "Courier New",
                                            "FreeMono" };
 
 
