--- a/proc/escape.c
+++ b/proc/escape.c
@@ -21,6 +21,7 @@
 #include <sys/types.h>
 #include <string.h>
 #include <limits.h>
+#include <langinfo.h>
 #include "procps.h"
 #include "escape.h"
 #include "readproc.h"
--- a/w.c
+++ b/w.c
@@ -55,10 +55,9 @@
 #include <termios.h>
 #include <time.h>
 #include <unistd.h>
+#include <utmp.h>
 #ifdef HAVE_UTMPX_H
 #	include <utmpx.h>
-#else
-#	include <utmp.h>
 #endif
 #include <arpa/inet.h>
 
