--- a/crates/zed/resources/zed.desktop.in
+++ b/crates/zed/resources/zed.desktop.in
@@ -1,18 +1,18 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Name=$APP_NAME
+Name=Zed
 GenericName=Text Editor
 Comment=A high-performance, multiplayer code editor.
-TryExec=$APP_CLI
-StartupNotify=$DO_STARTUP_NOTIFY
-Exec=$APP_CLI $APP_ARGS
-Icon=$APP_ICON
+TryExec=zeditor
+StartupNotify=true
+Exec=zeditor %U
+Icon=zed
 Categories=Utility;TextEditor;Development;IDE;
 Keywords=zed;
 MimeType=text/plain;application/x-zerosize;x-scheme-handler/zed;
 Actions=NewWorkspace;
 
 [Desktop Action NewWorkspace]
-Exec=$APP_CLI --new $APP_ARGS
+Exec=zeditor --new %U
 Name=Open a new workspace
