From c579fbe0c10d2b761cfe1fe4e825aaa19fe81c77 Mon Sep 17 00:00:00 2001
From: Alice Mikhaylenko <alicem@gnome.org>
Date: Mon, 25 Sep 2023 02:00:32 +0400
Subject: [PATCH] about-window: Fix build with newer libappstream

---
 src/adw-about-window.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/adw-about-window.c b/src/adw-about-window.c
index e861dcd09..dd8da0f08 100644
--- a/src/adw-about-window.c
+++ b/src/adw-about-window.c
@@ -2004,7 +2004,11 @@ adw_about_window_new_from_appdata (const char *resource_path,
     g_free (appid_desktop);
   }
 
+#if AS_CHECK_VERSION (1, 0, 0)
+  releases = as_component_get_releases_plain (component);
+#else
   releases = as_component_get_releases (component);
+#endif
 
   if (release_notes_version) {
     guint release_index = 0;
-- 
GitLab

