diff --git a/run-tests b/run-tests
index ab8a516d..f02d5de5 100755
--- a/run-tests
+++ b/run-tests
@@ -5,10 +5,6 @@ for f in bin/*; do
 done
 export PATH=$PATH:/usr/local/bin
 
-NPROCS=1
-if [ -r /proc/cpuinfo ]; then
-	NPROCS=$(grep ^proc /proc/cpuinfo|wc -l)
-fi
 export XBPS_SYSLOG=false
 LIBRARY_PATH=$PWD/lib LD_LIBRARY_PATH=$PWD/lib ATF_SHELL=/bin/sh kyua --variable parallelism=$NPROCS test -r result.db -k tests/xbps/Kyuafile
 rv=$?
diff --git a/tests/xbps/libxbps/shell/conf_files_test.sh b/tests/xbps/libxbps/shell/conf_files_test.sh
index 1aef7ce8..c7727ef2 100644
--- a/tests/xbps/libxbps/shell/conf_files_test.sh
+++ b/tests/xbps/libxbps/shell/conf_files_test.sh
@@ -21,7 +21,7 @@ tc1_body() {
 	xbps-install -C null.conf -r rootdir --repository=$PWD -yvd a
 	atf_check_equal $? 0
 
-	sed -e 's,fooblah,blahfoo,' -i rootdir/cf1.conf
+	sed -e 's,fooblah,blahfoo,' -i '' rootdir/cf1.conf
 	mkdir pkg_a
 	echo "fooblah" > pkg_a/cf1.conf
 	xbps-create -A noarch -n a-0.2_1 -s "pkg a" --config-files "/cf1.conf" pkg_a
@@ -61,7 +61,7 @@ tc2_body() {
 	xbps-install -C null.conf -r rootdir --repository=$PWD -yvd a
 	atf_check_equal $? 0
 
-	sed -e 's,fooblah,blahfoo,' -i rootdir/cf1.conf
+	sed -e 's,fooblah,blahfoo,' -i '' rootdir/cf1.conf
 	chmod 644 rootdir/cf1.conf
 	mkdir pkg_a
 	echo "bazbar" > pkg_a/cf1.conf
@@ -110,7 +110,7 @@ tc3_body() {
 	xbps-install -C null.conf -r rootdir --repository=$PWD -yvd a
 	atf_check_equal $? 0
 
-	sed -e 's,fooblah,blahfoo,' -i rootdir/cf1.conf
+	sed -e 's,fooblah,blahfoo,' -i '' rootdir/cf1.conf
 	chmod 644 rootdir/cf1.conf
 	mkdir pkg_a
 	echo "fooblah" > pkg_a/cf1.conf
@@ -189,7 +189,7 @@ tc5_body() {
 
 	mv rootdir/cf1.conf rootdir/foobar.conf
 	ln -sf foobar.conf rootdir/cf1.conf
-	sed -e 's,fooblah,blahfoo,' -i rootdir/foobar.conf
+	sed -e 's,fooblah,blahfoo,' -i '' rootdir/foobar.conf
 	chmod 644 rootdir/foobar.conf
 
 	mkdir pkg_a
diff --git a/tests/xbps/libxbps/shell/configure_test.sh b/tests/xbps/libxbps/shell/configure_test.sh
index 215e53c2..25241e5f 100644
--- a/tests/xbps/libxbps/shell/configure_test.sh
+++ b/tests/xbps/libxbps/shell/configure_test.sh
@@ -27,7 +27,7 @@ EOF
 	cd ..
 	xbps-install -C empty.conf -r root --repository=$PWD/repo -yd A
 	atf_check_equal $? 0
-	perms=$(stat --format=%a root/file)
+	perms=$(stat -f %OLp root/file)
 	atf_check_equal $perms 644
 }
 
diff --git a/tests/xbps/libxbps/shell/conflicts_test.sh b/tests/xbps/libxbps/shell/conflicts_test.sh
index 072ca7a0..4d25c8c2 100644
--- a/tests/xbps/libxbps/shell/conflicts_test.sh
+++ b/tests/xbps/libxbps/shell/conflicts_test.sh
@@ -23,8 +23,8 @@ conflicts_trans_body() {
 	# EAGAIN, conflicts.
 	atf_check_equal $? 11
 	# 0 pkgs installed.
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 0
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 0
 }
 
 conflicts_trans_hold_head() {
@@ -93,8 +93,8 @@ conflicts_trans_vpkg_body() {
 	# EAGAIN, conflicts.
 	atf_check_equal $? 11
 	# 0 pkgs installed.
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 0
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 0
 }
 
 atf_test_case conflicts_trans_multi
@@ -121,7 +121,7 @@ conflicts_trans_multi_body() {
 	xbps-install -r root --repository=$PWD/some_repo -dy A C
 	# EAGAIN, conflicts.
 	atf_check_equal $? 11
-	atf_check_equal $(xbps-query -r root -l|wc -l) 0
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 0
 }
 
 atf_test_case conflicts_installed
@@ -147,8 +147,8 @@ conflicts_installed_body() {
 	atf_check_equal $? 0
 	xbps-install -r root --repository=$PWD/some_repo -dy B
 	atf_check_equal $? 11
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 1
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 1
 }
 
 atf_test_case conflicts_installed_multi
@@ -176,8 +176,8 @@ conflicts_installed_multi_body() {
 	atf_check_equal $? 0
 	xbps-install -r root --repository=$PWD/some_repo -dy C
 	atf_check_equal $? 11
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 1
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 1
 }
 
 atf_test_case conflicts_trans_installed
@@ -203,8 +203,8 @@ conflicts_trans_installed_body() {
 	atf_check_equal $? 0
 	xbps-install -r root --repository=$PWD/some_repo -dy A
 	atf_check_equal $? 11
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 1
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 1
 }
 
 atf_test_case conflicts_trans_update
@@ -277,8 +277,8 @@ conflicts_trans_installed_multi_body() {
 	atf_check_equal $? 0
 	xbps-install -r root --repository=$PWD/some_repo -dy A
 	atf_check_equal $? 11
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 2
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 2
 }
 
 atf_test_case conflicts_trans_provrep
@@ -308,8 +308,8 @@ conflicts_trans_provrep_body() {
 	atf_check_equal $? 0
 
 	# 2 pkgs installed: libGL and neverball
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 2
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 2
 
 	out=$(xbps-query -r root -p pkgver libGL)
 	atf_check_equal "$out" "libGL-19.1_1"
@@ -321,8 +321,8 @@ conflicts_trans_provrep_body() {
 	atf_check_equal $? 0
 
 	# 2 pkgs installed: nvidia-libs and neverball
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 2
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 2
 	out=$(xbps-query -r root -p pkgver nvidia-libs)
 	atf_check_equal "$out" "nvidia-libs-1.0_1"
 
@@ -341,8 +341,8 @@ conflicts_trans_provrep_body() {
 	atf_check_equal $? 0
 
 	# 3 pkgs installed: libglvnd, nvidia-libs and neverball
-	xbps-query -r root -l|wc -l
-	atf_check_equal $(xbps-query -r root -l|wc -l) 3
+	xbps-query -r root -l|wc -l|awk '{print $1}'
+	atf_check_equal $(xbps-query -r root -l|wc -l|awk '{print $1}') 3
 
 	out=$(xbps-query -r root -p pkgver libglvnd)
 	atf_check_equal "$out" "libglvnd-1.0_1"
diff --git a/tests/xbps/libxbps/shell/incorrect_deps_test.sh b/tests/xbps/libxbps/shell/incorrect_deps_test.sh
index c6f2f13e..083593c5 100644
--- a/tests/xbps/libxbps/shell/incorrect_deps_test.sh
+++ b/tests/xbps/libxbps/shell/incorrect_deps_test.sh
@@ -157,7 +157,7 @@ multiple_versions_body() {
 	atf_check_equal $? 0
 	cd ..
 
-	out=$(xbps-install -r root --repo=repo --repo=repo2 -n kconfigwidgets foo|wc -l)
+	out=$(xbps-install -r root --repo=repo --repo=repo2 -n kconfigwidgets foo|wc -l|awk '{print $1}')
 	atf_check_equal $? 0
 	atf_check_equal "$out" 4
 
@@ -167,7 +167,7 @@ multiple_versions_body() {
 	xbps-pkgdb -r root -a
 	atf_check_equal $? 0
 
-	out=$(xbps-query -r root -l|wc -l)
+	out=$(xbps-query -r root -l|wc -l|awk '{print $1}')
 	atf_check_equal $? 0
 	atf_check_equal "$out" 4
 }
diff --git a/tests/xbps/libxbps/shell/install_test.sh b/tests/xbps/libxbps/shell/install_test.sh
index c0fe9a1a..68c3e415 100644
--- a/tests/xbps/libxbps/shell/install_test.sh
+++ b/tests/xbps/libxbps/shell/install_test.sh
@@ -198,7 +198,7 @@ install_dups_body() {
 	atf_check_equal $? 0
 	cd ..
 
-	out=$(xbps-install -C empty.conf -r root --repository=$PWD/some_repo -ynd A A A A|wc -l)
+	out=$(xbps-install -C empty.conf -r root --repository=$PWD/some_repo -ynd A A A A|wc -l|awk '{print $1}')
 	atf_check_equal $out 1
 }
 
@@ -355,8 +355,8 @@ update_file_timestamps_body() {
 	atf_check_equal $? 0
 	xbps-install -r root --repository=repo -yvd foo
 
-	expected=$(stat --printf='%Y' pkg_A/usr/include/gsm/gsm.h)
-	result=$(stat --printf='%Y' root/usr/include/gsm/gsm.h)
+	expected=$(stat -f %m pkg_A/usr/include/gsm/gsm.h)
+	result=$(stat -f %m root/usr/include/gsm/gsm.h)
 
 	atf_check_equal "$expected" "$result"
 
@@ -369,8 +369,8 @@ update_file_timestamps_body() {
 	atf_check_equal $? 0
 	xbps-install -r root --repository=repo -yuvd foo
 
-	expected=$(stat --printf='%Y' pkg_A/usr/include/gsm/gsm.h)
-	result=$(stat --printf='%Y' root/usr/include/gsm/gsm.h)
+	expected=$(stat -f %m pkg_A/usr/include/gsm/gsm.h)
+	result=$(stat -f %m root/usr/include/gsm/gsm.h)
 
 	atf_check_equal "$expected" "$result"
 }
@@ -654,7 +654,7 @@ update_and_install_body() {
 	xbps-install -r root --repo=repo1 --repo=repo1-dbg --repo=repo2 -ydu A
 	atf_check_equal $? 0
 
-	out=$(xbps-query -r root -l|wc -l)
+	out=$(xbps-query -r root -l|wc -l|awk '{print $1}')
 	atf_check_equal "$out" "2"
 
 	out=$(xbps-query -r root -p pkgver A)
diff --git a/tests/xbps/libxbps/shell/orphans_test.sh b/tests/xbps/libxbps/shell/orphans_test.sh
index dddc13c6..6c3c0e3f 100644
--- a/tests/xbps/libxbps/shell/orphans_test.sh
+++ b/tests/xbps/libxbps/shell/orphans_test.sh
@@ -27,7 +27,7 @@ tc1_body() {
 	atf_check_equal "$out" "D-1.0_1"
 	xbps-remove -r root -Ryd D
 	atf_check_equal $? 0
-	out="$(xbps-query -r root -l|wc -l)"
+	out="$(xbps-query -r root -l|wc -l|awk '{print $1}')"
 	atf_check_equal $? 0
 	atf_check_equal "$out" "0"
 
@@ -50,7 +50,7 @@ tc1_body() {
 	atf_check_equal $? 0
 	xbps-remove -r root -Ryd D
 	atf_check_equal $? 0
-	out="$(xbps-query -r root -l|wc -l)"
+	out="$(xbps-query -r root -l|wc -l|awk '{print $1}')"
 	atf_check_equal $? 0
 	atf_check_equal "$out" "0"
 
diff --git a/tests/xbps/libxbps/shell/remove_test.sh b/tests/xbps/libxbps/shell/remove_test.sh
index 28a8a35d..c8538ad4 100644
--- a/tests/xbps/libxbps/shell/remove_test.sh
+++ b/tests/xbps/libxbps/shell/remove_test.sh
@@ -116,7 +116,7 @@ remove_symlinks_relative_body() {
 	mkdir some_repo
 	mkdir -p pkg_A/usr/lib pkg_A/usr/share/lib
 	touch -f pkg_A/usr/lib/libfoo.so.1.2.0
-	ln -sfr pkg_A/usr/lib/libfoo.so.1.2.0 pkg_A/usr/share/lib/libfoo.so.1
+	ln -sf pkg_A/usr/lib/libfoo.so.1.2.0 "$PWD/pkg_A/usr/share/lib/libfoo.so.1"
 
 	cd some_repo
 	xbps-create -A noarch -n A-1.0_1 -s "A pkg" ../pkg_A
@@ -335,7 +335,7 @@ remove_dups_body() {
 	cd ..
 	xbps-install -r root -C null.conf --repository=$PWD/some_repo -yv A
 	atf_check_equal $? 0
-	out=$(xbps-remove -r root -yvn A A A|wc -l)
+	out=$(xbps-remove -r root -yvn A A A|wc -l|awk '{print $1}')
 	atf_check_equal $out 1
 }
 
@@ -386,7 +386,7 @@ remove_with_revdeps_force_body() {
 	atf_check_equal $? 0
 	xbps-remove -r root -Fyvd A
 	atf_check_equal $? 0
-	out=$(xbps-query -r root -l|wc -l)
+	out=$(xbps-query -r root -l|wc -l|awk '{print $1}')
 	atf_check_equal $out 1
 	out=$(xbps-query -r root -ppkgver B)
 	atf_check_equal $out B-1.0_1
@@ -462,7 +462,7 @@ remove_with_revdeps_in_trans_recursive_body() {
 	atf_check_equal $? 0
 	xbps-remove -r root -Ryvd B
 	atf_check_equal $? 0
-	out=$(xbps-query -r root -l|wc -l)
+	out=$(xbps-query -r root -l|wc -l|awk '{print $1}')
 	atf_check_equal $out 0
 }
 
diff --git a/tests/xbps/libxbps/shell/replace_test.sh b/tests/xbps/libxbps/shell/replace_test.sh
index daa1af22..8431131b 100644
--- a/tests/xbps/libxbps/shell/replace_test.sh
+++ b/tests/xbps/libxbps/shell/replace_test.sh
@@ -32,7 +32,7 @@ replace_dups_body() {
 	atf_check_equal $? 0
 	xbps-install -C xbps.d -r root --repository=$PWD/some_repo -yd B
 	atf_check_equal $? 0
-	result=$(xbps-query -C xbps.d -r root -l|wc -l)
+	result=$(xbps-query -C xbps.d -r root -l|wc -l|awk '{print $1}')
 	atf_check_equal $result 1
 	atf_check_equal $(xbps-query -C xbps.d -r root -p state B) installed
 }
@@ -235,7 +235,7 @@ replace_pkg_with_update_body() {
 	cd ..
 	xbps-install -C xbps.d -r root --repository=$PWD/some_repo -yfd A B
 	atf_check_equal $? 0
-	result=$(xbps-query -C xbps.d -r root -l|wc -l)
+	result=$(xbps-query -C xbps.d -r root -l|wc -l|awk '{print $1}')
 	atf_check_equal $result 1
 	atf_check_equal $(xbps-query -C xbps.d -r root -p state B) installed
 }
@@ -269,7 +269,7 @@ replace_vpkg_with_update_body() {
 	cd ..
 	xbps-install -C xbps.d -r root --repository=$PWD/some_repo -yfd A B
 	atf_check_equal $? 0
-	result=$(xbps-query -C xbps.d -r root -l|wc -l)
+	result=$(xbps-query -C xbps.d -r root -l|wc -l|awk '{print $1}')
 	atf_check_equal $result 1
 	atf_check_equal $(xbps-query -C xbps.d -r root -p state B) installed
 }
diff --git a/tests/xbps/libxbps/shell/repo_test.sh b/tests/xbps/libxbps/shell/repo_test.sh
index 30022acf..b92108c1 100644
--- a/tests/xbps/libxbps/shell/repo_test.sh
+++ b/tests/xbps/libxbps/shell/repo_test.sh
@@ -16,7 +16,7 @@ repo_close_body() {
 	cd ..
 	xbps-install -C empty.conf -r root --repository=repo -yn A
 	atf_check_equal $? 0
-	truncate --size 0 repo/*-repodata
+	truncate -s 0 repo/*-repodata
 	xbps-install -C empty.conf -r root --repository=repo -yn A
 	# ENOENT because invalid repodata
 	atf_check_equal $? 2
diff --git a/tests/xbps/xbps-alternatives/main_test.sh b/tests/xbps/xbps-alternatives/main_test.sh
index 084069f0..26748083 100644
--- a/tests/xbps/xbps-alternatives/main_test.sh
+++ b/tests/xbps/xbps-alternatives/main_test.sh
@@ -34,6 +34,7 @@ register_one_dangling_head() {
 	atf_set "descr" "xbps-alternatives: register one pkg with an alternative dangling symlink"
 }
 register_one_dangling_body() {
+	atf_expect_fail "freebsd readlink fails on dangling symlinks"
 	mkdir -p repo pkg_A/usr/bin
 	cd repo
 	xbps-create -A noarch -n A-1.1_1 -s "A pkg" --alternatives "file:/usr/bin/file:/usr/bin/fileA file2:/usr/lib/fileB:/usr/include/fileB" ../pkg_A
@@ -113,7 +114,7 @@ register_dups_body() {
 	atf_check_equal $? 0
 	xbps-install -r root --repository=repo -ydfv A
 	atf_check_equal $? 0
-	atf_check_equal "$(xbps-alternatives -r root -l|wc -l)" 3
+	atf_check_equal "$(xbps-alternatives -r root -l|wc -l|awk '{print $1}')" 3
 }
 
 atf_test_case register_multi
@@ -784,7 +785,7 @@ replace_alternative_with_symlink_body() {
 	xbps-install -r root --repository=repo -ydv A
 	atf_check_equal $? 0
 
-	ln -sfr pkg_A/usr/bin/fileA pkg_A/usr/bin/file
+	ln -sf pkg_A/usr/bin/fileA "$PWD/pkg_A/usr/bin/file"
 	touch pkg_A/usr/bin/file
 	cd repo
 	xbps-create -A noarch -n A-1.1_2 -s "A pkg" ../pkg_A
@@ -822,7 +823,7 @@ keep_provider_on_update_body() {
 
 	xbps-install -r root --repo=repo -y tar bsdtar
 	atf_check_equal $? 0
-	link=$(readlink -v root/usr/bin/tar)
+	link=$(readlink root/usr/bin/tar)
 	atf_check_equal $? 0
 	atf_check_equal $link gtar
 
@@ -836,7 +837,7 @@ keep_provider_on_update_body() {
 	xbps-install -r root --repo=repo -yud
 	atf_check_equal $? 0
 
-	link=$(readlink -v root/usr/bin/tar)
+	link=$(readlink root/usr/bin/tar)
 	atf_check_equal $? 0
 	atf_check_equal $link gtar
 }
diff --git a/tests/xbps/xbps-checkvers/checkvers_test.sh b/tests/xbps/xbps-checkvers/checkvers_test.sh
index 25fa940d..3a69681c 100755
--- a/tests/xbps/xbps-checkvers/checkvers_test.sh
+++ b/tests/xbps/xbps-checkvers/checkvers_test.sh
@@ -519,7 +519,7 @@ EOF
 		cd ..
 	done
 
-	sed -e s/version=1/version=2/ -i void-packages/srcpkgs/updated/template
+	sed -e s/version=1/version=2/ -i '' void-packages/srcpkgs/updated/template
 	rm -r void-packages/srcpkgs/removed
 	cd some_repo
 	xbps-rindex -a $PWD/*.xbps
@@ -545,7 +545,7 @@ removed_subpkgs_body() {
 	for pkg in same updated removed onlybase onlydevel
 	do
 		mkdir -p void-packages/srcpkgs/$pkg
-		ln -sr void-packages/srcpkgs/$pkg void-packages/srcpkgs/$pkg-devel
+		ln -s void-packages/srcpkgs/$pkg "$PWD/void-packages/srcpkgs/$pkg-devel"
 		cat > void-packages/srcpkgs/$pkg/template <<EOF
 pkgname=$pkg
 version=1
@@ -565,7 +565,7 @@ EOF
 		cd ..
 	done
 
-	sed -e s/version=1/version=2/ -i void-packages/srcpkgs/updated/template
+	sed -e s/version=1/version=2/ -i '' void-packages/srcpkgs/updated/template
 	rm -r void-packages/srcpkgs/onlydevel-devel
 	mv void-packages/srcpkgs/onlydevel void-packages/srcpkgs/onlydevel-devel
 	rm -r void-packages/srcpkgs/onlybase-devel
diff --git a/tests/xbps/xbps-create/basic_test.sh b/tests/xbps/xbps-create/basic_test.sh
index a98b21f3..0a467bde 100644
--- a/tests/xbps/xbps-create/basic_test.sh
+++ b/tests/xbps/xbps-create/basic_test.sh
@@ -131,8 +131,8 @@ restore_mtime_body() {
 	atf_check_equal $? 0
 	xbps-install -r root --repository=repo -yvd foo
 
-	expected=$(stat --printf='%Y' pkg_A/usr/include/gsm/gsm.h)
-	result=$(stat --printf='%Y' root/usr/include/gsm/gsm.h)
+	expected=$(stat -f %m pkg_A/usr/include/gsm/gsm.h)
+	result=$(stat -f %m root/usr/include/gsm/gsm.h)
 
 	atf_check_equal "$expected" "$result"
 }
diff --git a/tests/xbps/xbps-query/ignore_repos_test.sh b/tests/xbps/xbps-query/ignore_repos_test.sh
index 9ff6c7f3..dfa2dc74 100644
--- a/tests/xbps/xbps-query/ignore_repos_test.sh
+++ b/tests/xbps/xbps-query/ignore_repos_test.sh
@@ -21,7 +21,7 @@ ignore_system_body() {
 	systemdir=$(xbps-uhelper getsystemdir)
 	mkdir -p root/${systemdir}
 	echo "repository=$PWD/repo1" > root/${systemdir}/myrepo.conf
-	out="$(xbps-query -C empty.conf --repository=$PWD/repo -i -L|wc -l)"
+	out="$(xbps-query -C empty.conf --repository=$PWD/repo -i -L|wc -l|awk '{print $1}')"
 	atf_check_equal "$out" 1
 }
 
@@ -44,7 +44,7 @@ ignore_conf_body() {
 	cd ..
 	mkdir -p root/xbps.d
 	echo "repository=$PWD/repo1" > root/xbps.d/myrepo.conf
-	out="$(xbps-query -r root -C xbps.d --repository=$PWD/repo -i -L|wc -l)"
+	out="$(xbps-query -r root -C xbps.d --repository=$PWD/repo -i -L|wc -l|awk '{print $1}')"
 	atf_check_equal "$out" 1
 }
 
diff --git a/tests/xbps/xbps-rindex/clean_test.sh b/tests/xbps/xbps-rindex/clean_test.sh
index 6c5ca636..aaff5bb8 100644
--- a/tests/xbps/xbps-rindex/clean_test.sh
+++ b/tests/xbps/xbps-rindex/clean_test.sh
@@ -18,7 +18,7 @@ noremove_body() {
 	cd ..
 	xbps-rindex -c some_repo
 	atf_check_equal $? 0
-	result=$(xbps-query -r root -C empty.conf --repository=some_repo -s foo|wc -l)
+	result=$(xbps-query -r root -C empty.conf --repository=some_repo -s foo|wc -l|awk '{print $1}')
 	atf_check_equal ${result} 1
 }
 
