commit 9e16b17796940fc5e07044e7cfa38efdbeae8e60
Author: q66 <q66@chimera-linux.org>
Date:   Sun Oct 15 02:42:35 2023 +0200

    disable libnkutils module tests

diff --git a/subprojects/libnkutils/meson.build b/subprojects/libnkutils/meson.build
index b35ee68..9dcc174 100644
--- a/subprojects/libnkutils/meson.build
+++ b/subprojects/libnkutils/meson.build
@@ -218,6 +218,7 @@ if bindings
     endif
 endif
 
+if false
 foreach m : nk_modules
     if get_variable(m.underscorify())
         nk_test = executable('nk-@0@.test'.format(m), files('tests/@0@.c'.format(m)), c_args: nk_args, dependencies: libnkutils)
@@ -228,6 +229,7 @@ foreach m : nk_modules
         )
     endif
 endforeach
+endif
 
 foreach m : nk_special_modules
     if get_variable(m.underscorify())
