commit 975e8ee276a1c0983fe54b5fd1c90ce05c2af51a
Author: q66 <q66@chimera-linux.org>
Date:   Tue Apr 12 04:19:10 2022 +0200

    build llvm-omp-device-info standalone

diff --git a/offload/tools/deviceinfo/CMakeLists.txt b/offload/tools/deviceinfo/CMakeLists.txt
index a39de88..91cd5be 100644
--- a/offload/tools/deviceinfo/CMakeLists.txt
+++ b/offload/tools/deviceinfo/CMakeLists.txt
@@ -1,8 +1,4 @@
-message(STATUS "Building the llvm-omp-device-info tool")
-
-add_openmp_tool(llvm-omp-device-info llvm-omp-device-info.cpp)
-
-llvm_update_compile_flags(llvm-omp-device-info)
+add_executable(llvm-omp-device-info llvm-omp-device-info.cpp)
 
 target_include_directories(llvm-omp-device-info PRIVATE
   ${LIBOMPTARGET_INCLUDE_DIR}
@@ -11,3 +7,5 @@ target_link_libraries(llvm-omp-device-info PRIVATE
   omp
   omptarget
 )
+
+install(TARGETS llvm-omp-device-info DESTINATION bin)
