From 5dedd4990ade3792f918265219a28c0ab905b244 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sun, 27 Jun 2021 20:15:47 +0200
Subject: [PATCH] allow building with bmake

---
 warnings.mk | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/warnings.mk b/warnings.mk
index c07ac37..7eff71a 100644
--- a/warnings.mk
+++ b/warnings.mk
@@ -46,12 +46,4 @@ HIGHER_WARNINGS?= ${HIGH_WARNINGS} \
     -Wpadded \
     -Wstrict-aliasing
 
-ifeq "${LIBXO_WARNINGS}" "HIGH"
-WARNINGS += ${HIGH_WARNINGS}
-else
 WARNINGS += ${LOW_WARNINGS}
-endif
-
-ifeq "${GCC_WARNINGS}" "yes"
-WARNINGS += -fno-inline-functions-called-once
-endif
-- 
2.32.0

