diff --git a/scripts/fsadm.sh b/scripts/fsadm.sh
index 53afacca7..14ec1beaf 100755
--- a/scripts/fsadm.sh
+++ b/scripts/fsadm.sh
@@ -261,7 +261,7 @@ detect_fs() {
 		MINOR=${MAJORMINOR##*:}
 		;;
 	  *)
-		STAT=$(stat --format "MAJOR=\$((0x%t)) MINOR=\$((0x%T))" "$RVOLUME")
+		STAT=$(stat -f "MAJOR=\$((0x%XHr)) MINOR=\$((0x%XLr))" "$RVOLUME")
 		test -n "$STAT" || error "Cannot get major:minor for \"$VOLUME\"."
 		eval "$STAT"
 		MAJORMINOR="${MAJOR}:${MINOR}"
@@ -316,7 +316,7 @@ check_valid_mounted_device() {
 		read -r <"/sys/block/${VOL#/dev/}/dev" MOUNTEDMAJORMINOR 2>&1 || error "Cannot get major:minor for \"$VOLUME\"."
 		;;
 	  *)
-		STAT=$(stat --format "MOUNTEDMAJORMINOR=\$((0x%t)):\$((0x%T))" "$VOL")
+		STAT=$(stat -f "MOUNTEDMAJORMINOR=\$((0x%XHr)):\$((0x%XLr))" "$VOL")
 		test -n "$STAT" || error "Cannot get major:minor for \"$VOLUME\"."
 		eval "$STAT"
 		;;
@@ -393,7 +393,7 @@ detect_mounted_with_proc_mounts() {
 			MOUNTDEV=$(echo -n -e "${i%% *}")
 			MOUNTED=${i#* }
 			MOUNTED=$(echo -n -e "${MOUNTED%% *}")
-			STAT=$(stat --format "%d" "$MOUNTED")
+			STAT=$(stat -f "%d" "$MOUNTED")
 			validate_mounted_major_minor "$(decode_major_minor "$STAT")"
 		done < "$PROCMOUNTS"
 	fi
@@ -750,7 +750,7 @@ resize() {
 #  only one supported
 ####################################
 diff_dates() {
-	echo "$(( $("$DATE" -u -d"$1" +%s 2>"$NULL") - $("$DATE" -u -d"$2" +%s 2>"$NULL") ))"
+	echo $(( $("$DATE" -j -f "%a %b %e %H:%M:%S %Y" "$1" +%s 2>"$NULL") - $("$DATE" -j -f "%a %b %e %H:%M:%S %Y" "$2" +%s 2>"$NULL") ))
 }
 
 check_luks() {
@@ -847,7 +847,7 @@ done
 TEST64BIT=$(( 1000 * 1000000000000 ))
 test "$TEST64BIT" -eq 1000000000000000 || error "Shell does not handle 64bit arithmetic."
 echo Y | "$GREP" Y >"$NULL" || error "Grep does not work properly."
-test "$("$DATE" -u -d"Jan 01 00:00:01 1970" +%s)" -eq 1 || error "Date translation does not work."
+test "$("$DATE" -j -f "%b %e %H:%M:%S %Y" "Jan 1 00:00:01 1970" +%s)" -eq 1 || error "Date translation does not work."
 
 
 if [ "$#" -eq 0 ]; then
diff --git a/scripts/lvm_import_vdo.sh b/scripts/lvm_import_vdo.sh
index d581f323d..16fee8225 100755
--- a/scripts/lvm_import_vdo.sh
+++ b/scripts/lvm_import_vdo.sh
@@ -313,7 +313,7 @@ detect_lv_() {
 		DEVMINOR=${MAJORMINOR##*:}
 		;;
 	  *)
-		RSTAT=$("$STAT" --format "DEVMAJOR=\$((0x%t)) DEVMINOR=\$((0x%T))" "$RDEVICE" || true)
+		RSTAT=$("$STAT" -f "DEVMAJOR=\$((0x%XHr)) DEVMINOR=\$((0x%XLr))" "$RDEVICE" || true)
 		[ -n "$RSTAT" ] || error "Cannot get major:minor for \"$DEVICE\"."
 		eval "$RSTAT"
 		;;
@@ -610,7 +610,7 @@ convert2lvm_() {
 	for i in $(awk '/.*device:/ {print $2}' "$TEMPDIR/vdoconf.yml"); do
 		local DEV
 		DEV=$("$READLINK" $READLINK_E "$i") || continue
-		RSTAT=$("$STAT" --format "MAJOR=\$((0x%t)) MINOR=\$((0x%T))" "$DEV" 2>/dev/null) || continue
+		RSTAT=$("$STAT" -f "MAJOR=\$((0x%XHr)) MINOR=\$((0x%XLr))" "$DEV" 2>/dev/null) || continue
 		eval "$RSTAT"
 		if [ "$MAJOR" = "$DEVMAJOR" ] && [ "$MINOR" = "$DEVMINOR" ]; then
 			[ -z "$FOUND" ] || error "VDO configuration contains duplicate entries $FOUND and $i."
@@ -725,4 +725,6 @@ test ${#IMPORT_NAME} -lt 100 || error "Random name \"$IMPORT_NAME\" is too long!
 
 [ -n "$DEVICE" ] || error "Device name is not specified. (see: $TOOL --help)"
 
+"$READLINK" -e / >/dev/null 2>&1 || READLINK_E="-f"
+
 convert2lvm_
diff --git a/scripts/lvmdump.sh.in b/scripts/lvmdump.sh.in
index 2b09e1a00..4e556619f 100644
--- a/scripts/lvmdump.sh.in
+++ b/scripts/lvmdump.sh.in
@@ -98,7 +98,7 @@ while getopts :acd:hlpmus opt; do
 	esac
 done
 
-NOW=$("$DATE" -u +%G%m%d%k%M%S | "$TR" -d ' ')
+NOW=$("$DATE" -u +%G%m%d%_H%M%S | "$TR" -d ' ')
 if test -n "$userdir"; then
 	dir=$userdir
 else
@@ -233,7 +233,7 @@ log "$TAIL -n 75 /var/log/messages >> \"$dir/messages\" 2>> \"$log\""
 
 myecho "Gathering /etc/lvm info..."
 log "$LS -laR /etc/lvm >> \"$dir/etc_lvm_listing\" 2>> \"$log\""
-log "$CP -RL --preserve=all /etc/lvm \"$dir/lvm\" 2>> \"$log\""
+log "$CP -RLp /etc/lvm \"$dir/lvm\" 2>> \"$log\""
 log "$LVM dumpconfig --type diff --file \"$dir/config_diff\" 2>> \"$log\""
 log "$LVM dumpconfig --type missing --file \"$dir/config_missing\" 2>> \"$log\""
 
@@ -321,8 +321,8 @@ if (( udev )); then
 	log "$UDEVADM info --export-db >> \"$udev_dir/db\" 2>> \"$log\""
 	log "$CP -a /etc/udev/udev.conf \"$udev_dir/conf\" 2>> \"$log\""
 	log "$LS -la /lib/udev >> \"$udev_dir/lib_dir\" 2>> \"$log\""
-	log "$CP -RL --preserve=all /etc/udev/rules.d \"$udev_dir/rules_etc\" 2>> \"$log\""
-	log "$CP -RL --preserve=all /lib/udev/rules.d \"$udev_dir/rules_lib\" 2>> \"$log\""
+	log "$CP -RLp /etc/udev/rules.d \"$udev_dir/rules_etc\" 2>> \"$log\""
+	log "$CP -RLp /lib/udev/rules.d \"$udev_dir/rules_lib\" 2>> \"$log\""
 fi
 
 if (( lvmetad )); then
diff --git a/scripts/lvmpersist.sh b/scripts/lvmpersist.sh
index 4cc0ba300..d20af389a 100755
--- a/scripts/lvmpersist.sh
+++ b/scripts/lvmpersist.sh
@@ -1127,7 +1127,7 @@ if [ "$UID" != 0 ] && [ "$EUID" != 0 ] && [ "$CMD" != "help" ]; then
 	exit 1
 fi
 
-GETOPT="getopt"
+GETOPT="ugetopt"
 
 OPTIONS=$("$GETOPT" -o h -l help,ourkey:,removekey:,key:,prtype:,access:,ptpl,debug,device:,vg: -n "${SCRIPTNAME}" -- "$@")
 eval set -- "$OPTIONS"
diff --git a/scripts/lvresize_fs_helper.sh b/scripts/lvresize_fs_helper.sh
index 1eeb209f0..41aa9714f 100755
--- a/scripts/lvresize_fs_helper.sh
+++ b/scripts/lvresize_fs_helper.sh
@@ -15,7 +15,7 @@
 set -eu
 
 PATH="/sbin:/usr/sbin:/bin:/usr/bin:$PATH"
-GETOPT="getopt"
+GETOPT="ugetopt"
 SCRIPTNAME=$(basename "$0")
 DM_DEV_DIR="${DM_DEV_DIR:-/dev}"
 
@@ -142,7 +142,7 @@ accept_e2fsck() {
 btrfs_path_major_minor() {
 	local STAT
 
-	STAT=$(stat --format "echo \$((0x%t)):\$((0x%T))" "$(readlink -e "$1")") || \
+	STAT=$(stat -f "echo \$((0x%XHr)):\$((0x%XLr))" "$(readlink -e "$1")") || \
 		errorexit "Cannot get major:minor for \"$1\"."
 	eval "$STAT"
 }
@@ -599,7 +599,7 @@ if [ "$DO_FSCK" -eq 1 ] && [ "$FSTYPE" = "xfs" ]; then
 fi
 
 if [ "$DO_MOUNT" -eq 1 ]; then
-	TMPDIR=$(mktemp --suffix _lvresize_$$ -d -p /tmp)
+	TMPDIR=$(mktemp -d -t tmp_lvresize_$$)
 	if [ ! -e "$TMPDIR" ]; then
 		errorexit "Failed to create temp dir."
 	fi
diff --git a/scripts/vgimportclone.sh b/scripts/vgimportclone.sh
index 3d795048a..aa7228a84 100755
--- a/scripts/vgimportclone.sh
+++ b/scripts/vgimportclone.sh
@@ -22,7 +22,7 @@ RM="rm"
 BASENAME="basename"
 MKTEMP="mktemp"
 READLINK="readlink"
-GETOPT="getopt"
+GETOPT="ugetopt"
 
 # user may override lvm location by setting LVM_BINARY
 LVM=${LVM_BINARY:-lvm}
