w doesnt exist anymore
--
--- a/aom_dsp/arm/highbd_convolve8_sve.c
+++ b/aom_dsp/arm/highbd_convolve8_sve.c
@@ -294,7 +294,6 @@
                                    const int16_t *filter_y, int y_step_q4,
                                    int width, int height, int bd) {
   assert(y_step_q4 == 16);
-  assert(w >= 4 && h >= 4);
   (void)filter_x;
   (void)y_step_q4;
   (void)x_step_q4;
--- a/av1/common/arm/highbd_convolve_sve2.c
+++ b/av1/common/arm/highbd_convolve_sve2.c
@@ -644,7 +644,6 @@
                                     uint16_t *dst, ptrdiff_t dst_stride,
                                     int width, int height,
                                     const int16_t *filter_y, int bd) {
-  assert(w >= 4 && h >= 4);
 
   const int16x8_t y_filter = vld1q_s16(filter_y);
 
@@ -813,7 +812,6 @@
                                     uint16_t *dst, ptrdiff_t dst_stride,
                                     int width, int height,
                                     const int16_t *filter_y, int bd) {
-  assert(w >= 4 && h >= 4);
 
   const int16x8_t y_filter =
       vcombine_s16(vld1_s16(filter_y + 2), vdup_n_s16(0));
@@ -1430,7 +1428,6 @@
                                           int height, const int16_t *filter_y,
                                           ConvolveParams *conv_params, int bd,
                                           const int y_offset) {
-  assert(w >= 4 && h >= 4);
   const int64x2_t offset = vdupq_n_s64(y_offset);
   const int32x4_t shift = vdupq_n_s32(-conv_params->round_1);
   const int16x8_t y_filter = vld1q_s16(filter_y);
@@ -1620,7 +1617,6 @@
                                           int height, const int16_t *filter_y,
                                           ConvolveParams *conv_params, int bd,
                                           const int y_offset) {
-  assert(w >= 4 && h >= 4);
   const int64x2_t offset = vdupq_n_s64(y_offset);
   const int32x4_t shift = vdupq_n_s32(-conv_params->round_1);
 
