diff -ruN ../klibc-kinit-standalone-0.0.1.orig/do_mounts_md.c ./do_mounts_md.c
--- ../klibc-kinit-standalone-0.0.1.orig/do_mounts_md.c	2022-07-17 11:08:11.000000000 -0500
+++ ./do_mounts_md.c	2024-04-02 12:47:09.967466096 -0500
@@ -189,7 +189,7 @@
 		md_setup_ents++;
 	switch (get_option(&str, &level)) {	/* RAID level */
 	case 2:		/* could be 0 or -1.. */
-		if (level == 0 || level == LEVEL_LINEAR) {
+		if (level == 0) {
 			if (get_option(&str, &factor) != 2 ||	/* Chunk Size */
 			    get_option(&str, &fault) != 2) {
 				fprintf(stderr,
@@ -198,11 +198,7 @@
 			}
 			md_setup_args[ent].level = level;
 			md_setup_args[ent].chunk = 1 << (factor + 12);
-			if (level == LEVEL_LINEAR)
-				pername = "linear";
-			else
-				pername = "raid0";
-			break;
+			pername = "raid0";
 		}
 		/* FALL THROUGH */
 	case 1:		/* the first device is numeric */
