diff --git a/meson.build.orig b/meson.build
index 2697887..6d8ce4a 100644
--- a/meson.build.orig
+++ b/meson.build
@@ -1,4 +1,4 @@
-project('wofi', 'c', version : 'v1.3', default_options : ['c_std=c99', 'buildtype=release', 'warning_level=2'])
+project('wofi', 'c', version : '1.3', default_options : ['c_std=c99', 'buildtype=release', 'warning_level=2'])
 cc = meson.get_compiler('c')
 
 pkgcfg = import('pkgconfig')
