From 9a003b6a80fd7e0dae118dd9790ac784d2a4a5b4 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Thu, 25 Nov 2021 02:47:44 +0100
Subject: [PATCH] use external libunwind for standalone libcxxabi build

---
 libcxxabi/src/CMakeLists.txt                             | 2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libcxxabi/src/CMakeLists.txt b/libcxxabi/src/CMakeLists.txt
index f07d433..61be851 100644
--- a/libcxxabi/src/CMakeLists.txt
+++ b/libcxxabi/src/CMakeLists.txt
@@ -280,7 +280,7 @@ if (LIBCXXABI_ENABLE_STATIC)
         --ar "${CMAKE_AR}"
         ${MERGE_ARCHIVES_LIBTOOL}
         "$<TARGET_LINKER_FILE:cxxabi_static>"
-        "$<TARGET_LINKER_FILE:unwind_static>"
+        "${CMAKE_SYSROOT}/usr/lib/libunwind.a"
       WORKING_DIRECTORY ${LIBCXXABI_BUILD_DIR}
       DEPENDS unwind_static
     )
-- 
2.34.0
