From 3f727cfb8f9bd6b3e3ac1ebf91ae7b9602f3d351 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=BCrg=20Billeter?= <j@bitron.ch>
Date: Tue, 28 Nov 2023 21:24:01 +0100
Subject: [PATCH] sanei_usb: Fix build with libxml2 2.12

---
 sanei/sanei_usb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sanei/sanei_usb.c b/sanei/sanei_usb.c
index 022f6fab9..850c3b369 100644
--- a/sanei/sanei_usb.c
+++ b/sanei/sanei_usb.c
@@ -65,6 +65,7 @@
 #include <time.h>
 
 #if WITH_USB_RECORD_REPLAY
+#include <libxml/parser.h>
 #include <libxml/tree.h>
 #endif
 
-- 
GitLab

