there is no float128 on musl/ppc*

diff -ruN a/numpy/_core/tests/test_scalarprint.py b/numpy/_core/tests/test_scalarprint.py
--- a/numpy/_core/tests/test_scalarprint.py	2025-06-07 13:57:13.000000000 +0200
+++ b/numpy/_core/tests/test_scalarprint.py	2025-06-09 22:53:05.340890370 +0200
@@ -247,7 +247,7 @@
     @pytest.mark.parametrize("tp", available_float_dtypes)
     def test_dragon4_positional_interface(self, tp):
         # test is flaky for musllinux on np.float128
-        if IS_MUSL and tp == np.float128:
+        if IS_MUSL and hasattr(np, 'float128') and tp == np.float128:
             pytest.skip("Skipping flaky test of float128 on musllinux")
 
         fpos = np.format_float_positional
@@ -264,7 +264,7 @@
     @pytest.mark.parametrize("tp", available_float_dtypes)
     def test_dragon4_positional_interface_trim(self, tp):
         # test is flaky for musllinux on np.float128
-        if IS_MUSL and tp == np.float128:
+        if IS_MUSL and hasattr(np, 'float128') and tp == np.float128:
             pytest.skip("Skipping flaky test of float128 on musllinux")
 
         fpos = np.format_float_positional
@@ -296,7 +296,7 @@
     @pytest.mark.parametrize("pad_val", [10**5, np.iinfo("int32").max])
     def test_dragon4_positional_interface_overflow(self, tp, pad_val):
         # test is flaky for musllinux on np.float128
-        if IS_MUSL and tp == np.float128:
+        if IS_MUSL and hasattr(np, 'float128') and tp == np.float128:
             pytest.skip("Skipping flaky test of float128 on musllinux")
 
         fpos = np.format_float_positional
@@ -317,7 +317,7 @@
     @pytest.mark.parametrize("tp", available_float_dtypes)
     def test_dragon4_scientific_interface(self, tp):
         # test is flaky for musllinux on np.float128
-        if IS_MUSL and tp == np.float128:
+        if IS_MUSL and hasattr(np, 'float128') and tp == np.float128:
             pytest.skip("Skipping flaky test of float128 on musllinux")
 
         fsci = np.format_float_scientific
