don't include it if we're not using it

--- a/numpy/_core/src/umath/loops_trigonometric.dispatch.cpp
+++ b/numpy/_core/src/umath/loops_trigonometric.dispatch.cpp
@@ -3,8 +3,10 @@
 #include "loops_utils.h"
 
 #include "simd/simd.h"
+#if NPY_SIMD_FMA3
 #include <hwy/highway.h>
 namespace hn = hwy::HWY_NAMESPACE;
+#endif
 
 /*
  * Vectorized approximate sine/cosine algorithms: The following code is a
