diff --git a/tests/kded/configtest.cpp b/tests/kded/configtest.cpp
index d23dd88..f3d4706 100644
--- a/tests/kded/configtest.cpp
+++ b/tests/kded/configtest.cpp
@@ -115,7 +115,6 @@ void TestConfig::testSimpleConfig()
     QCOMPARE(output->isEnabled(), true);
     QCOMPARE(output->rotation(), KScreen::Output::None);
     QCOMPARE(output->pos(), QPoint(0, 0));
-    QCOMPARE(output->priority(), 0);
 
     auto screen = config->screen();
     QCOMPARE(screen->currentSize(), QSize(1920, 1280));
@@ -138,7 +137,6 @@ void TestConfig::testTwoScreenConfig()
     QCOMPARE(output->isEnabled(), true);
     QCOMPARE(output->rotation(), KScreen::Output::None);
     QCOMPARE(output->pos(), QPoint(0, 0));
-    QCOMPARE(output->priority(), 0);
 
     output = config->connectedOutputs().last();
     QCOMPARE(output->name(), QLatin1String("OUTPUT-2"));
@@ -147,7 +145,6 @@ void TestConfig::testTwoScreenConfig()
     QCOMPARE(output->isEnabled(), true);
     QCOMPARE(output->rotation(), KScreen::Output::None);
     QCOMPARE(output->pos(), QPoint(1920, 0));
-    QCOMPARE(output->priority(), 1);
 
     auto screen = config->screen();
     QCOMPARE(screen->currentSize(), QSize(3200, 1280));
@@ -170,7 +167,6 @@ void TestConfig::testRotatedScreenConfig()
     QCOMPARE(output->isEnabled(), true);
     QCOMPARE(output->rotation(), KScreen::Output::None);
     QCOMPARE(output->pos(), QPoint(0, 0));
-    QCOMPARE(output->priority(), 0);
 
     output = config->connectedOutputs().last();
     QCOMPARE(output->name(), QLatin1String("OUTPUT-2"));
@@ -179,7 +175,6 @@ void TestConfig::testRotatedScreenConfig()
     QCOMPARE(output->isEnabled(), true);
     QCOMPARE(output->rotation(), KScreen::Output::Left);
     QCOMPARE(output->pos(), QPoint(1920, 0));
-    QCOMPARE(output->priority(), 1);
 
     auto screen = config->screen();
     QCOMPARE(screen->currentSize(), QSize(2944, 1280));
@@ -202,7 +197,6 @@ void TestConfig::testDisabledScreenConfig()
     QCOMPARE(output->isEnabled(), true);
     QCOMPARE(output->rotation(), KScreen::Output::None);
     QCOMPARE(output->pos(), QPoint(0, 0));
-    QCOMPARE(output->priority(), 0);
 
     output = config->connectedOutputs().last();
     QCOMPARE(output->name(), QLatin1String("OUTPUT-2"));
diff --git a/tests/kded/testgenerator.cpp b/tests/kded/testgenerator.cpp
index 9b45429..5a1f2be 100644
--- a/tests/kded/testgenerator.cpp
+++ b/tests/kded/testgenerator.cpp
@@ -90,7 +90,6 @@ void testScreenConfig::singleOutput()
 
     QCOMPARE(output->currentModeId(), QLatin1String("3"));
     QCOMPARE(output->isEnabled(), true);
-    QCOMPARE(output->priority(), 0);
     QCOMPARE(output->pos(), QPoint(0, 0));
 }
 
@@ -108,12 +107,10 @@ void testScreenConfig::laptopLidOpenAndExternal()
     OutputPtr external = config->outputs().value(2);
 
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
 
     QCOMPARE(external->currentModeId(), QLatin1String("4"));
-    QCOMPARE(external->priority(), 1);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(1280, 0));
 }
@@ -133,17 +130,14 @@ void testScreenConfig::laptopLidOpenAndTwoExternal()
     OutputPtr hdmi2 = config->outputs().value(3);
 
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
 
     QCOMPARE(hdmi1->currentModeId(), QLatin1String("4"));
-    QCOMPARE(hdmi1->priority(), 1);
     QCOMPARE(hdmi1->isEnabled(), true);
     QCOMPARE(hdmi1->pos(), QPoint(hdmi2->pos().x() + hdmi2->currentMode()->size().width(), 0));
 
     QCOMPARE(hdmi2->currentModeId(), QLatin1String("4"));
-    QCOMPARE(hdmi2->priority(), 1);
     QCOMPARE(hdmi2->isEnabled(), true);
     QCOMPARE(hdmi2->pos(), QPoint(1280, 0));
 }
@@ -163,11 +157,8 @@ void testScreenConfig::laptopLidClosedAndExternal()
     OutputPtr external = config->outputs().value(2);
 
     QCOMPARE(laptop->isEnabled(), false);
-    QCOMPARE(laptop->priority(), 1);
 
     QCOMPARE(external->currentModeId(), QLatin1String("4"));
-    QCOMPARE(external->priority(), 0);
-    QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 }
 
@@ -188,20 +179,16 @@ void testScreenConfig::laptopLidClosedAndThreeExternal()
     OutputPtr primary = config->outputs().value(4);
 
     QCOMPARE(laptop->isEnabled(), false);
-    QCOMPARE(laptop->priority(), 1);
 
     QCOMPARE(hdmi1->isEnabled(), true);
-    QCOMPARE(hdmi1->priority(), 1);
     QCOMPARE(hdmi1->currentModeId(), QLatin1String("4"));
     QCOMPARE(hdmi1->pos(), QPoint(primary->currentMode()->size().width(), 0));
 
     QCOMPARE(hdmi2->isEnabled(), true);
-    QCOMPARE(hdmi2->priority(), 1);
     QCOMPARE(hdmi2->currentModeId(), QLatin1String("3"));
     QCOMPARE(hdmi2->pos(), QPoint(hdmi1->pos().x() + hdmi1->currentMode()->size().width(), 0));
 
     QCOMPARE(primary->isEnabled(), true);
-    QCOMPARE(primary->priority(), 0);
     QCOMPARE(primary->currentModeId(), QLatin1String("4"));
     QCOMPARE(primary->pos(), QPoint(0, 0));
 }
@@ -222,12 +209,10 @@ void testScreenConfig::laptopDockedLidOpenAndExternal()
     OutputPtr external = config->outputs().value(2);
 
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 1);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
 
     QCOMPARE(external->currentModeId(), QLatin1String("4"));
-    QCOMPARE(external->priority(), 0);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(1280, 0));
 }
@@ -248,11 +233,8 @@ void testScreenConfig::laptopDockedLidClosedAndExternal()
     OutputPtr external = config->outputs().value(2);
 
     QCOMPARE(laptop->isEnabled(), false);
-    QCOMPARE(laptop->priority(), 1);
 
     QCOMPARE(external->currentModeId(), QLatin1String("4"));
-    QCOMPARE(external->priority(), 0);
-    QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 }
 
@@ -304,12 +286,10 @@ void testScreenConfig::workstationTwoExternalSameSize()
     OutputPtr external1 = config->output(1);
     OutputPtr external2 = config->output(2);
 
-    QCOMPARE(external1->priority(), 0);
     QCOMPARE(external1->isEnabled(), true);
     QCOMPARE(external1->currentModeId(), QLatin1String("3"));
     QCOMPARE(external1->pos(), QPoint(0, 0));
 
-    QCOMPARE(external2->priority(), 1);
     QCOMPARE(external2->isEnabled(), true);
     QCOMPARE(external2->currentModeId(), QLatin1String("3"));
     QCOMPARE(external2->pos(), QPoint(external1->currentMode()->size().width(), 0));
@@ -329,12 +309,10 @@ void testScreenConfig::workstationFallbackMode()
     OutputPtr external1 = config->output(1);
     OutputPtr external2 = config->output(2);
 
-    QCOMPARE(external1->priority(), 0);
     QCOMPARE(external1->isEnabled(), true);
     QCOMPARE(external1->currentModeId(), QLatin1String("1"));
     QCOMPARE(external1->pos(), QPoint(0, 0));
 
-    QCOMPARE(external2->priority(), 1);
     QCOMPARE(external2->isEnabled(), true);
     QCOMPARE(external2->currentModeId(), QLatin1String("1"));
     QCOMPARE(external2->pos(), QPoint(0, 0));
@@ -354,12 +332,10 @@ void testScreenConfig::workstationTwoExternalDiferentSize()
     OutputPtr external1 = config->output(1);
     OutputPtr external2 = config->output(2);
 
-    QCOMPARE(external1->priority(), 1);
     QCOMPARE(external1->isEnabled(), true);
     QCOMPARE(external1->currentModeId(), QLatin1String("3"));
     QCOMPARE(external1->pos(), QPoint(external2->currentMode()->size().width(), 0));
 
-    QCOMPARE(external2->priority(), 0);
     QCOMPARE(external2->isEnabled(), true);
     QCOMPARE(external2->currentModeId(), QLatin1String("4"));
 }
@@ -381,11 +357,9 @@ void testScreenConfig::switchDisplayTwoScreens()
     OutputPtr laptop = config->outputs().value(1);
     OutputPtr external = config->outputs().value(2);
     QCOMPARE(laptop->currentModeId(), QLatin1String("2"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
     QCOMPARE(external->currentModeId(), QLatin1String("3"));
-    QCOMPARE(external->priority(), 1);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 
@@ -394,11 +368,9 @@ void testScreenConfig::switchDisplayTwoScreens()
     laptop = config->outputs().value(1);
     external = config->outputs().value(2);
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(1920, 0));
     QCOMPARE(external->currentModeId(), QLatin1String("5"));
-    QCOMPARE(external->priority(), 1);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 
@@ -409,7 +381,6 @@ void testScreenConfig::switchDisplayTwoScreens()
     ;
     QCOMPARE(laptop->isEnabled(), false);
     QCOMPARE(external->currentModeId(), QLatin1String("5"));
-    QCOMPARE(external->priority(), 0);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 
@@ -419,7 +390,6 @@ void testScreenConfig::switchDisplayTwoScreens()
     external = config->outputs().value(2);
     ;
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
     ;
@@ -430,11 +400,9 @@ void testScreenConfig::switchDisplayTwoScreens()
     laptop = config->outputs().value(1);
     external = config->outputs().value(2);
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
     QCOMPARE(external->currentModeId(), QLatin1String("5"));
-    QCOMPARE(external->priority(), 1);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(1280, 0));
 }
@@ -469,12 +437,10 @@ void testScreenConfig::switchDisplayTwoScreensOneRotated()
     OutputPtr laptop = config->outputs().value(1);
     OutputPtr external = config->outputs().value(2);
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(1920, 0));
     QCOMPARE(laptop->rotation(), KScreen::Output::Right);
     QCOMPARE(external->currentModeId(), QLatin1String("5"));
-    QCOMPARE(external->priority(), 1);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 
@@ -485,7 +451,6 @@ void testScreenConfig::switchDisplayTwoScreensOneRotated()
     ;
     QCOMPARE(laptop->isEnabled(), false);
     QCOMPARE(external->currentModeId(), QLatin1String("5"));
-    QCOMPARE(external->priority(), 0);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 
@@ -495,7 +460,6 @@ void testScreenConfig::switchDisplayTwoScreensOneRotated()
     external = config->outputs().value(2);
     ;
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
     QCOMPARE(laptop->rotation(), KScreen::Output::Right);
@@ -507,12 +471,10 @@ void testScreenConfig::switchDisplayTwoScreensOneRotated()
     laptop = config->outputs().value(1);
     external = config->outputs().value(2);
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
     QCOMPARE(laptop->rotation(), KScreen::Output::Right);
     QCOMPARE(external->currentModeId(), QLatin1String("5"));
-    QCOMPARE(external->priority(), 1);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(800, 0));
 }
@@ -529,11 +491,9 @@ void testScreenConfig::switchDisplayTwoScreensNoCommonMode()
     OutputPtr external = config->outputs().value(2);
 
     QCOMPARE(laptop->currentModeId(), QLatin1String("3"));
-    QCOMPARE(laptop->priority(), 0);
     QCOMPARE(laptop->isEnabled(), true);
     QCOMPARE(laptop->pos(), QPoint(0, 0));
     QCOMPARE(external->currentModeId(), QLatin1String("5"));
-    QCOMPARE(external->priority(), 1);
     QCOMPARE(external->isEnabled(), true);
     QCOMPARE(external->pos(), QPoint(0, 0));
 }
