diff --git a/CMakeLists.txt b/CMakeLists.txt
index 78f6531..c2ae249 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.8.0)
-project(Ptex)
+project(Ptex VERSION 2.4.3)
 
 option(PTEX_BUILD_STATIC_LIBS "Enable building Ptex static libraries" ON)
 option(PTEX_BUILD_SHARED_LIBS "Enable building Ptex shared libraries" ON)
diff --git a/src/build/CMakeLists.txt b/src/build/CMakeLists.txt
index deb1441..86ca917 100644
--- a/src/build/CMakeLists.txt
+++ b/src/build/CMakeLists.txt
@@ -38,4 +38,4 @@ else()
 endif()
 configure_file("${PROJECT_NAME_LOWERCASE}.pc.in" "${PROJECT_NAME_LOWERCASE}.pc" @ONLY)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME_LOWERCASE}.pc"
-    COMPONENT devel DESTINATION share/pkgconfig)
+    COMPONENT devel DESTINATION lib/pkgconfig)
diff --git a/src/build/ptex.pc.in b/src/build/ptex.pc.in
index b81a2fd..c55912b 100644
--- a/src/build/ptex.pc.in
+++ b/src/build/ptex.pc.in
@@ -13,7 +13,7 @@ includedir="@CMAKE_INSTALL_PKGCONFIG_INCLUDEDIR@"
 Name: @PROJECT_NAME@
 Description: @CMAKE_PROJECT_DESCRIPTION@
 URL: @CMAKE_PROJECT_HOMEPAGE_URL@
-Version: @PROJECT_VERSION@
+Version: @CMAKE_PROJECT_VERSION@
 Requires: @pc_req_public@
 Requires.private: @pc_req_private@
 Cflags: -I"${includedir}"
