commit 1b18bfa7c5c3b1b3e300baceb1467badad648a59
Author: q66 <q66@chimera-linux.org>
Date:   Sat Apr 9 06:00:01 2022 +0200

    fix examples build

diff --git a/examples/Makefile.in b/examples/Makefile.in
index bf3147d..6b04970 100644
--- a/examples/Makefile.in
+++ b/examples/Makefile.in
@@ -133,7 +133,7 @@ am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/nft-buffer.Po \
 	./$(DEPDIR)/nft-json-file.Po
 am__mv = mv -f
-COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
+COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) -I$(top_srcdir)/include $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 LTCOMPILE = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) \
