--- a/tools/faust2appls/faust2faustvst
+++ b/tools/faust2appls/faust2faustvst
@@ -47,7 +47,7 @@ sdkpaths="/usr/src/VST* /usr/src/vst* /usr/include/VST* /usr/include/vst* /opt/l
 
 # The part below is obsolete
 # architecture location is now defined by faustpath and stored in a variable named FAUSTARCH
-#[ -z "$FAUSTLIB" ] && FAUSTLIB=$(dirname "$((ls -f /usr/share/faust/faustvstqt.h /usr/local/share/faust/faustvstqt.h /opt/local/share/faust/faustvstqt.h "$PWD/faustvstqt.h" 2>/dev/null)|tail -1)")
+#[ -z "$FAUSTLIB" ] && FAUSTLIB=$(dirname "$((ls -f /usr/share/faust/faustvstqt.h /usr/local/share/faust/faustvstqt.h /opt/local/share/faust/faustvstqt.h "$PWD/faustvstqt.h" 2>/dev/null)|tail -n 1)")
 #[ -z "$FAUSTLIB" ] && FAUSTLIB="$PWD"
 
 # defaults (these can be changed with the options listed below)
@@ -177,7 +177,7 @@ if [ ! -f "$FAUSTINC/faust/gui/QTUI.h" ]; then echo "$0: faust include files not
 if [ ! -f "$FAUSTARCH/faustvstqt.h" ]; then echo "$0: faust-vst library files not found" >&2; exit 1; fi
 
 # Determine the Qt version so that we can pick the needed compilation options.
-QTVERSION=$($QMAKE -v 2>/dev/null | tail -1 | sed 's/.*Qt version \([0-9]\).*/\1/')
+QTVERSION=$($QMAKE -v 2>/dev/null | tail -n 1 | sed 's/.*Qt version \([0-9]\).*/\1/')
 
 QTEXTRA=
 if [ $QTVERSION = 5 ]; then
diff --git a/tools/faust2appls/faust2lv2 b/tools/faust2appls/faust2lv2
index c904724..b0e4320 100755
--- a/tools/faust2appls/faust2lv2
+++ b/tools/faust2appls/faust2lv2
@@ -21,7 +21,7 @@
 # that you can run the script from the faust-lv2 source directory. You can
 # also specify this explicitly by setting the FAUSTLIB environment variable
 # accordingly.
-[ -z "$FAUSTLIB" ] && FAUSTLIB=$(dirname "$((ls -f /usr/share/faust/lv2ui.cpp /usr/local/share/faust/lv2ui.cpp /opt/local/share/faust/lv2ui.cpp "$PWD/lv2ui.cpp" 2>/dev/null)|tail -1)")
+[ -z "$FAUSTLIB" ] && FAUSTLIB=$(dirname "$((ls -f /usr/share/faust/lv2ui.cpp /usr/local/share/faust/lv2ui.cpp /opt/local/share/faust/lv2ui.cpp "$PWD/lv2ui.cpp" 2>/dev/null)|tail -n 1)")
 [ -z "$FAUSTLIB" ] && FAUSTLIB="$PWD"
 
 # defaults (these can be changed with the options listed below)
@@ -152,7 +152,7 @@ if [ ! -f "$FAUSTINC/gui/QTUI.h" ]; then echo "$0: faust include files not found
 if [ ! -f "$FAUSTLIB/lv2ui.cpp" ]; then echo "$0: faust-lv2 library files not found" >&2; exit 1; fi
 
 # Determine the Qt version so that we can edit the manifests accordingly.
-QTVERSION=$($QMAKE -v 2>/dev/null | tail -1 | sed 's/.*Qt version \([0-9]\).*/\1/')
+QTVERSION=$($QMAKE -v 2>/dev/null | tail -n 1 | sed 's/.*Qt version \([0-9]\).*/\1/')
 
 arch=lv2.cpp
 archui=lv2ui.cpp
diff --git a/tools/faust2appls/faust2pure b/tools/faust2appls/faust2pure
index 265e4c4..7cb1b4b 100755
--- a/tools/faust2appls/faust2pure
+++ b/tools/faust2appls/faust2pure
@@ -6,7 +6,7 @@
 # locations here, and fall back to the current directory if the file isn't
 # found. You can also specify this explicitly by setting the FAUSTLIB
 # environment variable accordingly.
-[ -z "$FAUSTLIB" ] && FAUSTLIB=$(dirname "$((ls -f /usr/share/faust/pure.cpp /usr/local/share/faust/pure.cpp /opt/local/share/faust/pure.cpp "$PWD/pure.cpp" 2>/dev/null)|tail -1)")
+[ -z "$FAUSTLIB" ] && FAUSTLIB=$(dirname "$((ls -f /usr/share/faust/pure.cpp /usr/local/share/faust/pure.cpp /opt/local/share/faust/pure.cpp "$PWD/pure.cpp" 2>/dev/null)|tail -n 1)")
 [ -z "$FAUSTLIB" ] && FAUSTLIB="$PWD"
 
 # defaults (these can be changed with the options listed below)
