diff --git a/dnscrypt-proxy/example-dnscrypt-proxy.toml b/dnscrypt-proxy/example-dnscrypt-proxy.toml
index 8d53a795..7de6f36b 100644
--- a/dnscrypt-proxy/example-dnscrypt-proxy.toml
+++ b/dnscrypt-proxy/example-dnscrypt-proxy.toml
@@ -50,7 +50,7 @@ max_clients = 250
 ## Note (2): this feature is not compatible with systemd socket activation.
 ## Note (3): when using -pidfile, the PID file directory must be writable by the new user
 
-# user_name = 'nobody'
+user_name = '_dnscrypt'
 
 
 ###############################################################################
@@ -228,7 +228,7 @@ keepalive = 30
 ## This file is different from other log files, and will not be
 ## automatically rotated by the application.
 
-# log_file = 'dnscrypt-proxy.log'
+# log_file = '/var/log/dnscrypt-proxy/dnscrypt-proxy.log'
 
 
 ## When using a log file, only keep logs from the most recent launch.
@@ -443,9 +443,9 @@ reject_ttl = 10
 
 ## Route queries for specific domains to a dedicated set of servers
 
-## See the `example-forwarding-rules.txt` file for an example
+## See the `/usr/share/dnscrypt-proxy/forwarding-rules.txt` file for an example
 
-# forwarding_rules = 'forwarding-rules.txt'
+# forwarding_rules = '/usr/share/dnscrypt-proxy/forwarding-rules.txt'
 
 
 ###############################################################################
@@ -458,9 +458,9 @@ reject_ttl = 10
 ## If 'cloak_ptr' is set, then PTR (reverse lookups) are enabled
 ## for cloaking rules that do not contain wild cards.
 ##
-## See the `example-cloaking-rules.txt` file for an example
+## See the `/usr/share/dnscrypt-proxy/cloaking-rules.txt` file for an example
 
-# cloaking_rules = 'cloaking-rules.txt'
+# cloaking_rules = '/usr/share/dnscrypt-proxy/cloaking-rules.txt'
 
 ## TTL used when serving entries in cloaking-rules.txt
 
@@ -512,7 +512,7 @@ cache_neg_max_ttl = 600
 ## check for connectivity and captive portals, along with hard-coded
 ## IP addresses to return.
 
-# map_file = 'example-captive-portals.txt'
+# map_file = '/usr/share/dnscrypt-proxy/captive-portals.txt'
 
 
 ###############################################################################
@@ -556,7 +556,7 @@ cache_neg_max_ttl = 600
 ## Path to the query log file (absolute, or relative to the same directory as the config file)
 ## Can be set to /dev/stdout in order to log to the standard output.
 
-# file = 'query.log'
+# file = '/var/log/dnscrypt-proxy/query.log'
 
 
 ## Query log format (currently supported: tsv and ltsv)
@@ -587,7 +587,7 @@ format = 'tsv'
 
 ## Path to the query log file (absolute, or relative to the same directory as the config file)
 
-# file = 'nx.log'
+# file = '/var/log/dnscrypt-proxy/nx.log'
 
 
 ## Query log format (currently supported: tsv and ltsv)
@@ -616,12 +616,12 @@ format = 'tsv'
 
 ## Path to the file of blocking rules (absolute, or relative to the same directory as the config file)
 
-# blocked_names_file = 'blocked-names.txt'
+# blocked_names_file = '/usr/share/dnscrypt-proxy/blocked-names.txt'
 
 
 ## Optional path to a file logging blocked queries
 
-# log_file = 'blocked-names.log'
+# log_file = '/var/log/dnscrypt-proxy/blocked-names.log'
 
 
 ## Optional log format: tsv or ltsv (default: tsv)
@@ -643,12 +643,12 @@ format = 'tsv'
 
 ## Path to the file of blocking rules (absolute, or relative to the same directory as the config file)
 
-# blocked_ips_file = 'blocked-ips.txt'
+# blocked_ips_file = '/usr/share/dnscrypt-proxy/blocked-ips.txt'
 
 
 ## Optional path to a file logging blocked queries
 
-# log_file = 'blocked-ips.log'
+# log_file = '/var/log/dnscrypt-proxy/blocked-ips.log'
 
 
 ## Optional log format: tsv or ltsv (default: tsv)
@@ -670,12 +670,12 @@ format = 'tsv'
 
 ## Path to the file of allow list rules (absolute, or relative to the same directory as the config file)
 
-# allowed_names_file = 'allowed-names.txt'
+# allowed_names_file = '/usr/share/dnscrypt-proxy/allowed-names.txt'
 
 
 ## Optional path to a file logging allowed queries
 
-# log_file = 'allowed-names.log'
+# log_file = '/var/log/dnscrypt-proxy/allowed-names.log'
 
 
 ## Optional log format: tsv or ltsv (default: tsv)
@@ -695,12 +695,12 @@ format = 'tsv'
 
 ## Path to the file of allowed ip rules (absolute, or relative to the same directory as the config file)
 
-# allowed_ips_file = 'allowed-ips.txt'
+# allowed_ips_file = '/usr/share/dnscrypt-proxy/allowed-ips.txt'
 
 
 ## Optional path to a file logging allowed queries
 
-# log_file = 'allowed-ips.log'
+# log_file = '/var/log/dnscrypt-proxy/allowed-ips.log'
 
 ## Optional log format: tsv or ltsv (default: tsv)
 
@@ -775,9 +775,9 @@ format = 'tsv'
 urls = [
   'https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/public-resolvers.md',
   'https://download.dnscrypt.info/resolvers-list/v3/public-resolvers.md',
-  'https://cdn.jsdelivr.net/gh/DNSCrypt/dnscrypt-resolvers@master/v3/public-resolvers.md'
+  'https://cdn.jsdelivr.net/gh/DNSCrypt/dnscrypt-resolvers@master/v3/public-resolvers.md',
 ]
-cache_file = 'public-resolvers.md'
+cache_file = '/var/cache/dnscrypt-proxy/public-resolvers.md'
 minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
 refresh_delay = 73
 prefix = ''
@@ -788,9 +788,9 @@ prefix = ''
 urls = [
   'https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/relays.md',
   'https://download.dnscrypt.info/resolvers-list/v3/relays.md',
-  'https://cdn.jsdelivr.net/gh/DNSCrypt/dnscrypt-resolvers@master/v3/relays.md'
+  'https://cdn.jsdelivr.net/gh/DNSCrypt/dnscrypt-resolvers@master/v3/relays.md',
 ]
-cache_file = 'relays.md'
+cache_file = '/var/cache/dnscrypt-proxy/relays.md'
 minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
 refresh_delay = 73
 prefix = ''
@@ -799,13 +799,13 @@ prefix = ''
 
 # [sources.odoh-servers]
 #   urls = ['https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/odoh-servers.md', 'https://download.dnscrypt.info/resolvers-list/v3/odoh-servers.md', 'https://cdn.jsdelivr.net/gh/DNSCrypt/dnscrypt-resolvers@master/v3/odoh-servers.md']
-#   cache_file = 'odoh-servers.md'
+#   cache_file = '/var/cache/dnscrypt-proxy/odoh-servers.md'
 #   minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
 #   refresh_delay = 73
 #   prefix = ''
 # [sources.odoh-relays]
 #   urls = ['https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/odoh-relays.md', 'https://download.dnscrypt.info/resolvers-list/v3/odoh-relays.md', 'https://cdn.jsdelivr.net/gh/DNSCrypt/dnscrypt-resolvers@master/v3/odoh-relays.md']
-#   cache_file = 'odoh-relays.md'
+#   cache_file = '/var/cache/dnscrypt-proxy/odoh-relays.md'
 #   minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
 #   refresh_delay = 73
 #   prefix = ''
@@ -815,7 +815,7 @@ prefix = ''
 # [sources.quad9-resolvers]
 #   urls = ['https://quad9.net/dnscrypt/quad9-resolvers.md']
 #   minisign_key = 'RWQBphd2+f6eiAqBsvDZEBXBGHQBJfeG6G+wJPPKxCZMoEQYpmoysKUN'
-#   cache_file = 'quad9-resolvers.md'
+#   cache_file = '/var/cache/dnscrypt-proxy/quad9-resolvers.md'
 #   prefix = 'quad9-'
 
 ### Another example source, with resolvers censoring some websites not appropriate for children
@@ -823,7 +823,7 @@ prefix = ''
 
 # [sources.parental-control]
 #   urls = ['https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/parental-control.md', 'https://download.dnscrypt.info/resolvers-list/v3/parental-control.md', 'https://cdn.jsdelivr.net/gh/DNSCrypt/dnscrypt-resolvers@master/v3/parental-control.md']
-#   cache_file = 'parental-control.md'
+#   cache_file = '/var/cache/dnscrypt-proxy/parental-control.md'
 #   minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
 
 ### dnscry.pt servers - See https://www.dnscry.pt
@@ -831,7 +831,7 @@ prefix = ''
 #  [sources.dnscry-pt-resolvers]
 #    urls = ["https://www.dnscry.pt/resolvers.md"]
 #    minisign_key = "RWQM31Nwkqh01x88SvrBL8djp1NH56Rb4mKLHz16K7qsXgEomnDv6ziQ"
-#    cache_file = "dnscry.pt-resolvers.md"
+#    cache_file = "/var/cache/dnscrypt-proxy/dnscry.pt-resolvers.md"
 #    refresh_delay = 73
 #    prefix = "dnscry.pt-"
 
