--- a/data/Makefile
+++ b/data/Makefile
@@ -20,13 +20,13 @@ install-packed: files-install db-install-packed
 files-install: 
 	install -d $(PREFIX)/share/usb_modeswitch
 	install -d $(ETCDIR)/usb_modeswitch.d
-	install -D --mode=644 40-usb_modeswitch.rules $(RULESDIR)/40-usb_modeswitch.rules
+	install -D -m644 40-usb_modeswitch.rules $(RULESDIR)/40-usb_modeswitch.rules
 
 40-usb_modeswitch.rules:
 	./gen-rules.tcl
 
 db-install: files-install
-	install --mode=644 -t $(PREFIX)/share/usb_modeswitch ./usb_modeswitch.d/*
+	install -m644 -t $(PREFIX)/share/usb_modeswitch ./usb_modeswitch.d/*
 
 db-install-packed:
 	@# Create a compressed tar without gzip timestamp, so tar.gz
