--- a/Makefile
+++ b/Makefile
@@ -54,15 +54,15 @@ distclean: clean
 # If not, use the dispatcher directly from the udev rule as in previous versions
 
 install-common: $(PROG) $(DISPATCH)
-	install -D --mode=755 usb_modeswitch $(SBINDIR)/usb_modeswitch
-	install -D --mode=755 usb_modeswitch.sh $(UDEVDIR)/usb_modeswitch
-	install -D --mode=644 usb_modeswitch.conf $(ETCDIR)/usb_modeswitch.conf
-	install -D --mode=644 usb_modeswitch.1 $(MANDIR)/usb_modeswitch.1
-	install -D --mode=644 usb_modeswitch_dispatcher.1 $(MANDIR)/usb_modeswitch_dispatcher.1
-	install -D --mode=755 usb_modeswitch_dispatcher $(SBINDIR)/usb_modeswitch_dispatcher
+	install -D -m755 usb_modeswitch $(SBINDIR)/usb_modeswitch
+	install -D -m755 usb_modeswitch.sh $(UDEVDIR)/usb_modeswitch
+	install -D -m644 usb_modeswitch.conf $(ETCDIR)/usb_modeswitch.conf
+	install -D -m644 usb_modeswitch.1 $(MANDIR)/usb_modeswitch.1
+	install -D -m644 usb_modeswitch_dispatcher.1 $(MANDIR)/usb_modeswitch_dispatcher.1
+	install -D -m755 usb_modeswitch_dispatcher $(SBINDIR)/usb_modeswitch_dispatcher
 	install -d $(DESTDIR)/var/lib/usb_modeswitch
-	test -d $(UPSDIR) -a -e /sbin/initctl && install --mode=644 usb-modeswitch-upstart.conf $(UPSDIR) || test 1
-	test -d $(SYSDIR) -a \( -e /usr/bin/systemctl -o -e /bin/systemctl \) && install --mode=644 usb_modeswitch@.service $(SYSDIR) || test 1
+	test -d $(UPSDIR) -a -e /sbin/initctl && install -m644 usb-modeswitch-upstart.conf $(UPSDIR) || test 1
+	test -d $(SYSDIR) -a \( -e /usr/bin/systemctl -o -e /bin/systemctl \) && install -m644 usb_modeswitch@.service $(SYSDIR) || test 1
 
 install: install-script
 
