From bed88919bcc26e4c332457d16de5636f7bed0104 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Fri, 26 Nov 2021 02:18:12 +0100
Subject: [PATCH] ???

---
 Makefile.in  |  1 -
 configure.ac | 14 --------------
 2 files changed, 15 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index b1fd4aa..b67f6e5 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -168,7 +168,6 @@ ifeq ("@DEVTOOLS@", "yes")
 CXXFLAGS+=-DDEV_TOOLS
 endif
 
-CXXFLAGS+=@CXXOPTIMISE_FLAG@
 CXXFLAGS+=@CXXDEBUG_FLAG@
 CXXFLAGS+=@CXX_STRERROR_FLAG@
 CXXFLAGS+=@LFS_FLAGS@
diff --git a/configure.ac b/configure.ac
index 060118d..fe365a5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -128,20 +128,6 @@ if test x$GPROF = xyes; then
         CXXDEBUG_FLAG+=" -pg"
 fi
 
-################################################################################
-dnl -- Override optimisation
-AC_MSG_CHECKING(for C++ optimisation flag)
-CXXOPTIMISE_FLAG="-O8"
-AC_ARG_WITH(optimisation,
-            AC_HELP_STRING([--with-optimisation=OPT],
-                           [C++ optimisation flag [[OPT=-O8]]]),
-            CXXOPTIMISE_FLAG=$withval)
-AC_MSG_RESULT($CXXOPTIMISE_FLAG)
-
-if test x$CXXOPTIMISE_FLAG = xyes; then
-   CXXOPTIMISE_FLAG=-O8
-fi
-
 ################################################################################
 dnl -- Enable testing
 AC_MSG_CHECKING(whether to enable unit testing)
-- 
2.34.0

