diff --git a/runners/bookmarks/autotests/chrome/testchromebookmarks.cpp b/runners/bookmarks/autotests/chrome/testchromebookmarks.cpp
index b017c7e..a564b5f 100644
--- a/runners/bookmarks/autotests/chrome/testchromebookmarks.cpp
+++ b/runners/bookmarks/autotests/chrome/testchromebookmarks.cpp
@@ -23,7 +23,7 @@ void verifyMatch(BookmarkMatch &match, const QString &title, const QString &url)
 
 void TestChromeBookmarks::initTestCase()
 {
-    m_configHome = QFINDTESTDATA("chrome-config-home");
+    m_configHome = QFINDTESTDATA("../../runners/bookmarks/autotests/chrome/chrome-config-home");
 }
 
 void TestChromeBookmarks::bookmarkFinderShouldFindEachProfileDirectory()
diff --git a/runners/bookmarks/autotests/firefox/testfirefoxbookmarks.cpp b/runners/bookmarks/autotests/firefox/testfirefoxbookmarks.cpp
index eef904a..ba9716c 100644
--- a/runners/bookmarks/autotests/firefox/testfirefoxbookmarks.cpp
+++ b/runners/bookmarks/autotests/firefox/testfirefoxbookmarks.cpp
@@ -30,7 +30,7 @@ private Q_SLOTS:
 void TestFirefoxBookmarksMatch::initTestCase()
 {
     QStandardPaths::setTestModeEnabled(true);
-    m_firefox = std::make_unique<Firefox>(QFINDTESTDATA("firefox-config-home"));
+    m_firefox = std::make_unique<Firefox>(QFINDTESTDATA("../../runners/bookmarks/autotests/firefox/firefox-config-home"));
 }
 
 void TestFirefoxBookmarksMatch::testAllBookmarks()

