diff --git a/kcms/region_language/glibclocaleconstructor.cpp b/kcms/region_language/glibclocaleconstructor.cpp
index 46af7879e5..391d9ff179 100644
--- a/kcms/region_language/glibclocaleconstructor.cpp
+++ b/kcms/region_language/glibclocaleconstructor.cpp
@@ -155,7 +155,7 @@ void GlibcLocaleConstructor::constructGlibcLocaleMap()
 
 QString GlibcLocaleConstructor::localeFileDirPath()
 {
-    return QStringLiteral("/usr/share/i18n/locales");
+    return QStringLiteral("/usr/share/i18n/locales/musl");
 }
 
 QString GlibcLocaleConstructor::failedFindLocalesMessage()
