--- a/pyproject.toml	2025-05-14 22:11:10.045771211 +0100
+++ b/pyproject.toml	2025-05-14 22:12:29.615164540 +0100
@@ -103,7 +103,7 @@
 ignore_errors = true
 
 [build-system]
-requires = ['setuptools>=65.4.1', 'wheel', 'Cython>=3,<3.1.0', "poetry-core>=1.0.0"]
+requires = ['setuptools>=65.4.1', 'wheel', 'Cython>=3.1.0,<3.2.0', "poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.ruff]
