diff --color -ruN python-dbus-fast/pyproject.toml python-dbus-fast-patched/pyproject.toml
--- python-dbus-fast/pyproject.toml	2025-08-04 22:33:04.437995264 +0000
+++ python-dbus-fast-patched/pyproject.toml	2025-08-04 22:32:07.799066057 +0000
@@ -46,7 +46,6 @@
 
 [tool.poetry.group.dev.dependencies]
 pytest = ">=7,<9"
-pytest-cov = ">=3,<7"
 pytest-asyncio = ">=0.19,<1.2"
 pycairo = "^1.21.0"
 PyGObject = {version = ">=3.50,<3.51", python = "<4"}
@@ -63,7 +62,7 @@
 build_command = "pip install poetry && poetry build"
 
 [tool.pytest.ini_options]
-addopts = "-v -Wdefault --cov=dbus_fast --cov-report=term-missing:skip-covered"
+addopts = "-v -Wdefault"
 pythonpath = ["src"]
 
 [tool.coverage.run]
