--- a/audisp/plugins/af_unix/Makefile.am
+++ b/audisp/plugins/af_unix/Makefile.am
@@ -41,7 +41,7 @@
 install-data-hook:
 	mkdir -p -m 0750 ${DESTDIR}${plugin_confdir}
 	for i in $(CONF_FILES); do \
-		$(INSTALL_DATA) -D -m 640 ${srcdir}/"$$i" \
+		$(INSTALL_DATA) -m 640 ${srcdir}/"$$i" \
 			${DESTDIR}${plugin_confdir}; \
 	done
 
--- a/audisp/plugins/ids/Makefile.am
+++ b/audisp/plugins/ids/Makefile.am
@@ -42,8 +42,8 @@
 
 install-data-hook:
 	mkdir -p -m 0750 ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(prog_conf) ${DESTDIR}${prog_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(prog_conf) ${DESTDIR}${prog_confdir}
 
 uninstall-hook:
 	rm ${DESTDIR}${plugin_confdir}/$(plugin_conf)
--- a/audisp/plugins/statsd/Makefile.am
+++ b/audisp/plugins/statsd/Makefile.am
@@ -35,8 +35,8 @@
 
 install-data-hook:
 	mkdir -p -m 0750 ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(prog_conf) ${DESTDIR}${prog_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(prog_conf) ${DESTDIR}${prog_confdir}
 
 uninstall-hook:
 	rm ${DESTDIR}${plugin_confdir}/$(plugin_conf)
--- a/audisp/plugins/remote/Makefile.am
+++ b/audisp/plugins/remote/Makefile.am
@@ -44,8 +44,8 @@
 
 install-data-hook:
 	mkdir -p -m 0750 ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(prog_conf) ${DESTDIR}${prog_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(prog_conf) ${DESTDIR}${prog_confdir}
 
 uninstall-hook:
 	rm ${DESTDIR}${plugin_confdir}/$(plugin_conf)
--- a/audisp/plugins/filter/Makefile.am
+++ b/audisp/plugins/filter/Makefile.am
@@ -39,8 +39,8 @@
 
 install-data-hook:
 	mkdir -p -m 0750 ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(prog_conf) ${DESTDIR}${prog_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(prog_conf) ${DESTDIR}${prog_confdir}
 
 uninstall-hook:
 	rm ${DESTDIR}${plugin_confdir}/$(plugin_conf)
--- a/audisp/plugins/syslog/Makefile.am
+++ b/audisp/plugins/syslog/Makefile.am
@@ -38,7 +38,7 @@
 
 install-data-hook:
 	mkdir -p -m 0750 ${DESTDIR}${plugin_confdir}
-	$(INSTALL_DATA) -D -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
+	$(INSTALL_DATA) -m 640 ${srcdir}/$(plugin_conf) ${DESTDIR}${plugin_confdir}
 
 uninstall-hook:
 	rm ${DESTDIR}${plugin_confdir}/$(plugin_conf)
