commit f46f27b7d6ae53b21178312971b3a612d4cfad7a
Author: q66 <q66@chimera-linux.org>
Date:   Wed Mar 9 02:42:57 2022 +0100

    fix gsettings-desktop-schemas version requirement for apk

diff --git a/meson.build b/meson.build
index 4ad19c7..8f19b59 100644
--- a/meson.build
+++ b/meson.build
@@ -25,7 +25,7 @@ uprof_req = '>= 0.3'
 cairo_req = '>= 1.10.0'
 pangocairo_req = '>= 1.20'
 pixman_req = '>= 0.42'
-gsettings_desktop_schemas_req = '>= 47.beta'
+gsettings_desktop_schemas_req = '>= 47'
 x11_req = '>= 1.7.0'
 xcomposite_req = '>= 0.4'
 xkbcommon_req = '>= 0.4.3'
