--- a/src/backends/native/meta-onscreen-native.c
+++ b/src/backends/native/meta-onscreen-native.c
@@ -552,7 +552,7 @@ meta_onscreen_native_flip_crtc (CoglOnscreen           *onscreen,
                                                          render_gpu);
   switch (renderer_gpu_data->mode)
     {
-    case META_RENDERER_NATIVE_MODE_GBM:
+    case META_RENDERER_NATIVE_MODE_GBM: {
       graphene_rect_t src_rect;
       MtkRectangle dst_rect;
 
@@ -602,6 +602,7 @@ meta_onscreen_native_flip_crtc (CoglOnscreen           *onscreen,
                                                    rectangles, n_rectangles);
         }
       break;
+    }
     case META_RENDERER_NATIVE_MODE_SURFACELESS:
       g_assert_not_reached ();
       break;
