--- a/vendor/boringssl/include/openssl/target.h
+++ b/vendor/boringssl/include/openssl/target.h
@@ -54,6 +54,10 @@
 #define OPENSSL_32_BIT
 #elif defined(__myriad2__)
 #define OPENSSL_32_BIT
+#elif defined(__powerpc64__)
+#define OPENSSL_64_BIT
+#elif defined(__powerpc__)
+#define OPENSSL_32_BIT
 #else
 // The list above enumerates the platforms that BoringSSL supports. For these
 // platforms we keep a reasonable bar of not breaking them: automated test
--- a/vendor/e2fsprogs/contrib/android/perms.h
+++ b/vendor/e2fsprogs/contrib/android/perms.h
@@ -1,6 +1,7 @@
 #ifndef ANDROID_PERMS_H
 # define ANDROID_PERMS_H
 
+# include <linux/types.h>
 # include <ext2fs/ext2fs.h>
 
 typedef void (*fs_config_f)(const char *path, int dir,
