commit a363a5b04dbe879e5e82eb9a22036df1a735eef6
Author: q66 <q66@chimera-linux.org>
Date:   Fri Mar 17 06:11:26 2023 +0100

    use proper MAKE variable

diff --git a/Makefile.in b/Makefile.in
index 366776f..4ea8874 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -141,7 +141,7 @@ install: $(LIBNEWT) install-sh whiptail
 	install -m 644 $(LIBNEWT) $(instroot)/$(libdir)
 	install -m 755 whiptail $(instroot)/$(bindir)
 	install -m 644 whiptail.1 $(instroot)/$(man1dir)
-	make -C po datadir=$(instroot)/$(datadir) install
+	$(MAKE) -C po datadir=$(instroot)/$(datadir) install
 
 install-sh: sharedlib $(WHIPTCLSO) _snack.$(SOEXT)
 	[ -d $(instroot)/$(libdir) ] || install -m 755 -d $(instroot)/$(libdir)
