these tests require a postgresql installation

diff -ruN a/hscontrol/db/db_test.go b/hscontrol/db/db_test.go
--- a/hscontrol/db/db_test.go	2025-11-11 20:17:02.000000000 +0100
+++ b/hscontrol/db/db_test.go	2025-11-30 14:47:51.425469973 +0100
@@ -4,7 +4,6 @@
 	"database/sql"
 	"net/netip"
 	"os"
-	"os/exec"
 	"path/filepath"
 	"slices"
 	"strings"
@@ -843,10 +842,6 @@
 	}
 
 	for _, tt := range tests {
-		t.Run(tt.name+"-postgres", func(t *testing.T) {
-			db := newPostgresTestDB(t)
-			tt.run(t, db.DB.Debug())
-		})
 		t.Run(tt.name+"-sqlite", func(t *testing.T) {
 			db, err := newSQLiteTestDB()
 			if err != nil {
@@ -854,68 +849,6 @@
 			}
 
 			tt.run(t, db.DB.Debug())
-		})
-	}
-}
-
-// TestPostgresMigrationAndDataValidation tests specific PostgreSQL migration scenarios
-// and validates data integrity after migration. All migrations that require data validation
-// should be added here.
-//
-// TODO(kradalby): Convert to use plain text SQL dumps instead of binary .pssql dumps for consistency
-// with SQLite tests and easier version control.
-func TestPostgresMigrationAndDataValidation(t *testing.T) {
-	tests := []struct {
-		name     string
-		dbPath   string
-		wantFunc func(*testing.T, *HSDatabase)
-	}{
-		{
-			name:   "user-idx-breaking",
-			dbPath: "testdata/postgres/pre-24-postgresdb.pssql.dump",
-			wantFunc: func(t *testing.T, hsdb *HSDatabase) {
-				t.Helper()
-				users, err := Read(hsdb.DB, func(rx *gorm.DB) ([]types.User, error) {
-					return ListUsers(rx)
-				})
-				require.NoError(t, err)
-
-				for _, user := range users {
-					assert.NotEmpty(t, user.Name)
-					assert.Empty(t, user.ProfilePicURL)
-					assert.Empty(t, user.Email)
-				}
-			},
-		},
-	}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			u := newPostgresDBForTest(t)
-
-			pgRestorePath, err := exec.LookPath("pg_restore")
-			if err != nil {
-				t.Fatal("pg_restore not found in PATH. Please install it and ensure it is accessible.")
-			}
-
-			// Construct the pg_restore command
-			cmd := exec.Command(pgRestorePath, "--verbose", "--if-exists", "--clean", "--no-owner", "--dbname", u.String(), tt.dbPath)
-
-			// Set the output streams
-			cmd.Stdout = os.Stdout
-			cmd.Stderr = os.Stderr
-
-			// Execute the command
-			err = cmd.Run()
-			if err != nil {
-				t.Fatalf("failed to restore postgres database: %s", err)
-			}
-
-			db = newHeadscaleDBFromPostgresURL(t, u)
-
-			if tt.wantFunc != nil {
-				tt.wantFunc(t, db)
-			}
 		})
 	}
 }
