From 44cb98cb0620ea98c43d0f11ce92f5692ad57212 Mon Sep 17 00:00:00 2001
From: Rui Chen <rui@chenrui.dev>
Date: Sun, 9 Mar 2025 01:17:09 -0500
Subject: [PATCH] build: honor version set by ldflag

Signed-off-by: Rui Chen <rui@chenrui.dev>
---
 internal/version/version.go | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/internal/version/version.go b/internal/version/version.go
index 6fc0600637..f632d0ce7f 100644
--- a/internal/version/version.go
+++ b/internal/version/version.go
@@ -11,13 +11,14 @@ var (
 )
 
 func init() {
-	info, ok := debug.ReadBuildInfo()
-	if !ok || info.Main.Version == "(devel)" || info.Main.Version == "" {
-		version = "unknown"
-	} else {
-		if version == "" {
+	if version == "" {
+		info, ok := debug.ReadBuildInfo()
+		if !ok || info.Main.Version == "(devel)" || info.Main.Version == "" {
+			version = "unknown"
+		} else {
 			version = info.Main.Version
 		}
+
 		if sum == "" {
 			sum = info.Main.Sum
 		}
