diff --git a/utils/v4l2-dbg/meson.build b/utils/v4l2-dbg/meson.build
index c23bf8f3..c61af6a2 100644
--- a/utils/v4l2-dbg/meson.build
+++ b/utils/v4l2-dbg/meson.build
@@ -13,7 +13,7 @@ v4l2_dbg_sources = files(
 v4l2_dbg = executable('v4l2-dbg',
                       v4l2_dbg_sources,
                       install : true,
-                      install_dir : 'sbin',
+                      install_dir : 'bin',
                       include_directories : [
                           v4l2_utils_incdir,
                           utils_common_incdir,
