--- a/common/config.h.in
+++ b/common/config.h.in
@@ -234,13 +234,13 @@ void srandom(unsigned int seed);
 #endif
 
 /* Define if you have the __argz_count function.  */
-#define HAVE___ARGZ_COUNT 1
+/*#define HAVE___ARGZ_COUNT 1*/
 
 /* Define if you have the __argz_next function.  */
-#define HAVE___ARGZ_NEXT 1
+/*#define HAVE___ARGZ_NEXT 1*/
 
 /* Define if you have the __argz_stringify function.  */
-#define HAVE___ARGZ_STRINGIFY 1
+/*#define HAVE___ARGZ_STRINGIFY 1*/
 
 /* Define if you have the getcwd function.  */
 #define HAVE_GETCWD 1
@@ -288,7 +288,7 @@ void srandom(unsigned int seed);
 #define HAVE_ALLOCA_H 1
 
 /* Define if you have the <argz.h> header file.  */
-#define HAVE_ARGZ_H 1
+/*#define HAVE_ARGZ_H 1*/
 
 /* Define if you have the <dirent.h> header file.  */
 #define HAVE_DIRENT_H 1
