--- a/libobs-opengl/gl-egl-common.c
+++ b/libobs-opengl/gl-egl-common.c
@@ -387,11 +387,13 @@ bool gl_egl_query_dmabuf_modifiers_for_format(EGLDisplay egl_display, uint32_t d
 		blog(LOG_ERROR, "Unable to load eglQueryDmaBufModifiersEXT");
 		return false;
 	}
-	if (!query_dmabuf_modifiers(egl_display, drm_format, modifiers, n_modifiers)) {
+	EGLuint64KHR nmod;
+	if (!query_dmabuf_modifiers(egl_display, drm_format, modifiers, &nmod)) {
 		*n_modifiers = 0;
 		*modifiers = NULL;
 		return false;
 	}
+	*n_modifiers = (size_t)nmod;
 	return true;
 }
 
