diff --git a/test-output/src/tests/output.rs b/test-output/src/tests/output.rs
index 9833e8a..34be3aa 100644
--- a/test-output/src/tests/output.rs
+++ b/test-output/src/tests/output.rs
@@ -98,8 +98,8 @@ macro_rules! assert_output {
         let snapshot_name = snapshot_name(None, None, $project_name);
         insta::allow_duplicates! {
             assert_output!(&snapshot_name, Some(Target::Erlang), None, $project_name);
-            assert_output!(&snapshot_name, Some(Target::JavaScript), Some(Runtime::Bun), $project_name);
-            assert_output!(&snapshot_name, Some(Target::JavaScript), Some(Runtime::Deno), $project_name);
+            // assert_output!(&snapshot_name, Some(Target::JavaScript), Some(Runtime::Bun), $project_name);
+            // assert_output!(&snapshot_name, Some(Target::JavaScript), Some(Runtime::Deno), $project_name);
             assert_output!(&snapshot_name, Some(Target::JavaScript), Some(Runtime::NodeJs), $project_name);
         }
     };
@@ -108,8 +108,8 @@ macro_rules! assert_output {
         let snapshot_name = snapshot_name(Some($target), None, $project_name);
         match $target {
             Target::JavaScript => insta::allow_duplicates! {
-                assert_output!(&snapshot_name, Some($target), Some(Runtime::Bun), $project_name);
-                assert_output!(&snapshot_name, Some($target), Some(Runtime::Deno), $project_name);
+                // assert_output!(&snapshot_name, Some($target), Some(Runtime::Bun), $project_name);
+                // assert_output!(&snapshot_name, Some($target), Some(Runtime::Deno), $project_name);
                 assert_output!(&snapshot_name, Some($target), Some(Runtime::NodeJs), $project_name);
             },
             Target::Erlang => {
