diff -Nurp a/tests/conftest.py b/tests/conftest.py
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -8,7 +8,6 @@ import pytest
 from typing import TYPE_CHECKING
 
 import pytest
-import virtualenv
 
 from poetry.core.factory import Factory
 from poetry.core.utils._compat import WINDOWS
@@ -91,6 +90,7 @@ def temporary_directory() -> Iterator[Path]:
 
 
 @pytest.fixture
+@pytest.mark.skip(reason="wants unpackaged python-virtualenv")
 def venv(temporary_directory: Path) -> Path:
     venv_dir = temporary_directory / ".venv"
     virtualenv.cli_run(
