diff -Nurp a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-06-09 10:11:16.000000000 +0000
+++ b/CMakeLists.txt	2025-06-14 12:40:50.464198971 +0000
@@ -180,7 +180,7 @@ else ()
 
     if (NOT WITH_LIBLUA)
         ProcessPackage(LIBLUA LIBRARY "lua"
-                "lua-5.4"
+                "lua-5.4" "lua5.4"
                 LIB_SUFFIXES "lua5.4"
                 INCLUDE lua.h INCLUDE_SUFFIXES
                 "include/lua-5.4"
