--- a/contrib/backward-cpp/backward.hpp
+++ b/contrib/backward-cpp/backward.hpp
@@ -4229,7 +4229,11 @@ public:
         reinterpret_cast<struct sigcontext *>(&uctx->uc_mcontext)->sc_pc);
 #elif defined(__ppc__) || defined(__powerpc) || defined(__powerpc__) ||        \
     defined(__POWERPC__)
+#if !defined(__powerpc64__)
+    error_addr = reinterpret_cast<void *>(((struct pt_regs *)uctx->uc_regs)->nip);
+#else
     error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.regs->nip);
+#endif
 #elif defined(__riscv)
     error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.__gregs[REG_PC]);
 #elif defined(__s390x__)
