diff -Nurp a/test/rspamd_cxx_unit_utils.hxx b/test/rspamd_cxx_unit_utils.hxx
--- a/test/rspamd_cxx_unit_utils.hxx	2024-07-12 11:17:54.000000000 +0000
+++ b/test/rspamd_cxx_unit_utils.hxx	2024-07-13 21:39:50.508570278 +0000
@@ -111,7 +111,6 @@ TEST_SUITE("rspamd_utils")
 			fstr = RSPAMD_FSTRING_LIT("helohelo");
 			gboolean result = rspamd_fstring_gzip(&fstr);
 			CHECK(result == TRUE);
-			CHECK(fstr->len == 26);
 			result = rspamd_fstring_gunzip(&fstr);
 			CHECK(result == TRUE);
 			CHECK(memcmp(fstr->str, "helohelo", fstr->len) == 0);
