https://bugs.webkit.org/show_bug.cgi?id=299018

--- a/Source/JavaScriptCore/llint/LowLevelInterpreter.asm
+++ b/Source/JavaScriptCore/llint/LowLevelInterpreter.asm
@@ -2980,6 +2980,13 @@ _wasm_ipint_call_return_location_wide16:
 _wasm_ipint_call_return_location_wide32:
     crash()
 
+if C_LOOP and not JSVALUE64
+_op_instanceof_return_location:
+_op_instanceof_return_location_wide16:
+_op_instanceof_return_location_wide32:
+    crash()
+end
+
 end # WEBASSEMBLY
 
 include? LowLevelInterpreterAdditions
