commit 61a6aa1ebef368cd9b5a9d2432792564fcf3df75
Author: q66 <q66@chimera-linux.org>
Date:   Sun Nov 13 20:59:08 2022 +0100

    undef fortify when using cloop

diff --git a/Source/JavaScriptCore/CMakeLists.txt b/Source/JavaScriptCore/CMakeLists.txt
index d65c8ab..74bd94d 100644
--- a/Source/JavaScriptCore/CMakeLists.txt
+++ b/Source/JavaScriptCore/CMakeLists.txt
@@ -1491,6 +1491,11 @@ if (CMAKE_COMPILER_IS_GNUCXX AND GCC_OFFLINEASM_SOURCE_MAP)
         COMPILE_OPTIONS "-fno-lto")
 endif ()
 
+# fortify gets clang 15 stuck
+if (COMPILER_IS_CLANG AND ENABLE_C_LOOP)
+    WEBKIT_ADD_TARGET_CXX_FLAGS(LowLevelInterpreterLib -U_FORTIFY_SOURCE)
+endif ()
+
 # When building JavaScriptCore as an object library, we need to make sure the
 # lowlevelinterpreter lib objects get propogated.
 if (${JavaScriptCore_LIBRARY_TYPE} STREQUAL "OBJECT")
