--- a/Source/ThirdParty/skia/CMakeLists.txt
+++ b/Source/ThirdParty/skia/CMakeLists.txt
@@ -176,6 +176,7 @@ add_library(Skia STATIC
     src/core/SkBitmapProcState.cpp
     src/core/SkBitmapProcState_matrixProcs.cpp
     src/core/SkBitmapProcState_opts.cpp
+    src/core/SkBitmapProcState_opts_lasx.cpp
     src/core/SkBitmapProcState_opts_ssse3.cpp
     src/core/SkBlendMode.cpp
     src/core/SkBlendModeBlender.cpp
@@ -184,6 +185,7 @@ add_library(Skia STATIC
     src/core/SkBlitRow_D32.cpp
     src/core/SkBlitRow_opts.cpp
     src/core/SkBlitRow_opts_hsw.cpp
+    src/core/SkBlitRow_opts_lasx.cpp
     src/core/SkBlitter.cpp
     src/core/SkBlitter_A8.cpp
     src/core/SkBlitter_ARGB32.cpp
@@ -338,6 +340,7 @@ add_library(Skia STATIC
     src/core/SkSwizzle.cpp
     src/core/SkSwizzler_opts.cpp
     src/core/SkSwizzler_opts_hsw.cpp
+    src/core/SkSwizzler_opts_lasx.cpp
     src/core/SkSwizzler_opts_ssse3.cpp
     src/core/SkTaskGroup.cpp
     src/core/SkTextBlob.cpp
@@ -873,6 +876,7 @@ add_library(Skia STATIC
     src/utils/SkShadowUtils.cpp
 
     src/opts/SkOpts_hsw.cpp
+    src/opts/SkOpts_lasx.cpp
     src/opts/SkOpts_skx.cpp
 
     src/ports/SkGlobalInitialization_default.cpp
