commit 4630b5e76a527de0b10493ac9c3f35ac714138e9
Author: q66 <q66@chimera-linux.org>
Date:   Thu Sep 29 05:21:17 2022 +0200

    drop some cross-incompatible nonsense

diff --git a/configure b/configure
index d144000..1a458f0 100755
--- a/configure
+++ b/configure
@@ -12188,73 +12188,6 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_use_ipc" >&5
 $as_echo "$ac_cv_use_ipc" >&6; }
 
-if test $ac_cv_use_ipc = "sysv"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether SysV IPC message queues are actually working on the host" >&5
-$as_echo_n "checking whether SysV IPC message queues are actually working on the host... " >&6; }
-
-  ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
-ac_compiler_gnu=$ac_cv_c_compiler_gnu
-
-  if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <stdlib.h>
-#include <sys/types.h>
-#include <sys/ipc.h>
-#include <sys/msg.h>
-#include <time.h>
-#include <unistd.h>
-
-int main() {
-
-  srandom(time(NULL)+getpid()*33151);
-  key_t msg_key = random();
-  int msg_get = msgget(msg_key, IPC_CREAT|0600);
-
-  if (msg_get==-1) {
-    return 1;
-  } else {
-    msgctl(msg_get, IPC_RMID, NULL);
-    return 0;
-  }
-
-}
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  ac_cv_use_ipc=sysv
-else
-  ac_cv_use_ipc=tcp
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
-
-  if test $ac_cv_use_ipc = "tcp"; then
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: No, using TCP" >&5
-$as_echo "No, using TCP" >&6; }
-  else
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: Yes" >&5
-$as_echo "Yes" >&6; }
-  fi
-
-  ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
-ac_compiler_gnu=$ac_cv_c_compiler_gnu
-
-fi
-
 
 # Check whether --with-dbformat was given.
 if test "${with_dbformat+set}" = set; then :
