commit ef3e6b44
Author: Renato Botelho do Couto <renato@netgate.com>
Date:   Mon Jul 8 16:15:05 2024

    pear: Fix PHP shebang line

diff --git a/pear/Makefile.frag b/pear/Makefile.frag
index 9408757a..96d79f61 100644
--- a/pear/Makefile.frag
+++ b/pear/Makefile.frag
@@ -10,7 +10,7 @@ PEAR_SUFFIX = -ds a$(program_suffix)
 PEAR_INSTALLER_URL = https://pear.php.net/install-pear-nozlib.phar
 
 install-pear-installer: $(SAPI_CLI_PATH)
-	@$(top_builddir)/sapi/cli/php $(PEAR_INSTALL_FLAGS) pear/install-pear-nozlib.phar -d "$(peardir)" -b "$(bindir)" ${PEAR_PREFIX} ${PEAR_SUFFIX}
+	@$(top_builddir)/sapi/cli/php $(PEAR_INSTALL_FLAGS) pear/install-pear-nozlib.phar -p "$(bindir)/$(program_prefix)php$(program_suffix)" -d "$(peardir)" -b "$(bindir)" ${PEAR_PREFIX} ${PEAR_SUFFIX}
 
 install-pear:
 	@echo "Installing PEAR environment:      $(INSTALL_ROOT)$(peardir)/"
