--- a/src/meson.build
+++ b/src/meson.build
@@ -25,11 +25,10 @@
 
 # Setup cargo
 cargo = find_program('cargo')
-cargo_env = [ 'CARGO_HOME=' + meson.project_build_root() / 'cargo-home' ]
 cargo_options = [ '--manifest-path', meson.project_source_root() / 'Cargo.toml' ]
 cargo_options += [ '--target-dir', meson.project_build_root() / 'target' ]
 
-if get_option('profile') == 'default'
+if true
   cargo_options += [ '--release' ]
   rust_target = 'release'
   message('Building in release mode')
@@ -44,14 +43,9 @@
   build_always_stale: true,
   output: meson.project_name(),
   console: true,
-  install: true,
   install_dir: bindir,
   command: [
-    'env',
-    cargo_env,
     cargo, 'build',
     cargo_options,
-    '&&',
-    'cp', 'target' / rust_target / meson.project_name(), '@OUTPUT@',
   ]
-)
\ No newline at end of file
+)
