commit 3b19e984f953d4406ae16bd39d85761f85a43621
Author: q66 <q66@chimera-linux.org>
Date:   Mon Jan 23 00:34:58 2023 +0000

    allow commas in cflags

diff --git a/configure b/configure
index f977663..b0c3d2b 100755
--- a/configure
+++ b/configure
@@ -123,7 +123,7 @@ mkmkf() {
 	echo "  $2 <= $1"
 	sed -e "s,@DESTDIR@,$DESTDIR,g" -e "s,@SYSCONF@,$SYSCONF,g" \
 	    -e "s,@CROSS_COMPILE@,$CROSS_COMPILE,g" -e "s,@CC@,$CC,g" \
-	    -e "s,@CFLAGS@,$CFLAGS,g" $1 >$2
+	    -e "s|@CFLAGS@|$CFLAGS|g" $1 >$2
     fi
 }
 
