diff --git a/src/engine.cpp b/src/engine.cpp
index c9b9dcc..5f60522 100644
--- a/src/engine.cpp
+++ b/src/engine.cpp
@@ -131,7 +131,7 @@ public:
 
         auto hic_preedit = hangul_ic_get_preedit_string(context_.get());
         UString preedit = preedit_;
-        preedit.append(UString(hic_preedit));
+        preedit.append(UString((const char32_t *)hic_preedit));
         if (!preedit.empty()) {
             auto utf8 = ustringToUTF8(preedit);
             if (*engine_->config().wordCommit || *engine_->config().hanjaMode) {
@@ -357,9 +357,9 @@ public:
 
                 hic_preedit = hangul_ic_get_preedit_string(context_.get());
                 if (hic_preedit != nullptr && hic_preedit[0] != 0) {
-                    preedit_.append(UString(str));
+                    preedit_.append(UString((const char32_t *)str));
                 } else {
-                    preedit_.append(UString(str));
+                    preedit_.append(UString((const char32_t *)str));
                     if (!preedit_.empty()) {
                         auto commit = ustringToUTF8(preedit_);
                         if (!commit.empty()) {
@@ -370,7 +370,7 @@ public:
                 }
             } else {
                 if (str != nullptr && str[0] != 0) {
-                    auto commit = ustringToUTF8(str);
+                    auto commit = ustringToUTF8((const char32_t *)str);
                     if (!commit.empty()) {
                         ic_->commitString(commit);
                     }
@@ -408,7 +408,7 @@ public:
 
         auto str = hangul_ic_flush(context_.get());
 
-        preedit_ += str;
+        preedit_ += (const char32_t *)str;
 
         if (preedit_.empty())
             return;
@@ -430,7 +430,7 @@ public:
         std::string pre1 = ustringToUTF8(preedit_);
         std::string pre2;
         if (hic_preedit) {
-            pre2 = ustringToUTF8(hic_preedit);
+            pre2 = ustringToUTF8((const char32_t *)hic_preedit);
         }
 
         if (!pre1.empty() || !pre2.empty()) {
@@ -493,7 +493,7 @@ public:
 
         key_len = fcitx::utf8::length(std::string(key));
         preedit_len = preedit_.size();
-        hic_preedit_len = UString(hic_preedit).size();
+        hic_preedit_len = UString((const char32_t *)hic_preedit).size();
 
         bool surrounding = false;
         if (lastLookupMethod_ == LOOKUP_METHOD_PREFIX) {
diff --git a/src/engine.h b/src/engine.h
index b5bb60f..09e0325 100644
--- a/src/engine.h
+++ b/src/engine.h
@@ -85,7 +85,7 @@ typedef enum _LookupMethod {
 
 class HangulState;
 
-using UString = std::basic_string<uint32_t>;
+using UString = std::basic_string<char32_t>;
 
 class HangulEngine : public InputMethodEngine {
 public:
