From d3b2138d8204ef21b1a8d459afaeca4e07e2856b Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sat, 6 Aug 2022 04:12:18 +0200
Subject: [PATCH] use before= in sample service

---
 dinit-userservd | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dinit-userservd b/dinit-userservd
index b917171..ce3f338 100644
--- a/dinit-userservd
+++ b/dinit-userservd
@@ -1,6 +1,7 @@
 # dinit-userservd service
 
 type       = process
-depends-on = init
+before     = login.target
+depends-on = init.target
 command    = /usr/bin/dinit-userservd
 logfile    = /var/log/dinit-userservd.log
