--- generate-rules.sh	2024-09-22 02:10:25.521129020 -0300
+++ generate-rules.sh	2024-09-22 02:11:43.889844039 -0300
@@ -8,7 +8,7 @@
   echo "  -h, --help                              Print this help message"
 }
 
-for c in git jq tr awk sed; do
+for c in jq tr awk sed; do
   command -v $c > /dev/null
   if [[ $? > 0 ]]; then
     echo "Error: Command $c not found in \$PATH." >&2
@@ -23,7 +23,7 @@
 shopt -s globstar
 set -eu
 
-OTD_CONFIGURATIONS="${OTD_CONFIGURATIONS:="$(git rev-parse --show-toplevel)/OpenTabletDriver.Configurations/Configurations"}"
+OTD_CONFIGURATIONS="./OpenTabletDriver.Configurations/Configurations"
 
 while [ $# -gt 0 ]; do
   case "$1" in
