--- generate-rules.sh	2024-09-22 03:37:18.488334658 -0300
+++ generate-rules.sh	2024-09-22 03:37:26.436277278 -0300
@@ -55,7 +55,6 @@
 }
 
 echo \# OpenTabletDriver udev rules \(https://github.com/OpenTabletDriver/OpenTabletDriver\)
-echo KERNEL==\"uinput\", SUBSYSTEM==\"misc\", OPTIONS+=\"static_node=uinput\", TAG+=\"uaccess\", TAG+=\"udev-acl\"
 echo KERNEL==\"js[0-9]*\", SUBSYSTEM==\"input\", ATTRS{name}==\"OpenTabletDriver Virtual Tablet\", RUN+=\"/usr/bin/env rm %E{DEVNAME}\"
 
 IFS=':'

