--- a/Makefile	2024-08-07 17:09:40.000000000 -0300
+++ b/Makefile	2024-09-19 14:26:22.790244522 -0300
@@ -146,15 +146,12 @@
 	# binaries
 	$(call install_python_script,tuned.py,$(DESTDIR)/usr/sbin/tuned)
 	$(call install_python_script,tuned-adm.py,$(DESTDIR)/usr/sbin/tuned-adm)
-	$(call install_python_script,tuned-gui.py,$(DESTDIR)/usr/sbin/tuned-gui)
 
 	$(foreach file, diskdevstat netdevstat scomes, \
 		install -Dpm 0755 systemtap/$(file) $(DESTDIR)/usr/sbin/$(notdir $(file));)
 	$(call install_python_script, \
 		systemtap/varnetload, $(DESTDIR)/usr/sbin/varnetload)
 
-	# glade
-	install -Dpm 0644 tuned-gui.glade $(DESTDIR)$(DATADIR)/tuned/ui/tuned-gui.glade
 
 	# tools
 	$(call install_python_script, \
@@ -221,12 +218,7 @@
 		$(call install_python_script, \
 			$(file), $(DESTDIR)/usr/libexec/tuned/$(notdir $(file))))
 
-	# icon
-	install -Dpm 0644 icons/tuned.svg $(DESTDIR)$(DATADIR)/icons/hicolor/scalable/apps/tuned.svg
 
-	# desktop file
-	install -dD $(DESTDIR)$(DATADIR)/applications
-	desktop-file-install --dir=$(DESTDIR)$(DATADIR)/applications tuned-gui.desktop
 
 install-ppd: install
 	$(call install_python_script,tuned-ppd.py,$(DESTDIR)/usr/sbin/tuned-ppd)
