--- a/gnulib-tests/test-malloc-gnu.c
+++ b/gnulib-tests/test-malloc-gnu.c
@@ -27,19 +27,5 @@
 int
 main (int argc, char **argv)
 {
-  /* Check that malloc (0) is not a NULL pointer.  */
-  void *volatile p = malloc (0);
-  ASSERT (p != NULL);
-  free (p);
-
-  /* Check that malloc (n) fails when n exceeds PTRDIFF_MAX.  */
-  if (PTRDIFF_MAX < SIZE_MAX)
-    {
-      size_t one = argc != 12345;
-      p = malloc (PTRDIFF_MAX + one);
-      ASSERT (p == NULL);
-      ASSERT (errno == ENOMEM);
-    }
-
   return 0;
 }
