--- a/lxqt-session/src/procreaper.cpp
+++ b/lxqt-session/src/procreaper.cpp
@@ -121,10 +121,10 @@ void ProcReaper::stop(const std::set<int64_t> & excludedPids)
     pids_stack * stack = nullptr;
     while ((stack = procps_pids_get(info, PIDS_FETCH_TASKS_ONLY)))
     {
-        const int ppid = PIDS_VAL(rel_ppid, s_int, stack, info);
+        const int ppid = PIDS_VAL(rel_ppid, s_int, stack);
         if (ppid == my_pid)
         {
-            const int tgid = PIDS_VAL(rel_tgid, s_int, stack, info);
+            const int tgid = PIDS_VAL(rel_tgid, s_int, stack);
             children.push_back(tgid);
         }
     }
