--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -87,7 +87,7 @@
 endmacro()
 
 if(AVIF_ENABLE_GTEST)
-    check_avif_option(AVIF_GTEST TARGET GTest::gtest PKG_NAME Googletest)
+    check_avif_option(AVIF_GTEST TARGET GTest::gtest PKG_NAME GTest)
     add_library(avifincrtest_helpers OBJECT gtest/avifincrtest_helpers.cc)
     target_link_libraries(avifincrtest_helpers avif_internal GTest::gtest)
 endif()
