From 65593c162379ad1b9e9c131fd0b21fa185c7e875 Mon Sep 17 00:00:00 2001
From: Miko Larsson <mikoxyzzz@gmail.com>
Date: Sun, 10 Mar 2024 12:43:14 +0100
Subject: [PATCH 09/10] junction: define ALLPERMS if not already defined

Signed-off-by: Miko Larsson <mikoxyzzz@gmail.com>
---
 support/junction/junction.c | 5 +++++
 support/junction/nfs.c      | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/support/junction/junction.c b/support/junction/junction.c
index c1ec8ff8..b1684123 100644
--- a/support/junction/junction.c
+++ b/support/junction/junction.c
@@ -48,6 +48,11 @@
 #include "junction-internal.h"
 #include "xlog.h"
 
+/* ALLPERMS (mode 07777) is not specified in POSIX */
+#ifndef ALLPERMS
+#define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO)
+#endif
+
 /**
  * Open a file system object
  *
diff --git a/support/junction/nfs.c b/support/junction/nfs.c
index 73e3533b..1e48a53c 100644
--- a/support/junction/nfs.c
+++ b/support/junction/nfs.c
@@ -83,6 +83,11 @@
 #include "junction-internal.h"
 #include "xlog.h"
 
+/* ALLPERMS (mode 07777) is not specified in POSIX */
+#ifndef ALLPERMS
+#define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO)
+#endif
+
 /**
  * Tag name of NFS location element of a junction XML document
  */
-- 
2.45.2

