From 73374e53107d61d57f4e9717500210a5166818ff Mon Sep 17 00:00:00 2001
From: Miko Larsson <mikoxyzzz@gmail.com>
Date: Sun, 10 Mar 2024 12:21:15 +0100
Subject: [PATCH 07/10] junction: include linux/limits.h

Signed-off-by: Miko Larsson <mikoxyzzz@gmail.com>
---
 support/junction/path.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/support/junction/path.c b/support/junction/path.c
index c25cc52d..db59a8cf 100644
--- a/support/junction/path.c
+++ b/support/junction/path.c
@@ -26,6 +26,8 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
+#include <linux/limits.h>
+
 #include <stdbool.h>
 #include <unistd.h>
 #include <stdint.h>
-- 
2.45.2

