From 50ae69f20e7709eee4ed50be212a4a00a3eac946 Mon Sep 17 00:00:00 2001
From: Miko Larsson <mikoxyzzz@gmail.com>
Date: Sun, 10 Mar 2024 12:36:34 +0100
Subject: [PATCH 08/10] junction: define _GNU_SOURCE if not already defined

Signed-off-by: Miko Larsson <mikoxyzzz@gmail.com>
---
 support/junction/path.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/support/junction/path.c b/support/junction/path.c
index db59a8cf..3af5ad04 100644
--- a/support/junction/path.c
+++ b/support/junction/path.c
@@ -23,6 +23,10 @@
  *	http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
-- 
2.45.2

