commit 3bf24ff8720907d127170d4f603a296703ea22af
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Mon Nov 14 21:48:15 2022 +0100

    use libedit in place of readline

diff --git a/meson.build b/meson.build
index e55540f..6c2b35a 100644
--- a/meson.build
+++ b/meson.build
@@ -260,11 +260,7 @@ cdata.set('HAVE_DBUS', dbus_dep.found())
 sdl_dep = dependency('sdl2', required : get_option('sdl2'))
 summary({'SDL2 (video examples)': sdl_dep.found()}, bool_yn: true, section: 'Misc dependencies')
 drm_dep = dependency('libdrm', required : false)
-readline_dep = dependency('readline', required : get_option('readline'))
-
-if not readline_dep.found()
-  readline_dep = cc.find_library('readline', required : get_option('readline'))
-endif
+readline_dep = cc.find_library('edit')
 
 summary({'readline (for pw-cli)': readline_dep.found()}, bool_yn: true, section: 'Misc dependencies')
 cdata.set('HAVE_READLINE', readline_dep.found())
diff --git a/src/tools/pw-cli.c b/src/tools/pw-cli.c
index bb5ab99..c5f4dd8 100644
--- a/src/tools/pw-cli.c
+++ b/src/tools/pw-cli.c
@@ -36,8 +36,7 @@
 #include <getopt.h>
 #include <fnmatch.h>
 #ifdef HAVE_READLINE
-#include <readline/readline.h>
-#include <readline/history.h>
+#include <editline/readline.h>
 #endif
 #include <locale.h>
 
