commit 173e08fd2eccf3bb4be940a8123d60a161a924b3
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Apr 9 00:32:53 2022 +0200

    use libedit in place of readline

diff --git a/meson.build b/meson.build
index fb15702..06fcf08 100644
--- a/meson.build
+++ b/meson.build
@@ -259,11 +259,7 @@ cdata.set('HAVE_DBUS', dbus_dep.found())
 sdl_dep = dependency('sdl2', required : get_option('sdl2'))
 summary({'SDL2 (video examples)': sdl_dep.found()}, bool_yn: true, section: 'Misc dependencies')
 drm_dep = dependency('libdrm', required : false)
-readline_dep = dependency('readline', required : false)
-
-if not readline_dep.found()
-  readline_dep = cc.find_library('readline', required: false)
-endif
+readline_dep = cc.find_library('edit')
 
 summary({'readline (for pw-cli)': readline_dep.found()}, bool_yn: true, section: 'Misc dependencies')
 ncurses_dep = dependency('ncursesw', required : false)
diff --git a/src/tools/pw-cli.c b/src/tools/pw-cli.c
index d19e66b..cf3d2e3 100644
--- a/src/tools/pw-cli.c
+++ b/src/tools/pw-cli.c
@@ -33,8 +33,7 @@
 #endif
 #include <getopt.h>
 #include <fnmatch.h>
-#include <readline/readline.h>
-#include <readline/history.h>
+#include <editline/readline.h>
 #include <locale.h>
 
 #if !defined(FNM_EXTMATCH)
