--- a/src/renderer_GL_common.inl
+++ b/src/renderer_GL_common.inl
@@ -3870,7 +3870,7 @@
     return 0;  // FIXME: Handle errors better
 }
 
-static GPU_Image* CopyImageFromSurface(GPU_Renderer* renderer, SDL_Surface* surface, const GPU_Rect* surface_rect)
+static GPU_Image* CopyImageFromSurface(GPU_Renderer* renderer, SDL_Surface* surface, GPU_Rect* surface_rect)
 {
     GPU_FormatEnum format;
     GPU_Image* image;
