diff --git a/demos/3d/main.c b/demos/3d/main.c
index ba43e7d..141b19d 100644
--- a/demos/3d/main.c
+++ b/demos/3d/main.c
@@ -1,7 +1,7 @@
 #include "SDL.h"
 #include "SDL_gpu.h"
 #include "SDL_gpu_OpenGL_1.h"
-#include "glew.h"
+#include <GL/glew.h>
 #include <stdlib.h>
 
 
diff --git a/include/SDL_gpu_OpenGL_1.h b/include/SDL_gpu_OpenGL_1.h
index c14ab58..206e2a3 100644
--- a/include/SDL_gpu_OpenGL_1.h
+++ b/include/SDL_gpu_OpenGL_1.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
 
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
diff --git a/include/SDL_gpu_OpenGL_1_BASE.h b/include/SDL_gpu_OpenGL_1_BASE.h
index 5c33035..e8180f1 100644
--- a/include/SDL_gpu_OpenGL_1_BASE.h
+++ b/include/SDL_gpu_OpenGL_1_BASE.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
 
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
diff --git a/include/SDL_gpu_OpenGL_2.h b/include/SDL_gpu_OpenGL_2.h
index 32a0b72..4b90636 100644
--- a/include/SDL_gpu_OpenGL_2.h
+++ b/include/SDL_gpu_OpenGL_2.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
     
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
diff --git a/include/SDL_gpu_OpenGL_3.h b/include/SDL_gpu_OpenGL_3.h
index 83f5b59..56adcbd 100644
--- a/include/SDL_gpu_OpenGL_3.h
+++ b/include/SDL_gpu_OpenGL_3.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
     
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
diff --git a/include/SDL_gpu_OpenGL_4.h b/include/SDL_gpu_OpenGL_4.h
index 1840bfa..0f3f47d 100644
--- a/include/SDL_gpu_OpenGL_4.h
+++ b/include/SDL_gpu_OpenGL_4.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
     
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
