commit 06777c64ae12ffecd58d4c13ca1e864d4518d719
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Jun 17 02:34:05 2023 +0200

    fix clang 16 build

diff --git a/src/gd_bmp.c b/src/gd_bmp.c
index e186ac9..60155fa 100644
--- a/src/gd_bmp.c
+++ b/src/gd_bmp.c
@@ -31,6 +31,7 @@
 #include "gd.h"
 #include "gdhelpers.h"
 #include "bmp.h"
+#include "gd_errors.h"
 
 static int compress_row(unsigned char *uncompressed_row, int length);
 static int build_rle_packet(unsigned char *row, int packet_type, int length, unsigned char *data);
