commit 2f976834e0dd982caef702753a2d1c0155ed7329
Author: q66 <q66@chimera-linux.org>
Date:   Mon Dec 6 23:38:18 2021 +0100

    do not include cpuid.h on non-x86

diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index c383a61..0d4f2e0 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -23,6 +23,10 @@
 #include <config.h>
 #endif
 
+#if !defined(__x86_64__) && !defined(__i386__)
+#undef HAVE_CPUID_H
+#endif
+
 #include <math.h>
 #include <stdarg.h>
 #include <stdlib.h>
