commit b6cf5458e916e41fc72247568e798773d135ff33
Author: q66 <q66@chimera-linux.org>
Date:   Mon Jan 20 13:18:53 2025 +0100

    disable some broken but unused targets

diff --git a/arch/powerpc/boot/Makefile b/arch/powerpc/boot/Makefile
index 1ff6ad4..ef15a78 100644
--- a/arch/powerpc/boot/Makefile
+++ b/arch/powerpc/boot/Makefile
@@ -35,9 +35,7 @@ endif
 ifdef CONFIG_PPC64_BOOT_WRAPPER
 BOOTTARGETFLAGS	+= -m64
 BOOTTARGETFLAGS	+= -mabi=elfv2
-ifdef CONFIG_PPC64_ELF_ABI_V2
-BOOTTARGETFLAGS	+= $(call cc-option,-mabi=elfv2)
-endif
+BOOTCFLAGS	+= $(call cc-option,-mno-prefixed) $(call cc-option,-mno-pcrel)
 else
 BOOTTARGETFLAGS	:= -m32
 endif
@@ -73,8 +71,6 @@ BOOTCFLAGS	:= $(BOOTTARGETFLAGS) \
 		   -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
 		   -fno-strict-aliasing -O2 \
 		   -msoft-float -mno-altivec -mno-vsx \
-		   $(call cc-option,-mno-prefixed) \
-		   $(call cc-option,-mno-pcrel) \
 		   $(call cc-option,-mno-mma) \
 		   $(call cc-option,-mno-spe) $(call cc-option,-mspe=no) \
 		   -fomit-frame-pointer -fno-builtin -fPIC
@@ -274,17 +270,6 @@ quiet_cmd_wrap	= WRAP    $@
 		$(CROSSWRAP) $(if $3, -s $3)$(if $4, -d $4)$(if $5, -i $5) \
 		vmlinux
 
-image-$(CONFIG_PPC_PSERIES)		+= zImage.pseries
-image-$(CONFIG_PPC_POWERNV)		+= zImage.pseries
-image-$(CONFIG_PPC_IBM_CELL_BLADE)	+= zImage.pseries
-image-$(CONFIG_PPC_PS3)			+= dtbImage.ps3
-image-$(CONFIG_PPC_CHRP)		+= zImage.chrp
-image-$(CONFIG_PPC_EFIKA)		+= zImage.chrp
-image-$(CONFIG_PPC_PMAC)		+= zImage.pmac
-image-$(CONFIG_PPC_HOLLY)		+= dtbImage.holly
-image-$(CONFIG_DEFAULT_UIMAGE)		+= uImage
-image-$(CONFIG_EPAPR_BOOT)		+= zImage.epapr
-
 #
 # Targets which embed a device tree blob
 #
@@ -354,12 +339,6 @@ image-$(CONFIG_AMIGAONE)		+= cuImage.amigaone
 
 image-$(CONFIG_PPC_MICROWATT)		+= dtbImage.microwatt
 
-# For 32-bit powermacs, build the COFF and miboot images
-# as well as the ELF images.
-ifdef CONFIG_PPC32
-image-$(CONFIG_PPC_PMAC)	+= zImage.coff zImage.miboot
-endif
-
 # Allow extra targets to be added to the defconfig
 image-y	+= $(CONFIG_EXTRA_TARGETS)
 
@@ -441,9 +420,7 @@ $(obj)/zImage.initrd:	$(addprefix $(obj)/, $(initrd-y))
 
 # anything not in $(targets)
 clean-files += $(image-) $(initrd-) cuImage.* dtbImage.* treeImage.* \
-	zImage zImage.initrd zImage.chrp zImage.coff zImage.holly \
-	zImage.miboot zImage.pmac zImage.pseries \
-	simpleImage.* otheros.bld
+	zImage
 
 # clean up files cached by wrapper
 clean-kernel-base := vmlinux.strip vmlinux.bin
