--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,13 +5,6 @@ requires = [
     # self-typechecking :/
     "setuptools >= 40.6.2",
     "wheel >= 0.30.0",
-    # the following is from mypy-requirements.txt/setup.py
-    "typing_extensions>=4.6.0",
-    "mypy_extensions>=1.0.0",
-    "tomli>=1.1.0; python_version<'3.11'",
-    # the following is from build-requirements.txt
-    "types-psutil",
-    "types-setuptools",
 ]
 build-backend = "setuptools.build_meta"
 
