--- a/prnt/hpps/pserror.c
+++ b/prnt/hpps/pserror.c
@@ -24,7 +24,7 @@
 void message(int flags, char *format, ...)
 {
   va_list args ;
-  static column = 0 ;		/* current screen column for message wrap */
+  static int column = 0 ;		/* current screen column for message wrap */
   char msgbuf[MAX_MESSAGE] ;	/* buffer in which to put the message */
   char *bufptr = msgbuf ;	/* message buffer pointer */
 
--- a/prnt/hpps/hppsfilter.c
+++ b/prnt/hpps/hppsfilter.c
@@ -105,7 +105,6 @@
     if(ptempbooklet_file == NULL)
     {
             fprintf(stderr, "ERROR: Unable to open temp file %s\n", temp_filename);
-            return 1;
     }  
     chmod(temp_filename, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
 
