no register in c++17 or later
--- a/prnt/hpcups/ModeDeltaPlus.cpp
+++ b/prnt/hpcups/ModeDeltaPlus.cpp
@@ -241,8 +241,8 @@
                               const     uint32_t    inheight,
                               uint32_t  horz_ht_dist)
 {
-    register    BYTE     *outptr = outmem;
-    register    uint32_t    col;
+    BYTE     *outptr = outmem;
+    uint32_t    col;
     const       BYTE     *seedrow;
     uint32_t                seedrow_count = 0;
     uint32_t                location = 0;
--- a/prnt/hpcups/Mode9.cpp
+++ b/prnt/hpcups/Mode9.cpp
@@ -203,7 +203,7 @@
     unsigned int    offset,byte_count,rem_count;
     Mode9_comtype       command;
     char* dest=    (char*) compressBuf;
-    register char *dptr=dest;
+    char *dptr=dest;
 
     while ( size > 0 )
     {
--- a/prnt/hpijs/compression.cpp
+++ b/prnt/hpijs/compression.cpp
@@ -266,7 +266,7 @@
     unsigned int    offset,byte_count,rem_count;
     Mode9_comtype       command;
     char* dest=    (char*) compressBuf;
-    register char *dptr=dest;
+    char *dptr=dest;
 
     while ( size > 0 )
     {
--- a/prnt/hpijs/ljfastraster.cpp
+++ b/prnt/hpijs/ljfastraster.cpp
@@ -919,8 +919,8 @@
                               const     uint32_t    inheight,
                               uint32_t  horz_ht_dist)
 {
-	register    HPUInt8     *outptr = outmem;
-	register    uint32_t    col;
+	HPUInt8     *outptr = outmem;
+	uint32_t    col;
     const       HPUInt8     *seedrow;
     uint32_t                seedrow_count = 0;
     uint32_t                location = 0;
