--- a/c/CMakeLists.txt
+++ b/c/CMakeLists.txt
@@ -240,7 +240,11 @@ if(BLAKE3_USE_TBB)
       PUBLIC
         BLAKE3_USE_TBB)
   endif()
-  list(APPEND PKG_CONFIG_REQUIRES "tbb >= ${TBB_VERSION}")
+  if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+    list(APPEND PKG_CONFIG_REQUIRES "tbb >= ${TBB_VERSION}")
+  else()
+    list(APPEND PKG_CONFIG_REQUIRES "tbb32 >= ${TBB_VERSION}")
+  endif()
   list(APPEND PKG_CONFIG_CFLAGS -DBLAKE3_USE_TBB)
   include(CheckCXXSymbolExists)
   check_cxx_symbol_exists(_LIBCPP_VERSION "version" BLAKE3_HAVE_LIBCPP)
