https://github.com/chimera-linux/cports/pull/1273#discussion_r1466266497
diff -ruN a/schism/version.c b/schism/version.c
--- a/schism/version.c	2025-04-16 01:08:37.000000000 +0200
+++ b/schism/version.c	2025-05-12 09:48:01.156515459 +0200
@@ -48,11 +48,12 @@
 		on gcc-like compilers. (not like anyone's really using schism with anything else...)
 
 */
-#if !defined(EMPTY_VERSION) && defined(VERSION)
-# define TOP_BANNER_NORMAL "Schism Tracker " VERSION
+#if 1
+# define TOP_BANNER_NORMAL "Schism Tracker " VERSION2
 #else
 # define TOP_BANNER_NORMAL "Schism Tracker built " __DATE__ " " __TIME__
 #endif
+
 	;
 
 /* -------------------------------------------------------- */
@@ -200,12 +201,12 @@
 // Tries multiple methods to get a reasonable date to start with.
 static inline int get_version_date(int *pyear, int *pmonth, int *pday)
 {
-#if !defined(EMPTY_VERSION) && defined(VERSION)
+#if 1
 	{
 		int year, month, day;
 
 		// by the time we reach the year 10000 nobody will care that this breaks
-		if (sscanf(VERSION, "%04d%02d%02d", &year, &month, &day) == 3) {
+		if (sscanf(VERSION2, "%04d%02d%02d", &year, &month, &day) == 3) {
 			*pyear = year;
 			*pmonth = month - 1;
 			*pday = day;
