diff --git a/tools/libwacom-update-db.py b/tools/libwacom-update-db.py
index f9317de..05e175d 100755
--- a/tools/libwacom-update-db.py
+++ b/tools/libwacom-update-db.py
@@ -205,7 +205,7 @@ if __name__ == "__main__":
         "--skip-systemd-hwdb-update",
         action="store_true",
         default=False,
-        help="Do not run systemd-hwdb --update (Note: updates to tablet files will not be reflected in udev)",
+        help="Do not run udev-hwdb --update (Note: updates to tablet files will not be reflected in udev)",
     )
     parser.add_argument(
         "--udev-base-dir",
@@ -243,7 +243,7 @@ if __name__ == "__main__":
 
             if not ns.skip_systemd_hwdb_update:
                 subprocess.run(
-                    ["systemd-hwdb", "update"],
+                    ["udev-hwdb", "update"],
                     capture_output=True,
                     check=True,
                     text=True,
