https://bugzilla.mozilla.org/show_bug.cgi?id=1833064

diff --git a/mail/app-system-headers.mozbuild b/mail/app-system-headers.mozbuild
--- a/comm/mail/app-system-headers.mozbuild
+++ b/comm/mail/app-system-headers.mozbuild
@@ -8,5 +8,11 @@
 
 # system_headers += [
 #    'file1.h',
 #    'file2.h',
 # ]
+
+if CONFIG['MOZ_SYSTEM_ICU']:
+    system_headers += [
+        "unicode/strenum.h",
+        "unicode/vtzone.h",
+    ]
