From d8b94298b472daa2051e37d12a0500729ed7020f Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Fri, 30 May 2025 12:38:56 +0200
Subject: [PATCH 14/14] disable f16 for riscv64

---
 vendor/compiler_builtins-0.1.158/configure.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/vendor/compiler_builtins-0.1.158/configure.rs b/vendor/compiler_builtins-0.1.158/configure.rs
index d825f35a9..53ef62d6d 100644
--- a/vendor/compiler_builtins-0.1.158/configure.rs
+++ b/vendor/compiler_builtins-0.1.158/configure.rs
@@ -97,6 +97,7 @@ pub fn configure_f16_f128(target: &Target) {
         "csky" => false,
         "hexagon" => false,
         "powerpc" | "powerpc64" => false,
+        "riscv64" | "riscv64gc" => false,
         "sparc" | "sparc64" => false,
         "wasm32" | "wasm64" => false,
         // Most everything else works as of LLVM 19
-- 
2.49.0

