diff --git a/src/meson.build.bck b/src/meson.build
index c041cfe7..e9c183bc 100644
--- a/src/meson.build.bck
+++ b/src/meson.build
@@ -222,7 +222,7 @@ pkgc.generate(
 # builds using libappstream. Fix this issue by post-processing the file.
 sed_prog = find_program('sed')
 pc_fixup = run_command(sed_prog,
-                       '-i',
+                       '-i', '',
                        '/^Requires.private\|^Libs.private/ d',
                        join_paths(meson.project_build_root(), 'meson-private', 'appstream.pc'),
                        check: false)
