this detects ppc64le as BE
--
--- a/core/libs/rawengine/libraw/libraw/libraw_types.h
+++ b/core/libs/rawengine/libraw/libraw/libraw_types.h
@@ -1141,31 +1141,12 @@
 
 
 /* Byte order */
-#if defined(__POWERPC__)
-#define LibRawBigEndian 1
-
-#elif defined(__INTEL__)
+#if defined(__LITTLE_ENDIAN__)
 #define LibRawBigEndian 0
 
-#elif defined(_M_IX86) || defined(__i386__)
-#define LibRawBigEndian 0
-
-#elif defined(_M_X64) || defined(__amd64__) || defined(__x86_64__)
-#define LibRawBigEndian 0
-
-#elif defined(__LITTLE_ENDIAN__)
-#define LibRawBigEndian 0
-
 #elif defined(__BIG_ENDIAN__)
 #define LibRawBigEndian 1
-#elif defined(_ARM_)
-#define LibRawBigEndian 0
 
-#elif __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-#define LibRawBigEndian 0
-
-#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
-#define LibRawBigEndian 1
 #else
 #ifndef qXCodeRez
 #error Unable to figure out byte order.
