commit ea719bd1c9f1b07840a509cf245b2e10dddea413
Author: q66 <q66@chimera-linux.org>
Date:   Wed Apr 2 23:52:55 2025 +0200

    ble

diff --git a/core/libs/widgets/text/localizeselector.cpp b/core/libs/widgets/text/localizeselector.cpp
index b284b27..f2af094 100644
--- a/core/libs/widgets/text/localizeselector.cpp
+++ b/core/libs/widgets/text/localizeselector.cpp
@@ -326,7 +326,7 @@ bool s_inlineTranslateString(const QString& text, const QString& trCode, QString
     }
     else
     {
-        error = QChar(trengine->error());
+        error = QChar(int(trengine->error()));
     }
 
     return false;
