From 2cef459b694cfbdb8a7f6904ea7c270a20c047d3 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Wed, 31 Jul 2024 18:04:11 +0200
Subject: [PATCH] ignore mdadm exit status, may fail

---
 early/scripts/mdadm.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/early/scripts/mdadm.sh b/early/scripts/mdadm.sh
index 9c43cc6..be85504 100644
--- a/early/scripts/mdadm.sh
+++ b/early/scripts/mdadm.sh
@@ -22,4 +22,4 @@ if ! grep -q "^ARRAY" "$CONFIG"; then
     exit 0
 fi
 
-exec mdadm -As
+mdadm -As || :
