From 8ddfbb7b5249d74453f21e8548bb2c714ced0c59 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Fri, 13 Oct 2023 03:17:23 +0200
Subject: [PATCH] modules.sh: remove spurious line

---
 early/scripts/modules.sh | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/early/scripts/modules.sh b/early/scripts/modules.sh
index 5859f4f..d7dca52 100644
--- a/early/scripts/modules.sh
+++ b/early/scripts/modules.sh
@@ -8,9 +8,6 @@ DINIT_NO_CONTAINER=1
 # no modules support
 [ -e /proc/modules ] || exit 0
 
-# no modules file
-[ -r /etc/modules ] || exit 0
-
 {
     # Parameters passed as modules-load= or rd.modules-load= in kernel command line.
     sed -nr 's/,/\n/g;s/(.* |^)(rd\.)?modules-load=([^ ]*).*/\3/p' /proc/cmdline
