commit c6579426a15f69d1b63f43226531b7133f354b3a
Author: q66 <q66@chimera-linux.org>
Date:   Mon Mar 24 22:44:12 2025 +0100

    devclient: fix zero-initialization of handshake packet

diff --git a/early/helpers/devclient.cc b/early/helpers/devclient.cc
index 2b6a5c5..c224744 100644
--- a/early/helpers/devclient.cc
+++ b/early/helpers/devclient.cc
@@ -99,9 +99,10 @@ int main(int argc, char **argv) {
 
     /* handshake sequence */
     unsigned char wz[8 + sizeof(unsigned short)];
+    std::memset(wz, 0, sizeof(wz));
     wz[0] = 0xDD;
-    memcpy(&wz[1], type, std::strlen(type));
-    memcpy(&wz[8], &devlen, sizeof(devlen));
+    std::memcpy(&wz[1], type, std::strlen(type));
+    std::memcpy(&wz[8], &devlen, sizeof(devlen));
 
     if (connect(sock, reinterpret_cast<sockaddr const *>(&saddr), sizeof(saddr)) < 0) {
         err(1, "connect failed");
