From e905a40fc456a7b1311a129e7d186099862c128a Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sun, 23 Mar 2025 17:05:38 +0100
Subject: [PATCH] fix device monitor connection dropping

---
 early/helpers/devmon.cc | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/early/helpers/devmon.cc b/early/helpers/devmon.cc
index 5a32585..372baac 100644
--- a/early/helpers/devmon.cc
+++ b/early/helpers/devmon.cc
@@ -1176,6 +1176,14 @@ int main(void) {
             }
             if (fds[i].revents & POLLHUP) {
                 std::printf("devmon: term %d\n", fds[i].fd);
+                /* look up the connection so we can nuke it */
+                for (auto &cnc: conns) {
+                    if (cnc.fd == fds[i].fd) {
+                        nc = &cnc;
+                        break;
+                    }
+                }
+                /* now terminate */
                 goto bad_msg;
             }
             if (fds[i].revents & POLLIN) {
-- 
2.48.1

