commit f7645ab9dade62404c0de00e32c6c6faabaa39e1
Author: q66 <q66@chimera-linux.org>
Date:   Mon Mar 24 22:58:14 2025 +0100

    devmon: fix dinit events randomly getting stuck

diff --git a/early/helpers/devmon.cc b/early/helpers/devmon.cc
index 372baac..109243a 100644
--- a/early/helpers/devmon.cc
+++ b/early/helpers/devmon.cc
@@ -1151,19 +1151,22 @@ int main(void) {
             break;
         }
 #endif
-        if (fds[++ni].revents) {
-            for (;;) {
-                auto nev = dinitctl_dispatch(dctl, 0, nullptr);
-                if (nev < 0) {
-                    if (errno == EINTR) {
-                        continue;
-                    }
-                    warn("dinitctl_dispatch failed");
-                    ret = 1;
-                    goto do_compact;
-                } else if (!nev) {
-                    break;
+        /* we don't check fd revents here; we need to dispatch anyway
+         * to send out any requests that may be in the write buffer
+         * from e.g. udev monitor events
+         */
+        ++ni; /* skip over the dinit fd */
+        for (;;) {
+            auto nev = dinitctl_dispatch(dctl, 0, nullptr);
+            if (nev < 0) {
+                if (errno == EINTR) {
+                    continue;
                 }
+                warn("dinitctl_dispatch failed");
+                ret = 1;
+                goto do_compact;
+            } else if (!nev) {
+                break;
             }
         }
         /* handle connections */
