From a0c1a7e64b86c63fa6a487282f72d25267af7219 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 28 Feb 2026 22:02:30 +0100
Subject: [PATCH] don't fail for unsupported filesystems for optional mounts

---
 early/helpers/mnt.cc | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/early/helpers/mnt.cc b/early/helpers/mnt.cc
index 999072d..14cae8e 100644
--- a/early/helpers/mnt.cc
+++ b/early/helpers/mnt.cc
@@ -648,7 +648,18 @@ static int do_try_maybe(
     if (stat(tgt, &st) || !S_ISDIR(st.st_mode)) {
         return 0;
     }
-    return do_try(tgt, src, fstype, opts);
+    int ret = do_try(tgt, src, fstype, opts);
+    if (ret < 0) {
+        switch (errno) {
+            case ENODEV:
+            case ENOTSUP:
+                /* filesystem type not known or supported */
+                return 0;
+            default:
+                break;
+        }
+    }
+    return ret;
 }
 
 static int do_remount(char const *tgt, char *opts) {
-- 
2.52.0
