commit 57f4ad3ac4c69bdf1bef00f403eda54803370f7c
Author: q66 <q66@chimera-linux.org>
Date:   Mon Mar 11 14:19:25 2024 +0100

    root-remount: make root a shared mount by default

diff --git a/early/scripts/root-remount.sh b/early/scripts/root-remount.sh
index 57ff976..c037132 100644
--- a/early/scripts/root-remount.sh
+++ b/early/scripts/root-remount.sh
@@ -5,4 +5,4 @@ DINIT_NO_CONTAINER=1
 
 . ./early/scripts/common.sh
 
-exec mount -o remount,${dinit_early_root_remount:-ro} /
+exec mount -o remount,${dinit_early_root_remount:-ro,rshared} /
