commit 2eb7d46927479e4d2dbe9c7b75c8b343995f576c
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Fri Feb 24 03:43:18 2023 +0100

    use udevd from libexec

diff --git a/system-services/init-udevd b/system-services/init-udevd
index ea466d3..876b1f8 100644
--- a/system-services/init-udevd
+++ b/system-services/init-udevd
@@ -2,7 +2,7 @@
 # replaced with a real supervised service once everything else is up
 
 type         = scripted
-command      = /usr/bin/udevd --daemon
+command      = /usr/libexec/udevd --daemon
 stop-command = /usr/bin/sh -c "/usr/bin/udevadm control -e || :"
 depends-on   = init-prepare.target
 depends-on   = init-modules-early
