diff --git a/configure b/configure
index df6756c..f2b83e7 100755
--- a/configure
+++ b/configure
@@ -13864,7 +13864,7 @@ printf %s "checking gettext in ${GETTEXT_PREFIX}... " >&6; }
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <libintl.h>
-             int main() {char *cp = dgettext(NULL, "xx"); return 0; }
+             int main() {char *cp = dgettext(0, "xx"); return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
 then :
@@ -13889,7 +13889,7 @@ printf %s "checking gettext in ${GETTEXT_PREFIX}... " >&6; }
       cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <libintl.h>
-                 int main() {char *cp = dgettext(NULL, "xx"); return 0; }
+                 int main() {char *cp = dgettext(0, "xx"); return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
 then :
@@ -13915,7 +13915,7 @@ printf %s "checking gettext in ${GETTEXT_PREFIX}... " >&6; }
       cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <libintl.h>
-                 int main() {char *cp = dgettext(NULL, "xx"); return 0; }
+                 int main() {char *cp = dgettext(0, "xx"); return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
 then :
