From 95ada8a3ed98f8d5c292787d952e237268bc5953 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?JC=20Gr=C3=BCnhage?= <jan.christian@gruenhage.xyz>
Date: Sat, 31 Jan 2026 12:07:42 +0100
Subject: [PATCH 1/5] cni plugins path

---
 pkg/config/config_unix.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/config/config_unix.go b/pkg/config/config_unix.go
index 91a6a572f..9f9d2d17d 100644
--- a/pkg/config/config_unix.go
+++ b/pkg/config/config_unix.go
@@ -5,7 +5,7 @@ package config
 // Defaults for linux/unix if none are specified.
 const (
 	cniConfigDir             = "/etc/cni/net.d/"
-	cniBinDir                = "/opt/cni/bin/"
+	cniBinDir                = "/usr/libexec/cni/"
 	containerExitsDir        = "/var/run/crio/exits"
 	ContainerAttachSocketDir = "/var/run/crio"
 
-- 
2.52.0

