Fix from Kaz Kylheku.

diff --git a/lib.c b/lib.c
index 0e374c70..b1ce9e9c 100644
--- a/lib.c
+++ b/lib.c
@@ -1195,6 +1195,7 @@ val seq_begin(val obj)
   iter = si->ui.iter;
   si_obj = cobj(coerce(mem_t *, si), seq_iter_cls, &seq_iter_ops);
   gc_hint(iter);
+  gc_hint(obj);
   return si_obj;
 }

@@ -1250,6 +1251,7 @@ val iter_begin(val obj)
         iter = si->ui.iter;
         si_obj = cobj(coerce(mem_t *, si), seq_iter_cls, &seq_iter_ops);
         gc_hint(iter);
+        gc_hint(obj);
         return si_obj;
       }
     }
