From ba4417ee5c8944e2202264518043248912e9499f Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 4 Dec 2021 15:05:27 +0100
Subject: [PATCH] fix typo

---
 src/common/pseries_platform.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/pseries_platform.h b/src/common/pseries_platform.h
index 440c3e6..5fa12ff 100644
--- a/src/common/pseries_platform.h
+++ b/src/common/pseries_platform.h
@@ -19,7 +19,7 @@
  */
 
 #ifndef PLATFORM_H
-#define PLARFORM_H
+#define PLATFORM_H
 
 #define PLATFORM_FILE	"/proc/cpuinfo"
 
-- 
2.34.0

