--- a/acconfig.h
+++ b/acconfig.h
@@ -191,7 +191,6 @@
  * If screen is installed with permissions to update /etc/utmp (such
  * as if it is installed set-uid root), define UTMPOK.
  */
-#define UTMPOK
 
 /* Set LOGINDEFAULT to one (1)
  * if you want entries added to /etc/utmp by default, else set it to
--- a/utmp.c
+++ b/utmp.c
@@ -882,7 +882,7 @@ getlogin()
 }
 # endif /* BUGGYGETLOGIN */
 
-#if defined(linux) && defined(GETUTENT)
+#if defined(linux) && defined(GETUTENT) && defined(UTMPOK)
 # undef pututline
 
 /* aargh, linux' pututline returns void! */
