--- a/subprojects/simdutf/include/simdutf/internal/isadetection.h
+++ b/subprojects/simdutf/include/simdutf/internal/isadetection.h
@@ -73,8 +73,8 @@ struct simdutf_riscv_hwprobe {
 #if defined(__loongarch__) && defined(__linux__)
   #include <sys/auxv.h>
 // bits/hwcap.h
-// #define HWCAP_LOONGARCH_LSX             (1 << 4)
-// #define HWCAP_LOONGARCH_LASX            (1 << 5)
+  #define HWCAP_LOONGARCH_LSX             (1 << 4)
+  #define HWCAP_LOONGARCH_LASX            (1 << 5)
 #endif
 
 namespace simdutf {
