From 063772d698a08830d690a158fa7f4b28df2e0a83 Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Thu, 19 Sep 2024 12:17:49 -0400
Subject: [PATCH] Add GNOME 47 to version list

---
 data/metadata.json.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/metadata.json.in b/data/metadata.json.in
index 63becbfb2..60d577688 100644
--- a/data/metadata.json.in
+++ b/data/metadata.json.in
@@ -3,6 +3,6 @@
     "name": "GSConnect",
     "description": "GSConnect is a complete implementation of KDE Connect especially for GNOME Shell with Nautilus, Chrome and Firefox integration. It does not rely on the KDE Connect desktop application and will not work with it installed.\n\nKDE Connect allows devices to securely share content like notifications or files and other features like SMS messaging and remote control. The KDE Connect team has applications for Linux, BSD, Android, Sailfish, iOS, macOS and Windows.\n\nPlease report issues on Github!",
     "version": @PACKAGE_VERSION@,
-    "shell-version": [ "46" ],
+    "shell-version": [ "46", "47" ],
     "url": "@PACKAGE_URL@/wiki"
 }
