diff --git a/src/itx_tmpl.c b/src/itx_tmpl.c
index bafe0a8..319968f 100644
--- a/src/itx_tmpl.c
+++ b/src/itx_tmpl.c
@@ -206,8 +206,6 @@ static void inv_txfm_add_wht_wht_4x4_c(pixel *dst, const ptrdiff_t stride,
 #if HAVE_ASM
 #if ARCH_AARCH64 || ARCH_ARM
 #include "src/arm/itx.h"
-#elif ARCH_LOONGARCH64
-#include "src/loongarch/itx.h"
 #elif ARCH_PPC64LE
 #include "src/ppc/itx.h"
 #elif ARCH_RISCV
@@ -292,9 +290,6 @@ COLD void bitfn(dav1d_itx_dsp_init)(Dav1dInvTxfmDSPContext *const c, int bpc) {
 #if ARCH_AARCH64 || ARCH_ARM
     itx_dsp_init_arm(c, bpc, &all_simd);
 #endif
-#if ARCH_LOONGARCH64
-    itx_dsp_init_loongarch(c, bpc);
-#endif
 #if ARCH_PPC64LE
     itx_dsp_init_ppc(c, bpc);
 #endif
diff --git a/src/loopfilter_tmpl.c b/src/loopfilter_tmpl.c
index 0a2baf1..0de8255 100644
--- a/src/loopfilter_tmpl.c
+++ b/src/loopfilter_tmpl.c
@@ -247,8 +247,6 @@ static void loop_filter_v_sb128uv_c(pixel *dst, const ptrdiff_t stride,
 #if HAVE_ASM
 #if ARCH_AARCH64 || ARCH_ARM
 #include "src/arm/loopfilter.h"
-#elif ARCH_LOONGARCH64
-#include "src/loongarch/loopfilter.h"
 #elif ARCH_PPC64LE
 #include "src/ppc/loopfilter.h"
 #elif ARCH_X86
@@ -265,8 +263,6 @@ COLD void bitfn(dav1d_loop_filter_dsp_init)(Dav1dLoopFilterDSPContext *const c)
 #if HAVE_ASM
 #if ARCH_AARCH64 || ARCH_ARM
     loop_filter_dsp_init_arm(c);
-#elif ARCH_LOONGARCH64
-    loop_filter_dsp_init_loongarch(c);
 #elif ARCH_PPC64LE
     loop_filter_dsp_init_ppc(c);
 #elif ARCH_X86
diff --git a/src/meson.build b/src/meson.build
index 04dea4d..e5ac7db 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -242,11 +242,9 @@ if is_asm_enabled
             'loongarch/cdef.S',
             'loongarch/ipred.S',
             'loongarch/mc.S',
-            'loongarch/loopfilter.S',
             'loongarch/looprestoration.S',
             'loongarch/msac.S',
             'loongarch/refmvs.S',
-            'loongarch/itx.S',
         )
         libdav1d_asm_objs += libdav1d_sources_asm
     elif host_machine.cpu() == 'ppc64le'
