commit ad576a9558d27d7ae7609d6acac0614bcbf31fd8
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Thu May 18 00:41:09 2023 +0200

    remove some more binary/unary_function

diff --git a/src/libnrtype/FontFactory.h b/src/libnrtype/FontFactory.h
index e3bf1d2..33ce1c8 100644
--- a/src/libnrtype/FontFactory.h
+++ b/src/libnrtype/FontFactory.h
@@ -46,10 +46,10 @@ namespace Glib
 
 // the font_factory keeps a hashmap of all the loaded font_instances, and uses the PangoFontDescription
 // as index (nota: since pango already does that, using the PangoFont could work too)
-struct font_descr_hash : public std::unary_function<PangoFontDescription*,size_t> {
+struct font_descr_hash {
     size_t operator()(PangoFontDescription *const &x) const;
 };
-struct font_descr_equal : public std::binary_function<PangoFontDescription*, PangoFontDescription*, bool> {
+struct font_descr_equal {
     bool operator()(PangoFontDescription *const &a, PangoFontDescription *const &b) const;
 };
 
