From f3bc15a4a141b8193aea0c91f45e4294bce68546 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sun, 1 Jan 2023 16:57:49 +0000
Subject: [PATCH] fix cfi violations

---
 cmdbuf.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmdbuf.c b/cmdbuf.c
index 5b57046..f3bc201 100644
--- a/cmdbuf.c
+++ b/cmdbuf.c
@@ -1563,7 +1563,7 @@ read_cmdhist(action, uparam, skip_search, skip_shell)
 addhist_init(uparam, ml, string)
 	void *uparam;
 	struct mlist *ml;
-	char constant *string;
+	char *string;
 {
 	if (ml != NULL)
 		cmd_addhist(ml, string, 0);
@@ -1648,7 +1648,7 @@ struct save_ctx
 copy_hist(uparam, ml, string)
 	void *uparam;
 	struct mlist *ml;
-	char constant *string;
+	char *string;
 {
 	struct save_ctx *ctx = (struct save_ctx *) uparam;
 
-- 
2.34.1

