commit d205f812f20380398d59aff0a85390a26f18e0af
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Jul 17 04:18:47 2021 +0200

    pass CFLAGS during linking

diff --git a/Makefile.in b/Makefile.in
index c9cbf8e..2e064c0 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -54,13 +54,13 @@ OBJ = \
 all: less$(EXEEXT) lesskey$(EXEEXT) lessecho$(EXEEXT)
 
 less$(EXEEXT): ${OBJ}
-	${CC} ${LDFLAGS} -o $@ ${OBJ} ${LIBS}
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ ${OBJ} ${LIBS}
 
 lesskey$(EXEEXT): lesskey.${O} version.${O}
-	${CC} ${LDFLAGS} -o $@ lesskey.${O} version.${O}
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ lesskey.${O} version.${O}
 
 lessecho$(EXEEXT): lessecho.${O} version.${O}
-	${CC} ${LDFLAGS} -o $@ lessecho.${O} version.${O}
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ lessecho.${O} version.${O}
 
 charset.${O}: compose.uni ubin.uni wide.uni
 
