commit 04e3ea0878b7ce0f2eba0f035bb51736f73f52f8
Author: q66 <q66@chimera-linux.org>
Date:   Thu Nov 18 05:49:50 2021 +0100

    pass CFLAGS during linking

diff --git a/Makefile.in b/Makefile.in
index d087e9e..139fac9 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -59,13 +59,13 @@ endif
 all: less$(EXEEXT) lesskey$(EXEEXT) lessecho$(EXEEXT)
 
 less$(EXEEXT): ${OBJ}
-	${CC} ${LDFLAGS} -o $@ ${OBJ} ${LIBS}
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ ${OBJ} ${LIBS}
 
 lesskey$(EXEEXT): lesskey.${O} lesskey_parse.${O} xbuf.${O} version.${O}
-	${CC} ${LDFLAGS} -o $@ lesskey.${O} lesskey_parse.${O} xbuf.${O} version.${O}
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ lesskey.${O} lesskey_parse.${O} xbuf.${O} version.${O}
 
 lessecho$(EXEEXT): lessecho.${O} version.${O}
-	${CC} ${LDFLAGS} -o $@ lessecho.${O} version.${O}
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ lessecho.${O} version.${O}
 
 charset.${O}: compose.uni ubin.uni wide.uni
 
