diff -ruN a/src/libOpenImageIO/imageinout_test.cpp b/src/libOpenImageIO/imageinout_test.cpp
--- a/src/libOpenImageIO/imageinout_test.cpp	2024-11-08 19:15:11.000000000 +0100
+++ b/src/libOpenImageIO/imageinout_test.cpp	2024-11-09 14:12:21.004054137 +0100
@@ -537,7 +537,7 @@
     getargs(argc, argv);
 
     if (enable_fpe) {
-#if defined(__linux__)
+#if 0
         fprintf(stderr, "Enable floating point exceptions.\n");
         feenableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
 #else
