--- a/include/mtxclient/http/errors.hpp
+++ b/include/mtxclient/http/errors.hpp
@@ -87,7 +89,7 @@
     // Formats the point p using the parsed format specification (presentation)
     // stored in this formatter.
     template<typename FormatContext>
-    auto format(const mtx::http::ClientError &e, FormatContext &ctx) -> decltype(ctx.out())
+    auto format(const mtx::http::ClientError &e, FormatContext &ctx) const -> decltype(ctx.out())
     {
         // ctx.out() is an output iterator to write to.
         bool prepend_comma = false;
@@ -132,7 +134,7 @@
 {
     // parse is inherited from formatter<string_view>.
     template<typename FormatContext>
-    auto format(std::optional<mtx::http::ClientError> c, FormatContext &ctx)
+    auto format(std::optional<mtx::http::ClientError> c, FormatContext &ctx) const
     {
         if (!c)
             return fmt::format_to(ctx.out(), "(no error)");
