--- a/configure.ac
+++ b/configure.ac
@@ -347,13 +347,13 @@ if test -d $AuxPrefix/lib64; then
     BITS=$(getconf LONG_BIT 2>/dev/null)
     if test $? = 0; then
         if test $BITS = 64; then
-            SADC_DIR=$AuxPrefix/lib64
+            SADC_DIR=$AuxPrefix/lib
         fi
     else
         # Fallback: look for lm (long mode) flag to know if CPU is 64 bit
         grep " lm " /proc/cpuinfo >/dev/null 2>&1
         if test $? = 0; then
-            SADC_DIR=$AuxPrefix/lib64
+            SADC_DIR=$AuxPrefix/lib
         fi
     fi
 fi
