type that makes the link empty
--- a/cmake-proxies/cmake-modules/Findopusfile.cmake
+++ b/cmake-proxies/cmake-modules/Findopusfile.cmake
@@ -24,7 +24,7 @@
          add_library( opusfile::opusfile INTERFACE IMPORTED GLOBAL )
 
          target_include_directories( opusfile::opusfile INTERFACE ${opusfile_INCLUDE_DIR} "${opusfile_INCLUDE_DIR}/opus" )
-         target_link_libraries( opusfile::opusfile INTERFACE ${Oopusfile_LIBRARIES} )
+         target_link_libraries( opusfile::opusfile INTERFACE ${opusfile_LIBRARIES} )
       endif()
 
       mark_as_advanced(
