--- a/meson.build
+++ b/meson.build
@@ -112,8 +112,7 @@
     config_h.set('PATH_WTMP', '_PATH_WTMPX')
   else
     assert(not meson.is_cross_build(), 'Cannot determine wtmp for this cross compile, please specify -Dwtmpfile=')
-    path_wtmp = '/var/log/utx.log'
-    assert(run_command('test', '-e', path_wtmp, check: false).returncode() == 0, 'Do not know which filename to watch for wtmp changes')
+    path_wtmp = '/var/log/wtmp'
     config_h.set_quoted('PATH_WTMP', path_wtmp)
   endif
 else
