commit 22daf93a551320d5c9d5c72aa459d9d3b2498fd4
Author: q66 <q66@chimera-linux.org>
Date:   Fri Mar 17 04:34:24 2023 +0100

    disable tests because mocklibc is missing

diff --git a/meson.build b/meson.build
index f995d28..d25e527 100644
--- a/meson.build
+++ b/meson.build
@@ -231,8 +231,6 @@ if get_option('gtk_doc')
   subdir('doc/libaccountsservice')
 endif
 
-subdir('tests')
-
 configure_file(
   output: 'config.h',
   configuration: config_h,
