--- a/debian/functions
+++ b/debian/functions
@@ -596,6 +596,14 @@ _resolve_device() {
         MIN="$min"
         return 0
     else
+        # for pooled filesystems like zfs, don't raise this confusing
+        # error, as the user is expected to provide the luks device in
+        # their crypttab manually (marked initramfs); as the / filesystem
+        # may refer to a pool of multiple devices, we fundamentally cannot
+        # resolve just one of them
+        case $(awk "{if (\$1 == \"$spec\") print \$3}" /proc/mounts 2>/dev/null) in
+            zfs|btrfs) return 1 ;;
+        esac
         cryptsetup_message "ERROR: Couldn't resolve device $spec"
     fi
     return 1
