diff --git a/debian/functions b/debian/functions
index 63ecf5d..8b553e1 100644
--- a/debian/functions
+++ b/debian/functions
@@ -596,6 +596,14 @@ _resolve_device() {
         MIN="$min"
         return 0
     else
+        # for pooled filesystems like zfs, don't raise this confusing
+        # error, as the user is expected to provide the luks device in
+        # their crypttab manually (marked initramfs); as the / filesystem
+        # may refer to a pool of multiple devices, we fundamentally cannot
+        # resolve just one of them
+        case $(findmnt -n -o FSTYPE "$spec" 2>/dev/null) in
+            zfs|btrfs) return 1 ;;
+        esac
         cryptsetup_message "ERROR: Couldn't resolve device $spec"
     fi
     return 1
diff --git a/debian/initramfs/hooks/cryptroot b/debian/initramfs/hooks/cryptroot
index ca759bd..d8f9741 100644
--- a/debian/initramfs/hooks/cryptroot
+++ b/debian/initramfs/hooks/cryptroot
@@ -349,6 +349,7 @@ manual_add_modules dm_mod dm_crypt
 
 copy_exec /usr/bin/cryptsetup
 copy_exec /usr/bin/dmsetup
+copy_exec /usr/bin/findmnt
 
 [ "$ASKPASS" = n ] || copy_exec /usr/lib/cryptsetup/askpass
 
