commit 731d941924d070ae293d9efa8c7dbdf523451e03
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Thu Aug 25 23:07:34 2022 +0200

    fix build

diff --git a/ip/ipstats.c b/ip/ipstats.c
index 5cdd15a..2f500fc 100644
--- a/ip/ipstats.c
+++ b/ip/ipstats.c
@@ -6,6 +6,10 @@
 #include "utils.h"
 #include "ip_common.h"
 
+#ifndef MIN
+#define MIN(a, b) ((a) < (b) ? (a) : (b))
+#endif
+
 struct ipstats_stat_dump_filters {
 	/* mask[0] filters outer attributes. Then individual nests have their
 	 * filtering mask at the index of the nested attribute.
