commit 0d14940ac1b89f9c66a417b707fa9ff8407d2e3a
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Mon Nov 14 03:27:03 2022 +0100

    drop -latomic (none in pure-llvm env)

diff --git a/node.gyp b/node.gyp
index e8e1d9f9..8ec0d178 100644
--- a/node.gyp
+++ b/node.gyp
@@ -117,9 +117,6 @@
           '-Wl,-bnoerrmsg',
         ],
       }],
-      ['OS == "linux" and llvm_version != "0.0"', {
-        'libraries': ['-latomic'],
-      }],
     ],
   },
 
diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 42e26cd9..b321382e 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -977,13 +977,6 @@
         ['v8_enable_third_party_heap', {
           # TODO(targos): add values from v8_third_party_heap_libs to link_settings.libraries
         }],
-        # Platforms that don't have Compare-And-Swap (CAS) support need to link atomic library
-        # to implement atomic memory access
-        ['v8_current_cpu in ["mips", "mipsel", "mips64", "mips64el", "ppc", "arm", "riscv64", "loong64"]', {
-          'link_settings': {
-            'libraries': ['-latomic', ],
-          },
-        }],
       ],
     },  # v8_base_without_compiler
     {
