commit ae176742ff18cd283d21b71a8c844a3ca66974be
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Jan 29 15:59:28 2022 +0100

    drop -latomic (none in pure-llvm env)

diff --git a/node.gyp b/node.gyp
index 2052917..3e7104a 100644
--- a/node.gyp
+++ b/node.gyp
@@ -116,9 +116,6 @@
           '-Wl,-bnoerrmsg',
         ],
       }],
-      ['OS == "linux" and llvm_version != "0.0"', {
-        'libraries': ['-latomic'],
-      }],
     ],
   },
 
diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 73099a0..b172812 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -945,13 +945,6 @@
         ['v8_enable_third_party_heap', {
           # TODO(targos): add values from v8_third_party_heap_libs to link_settings.libraries
         }],
-        # Platforms that don't have Compare-And-Swap (CAS) support need to link atomic library
-        # to implement atomic memory access
-        ['v8_current_cpu in ["mips", "mipsel", "mips64", "mips64el", "ppc", "arm", "riscv64"]', {
-          'link_settings': {
-            'libraries': ['-latomic', ],
-          },
-        }],
       ],
     },  # v8_base_without_compiler
     {
