diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0dd4f0a..dfad117 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -459,7 +459,7 @@ if(HAVE_MULTIVIEW)
     add_definitions(-DOSG_HAS_MULTIVIEW)
 endif(HAVE_MULTIVIEW)
 
-set(BOOST_COMPONENTS iostreams program_options system)
+set(BOOST_COMPONENTS iostreams program_options)
 
 find_package(Boost 1.70.0 CONFIG REQUIRED COMPONENTS ${BOOST_COMPONENTS} OPTIONAL_COMPONENTS ${BOOST_OPTIONAL_COMPONENTS})
 
diff --git a/components/CMakeLists.txt b/components/CMakeLists.txt
index 7659de0..f03a1c8 100644
--- a/components/CMakeLists.txt
+++ b/components/CMakeLists.txt
@@ -586,7 +586,6 @@ target_link_libraries(components
     ${OSG_LIBRARIES}
     ${OPENTHREADS_LIBRARIES}
 
-    Boost::system
     Boost::program_options
     Boost::iostreams
 
