diff -ruN a/main/gui/source/messaging/linux.cpp b/main/gui/source/messaging/linux.cpp
--- a/main/gui/source/messaging/linux.cpp	2025-02-18 11:18:52.000000000 +0100
+++ b/main/gui/source/messaging/linux.cpp	2025-02-19 10:08:02.982215123 +0100
@@ -3,6 +3,7 @@
 #include <stdexcept>
 #include <fcntl.h>
 #include <sys/file.h>
+#include <unistd.h>
 
 #include <hex/helpers/logger.hpp>
 #include <hex/api/events/events_lifecycle.hpp>
