diff -ruN a/source/blender/blenlib/BLI_build_config.h b/source/blender/blenlib/BLI_build_config.h
--- a/source/blender/blenlib/BLI_build_config.h	2025-02-05 14:43:27.000000000 +0100
+++ b/source/blender/blenlib/BLI_build_config.h	2025-03-23 15:24:46.273040428 +0100
@@ -354,6 +354,30 @@
 #  else
 #    define ARCH_CPU_BIG_ENDIAN 1
 #  endif
+#elif defined(__riscv)
+#  if __riscv_xlen == 64
+#    define ARCH_CPU_RISCV_FAMILY 1
+#    define ARCH_CPU_RISCV64 1
+#    define ARCH_CPU_64_BITS 1
+#    define ARCH_CPU_LITTLE_ENDIAN 1
+#  elif __riscv_xlen == 32
+#    define ARCH_CPU_RISCV_FAMILY 1
+#    define ARCH_CPU_RISCV 1
+#    define ARCH_CPU_32_BITS 1
+#    define ARCH_CPU_LITTLE_ENDIAN 1
+#  endif
+#elif defined(__loongarch__)
+#  if defined(__loongarch64)
+#    define ARCH_CPU_LOONGARCH_FAMILY 1
+#    define ARCH_CPU_LOONGARCH64 1
+#    define ARCH_CPU_64_BITS 1
+#    define ARCH_CPU_LITTLE_ENDIAN 1
+#  else
+#    define ARCH_CPU_LOONGARCH_FAMILY 1
+#    define ARCH_CPU_LOONGARCH 1
+#    define ARCH_CPU_32_BITS 1
+#    define ARCH_CPU_LITTLE_ENDIAN 1
+#  endif
 #else
 #  error Please add support for your architecture in BLI_build_config.h
 #endif
