commit 563bf0e688efea932ead4c4b8b2581dc5831fb07
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Tue Jan 25 05:35:29 2022 +0100

    update dependencies

diff --git a/gnome-initial-setup/meson.build b/gnome-initial-setup/meson.build
index 1d2388b..c1804ee 100644
--- a/gnome-initial-setup/meson.build
+++ b/gnome-initial-setup/meson.build
@@ -35,7 +35,7 @@ dependencies = [
     dependency ('gnome-desktop-3.0', version: '>= 3.7.5'),
     dependency ('gsettings-desktop-schemas', version: '>= 3.37.1'),
     dependency ('fontconfig'),
-    dependency ('gweather-3.0'),
+    dependency ('gweather4'),
     dependency ('goa-1.0'),
     dependency ('goa-backend-1.0'),
     dependency ('gtk+-3.0', version: '>= 3.11.3'),
@@ -46,7 +46,7 @@ dependencies = [
     dependency ('libgeoclue-2.0', version: '>= 2.3.1'),
     cc.find_library('m', required: false),
     dependency ('pango', version: '>= 1.32.5'),
-    dependency ('rest-0.7'),
+    dependency ('rest-1.0'),
     dependency ('json-glib-1.0'),
     dependency ('krb5'),
     dependency ('libsecret-1', version: '>= 0.18.8'),
@@ -51,7 +51,7 @@ dependencies = [
     dependency ('krb5'),
     dependency ('libsecret-1', version: '>= 0.18.8'),
     dependency ('pwquality'),
-    dependency ('webkit2gtk-4.0', version: '>= 2.26.0'),
+    dependency ('webkit2gtk-4.1', version: '>= 2.26.0'),
     cheese_dep,
     cheese_gtk_dep,
     ibus_dep,
