commit 6ae94614eb41bd1a895c21e9fb96a377e9e81464
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Thu Aug 25 03:00:36 2022 +0200

    use geocode-glib-2.0

diff --git a/gnome-initial-setup/meson.build b/gnome-initial-setup/meson.build
index 4ee9ff3..b05a08e 100644
--- a/gnome-initial-setup/meson.build
+++ b/gnome-initial-setup/meson.build
@@ -42,7 +42,7 @@ dependencies = [
     dependency ('glib-2.0', version: '>= 2.63.1'),
     dependency ('gio-unix-2.0', version: '>= 2.53.0'),
     dependency ('gdm', version: '>= 3.8.3'),
-    dependency ('geocode-glib-1.0'),
+    dependency ('geocode-glib-2.0'),
     dependency ('libgeoclue-2.0', version: '>= 2.3.1'),
     dependency ('libhandy-1', version: '>= 1.5.90', fallback: ['libhandy', 'libhandy_dep']),
     cc.find_library('m', required: false),
