commit a34727348140395cb2458abb1454159f33b7d83a
Author: q66 <q66@chimera-linux.org>
Date:   Tue Jan 25 05:37:40 2022 +0100

    heimdal compat

diff --git a/gnome-initial-setup/pages/account/um-realm-manager.c b/gnome-initial-setup/pages/account/um-realm-manager.c
index bc4fd33..03a0416 100644
--- a/gnome-initial-setup/pages/account/um-realm-manager.c
+++ b/gnome-initial-setup/pages/account/um-realm-manager.c
@@ -23,7 +23,7 @@
 
 #include "um-realm-manager.h"
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 #include <glib.h>
 #include <glib/gi18n.h>
@@ -706,9 +706,6 @@ login_perform_kinit (krb5_context k5,
         code = krb5_get_init_creds_opt_alloc (k5, &opts);
         g_return_val_if_fail (code == 0, code);
 
-        code = krb5_get_init_creds_opt_set_out_ccache (k5, opts, ccache);
-        g_return_val_if_fail (code == 0, code);
-
         code = krb5_get_init_creds_password (k5, &creds, principal,
                                              (char *)password,
                                              NULL, 0, 0, NULL, opts);
