From ffe7e1468a4256e5862687741279efd4452e0c16 Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Sun, 30 Jun 2024 02:25:04 +0200
Subject: [PATCH 16/22] avoid redefinition of prctl_mm_map structure

---
 src/basic/missing_prctl.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/basic/missing_prctl.h b/src/basic/missing_prctl.h
index 2c9f9f6..65a984b 100644
--- a/src/basic/missing_prctl.h
+++ b/src/basic/missing_prctl.h
@@ -1,7 +1,9 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+#ifdef __GLIBC__
 #include <linux/prctl.h>
+#endif
 
 #include "macro.h"
 
-- 
2.45.2

