commit fdb40aba64a42980ba9d6f51ba40555a9501c45a
Author: q66 <q66@chimera-linux.org>
Date:   Sun Jul 30 12:20:31 2023 +0200

    remove malloc_trim, malloc_info

diff --git a/src/libsystemd/sd-event/sd-event.c b/src/libsystemd/sd-event/sd-event.c
index aba4581..b7fbeca 100644
--- a/src/libsystemd/sd-event/sd-event.c
+++ b/src/libsystemd/sd-event/sd-event.c
@@ -1888,7 +1888,7 @@ _public_ int sd_event_trim_memory(void) {
 
         usec_t before_timestamp = now(CLOCK_MONOTONIC);
         hashmap_trim_pools();
-        r = malloc_trim(0);
+        r = 0;
         usec_t after_timestamp = now(CLOCK_MONOTONIC);
 
         if (r > 0)
diff --git a/src/shared/bus-util.c b/src/shared/bus-util.c
index c42a144..5e3741e 100644
--- a/src/shared/bus-util.c
+++ b/src/shared/bus-util.c
@@ -625,10 +625,6 @@ static int method_dump_memory_state_by_fd(sd_bus_message *message, void *userdat
         if (!f)
                 return -ENOMEM;
 
-        r = RET_NERRNO(malloc_info(/* options= */ 0, f));
-        if (r < 0)
-                return r;
-
         r = memstream_finalize(&m, &dump, &dump_size);
         if (r < 0)
                 return r;
diff --git a/src/shared/common-signal.c b/src/shared/common-signal.c
index e4edd18..78a4b5e 100644
--- a/src/shared/common-signal.c
+++ b/src/shared/common-signal.c
@@ -66,11 +66,6 @@ int sigrtmin18_handler(sd_event_source *s, const struct signalfd_siginfo *si, vo
                         break;
                 }
 
-                if (malloc_info(0, f) < 0) {
-                        log_error_errno(errno, "Failed to invoke malloc_info(): %m");
-                        break;
-                }
-
                 (void) memstream_dump(LOG_INFO, &m);
                 break;
         }
