--- a/src/basic/syslog-util.c
+++ b/src/basic/syslog-util.c
@@ -56,7 +56,7 @@
         return 1;
 }
 
-static const char *const log_facility_unshifted_table[LOG_NFACILITIES] = {
+static const char *const log_facility_unshifted_table[LOG_FACMASK] = {
         [LOG_FAC(LOG_KERN)]     = "kern",
         [LOG_FAC(LOG_USER)]     = "user",
         [LOG_FAC(LOG_MAIL)]     = "mail",
--- a/src/journal/journald-server.c
+++ b/src/journal/journald-server.c
@@ -1167,7 +1167,6 @@
         m = N_IOVEC_META_FIELDS + 5 + N_IOVEC_PAYLOAD_FIELDS + client_context_extra_fields_n_iovec(s->my_context) + N_IOVEC_OBJECT_FIELDS;
         iovec = newa(struct iovec, m);
 
-        assert_cc(3 == LOG_FAC(LOG_DAEMON));
         iovec[n++] = IOVEC_MAKE_STRING("SYSLOG_FACILITY=3");
         iovec[n++] = IOVEC_MAKE_STRING("SYSLOG_IDENTIFIER=systemd-journald");
 
--- a/src/journal/journald-audit.c
+++ b/src/journal/journald-audit.c
@@ -385,7 +385,6 @@
         sprintf(id_field, "_AUDIT_ID=%" PRIu64, id);
         iovec[n++] = IOVEC_MAKE_STRING(id_field);
 
-        assert_cc(4 == LOG_FAC(LOG_AUTH));
         iovec[n++] = IOVEC_MAKE_STRING("SYSLOG_FACILITY=4");
         iovec[n++] = IOVEC_MAKE_STRING("SYSLOG_IDENTIFIER=audit");
 
