diff --git a/Makefile b/Makefile
index 1b1b284..4408d1f 100644
--- a/Makefile
+++ b/Makefile
@@ -207,21 +207,27 @@ clean-wayland-protocols:
 install: install-dunst install-dunstctl install-dunstrc install-service
 
 install-dunst: dunst doc
+	install -d ${DESTDIR}${BINDIR}
 	install -Dm755 dunst ${DESTDIR}${BINDIR}/dunst
+	install -d ${DESTDIR}${MANPREFIX}/man1
+	install -d ${DESTDIR}${MANPREFIX}/man5
 	install -Dm644 docs/dunst.1 ${DESTDIR}${MANPREFIX}/man1/dunst.1
 	install -Dm644 docs/dunst.5 ${DESTDIR}${MANPREFIX}/man5/dunst.5
 	install -Dm644 docs/dunstctl.1 ${DESTDIR}${MANPREFIX}/man1/dunstctl.1
 
 install-dunstctl: dunstctl
+	install -d ${DESTDIR}${BINDIR}
 	install -Dm755 dunstctl ${DESTDIR}${BINDIR}/dunstctl
 
 ifeq (1,${SYSCONF_FORCE_NEW})
 install-dunstrc:
+	install -d ${DESTDIR}${SYSCONFDIR}/dunst
 	install -Dm644 dunstrc ${DESTDIR}${SYSCONFFILE}
 endif
 
 install-service: install-service-dbus
 install-service-dbus: service-dbus
+	install -d ${DESTDIR}${SERVICEDIR_DBUS}
 	install -Dm644 org.knopwob.dunst.service ${DESTDIR}${SERVICEDIR_DBUS}/org.knopwob.dunst.service
 ifneq (0,${SYSTEMD})
 install-service: install-service-systemd
@@ -232,22 +238,27 @@ endif
 ifneq (0,${DUNSTIFY})
 install: install-dunstify
 install-dunstify: dunstify
+	install -d ${DESTDIR}${BINDIR}
 	install -Dm755 dunstify ${DESTDIR}${BINDIR}/dunstify
 endif
 
 ifneq (0,${COMPLETIONS})
 install: install-completions
 install-completions:
+	install -d ${DESTDIR}${BASHCOMPLETIONDIR}
 	install -Dm644 completions/dunst.bashcomp ${DESTDIR}${BASHCOMPLETIONDIR}/dunst
 	install -Dm644 completions/dunstctl.bashcomp ${DESTDIR}${BASHCOMPLETIONDIR}/dunstctl
+	install -d ${DESTDIR}${ZSHCOMPLETIONDIR}
 	install -Dm644 completions/_dunst.zshcomp ${DESTDIR}${ZSHCOMPLETIONDIR}/_dunst
 	install -Dm644 completions/_dunstctl.zshcomp ${DESTDIR}${ZSHCOMPLETIONDIR}/_dunstctl
+	install -d ${DESTDIR}${FISHCOMPLETIONDIR}
 	install -Dm644 completions/dunst.fishcomp ${DESTDIR}${FISHCOMPLETIONDIR}/dunst
 	install -Dm644 completions/dunstctl.fishcomp ${DESTDIR}${FISHCOMPLETIONDIR}/dunstctl
 
 ifneq (0,${DUNSTIFY})
 install: install-completions-dunstify
 install-completions-dunstify:
+	install -d ${DESTDIR}${FISHCOMPLETIONDIR}
 	install -Dm644 completions/dunstify.fishcomp ${DESTDIR}${FISHCOMPLETIONDIR}/dunstify
 endif
 endif
