we don't need to check this, we guarantee this

--- a/src/backend/meson.build
+++ b/src/backend/meson.build
@@ -1,25 +1,10 @@
 backend_inc = include_directories('.')
 
-gst_inspect = find_program(
-  'gst-inspect-1.0',
-  join_paths(gst_dep.get_pkgconfig_variable('toolsdir'), 'gst-inspect-1.0'),
-  required: false
-)
-
-assert(gst_inspect.found(),
-       'Cannot find required GStreamer-1.0 tool "gst-inspect-1.0". It should be part of gstreamer-1_0-utils. Please install it.')
-
 gst_base_plugins = [
   'playbin',
   'videoscale'
 ]
 
-foreach plugin: gst_base_plugins
-  r = run_command(gst_inspect, plugin, check: false)
-  assert(r.returncode() == 0,
-         'Cannot find required GStreamer-1.0 plugin "' + plugin + '". It should be part of gst-plugins-base. Please install it.')
-endforeach
-
 gst_good_plugins = [
   'autoaudiosink',
   'scaletempo',
@@ -27,12 +12,6 @@ gst_good_plugins = [
   'glsinkbin'
 ]
 
-foreach plugin: gst_good_plugins
-  r = run_command(gst_inspect, plugin, check: false)
-  assert(r.returncode() == 0,
-         'Cannot find required GStreamer-1.0 plugin "' + plugin + '". It should be part of gst-plugins-good. Please install it.')
-endforeach
-
 sources = files(
   'bacon-time-label.c',
   'bacon-video-widget-gst-missing-plugins.c',
