diff --git a/Cargo.toml b/Cargo.toml
index 7dad3e4..c0eb483 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,7 +47,7 @@ clap_complete = "4"
 clap_mangen = "0.2"
 dirs = { version = ">=5, <7" }
 roff = "0.2.1"
-vergen = { version = "8", features = [ "git", "git2" ] }
+vergen = { version = "8", features = [ ] }
 sequoia-directories = "0.1"
 sequoia-man = "0.3"
 
diff --git a/build.rs b/build.rs
index 527f9b8..3806857 100644
--- a/build.rs
+++ b/build.rs
@@ -28,30 +28,10 @@ pub mod openpgp {
 fn main() {
     let mut cli = cli::build(false);
 
-    git_version();
     completions();
     generate_man_pages(&mut cli).expect("can generate man pages");
 }
 
-fn git_version() {
-    // Emit the "cargo:" instructions including
-    // "cargo:rustc-env=VERGEN_GIT_DESCRIBE=<DESCRIBE>".
-    //
-    // If the source directory does not contain a git repository,
-    // e.g., because the code was extracted from a tarball, this
-    // produces an `Error` result, which we ignore, and
-    // `VERGEN_GIT_DESCRIBE` is not set.  That's okay, because we are
-    // careful to only use `VERGEN_GIT_DESCRIBE` if it is actually
-    // set.
-    let _ = vergen::EmitBuilder::builder()
-        // VERGEN_GIT_DESCRIBE
-        .git_describe(/* dirty */ true, /* tags */ false, None)
-        // Don't emit placeholder values for the git version if the
-        // git repository is not present.
-        .fail_on_error()
-        .emit();
-}
-
 fn completions() {
     // Generate shell completions
     let outdir = PathBuf::from("completions");
