--- a/include/private/gcconfig.h
+++ b/include/private/gcconfig.h
@@ -1720,8 +1720,8 @@ EXTERN_C_BEGIN
 #     pragma weak __data_start
       extern int __data_start[];
 #     define DATASTART ((ptr_t)(__data_start))
-#     define CPP_WORDSZ _LOONGARCH_SZPTR
-#     define ALIGNMENT (_LOONGARCH_SZPTR/8)
+#     define CPP_WORDSZ __loongarch_grlen
+#     define ALIGNMENT (__loongarch_grlen/8)
 #   endif
 # endif /* LOONGARCH */
 
