--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -52,7 +52,7 @@
 
 # Yes, 'task' (and hence libshared) is included twice, otherwise linking fails on assorted OSes.
 # Similarly for `tc`.
-target_link_libraries (task_executable task tc commands tc columns libshared task libshared ${TASK_LIBRARIES})
-target_link_libraries (calc_executable task tc commands tc columns libshared task libshared ${TASK_LIBRARIES})
-target_link_libraries (lex_executable  task tc commands tc columns libshared task libshared ${TASK_LIBRARIES})
+target_link_libraries (calc_executable task tc commands tc columns libshared task libshared ${TASK_LIBRARIES} sqlite3)
+target_link_libraries (lex_executable  task tc commands tc columns libshared task libshared ${TASK_LIBRARIES} sqlite3)
+target_link_libraries (task_executable task tc commands tc columns libshared task libshared ${TASK_LIBRARIES} sqlite3)
 if (DARWIN)
