--- a/source/common/putil.cpp
+++ b/source/common/putil.cpp
@@ -39,6 +39,8 @@
 ******************************************************************************
 */
 
+#define ICU_DATA_DIR "/usr/share/icu/78.2"
+
 // Defines _XOPEN_SOURCE for access to POSIX functions.
 // Must be before any other #includes.
 #include "uposixdefs.h"
--- a/source/config/mh-linux
+++ b/source/config/mh-linux
@@ -23,7 +23,7 @@ LD_RPATH= -Wl,-zorigin,-rpath,'$$'ORIGIN
 LD_RPATH_PRE = -Wl,-rpath,
 
 ## These are the library specific LDFLAGS
-LDFLAGSICUDT=-nodefaultlibs -nostdlib
+LDFLAGSICUDT=-nodefaultlibs
 
 ## Compiler switch to embed a library name
 # The initial tab in the next line is to prevent icu-config from reading it.
