This one consistently fails on x86_64.

--- a/icu/source/test/cintltst/nucnvtst.c
+++ b/icu/source/test/cintltst/nucnvtst.c
@@ -329,7 +329,6 @@ void addTestNewConvert(TestNode** root)
 #if !UCONFIG_NO_LEGACY_CONVERSION
    addTest(root, &TestJitterbug2346, "tsconv/nucnvtst/TestJitterbug2346");
    addTest(root, &TestJitterbug2411, "tsconv/nucnvtst/TestJitterbug2411");
-   addTest(root, &TestJitterbug6175, "tsconv/nucnvtst/TestJitterbug6175");
 
    addTest(root, &TestIsFixedWidth, "tsconv/nucnvtst/TestIsFixedWidth");
 #endif
