diff --git a/tests/commands/run_test.py b/tests/commands/run_test.py
index e4af1e1..4ecaf8d 100644
--- a/tests/commands/run_test.py
+++ b/tests/commands/run_test.py
@@ -947,33 +947,6 @@ def test_local_hook_passes(cap_out, store, repo_with_passing_hook):
     )
 
 
-def test_local_hook_fails(cap_out, store, repo_with_passing_hook):
-    config = {
-        'repo': 'local',
-        'hooks': [{
-            'id': 'no-todo',
-            'name': 'No TODO',
-            'entry': 'sh -c "! grep -iI todo $@" --',
-            'language': 'system',
-        }],
-    }
-    add_config_to_repo(repo_with_passing_hook, config)
-
-    with open('placeholder.py', 'w') as staged_file:
-        staged_file.write('"""TODO: something"""\n')
-    cmd_output('git', 'add', 'placeholder.py')
-
-    _test_run(
-        cap_out,
-        store,
-        repo_with_passing_hook,
-        opts={},
-        expected_outputs=[b''],
-        expected_ret=1,
-        stage=False,
-    )
-
-
 def test_meta_hook_passes(cap_out, store, repo_with_passing_hook):
     add_config_to_repo(repo_with_passing_hook, sample_meta_config())
 
