--- /dev/null
+++ b/res/linux/Makefile
@@ -0,0 +1,55 @@
+
+# Install destination
+PREFIX ?= /usr/local
+
+# Files to be compiled
+SRC = ../../snibbetracker/src/CAllocator.c ../../snibbetracker/src/CEngine.c ../../snibbetracker/src/CInput.c ../../snibbetracker/src/CSynth.c ../../snibbetracker/src/cJSON/cJSON.c ../../snibbetracker/src/dir_posix.c ../../snibbetracker/src/main.c
+OBJ = $(SRC:.c=.o) # replaces the .c from SRC with .o
+DEP = $(SRC:.c=.d)
+
+# Compiler
+CC = gcc
+
+RM = rm -f
+
+# Include paths
+INCLUDE_PATHS = -I../../snibbetracker/src/cJSON
+
+# Library paths
+# LIBRARY_PATHS = -LC:\Downloads\SDL2-devel-2.0.4-mingw\i686-w64-mingw32\lib
+
+# Compiler flags
+COMPILER_FLAGS = -Wall -std=gnu99 -lasound -Wno-unused-function $(CFLAGS)
+
+# Linker flags
+LINKER_FLAGS = -lSDL2 -lm $(LDFLAGS) $(CFLAGS)
+
+# Executable file
+EXE = snibbetracker
+
+# Compile command
+#all:
+#	$(SRC)
+#	$(CC) $(SRC) $(INCLUDE_PATHS) $(LIBRARY_PATHS) $(COMPILER_FLAGS) $(LINKER_FLAGS) -o $(EXE)
+
+%.o: %.c         # combined w/ next line will compile recently changed .c files
+	$(CC) -MD -MP $(COMPILER_FLAGS) $(INCLUDE_PATHS) -o $@ -c $<
+
+.PHONY : all     # .PHONY ignores files named all
+all: $(EXE)      # all is dependent on $(EXE) to be complete
+
+.PHONY : install
+install: all
+	install -d $(DESTDIR)$(PREFIX)/bin
+	install -m755 snibbetracker $(DESTDIR)$(PREFIX)/bin/snibbetracker
+	install -d $(DESTDIR)$(PREFIX)/share/lundstroem/snibbetracker/demos
+	cp ../demos/*.snibb $(DESTDIR)$(PREFIX)/share/lundstroem/snibbetracker/demos/
+
+$(EXE): $(OBJ)   # $(EXE) is dependent on all of the files in $(OBJ) to exist
+	$(CC) $(OBJ) $(LINKER_FLAGS) -o $@
+
+.PHONY : clean   # .PHONY ignores files named clean
+clean:
+	-$(RM) $(OBJ) $(DEP) $(EXE)
+
+-include $(DEP)
