--- a/src/encryption/DeviceVerificationFlow.cpp
+++ b/src/encryption/DeviceVerificationFlow.cpp
@@ -9,6 +9,7 @@
 #include <QDateTime>
 #include <QTimer>
 
+#include <fmt/ranges.h>
 #include <nlohmann/json.hpp>
 
 #include "Cache.h"
--- a/src/Utils.cpp
+++ b/src/Utils.cpp
@@ -29,6 +29,8 @@
 #include <nlohmann/json.hpp>
 
 #include <cmark.h>
+
+#include <fmt/ranges.h>
 
 #include <mtx/responses/messages.hpp>
 
--- a/src/RegisterPage.cpp
+++ b/src/RegisterPage.cpp
@@ -9,6 +9,8 @@
 #include <mtx/responses/version.hpp>
 #include <mtx/responses/well-known.hpp>
 #include <mtxclient/http/client.hpp>
+
+#include <fmt/ranges.h>
 
 #include "Logging.h"
 #include "LoginPage.h"
