matches the changes in contrib/7zip
--
--- a/3rdparty/lzma/include/Threads.h
+++ b/3rdparty/lzma/include/Threads.h
@@ -20,6 +20,7 @@
 #endif
 
 #include <pthread.h>
+#include <sched.h>
 
 #endif
 
--- a/3rdparty/lzma/CMakeLists.txt
+++ b/3rdparty/lzma/CMakeLists.txt
@@ -87,9 +87,9 @@
 	src/XzIn.c
 )
 
-target_compile_definitions(pcsx2-lzma PRIVATE _7ZIP_ST)
+target_compile_definitions(pcsx2-lzma PRIVATE _7ZIP_ST -D_GNU_SOURCE)
 
 target_include_directories(pcsx2-lzma PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/include")
 target_include_directories(pcsx2-lzma INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/include")
 
-add_library(LZMA::LZMA ALIAS pcsx2-lzma)
\ No newline at end of file
+add_library(LZMA::LZMA ALIAS pcsx2-lzma)
--- a/3rdparty/lzma/src/CpuArch.c
+++ b/3rdparty/lzma/src/CpuArch.c
@@ -421,8 +421,6 @@
 
 #ifdef USE_HWCAP
 
-#include <asm/hwcap.h>
-
   #define MY_HWCAP_CHECK_FUNC_2(name1, name2) \
   BoolInt CPU_IsSupported_ ## name1() { return (getauxval(AT_HWCAP)  & (HWCAP_  ## name2)) ? 1 : 0; }
 
--- a/3rdparty/lzma/src/Threads.c
+++ b/3rdparty/lzma/src/Threads.c
@@ -257,7 +257,7 @@
       */
 
       // ret2 =
-      pthread_attr_setaffinity_np(&attr, sizeof(*cpuSet), cpuSet);
+      //pthread_attr_setaffinity_np(&attr, sizeof(*cpuSet), cpuSet);
       // if (ret2) ret = ret2;
       #endif
     }
@@ -267,14 +267,12 @@
     if (!ret)
     {
       p->_created = 1;
-      /*
       if (cpuSet)
       {
         // ret2 =
         pthread_setaffinity_np(p->_tid, sizeof(*cpuSet), cpuSet);
         // if (ret2) ret = ret2;
       }
-      */
     }
   }
   // ret2 =
