--- a/deps/v8/third_party/highway/src/hwy/detect_targets.h
+++ b/deps/v8/third_party/highway/src/hwy/detect_targets.h
@@ -271,7 +271,7 @@
      (HWY_COMPILER_GCC_ACTUAL && HWY_COMPILER_GCC_ACTUAL < 1300))
 #define HWY_BROKEN_RVV (HWY_RVV)
 #else
-#define HWY_BROKEN_RVV 0
+#define HWY_BROKEN_RVV (HWY_RVV)
 #endif
 
 // Allow the user to override this without any guarantee of success.
@@ -715,9 +715,9 @@
 #endif
 
 #if HWY_ARCH_RISCV && HWY_HAVE_RUNTIME_DISPATCH
-#define HWY_ATTAINABLE_RISCV HWY_RVV
+#define HWY_ATTAINABLE_RISCV 0
 #else
-#define HWY_ATTAINABLE_RISCV HWY_BASELINE_RVV
+#define HWY_ATTAINABLE_RISCV 0
 #endif
 
 #ifndef HWY_ATTAINABLE_TARGETS_X86  // allow override
