--- a/configure.ac
+++ b/configure.ac
@@ -1769,15 +1769,15 @@ if test x"$enable_grub_mount" = xno ; then
 fi
 
 if test x"$grub_mount_excuse" = x ; then
-  AC_CHECK_LIB([fuse], [fuse_main_real], [],
+  AC_CHECK_LIB([fuse3], [fuse_main_real], [],
                [grub_mount_excuse="need FUSE library"])
 fi
 
 if test x"$grub_mount_excuse" = x ; then
   # Check for fuse headers.
   SAVED_CPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="$CPPFLAGS -DFUSE_USE_VERSION=26"
-  AC_CHECK_HEADERS([fuse/fuse.h], [],
+  CPPFLAGS="$CPPFLAGS -I/usr/include/fuse3 -DFUSE_USE_VERSION=30"
+  AC_CHECK_HEADERS([fuse3/fuse.h], [],
   	[grub_mount_excuse=["need FUSE headers"]])
   CPPFLAGS="$SAVED_CPPFLAGS"
 fi
--- a/Makefile.util.am
+++ b/Makefile.util.am
@@ -195,7 +195,7 @@ CLEANFILES += grub-mount.1
 endif
 grub_mount_SOURCES  = util/grub-mount.c grub-core/kern/emu/hostfs.c grub-core/disk/host.c grub-core/osdep/init.c 
 nodist_grub_mount_SOURCES  = grub_fstest_init.c 
-grub_mount_LDADD  = libgrubmods.a libgrubgcry.a libgrubkern.a grub-core/lib/gnulib/libgnu.a $(LIBINTL) $(LIBDEVMAPPER) $(LIBZFS) $(LIBNVPAIR) $(LIBGEOM) -lfuse 
+grub_mount_LDADD  = libgrubmods.a libgrubgcry.a libgrubkern.a grub-core/lib/gnulib/libgnu.a $(LIBINTL) $(LIBDEVMAPPER) $(LIBZFS) $(LIBNVPAIR) $(LIBGEOM) -lfuse3 
 grub_mount_CFLAGS  = $(AM_CFLAGS) $(CFLAGS_PROGRAM) 
 grub_mount_LDFLAGS  = $(AM_LDFLAGS) $(LDFLAGS_PROGRAM) 
 grub_mount_CPPFLAGS  = $(AM_CPPFLAGS) $(CPPFLAGS_PROGRAM) 
--- a/Makefile.util.def
+++ b/Makefile.util.def
@@ -313,7 +313,7 @@ program = {
   ldadd = libgrubgcry.a;
   ldadd = libgrubkern.a;
   ldadd = grub-core/lib/gnulib/libgnu.a;
-  ldadd = '$(LIBINTL) $(LIBDEVMAPPER) $(LIBZFS) $(LIBNVPAIR) $(LIBGEOM) -lfuse';
+  ldadd = '$(LIBINTL) $(LIBDEVMAPPER) $(LIBZFS) $(LIBNVPAIR) $(LIBGEOM) -lfuse3';
   condition = COND_GRUB_MOUNT;
 };
 
--- a/util/grub-mount.c
+++ b/util/grub-mount.c
@@ -16,7 +16,7 @@
  *  You should have received a copy of the GNU General Public License
  *  along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
  */
-#define FUSE_USE_VERSION 26
+#define FUSE_USE_VERSION 30
 #include <config.h>
 #include <grub/types.h>
 #include <grub/emu/misc.h>
@@ -34,7 +34,7 @@
 #include <grub/command.h>
 #include <grub/zfs/zfs.h>
 #include <grub/i18n.h>
-#include <fuse/fuse.h>
+#include <fuse3/fuse.h>
 
 #include <stdio.h>
 #include <unistd.h>
@@ -147,7 +147,7 @@
 }
 
 static int
-fuse_getattr (const char *path, struct stat *st)
+fuse_getattr (const char *path, struct stat *st, struct fuse_file_info *fi)
 {
   struct fuse_getattr_ctx ctx;
   char *pathname, *path2;
@@ -330,13 +330,13 @@
   st.st_blocks = (st.st_size + 511) >> 9;
   st.st_atime = st.st_mtime = st.st_ctime
     = info->mtimeset ? info->mtime : 0;
-  ctx->fill (ctx->buf, filename, &st, 0);
+  ctx->fill (ctx->buf, filename, &st, 0, 0);
   return 0;
 }
 
 static int 
 fuse_readdir (const char *path, void *buf,
-	      fuse_fill_dir_t fill, off_t off, struct fuse_file_info *fi)
+	      fuse_fill_dir_t fill, off_t off, struct fuse_file_info *fi, enum fuse_readdir_flags fl)
 {
   struct fuse_readdir_ctx ctx = {
     .path = path,
