diff --git a/test/units/_internal/_encryption/test_crypt.py b/test/units/_internal/_encryption/test_crypt.py
index 661e2c6..37aa4f7 100644
--- a/test/units/_internal/_encryption/test_crypt.py
+++ b/test/units/_internal/_encryption/test_crypt.py
@@ -122,6 +122,7 @@ class TestCryptFacade:
         crypt_facade.crypt_gensalt(b'', 1, b'')
         mock_impl.assert_called_once_with(b'', 1, b'', 0)
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_gensalt_fail_errno(self, mocker: MockerFixture) -> None:
         """Test crypt_gensalt() setting failure errno raises OSError."""
         mocker.patch('ctypes.get_errno', return_value=errno.EBADFD)
diff --git a/test/units/utils/test_encrypt.py b/test/units/utils/test_encrypt.py
index 3c48bf6..a4141b9 100644
--- a/test/units/utils/test_encrypt.py
+++ b/test/units/utils/test_encrypt.py
@@ -294,18 +295,3 @@ class TestCryptHash:
         with pytest.raises(AnsibleError, match=r"crypt does not support 'sha256_crypt' algorithm"):
             crypt_hash.hash("123", salt="12345678")
 
-
-class TestPasslibHash:
-    """
-    Tests for the PasslibHash class.
-
-    These tests are hitting code paths that are otherwise impossible to reach
-    through integration tests, but necessary for more complete code coverage.
-    """
-
-    def test_invalid_instantiation(self, mocker: MockerFixture) -> None:
-        """Should not be able to instantiate a PasslibHash class without passlib."""
-        mocker.patch('ansible.utils.encrypt.PASSLIB_AVAILABLE', False)
-
-        with pytest.raises(AnsibleError, match=r"The passlib Python package must be installed to hash with the 'sha256_crypt' algorithm\."):
-            encrypt.PasslibHash("sha256_crypt")
