--- a/external/skia/UnpackedTarball_skia.mk
+++ b/external/skia/UnpackedTarball_skia.mk
@@ -39,6 +39,7 @@ skia_patches := \
     0001-AvoidCombiningExtrememelyLargeMeshes.patch.1 \
     sort-comparison-assumption.patch.0 \
     help-msvc-analyzer.patch \
+    ppc-skia-musttail.patch.1 \
 
 $(eval $(call gb_UnpackedTarball_set_patchlevel,skia,1))
 
--- a/external/libcmis/UnpackedTarball_libcmis.mk
+++ b/external/libcmis/UnpackedTarball_libcmis.mk
@@ -25,6 +25,8 @@
 	external/libcmis/0001-cppcheck-operatorEqVarError-in-src-libcmis-http-sess.patch \
 	external/libcmis/0001-Take-into-account-m_CurlInitProtocolsFunction-in-cop.patch \
 	external/libcmis/initprotocols.patch.1 \
+	external/libcmis/libcmis-libxml2.patch.1 \
+	external/libcmis/libcmis-boost-1.86.patch.1 \
 ))
 
 # vim: set noet sw=4 ts=4:
