commit 670bcfca01b1fb9647eb937cf71b9ed09c7328e7
Author: q66 <q66@chimera-linux.org>
Date:   Sat Aug 26 01:50:32 2023 +0200

    patch out gnu cp requirement

diff --git a/configure.ac b/configure.ac
index f59c1cf..1447177 100644
--- a/configure.ac
+++ b/configure.ac
@@ -11580,14 +11580,11 @@ elif $GNUCP --version 2>/dev/null | grep "GNU fileutils" >/dev/null 2>/dev/null;
     AC_MSG_RESULT([yes])
 else
     case "$build_os" in
-    darwin*|netbsd*|openbsd*|freebsd*|dragonfly*)
+    *)
         x_GNUCP=[\#]
         GNUCP=''
         AC_MSG_RESULT([no gnucp found - using the system's cp command])
         ;;
-    *)
-        AC_MSG_ERROR([no, GNU cp needed. install or specify with --with-gnu-cp=/path/to/it])
-        ;;
     esac
 fi
 
