commit d2d7ac65311616774254da1d05696ebf94943121
Author: q66 <q66@chimera-linux.org>
Date:   Tue Jun 6 15:42:57 2023 +0200

    get rid of msguniq dep

diff --git a/configure.ac b/configure.ac
index c7c9a4f..225dd02 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13946,18 +13946,7 @@ else
             fi
         fi
     fi
-    if test -z "$MSGUNIQ"; then
-        if test -n "$LODE_HOME" -a -x "$LODE_HOME/opt/bin/msguniq" ; then
-            MSGUNIQ="$LODE_HOME/opt/bin/msguniq"
-        elif test -x "/opt/lo/bin/msguniq"; then
-            MSGUNIQ="/opt/lo/bin/msguniq"
-        else
-            AC_CHECK_PROGS(MSGUNIQ, [msguniq])
-            if test -z "$MSGUNIQ"; then
-                AC_MSG_ERROR([msguniq not found. Install GNU gettext, or re-run without languages.])
-            fi
-        fi
-    fi
+    MSGUNIQ="/usr/bin/cat"
 fi
 AC_SUBST(MSGFMT)
 AC_SUBST(MSGUNIQ)
diff --git a/solenv/gbuild/AllLangMoTarget.mk b/solenv/gbuild/AllLangMoTarget.mk
index ec05985..5e6b548 100644
--- a/solenv/gbuild/AllLangMoTarget.mk
+++ b/solenv/gbuild/AllLangMoTarget.mk
@@ -37,7 +37,7 @@ $(call gb_MoTarget_get_target,%) : $(gb_Helper_MISCDUMMY)
 	$(call gb_Trace_StartRange,$*,MO )
 	$(call gb_Helper_abbreviate_dirs,\
 		mkdir -p $(dir $@) && \
-		$(MSGUNIQ) --force-po $(gb_POLOCATION)/$(LANGUAGE)/$(POLOCATION)/messages.po | $(MSGFMT) - -o $@)
+		cat $(gb_POLOCATION)/$(LANGUAGE)/$(POLOCATION)/messages.po | $(MSGFMT) - -o $@)
 	$(call gb_Trace_EndRange,$*,MO )
 
 #$(info $(call gb_MoTarget_get_target,$(1)))
