commit 5a3de72532d112b9cc8636b046832b874c6f8518
Author: q66 <q66@chimera-linux.org>
Date:   Sat Aug 26 11:46:17 2023 +0200

    cui: fix build without java

diff --git a/cui/source/options/optjava.cxx b/cui/source/options/optjava.cxx
index 17f128c..64ba41c 100644
--- a/cui/source/options/optjava.cxx
+++ b/cui/source/options/optjava.cxx
@@ -935,6 +935,7 @@ OUString SvxJavaClassPathDlg::GetClassPath() const
     return sPath.makeStringAndClear();
 }
 
+#if HAVE_FEATURE_JAVA
 void SvxJavaClassPathDlg::SetClassPath( const OUString& _rPath )
 {
     if ( m_sOldPath.isEmpty() )
@@ -963,5 +964,6 @@ void SvxJavaClassPathDlg::SetClassPath( const OUString& _rPath )
     }
     SelectHdl_Impl(*m_xPathList);
 }
+#endif
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
diff --git a/cui/source/options/optjava.hxx b/cui/source/options/optjava.hxx
index 097b7ff..929dfe7 100644
--- a/cui/source/options/optjava.hxx
+++ b/cui/source/options/optjava.hxx
@@ -200,7 +200,9 @@ public:
     void             SetFocus() { m_xPathList->grab_focus(); }
 
     OUString                GetClassPath() const;
+#if HAVE_FEATURE_JAVA
     void                    SetClassPath( const OUString& _rPath );
+#endif
 };
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
