--- a/Makefile
+++ b/Makefile
@@ -93,8 +93,10 @@
 	$(E) STRIP $(TARGET)
 	$(Q)$(STRIP) $(TARGET)
 	$(E) INSTALL $(TARGET)
-	$(Q)$(INSTALL) -D -m 0755 $(TARGET) $(PREFIX)/sbin/$(TARGET)
-	$(Q)$(INSTALL) -D -m 0644 iotop.8 $(PREFIX)/share/man/man8/iotop.8
+	$(Q)$(INSTALL) -d $(DESTDIR)$(PREFIX)/bin/
+	$(Q)$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/man/man8/
+	$(Q)$(INSTALL) -m 0755 $(TARGET) $(DESTDIR)$(PREFIX)/bin/$(TARGET)
+	$(Q)$(INSTALL) -m 0644 iotop.8 $(DESTDIR)$(PREFIX)/share/man/man8/iotop.8
 
 uninstall:
 	$(E) UNINSTALL $(TARGET)
