cpuinfo does not have object type

--- a/cmake/cpuinfo.cmake
+++ b/cmake/cpuinfo.cmake
@@ -79,7 +79,7 @@ if(REQUESTED_CPUINFO STREQUAL "LOCAL" OR (REQUESTED_CPUINFO STREQUAL "AUTO" AND
         )
     endif()
 
-    set(CPUINFO_LIBRARY_TYPE object CACHE STRING "")
+    set(CPUINFO_LIBRARY_TYPE static CACHE STRING "")
     option(CPUINFO_BUILD_TOOLS "" OFF)
     option(CPUINFO_BUILD_UNIT_TESTS "" OFF)
     option(CPUINFO_BUILD_MOCK_TESTS "" OFF)
