diff --git a/crates/uv/Cargo.toml b/crates/uv/Cargo.toml
index a243ba0..8b37db1 100644
--- a/crates/uv/Cargo.toml
+++ b/crates/uv/Cargo.toml
@@ -152,7 +152,7 @@ nix = { workspace = true }
 uv-unix = { workspace = true }

 [features]
-default = ["performance", "uv-distribution/static", "default-tests"]
+default = ["uv-distribution/static", "default-tests"]
 native-auth = []
 # Use better memory allocators, etc.
 performance = ["performance-memory-allocator"]
