diff --git a/crates/uv/Cargo.toml b/crates/uv/Cargo.toml
index eba275850..c7abfb04e 100644
--- a/crates/uv/Cargo.toml
+++ b/crates/uv/Cargo.toml
@@ -136,7 +136,7 @@ ignored = [
 ]
 
 [features]
-default = ["performance", "uv-distribution/static", "default-tests"]
+default = ["uv-distribution/static", "default-tests"]
 # Use better memory allocators, etc.
 performance = ["performance-memory-allocator"]
 performance-memory-allocator = ["dep:uv-performance-memory-allocator"]
