diff --git a/configure.ac b/configure.ac
index 1fc0d9d..fd849d1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -73,7 +73,7 @@ m4_define([libxfce4windowing_min_version], [4.19.2])
 XDT_CHECK_PACKAGE([GLIB], [glib-2.0], [glib_min_version])
 XDT_CHECK_PACKAGE([GIO], [gio-2.0], [glib_min_version])
 XDT_CHECK_PACKAGE([GTK], [gtk+-3.0], [gtk_min_version])
-XDT_CHECK_PACKAGE([DBUS_GLIB], [dbus-glib-1], [dbus_glib_min_version])
+XDT_CHECK_PACKAGE([DBUS_GLIB], [dbus-1], [dbus_glib_min_version])
 XDT_CHECK_PACKAGE([XFCONF], [libxfconf-0], [xfconf_min_version])
 XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-2], [libxfce4ui_min_version])
 XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0], [libxfce4util_min_version])
diff --git a/src/Makefile.am b/src/Makefile.am
index 8a89b9b..2cd8ae4 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -282,6 +282,7 @@ xfce4_screensaver_SOURCES = \
 	subprocs.h                  \
 	xfce-desktop-utils.c        \
 	xfce-desktop-utils.h        \
+	dbus-gmain.c \
 	$(BUILT_SOURCES)            \
 	$(NULL)
 
diff --git a/src/gs-listener-dbus.c b/src/gs-listener-dbus.c
index 6d96781..b0626b9 100644
--- a/src/gs-listener-dbus.c
+++ b/src/gs-listener-dbus.c
@@ -38,8 +38,7 @@
 #include <elogind/sd-login.h>
 #endif
 
-#include <dbus/dbus-glib-lowlevel.h>
-#include <dbus/dbus-glib.h>
+#include <dbus-gmain/dbus-gmain.h>
 #include <dbus/dbus.h>
 #include <libxfce4util/libxfce4util.h>
 
@@ -1703,7 +1702,7 @@ gs_listener_dbus_dbus_init (GSListenerDBus *listener) {
             return FALSE;
         }
 
-        dbus_connection_setup_with_g_main (listener->priv->connection, NULL);
+        dbus_gmain_set_up_connection (listener->priv->connection, NULL);
         dbus_connection_set_exit_on_disconnect (listener->priv->connection, FALSE);
     }
 
@@ -1718,7 +1717,7 @@ gs_listener_dbus_dbus_init (GSListenerDBus *listener) {
             return FALSE;
         }
 
-        dbus_connection_setup_with_g_main (listener->priv->system_connection, NULL);
+        dbus_gmain_set_up_connection (listener->priv->system_connection, NULL);
         dbus_connection_set_exit_on_disconnect (listener->priv->system_connection, FALSE);
     }
 
