--- a/waftools/lv2.py
+++ b/waftools/lv2.py
@@ -43,7 +43,7 @@ def lv2(bld, *k, **kw):
     tg = bld.shlib(features='strip', *k, **kw)
     cxxflags = []
     if not bld.env['OPT'] and bld.env['SSE2']:
-        cxxflags = [ "-msse2", "-mfpmath=sse"]
+        cxxflags = []
     lv2_add_common(tg, lv2_base, dst, ["LV2_SO"], cxxflags=cxxflags + ['-fvisibility=hidden','-Wl,-z,relro,-z,now','-Wl,--exclude-libs,ALL'])
     if bld.env['MODGUI']:
         bld.install_files(dst, bld.path.ant_glob('*.ttl'), relative_trick=True)
