commit 046621d1c951358ad389845203d77ddd27eedfab
Author: q66 <q66@chimera-linux.org>
Date:   Tue Dec 5 15:31:12 2023 +0100

    leave stdout/err intact

diff --git a/src/AudacityApp.cpp b/src/AudacityApp.cpp
index b048e29..a5e5877 100644
--- a/src/AudacityApp.cpp
+++ b/src/AudacityApp.cpp
@@ -904,14 +904,6 @@
 
    wxDISABLE_DEBUG_SUPPORT();
 
-   // Bug #1986 workaround - This doesn't actually reduce the number of
-   // messages, it simply hides them in Release builds. We'll probably
-   // never be able to get rid of the messages entirely, but we should
-   // look into what's causing them, so allow them to show in Debug
-   // builds.
-   freopen("/dev/null", "w", stdout);
-   freopen("/dev/null", "w", stderr);
-
    return wxEntry(argc, argv);
 }
 
