commit aaa01968247e3e51eeda1329f565021e3fce793e
Author: q66 <q66@chimera-linux.org>
Date:   Sat Dec 16 17:43:43 2023 +0100

    portable install

diff --git a/Makefile b/Makefile
index fae532f..adf6147 100644
--- a/Makefile
+++ b/Makefile
@@ -117,11 +117,14 @@ trans:
 	--package-name radeontop
 
 install: all
-	$(INSTALL) -D -m755 $(bin) $(DESTDIR)/$(PREFIX)/bin/$(bin)
+	$(INSTALL) -d $(DESTDIR)/$(PREFIX)/bin
+	$(INSTALL) -m755 $(bin) $(DESTDIR)/$(PREFIX)/bin/$(bin)
 ifeq ($(xcb), 1)
-	$(INSTALL) -D -m755 $(xcblib) $(DESTDIR)/$(PREFIX)/$(LIBDIR)/$(xcblib)
+	$(INSTALL) -d $(DESTDIR)/$(PREFIX)/$(LIBDIR)
+	$(INSTALL) -m755 $(xcblib) $(DESTDIR)/$(PREFIX)/$(LIBDIR)/$(xcblib)
 endif
-	$(INSTALL) -D -m644 radeontop.1 $(DESTDIR)/$(PREFIX)/$(MANDIR)/man1/radeontop.1
+	$(INSTALL) -d $(DESTDIR)/$(PREFIX)/$(MANDIR)/man1
+	$(INSTALL) -m644 radeontop.1 $(DESTDIR)/$(PREFIX)/$(MANDIR)/man1/radeontop.1
 ifeq ($(nls), 1)
 	$(MAKE) -C translations install PREFIX=$(PREFIX)
 endif
diff --git a/translations/Makefile b/translations/Makefile
index 4ae1139..9b39c22 100644
--- a/translations/Makefile
+++ b/translations/Makefile
@@ -15,6 +15,7 @@ $(obj): $(src)
 	$(MSGFMT) -o $@ $<
 
 %.inst: %.mo
-	$(INSTALL) -D -m 644 $< $(DESTDIR)/$(PREFIX)/share/locale/$*/LC_MESSAGES/radeontop.mo
+	$(INSTALL) -d $(DESTDIR)/$(PREFIX)/share/locale/$*/LC_MESSAGES
+	$(INSTALL) -m 644 $< $(DESTDIR)/$(PREFIX)/share/locale/$*/LC_MESSAGES/radeontop.mo
 
 install: all $(inst)
