commit 5509215b9c49487941664f5ed76b8240f098c1ae
Author: q66 <q66@chimera-linux.org>
Date:   Mon Mar 21 00:37:02 2022 +0100

    install(1) portability

diff --git a/Makefile b/Makefile
index 1195c52..b4af4a8 100644
--- a/Makefile
+++ b/Makefile
@@ -13,11 +13,13 @@ xmlmantohtml.1: xml/xmlmantohtml.1.xml
 	./xmltoman $< > $@
 
 install:
-	install -D xmltoman $(DESTDIR)/$(PREFIX)/bin/xmltoman
-	install -D xmlmantohtml $(DESTDIR)/$(PREFIX)/bin/xmlmantohtml
-	install -D -m 0644 xmltoman.dtd $(DESTDIR)/$(PREFIX)/share/xmltoman/xmltoman.dtd
-	install -D -m 0644 xmltoman.css $(DESTDIR)/$(PREFIX)/share/xmltoman/xmltoman.css
-	install -D -m 0644 xmltoman.xsl $(DESTDIR)/$(PREFIX)/share/xmltoman/xmltoman.xsl
+	install -d $(DESTDIR)/$(PREFIX)/bin
+	install xmltoman $(DESTDIR)/$(PREFIX)/bin/xmltoman
+	install xmlmantohtml $(DESTDIR)/$(PREFIX)/bin/xmlmantohtml
+	install -d $(DESTDIR)/$(PREFIX)/share/xmltoman
+	install -m 0644 xmltoman.dtd $(DESTDIR)/$(PREFIX)/share/xmltoman/xmltoman.dtd
+	install -m 0644 xmltoman.css $(DESTDIR)/$(PREFIX)/share/xmltoman/xmltoman.css
+	install -m 0644 xmltoman.xsl $(DESTDIR)/$(PREFIX)/share/xmltoman/xmltoman.xsl
 
 dist:
 	[ -d $(DISTNAME) ] && rm -rf $(DISTNAME) || true
