--- a/libxfce4util/abicheck.sh
+++ b/libxfce4util/abicheck.sh
@@ -19,6 +19,6 @@
 # Boston, MA 02110-1301 USA
 #
 
-cpp -P -DINCLUDE_INTERNAL_SYMBOLS -DINCLUDE_VARIABLES -DALL_FILES ${srcdir:-.}/libxfce4util.symbols | sed -e '/^$/d' -e 's/ G_GNUC.*$//' -e 's/ PRIVATE//' | sort > expected-abi
+clang-cpp -P -DINCLUDE_INTERNAL_SYMBOLS -DINCLUDE_VARIABLES -DALL_FILES ${srcdir:-.}/libxfce4util.symbols | sed -e '/^$/d' -e 's/ G_GNUC.*$//' -e 's/ PRIVATE//' | sort > expected-abi
 nm -D .libs/libxfce4util.so | grep " [TRGD] " | cut -d ' ' -f 3 | grep -v '^_.*' | grep -v '^ *$' | sort > actual-abi
 diff -u expected-abi actual-abi && rm expected-abi actual-abi
