diff --git a/func.d/tlp-func-stat b/func.d/tlp-func-stat
index 0eefb0f..a19bffd 100644
--- a/func.d/tlp-func-stat
+++ b/func.d/tlp-func-stat
@@ -14,7 +14,7 @@ readonly SESTATUS=sestatus
 readonly SMARTCTL=smartctl
 
 readonly RE_AC_QUIRK='^UNDEFINED$'
-readonly RE_ATA_ERROR='ata[0-9]+: SError: {.*CommWake }'
+readonly RE_ATA_ERROR='ata[0-9]+: SError: \{.*CommWake \}'
 
 # ----------------------------------------------------------------------------
 # Functions
@@ -174,8 +174,8 @@ print_file_modtime_and_age () {
     local mtime age
 
     if [ -f "$1" ]; then
-        mtime=$(date +%X -r "$1")
-        age=$(( $(date +%s) - $(date +%s -r "$1") ))
+        mtime=$(date -r "$1" +%X)
+        age=$(( $(date +%s) - $(date -r "$1" +%s) ))
         printf '%s, %d sec(s) ago' "$mtime" "$age"
     else
         printf "unknown"
