--- a/driver/others/CMakeLists.txt
+++ b/driver/others/CMakeLists.txt
@@ -54,6 +54,7 @@ if (DYNAMIC_ARCH)
     list(APPEND COMMON_SOURCES dynamic_power.c)
   elseif (RISCV64)
     list(APPEND COMMON_SOURCES dynamic_riscv64.c detect_riscv64.c)
+    set_source_files_properties(detect_riscv64.c PROPERTIES COMPILE_FLAGS -march=rv64imafdcv)
   elseif (LOONGARCH64)
     list(APPEND COMMON_SOURCES dynamic_loongarch64.c)
   else ()  
