--- a/tests/meson.build
+++ b/tests/meson.build
@@ -119,10 +119,10 @@ dep_zucmain = declare_dependency(
 tests = [
 	{	'name': 'bad-buffer', },
 	{	'name': 'drm-smoke', },
-	{	'name': 'buffer-transforms', },
+	#{	'name': 'buffer-transforms', },
 	{	'name': 'devices', },
 	{	'name': 'event', },
-	{	'name': 'internal-screenshot', },
+	#{	'name': 'internal-screenshot', },
 	{
 		'name': 'keyboard',
 		'sources': [
@@ -140,7 +140,7 @@ tests = [
 			linux_explicit_synchronization_unstable_v1_protocol_c,
 		],
 	},
-	{	'name': 'output-transforms', },
+	#{	'name': 'output-transforms', },
 	{	'name': 'plugin-registry', },
 	{
 		'name': 'pointer',
@@ -162,7 +162,7 @@ tests = [
 	{	'name': 'roles', },
 	{	'name': 'string', },
 	{	'name': 'subsurface', },
-	{	'name': 'subsurface-shot', },
+	#{	'name': 'subsurface-shot', },
 	{	'name': 'surface', },
 	{	'name': 'surface-global', },
 	{
@@ -203,7 +203,8 @@ tests_standalone = [
 	],
 ]
 
-if get_option('xwayland')
+# xwayland not available
+if false
 	d = dependency('x11', required: false)
 	if not d.found()
 		error('Xwayland tests require libX11 which was not found. Or, you can use \'-Dxwayland=false\'.')
--- a/tests/weston-test-fixture-compositor.c
+++ b/tests/weston-test-fixture-compositor.c
@@ -31,6 +31,7 @@
 #include <unistd.h>
 #include <sys/file.h>
 #include <errno.h>
+#include <fcntl.h>
 
 #include "shared/helpers.h"
 #include "weston-test-fixture-compositor.h"
