diff --git a/common/xfdesktop-common.h b/common/xfdesktop-common.h
index 4e1a52ef..65f26bcd 100644
--- a/common/xfdesktop-common.h
+++ b/common/xfdesktop-common.h
@@ -35,7 +35,7 @@
 #include <stdarg.h>
 
 #define XFDESKTOP_CHANNEL        "xfce4-desktop"
-#define DEFAULT_BACKDROP         DATADIR "/backgrounds/xfce/xfce-shapes.svg"
+#define DEFAULT_BACKDROP         DATADIR "/backgrounds/chimera/bg-l.svg"
 #define DEFAULT_ICON_FONT_SIZE   12
 #define DEFAULT_ICON_SIZE        48
 #define ITHEME_FLAGS             (GTK_ICON_LOOKUP_USE_BUILTIN)
