wat

diff --git a/tests/Makefile b/tests/Makefile
index 56455bd..3cdbae9 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -1,7 +1,7 @@
 MACHINE := $(shell uname -m)
 _FORTIFY_SOURCE ?= 3
 CFLAGS ?= -std=gnu99
-CFLAGS += -I../include/ -D_FORTIFY_SOURCE=$(_FORTIFY_SOURCE) -static -O2 -Wall -Wextra -Werror -Werror=pointer-arith
+CFLAGS += -I../include/ -D_FORTIFY_SOURCE=$(_FORTIFY_SOURCE) -O2 -Wall -Wextra
 CFLAGS += -Wno-format -Wno-array-bounds -Wno-shift-count-negative -Wno-unused-variable -Wno-unused-parameter
 CXXFLAGS += -D_FORTIFY_SOURCE=$(_FORTIFY_SOURCE) -O2
 CXXFLAGS += -Wno-format -Wno-array-bounds -Wno-shift-count-negative -Wno-unused-variable -Wno-unused-parameter
@@ -112,14 +112,10 @@ gcc: CFLAGS += -Wno-stringop-overread -Wno-stringop-overflow -ffreestanding
 gcc: $(RUNTIME_TARGETS)
 
 clang: CC=clang
-clang: CFLAGS+=-I/usr/include/$(MACHINE)-linux-musl
-clang: CFLAGS+=-I../$(MACHINE)-linux-musl-native/include/
-clang: CFLAGS+=-I$(MACHINE)-linux-musl-native/include/
+clang: CFLAGS+=-I/usr/include
 clang: CFLAGS+=-nostdinc
 clang: CXX=clang++
-clang: CXXFLAGS+=-I/usr/include/$(MACHINE)-linux-musl
-clang: CXXFLAGS+=-I../$(MACHINE)-linux-musl-native/include/
-clang: CXXFLAGS+=-I$(MACHINE)-linux-musl-native/include/
+clang: CXXFLAGS+=-I/usr/include
 clang: CXXFLAGS+=-nostdinc -Wno-fortify-source
 clang: $(RUNTIME_TARGETS) cpp
 
