wat

--- a/tests/Makefile
+++ b/tests/Makefile
@@ -1,4 +1,4 @@
-CFLAGS+=-I../include/ -D_FORTIFY_SOURCE=3 -static -O2
+CFLAGS+=-I../include/ -D_FORTIFY_SOURCE=3 -O2
 
 COMPTIME_TARGETS= \
 	test_memcpy_overwrite_under  \
@@ -142,9 +142,7 @@ gcc: CC=../x86_64-linux-musl-native/bin/gcc
 gcc: $(RUNTIME_TARGETS)
 
 clang: CC=clang
-clang: CFLAGS+=-I/usr/include/x86_64-linux-musl
-clang: CFLAGS+=-I../x86_64-linux-musl-native/include/
-clang: CFLAGS+=-Ix86_64-linux-musl-native/include/
+clang: CFLAGS+=-I/usr/include
 clang: CFLAGS+=-nostdinc
 clang: comptime $(RUNTIME_TARGETS)
 
