wat

diff --git a/tests/Makefile b/tests/Makefile
index 56455bd..3cdbae9 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -1,4 +1,4 @@
-CFLAGS+=-I../include/ -D_FORTIFY_SOURCE=3 -static -O2 -DFORTIFY_PEDANTIC_CHECKS -Wno-format -Werror=pointer-arith
+CFLAGS+=-I../include/ -D_FORTIFY_SOURCE=3 -O2
 
 COMPTIME_TARGETS= \
 	test_memcpy_overwrite_under  \
@@ -144,14 +144,10 @@
 gcc: $(RUNTIME_TARGETS)
 
 clang: CC=clang
-clang: CFLAGS+=-I/usr/include/x86_64-linux-musl
-clang: CFLAGS+=-I../x86_64-linux-musl-native/include/
-clang: CFLAGS+=-Ix86_64-linux-musl-native/include/
+clang: CFLAGS+=-I/usr/include
 clang: CFLAGS+=-nostdinc
 clang: CXX=clang++
-clang: CXXFLAGS+=-I/usr/include/x86_64-linux-musl
-clang: CXXFLAGS+=-I../x86_64-linux-musl-native/include/
-clang: CXXFLAGS+=-Ix86_64-linux-musl-native/include/
+clang: CXXFLAGS+=-I/usr/include
 clang: CXXFLAGS+=-nostdinc
 clang: comptime $(RUNTIME_TARGETS) cpp
 
