--- a/tests/Makefile
+++ b/tests/Makefile
@@ -120,11 +120,11 @@ $(RUNTIME_TARGETS): %: %.c
 
 cpp: test_compile.cc
 	$(CXX) $(CXXFLAGS) test_compile.cc -o ./test_compile_cc
-	timeout 1s ./test_compile_cc 1234567890 2 3 4 5 6 7 8 9 0 >/dev/null && echo "$(EXE) OK" || echo "$(EXE) FAIL"; \
+	if timeout 1s ./test_compile_cc 1234567890 2 3 4 5 6 7 8 9 0 >/dev/null; then echo "$(EXE) OK"; else echo "$(EXE) FAIL"; exit 1; fi; \
 
 run: $(RUNTIME_TARGETS)
 	$(foreach EXE, $(RUNTIME_TARGETS), \
-		timeout 1s ./$(EXE) 1234567890 2 3 4 5 6 7 8 9 0 >/dev/null && echo "$(EXE) OK" || echo "$(EXE) FAIL"; \
+		if timeout 1s ./$(EXE) 1234567890 2 3 4 5 6 7 8 9 0 >/dev/null; then echo "$(EXE) OK"; else echo "$(EXE) FAIL"; exit 1; fi; \
 	)
 
 clean:
