From a4b8913b5166bf403feaf73c9f6f9e77f0f42892 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sun, 10 Apr 2022 02:23:14 +0200
Subject: [PATCH] enable clang/musl builds

---
 Makefile                  | 20 ++++++++++----------
 usr/include/sys/sysinfo.h |  7 +++----
 2 files changed, 13 insertions(+), 14 deletions(-)

diff --git a/Makefile b/Makefile
index c99b962..586884a 100644
--- a/Makefile
+++ b/Makefile
@@ -18,14 +18,14 @@ include $(srctree)/scripts/Kbuild.include
 
 KLIBCROSS	?= $(CROSS_COMPILE)
 export KLIBCROSS
-export CC	:= $(KLIBCROSS)gcc
-export LD	:= $(KLIBCROSS)ld
-export AR	:= $(KLIBCROSS)ar
-export RANLIB	:= $(KLIBCROSS)ranlib
-export STRIP	:= $(KLIBCROSS)strip
-export NM	:= $(KLIBCROSS)nm
-export OBJCOPY  := $(KLIBCROSS)objcopy
-export OBJDUMP  := $(KLIBCROSS)objdump
+export CC	:= $(KLIBCROSS)clang
+export LD	:= $(KLIBCROSS)ld.bfd
+export AR	:= llvm-ar
+export RANLIB	:= llvm-ranlib
+export STRIP	:= llvm-strip
+export NM	:= llvm-nm
+export OBJCOPY  := llvm-objcopy
+export OBJDUMP  := llvm-objdump
 
 NOSTDINC_FLAGS := -nostdlib -nostdinc -isystem $(shell $(CC) -print-file-name=include)
 
@@ -37,7 +37,7 @@ ARCH	          := $(shell uname -m | sed -e s/i.86/i386/ \
 export KLIBCARCH  ?= $(ARCH)
 export KLIBCARCHDIR := $(shell echo $(KLIBCARCH) | sed -e s/s390x/s390/)
 
-export HOSTCC     := gcc
+export HOSTCC     := clang
 export HOSTCFLAGS := -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer
 export PERL       := perl
 
@@ -45,7 +45,7 @@ export PERL       := perl
 export prefix      = /usr
 export bindir      = $(prefix)/bin
 export libdir      = $(prefix)/lib
-export mandir      = $(prefix)/man
+export mandir      = $(prefix)/share/man
 export INSTALLDIR  = $(prefix)/lib/klibc
 export INSTALLROOT =
 
diff --git a/usr/include/sys/sysinfo.h b/usr/include/sys/sysinfo.h
index 8469ba0..574a27f 100644
--- a/usr/include/sys/sysinfo.h
+++ b/usr/include/sys/sysinfo.h
@@ -2,12 +2,11 @@
  * sys/sysinfo.h
  */
 
-#ifndef _SYS_SYSINFO_H
-#define _SYS_SYSINFO_H
+#ifndef _SYS_SYSINFO_H_
+#define _SYS_SYSINFO_H_
 
+#include </usr/include/sys/sysinfo.h>
 #include <sys/types.h>
 #include <linux/kernel.h>
 
-extern int sysinfo(struct sysinfo *info);
-
 #endif				/* _SYS_SYSINFO_H */
-- 
2.35.1

