diff --color -ruN nmap-7.97/configure.ac nmap-7.97-patched/configure.ac
--- nmap-7.97/configure.ac	2024-04-12 17:20:04.000000000 +0000
+++ nmap-7.97-patched/configure.ac	2024-07-09 06:08:34.372883728 +0000
@@ -471,7 +471,6 @@
 fi
 
 if test $have_libpcap != yes; then
-  AC_CONFIG_SUBDIRS(libpcap)
   if test "${LIBPCAP_INC+set}" = "set"; then
     CPPFLAGS="$CPPFLAGS -I$LIBPCAP_INC"
   else
@@ -539,7 +538,6 @@
 
 # If we still don't have it, we use our own
 if test $have_pcre != yes ; then
-  AC_CONFIG_SUBDIRS( libpcre )
   CPPFLAGS="-I\$(top_srcdir)/$LIBPCREDIR/src $CPPFLAGS"
   LIBPCRE_LIBS="$LIBPCREDIR/.libs/libpcre2-8.a"
   PCRE_BUILD="build-pcre"
@@ -616,7 +614,6 @@
   fi
 
   if test $have_libz != yes; then
-    AC_CONFIG_SUBDIRS(libz)
     # TODO: This doesn't work because libssh2's configure script is looking for
     # already-built libs.  Giving up for now: build libz first or install
     # headers/libs on your own if you want compression support for SSH.
@@ -702,7 +699,6 @@
   # If we still don't have it, we use our own
   if test $have_libssh2 != yes; then
     have_libssh2=yes
-    AC_CONFIG_SUBDIRS(libssh2)
     CPPFLAGS="-I\$(top_srcdir)/$LIBSSH2DIR/include $CPPFLAGS"
     LIBSSH2_LIBS="$LIBSSH2DIR/lib/libssh2.a"
     LIBSSH2_BUILD="build-libssh2"
