--- a/versioneer.py
+++ b/versioneer.py
@@ -339,9 +339,9 @@
     # configparser.NoOptionError (if it lacks "VCS="). See the docstring at
     # the top of versioneer.py for instructions on writing your setup.cfg .
     setup_cfg = os.path.join(root, "setup.cfg")
-    parser = configparser.SafeConfigParser()
+    parser = configparser.ConfigParser()
     with open(setup_cfg, "r") as f:
-        parser.readfp(f)
+        parser.read_file(f)
     VCS = parser.get("versioneer", "VCS")  # mandatory
 
     def get(parser, name):
--- a/setup.py
+++ b/setup.py
@@ -12,34 +12,36 @@
     "Topic :: System :: Networking",
     "Topic :: System :: Systems Administration",
     "Topic :: Utilities",
-    ]
+]
 
-setup(name="magic-wormhole-mailbox-server",
-      version=versioneer.get_version(),
-      description="Securely transfer data between computers",
-      long_description=open('README.md', 'rU').read(),
-      long_description_content_type='text/markdown',
-      author="Brian Warner",
-      author_email="warner-magic-wormhole@lothar.com",
-      license="MIT",
-      url="https://github.com/warner/magic-wormhole-mailbox-server",
-      classifiers=trove_classifiers,
-      package_dir={"": "src"},
-      packages=["wormhole_mailbox_server",
-                "wormhole_mailbox_server.test",
-                "twisted.plugins",
-                ],
-      package_data={"wormhole_mailbox_server": ["db-schemas/*.sql"]},
-      install_requires=[
-          "six",
-          "attrs >= 16.3.0", # 16.3.0 adds __attrs_post_init__
-          "twisted[tls] >= 17.5.0",
-          "autobahn[twisted] >= 0.14.1",
-      ],
-      extras_require={
-          ':sys_platform=="win32"': ["pywin32"],
-          "dev": ["mock", "treq", "tox", "pyflakes"],
-      },
-      test_suite="wormhole_mailbox_server.test",
-      cmdclass=commands,
-      )
+setup(
+    name="magic-wormhole-mailbox-server",
+    version=versioneer.get_version(),
+    description="Securely transfer data between computers",
+    long_description=open("README.md", "r").read(),
+    long_description_content_type="text/markdown",
+    author="Brian Warner",
+    author_email="warner-magic-wormhole@lothar.com",
+    license="MIT",
+    url="https://github.com/warner/magic-wormhole-mailbox-server",
+    classifiers=trove_classifiers,
+    package_dir={"": "src"},
+    packages=[
+        "wormhole_mailbox_server",
+        "wormhole_mailbox_server.test",
+        "twisted.plugins",
+    ],
+    package_data={"wormhole_mailbox_server": ["db-schemas/*.sql"]},
+    install_requires=[
+        "six",
+        "attrs >= 16.3.0",  # 16.3.0 adds __attrs_post_init__
+        "twisted[tls] >= 17.5.0",
+        "autobahn[twisted] >= 0.14.1",
+    ],
+    extras_require={
+        ':sys_platform=="win32"': ["pywin32"],
+        "dev": ["mock", "treq", "tox", "pyflakes"],
+    },
+    test_suite="wormhole_mailbox_server.test",
+    cmdclass=commands,
+)
