diff --git a/examples/multiple_runs.rs b/examples/multiple_runs.rs
index 23736e8..4c75c0b 100644
--- a/examples/multiple_runs.rs
+++ b/examples/multiple_runs.rs
@@ -13,10 +13,6 @@ fn main() {
             .build()
             .unwrap();
         let res = Skim::run_with(opts, None).unwrap();
-        #[cfg(target_os = "linux")]
-        unsafe {
-            nix::libc::malloc_trim(0);
-        }
         println!(
             "run {i}: {:?}, sleeping for 5 secs",
             res.selected_items.first().map(|x| x.output())
