commit a5279eeb047cecc587a3c94f0956d35dc3f0a50f
Author: q66 <q66@chimera-linux.org>
Date:   Sat Sep 24 03:50:40 2022 +0200

    use 1500000 baud for rk3399

diff --git a/plat/rockchip/rk3399/rk3399_def.h b/plat/rockchip/rk3399/rk3399_def.h
index ba83242..8d6ecfb 100644
--- a/plat/rockchip/rk3399/rk3399_def.h
+++ b/plat/rockchip/rk3399/rk3399_def.h
@@ -17,7 +17,7 @@
 /**************************************************************************
  * UART related constants
  **************************************************************************/
-#define RK3399_BAUDRATE			115200
+#define RK3399_BAUDRATE			1500000
 #define RK3399_UART_CLOCK		24000000
 
 /******************************************************************************
