--- a/autotests/addressformattest.cpp
+++ b/autotests/addressformattest.cpp
@@ -165,9 +165,9 @@ private Q_SLOTS:
         // the weird language mix is because we are in the wrong locale for KR here,
         // the important part is that the country is in the front when using Hangul
         s = AddressFormatter::format(address, QString(), QString(), fmt, AddressFormatStyle::SingleLineInternational);
-        QCOMPARE(s, QStringLiteral("Korea, Republic of 서울특별시 중구 한강대로 416 04637"));
+        QCOMPARE(s, QStringLiteral("South Korea 서울특별시 중구 한강대로 416 04637"));
         s = AddressFormatter::format(address, QString(), QString(), fmt, AddressFormatStyle::MultiLineInternational);
-        QCOMPARE(s, QStringLiteral("Korea, Republic of\n서울특별시 중구\n한강대로 416\n04637"));
+        QCOMPARE(s, QStringLiteral("South Korea\n서울특별시 중구\n한강대로 416\n04637"));
 
         // reverse order address lines
         address.setExtended(QStringLiteral("서울스퀘어 11층"));
