diff --git a/Cargo.toml b/Cargo.toml
index 792581f..1d9f1f5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,7 +43,7 @@ gix = { version = "0.66.0", default-features = false, features = [
     "index",
     "status",
 ] }
-gix-features = { version = "0.38.0", features = ["zlib-ng"] }
+gix-features = { version = "0.38.0", features = ["zlib-stock"] }
 globset = "0.4.15"
 human-panic = "2.0.1"
 image.workspace = true
