diff --git a/Cargo.toml b/Cargo.toml
index a0a2777..1a510cc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,7 +44,7 @@ gix = { version = "0.62.0", default-features = false, features = [
     "index",
     "status"
 ] }
-gix-features = { version = "0.38.0", features = ["zlib-ng"] }
+gix-features = { version = "0.38.0", features = ["zlib-stock"] }
 globset = "0.4.14"
 human-panic = "1.2.1"
 image.workspace = true
