From 2f168cecb034c37e08b4c4d1e10aa132292a4dcb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Christian=20Gr=C3=BCnhage?=
 <jan.christian@gruenhage.xyz>
Date: Sun, 17 Mar 2024 21:49:37 +0100
Subject: [PATCH] include stdlib header

---
 libpskc/global.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libpskc/global.c b/libpskc/global.c
index 1a51450..9ba55bb 100644
--- a/libpskc/global.c
+++ b/libpskc/global.c
@@ -20,6 +20,7 @@
  */
 
 #include <config.h>
+#include <stdlib.h>
 
 #include <pskc/pskc.h>
 
-- 
2.44.0

