fixes edge case of `li t1, 9223372036854775807` in riscv64 assembly
diff --git a/gas/config/tc-riscv.c b/gas/config/tc-riscv.c
index a4161420128..fdd2623ae40 100644
--- a/gas/config/tc-riscv.c
+++ b/gas/config/tc-riscv.c
@@ -1998,7 +1998,7 @@ load_const (int reg, expressionS *ep)
   bfd_vma upper_imm, sign = (bfd_vma) 1 << (RISCV_IMM_BITS - 1);
   expressionS upper = *ep, lower = *ep;
   lower.X_add_number = ((ep->X_add_number & (sign + sign - 1)) ^ sign) - sign;
-  upper.X_add_number -= lower.X_add_number;
+  upper.X_add_number = (bfd_vma)upper.X_add_number - lower.X_add_number;
 
   if (ep->X_op != O_constant)
     {
