diff --git a/src/core/random_func.cpp b/src/core/random_func.cpp
index cbf8cc93ec..b536741aee 100644
--- a/src/core/random_func.cpp
+++ b/src/core/random_func.cpp
@@ -25,7 +25,7 @@
 #	include <bcrypt.h>
 #elif defined(__APPLE__) || defined(__NetBSD__) || defined(__FreeBSD__)
 /* No includes required. */
-#elif defined(__GLIBC__) && ((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 25)))
+#elif defined(__linux__)
 #	include <sys/random.h>
 #elif defined(__EMSCRIPTEN__)
 #	include <emscripten.h>
@@ -99,7 +99,7 @@ void RandomBytesWithFallback(std::span<uint8_t> buf)
 #elif defined(__APPLE__) || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
 	arc4random_buf(buf.data(), buf.size());
 	return;
-#elif defined(__GLIBC__) && ((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 25)))
+#elif defined(__linux__)
 	auto res = getrandom(buf.data(), buf.size(), 0);
 	if (res > 0 && static_cast<size_t>(res) == buf.size()) return;
 #elif defined(__EMSCRIPTEN__)
