commit b04c363aebfe5ba79f6c9af2fca5e9f3e260d150
Author: q66 <q66@chimera-linux.org>
Date:   Sat Apr 2 18:56:12 2022 +0200

    don't use static runtime

diff --git a/Makefile.in b/Makefile.in
index 96089c8..f2d7d6a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -94,7 +94,7 @@ INSTALL_PKGCONFIG	  = $(INSTALL_PREFIX)/lib/pkgconfig
 
 # Flags required to tell the compiler to create a dynamically loadable object
 #
-DYNAMIC_LDFLAGS		= -static-libgcc -shared -Wl,-Bsymbolic
+DYNAMIC_LDFLAGS		= -shared -Wl,-Bsymbolic
 SDK_DYNAMIC_LDFLAGS	= $(DYNAMIC_LDFLAGS) -Wl,-soname=$(INSTALL_SDK_LINK_ABI)
 HOSTSDK_DYNAMIC_LDFLAGS	= $(DYNAMIC_LDFLAGS) -Wl,-soname=$(INSTALL_HOSTSDK_LINK_ABI)
 
