commit ff4b4e8825858d4ddb07848db424b4779eeddede
Author: q66 <q66@chimera-linux.org>
Date:   Sat Apr 2 18:54:17 2022 +0200

    don't run test target by default

diff --git a/configure b/configure
index 9137bec..73a777e 100755
--- a/configure
+++ b/configure
@@ -4076,7 +4076,7 @@ $as_echo "yes" >&6; }
 	have_sndfile="yes"
 fi
 	if test "x$have_sndfile" = "xyes" ; then
-				TARGETS="sdk plugins host rdfgen test"
+				TARGETS="sdk plugins host rdfgen"
 	else
 				as_fn_error $? "libsndfile not found - cannot build example Vamp host!
 If you don't want to build the host, configure with --disable-programs." "$LINENO" 5
