diff --git a/bin/pnpm.cjs b/bin/pnpm.cjs
index 8f162c2..75cf642 100755
--- a/bin/pnpm.cjs
+++ b/bin/pnpm.cjs
@@ -16,13 +16,13 @@ ${COMPATIBILITY_PAGE}`)
 try {
   // Use node.js 22 new API for better performance.
   if(!require('module')?.enableCompileCache?.())
-    require('v8-compile-cache');
+    require('/usr/lib/pnpm/v8-compile-cache');
 } catch {
   // We don't have/need to care about v8-compile-cache failed
 }
 
 global['pnpm__startedAt'] = Date.now()
-require('../dist/pnpm.cjs')
+require('/usr/lib/pnpm/pnpm.cjs')
 
 // if you want to debug at your local env, you can use this
 // require('../lib/pnpm')
diff --git a/bin/pnpx.cjs b/bin/pnpx.cjs
index 8deb37e..b135ce8 100755
--- a/bin/pnpx.cjs
+++ b/bin/pnpx.cjs
@@ -2,4 +2,4 @@
 
 process.argv = [...process.argv.slice(0, 2), 'dlx', ...process.argv.slice(2)]
 
-require('./pnpm.cjs')
+require('/usr/lib/pnmp/pnpm.cjs')
