From c5505a4a9b497d1a4e46887a6b406caf2438f674 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Fri, 17 Mar 2023 04:39:15 +0100
Subject: [PATCH] tests portability

diff --git a/tests/test-run.sh b/tests/test-run.sh
index 1003e8c..fe659ae 100755
--- a/tests/test-run.sh
+++ b/tests/test-run.sh
@@ -109,7 +109,7 @@ assert_file_has_content json-status.json '"child-pid": [0-9]'
 assert_file_has_content_literal json-status.json '"exit-code": 42'
 ok "info and json-status fd"
 
-DATA=$($RUN --proc /proc --unshare-all --info-fd 42 --json-status-fd 43 -- bash -c 'stat -L -c "%n %i" /proc/self/ns/*' 42>info.json 43>json-status.json 2>err.txt)
+DATA=$($RUN --proc /proc --unshare-all --info-fd 42 --json-status-fd 43 -- bash -c 'stat -L -f "%N %i" /proc/self/ns/*' 42>info.json 43>json-status.json 2>err.txt)
 
 for NS in "ipc" "mnt" "net" "pid" "uts"; do
 
@@ -339,9 +339,9 @@ $RUN \
     --bind "$(pwd -P)/some-dir" "$(pwd -P)/new-dir-mountpoint" \
     --bind "$(pwd -P)/some-file" "$(pwd -P)/new-file-mountpoint" \
     true
-command stat -c '%a' new-dir-mountpoint > new-dir-permissions
+command stat -f '%OLp' new-dir-mountpoint > new-dir-permissions
 assert_file_has_content new-dir-permissions 755
-command stat -c '%a' new-file-mountpoint > new-file-permissions
+command stat -f '%OLp' new-file-mountpoint > new-file-permissions
 assert_file_has_content new-file-permissions 444
 ok "Files and directories created as mount points have expected permissions"
 
@@ -367,15 +367,15 @@ $RUN \
     --chmod 1755 "$(pwd -P)/dir-chmod" \
     --dir "$(pwd -P)/new-dir-default-perms" \
     true
-command stat -c '%a' new-dir-default-perms > new-dir-permissions
+command stat -f '%OLp' new-dir-default-perms > new-dir-permissions
 assert_file_has_content new-dir-permissions '^755$'
-command stat -c '%a' new-dir-set-perms > new-dir-permissions
+command stat -f '%OMp%OLp' new-dir-set-perms > new-dir-permissions
 assert_file_has_content new-dir-permissions '^1741$'
-command stat -c '%a' dir-already-existed > dir-permissions
+command stat -f '%OLp' dir-already-existed > dir-permissions
 assert_file_has_content dir-permissions '^710$'
-command stat -c '%a' dir-already-existed2 > dir-permissions
+command stat -f '%OLp' dir-already-existed2 > dir-permissions
 assert_file_has_content dir-permissions '^754$'
-command stat -c '%a' dir-chmod > dir-permissions
+command stat -f '%OMp%OLp' dir-chmod > dir-permissions
 assert_file_has_content dir-permissions '^1755$'
 ok "Directories created explicitly have expected permissions"
 
@@ -407,31 +407,31 @@ $RUN \
     --perms 0604 --dir "$(pwd -P)"/parent-of-0604/dir \
     --perms 0000 --dir "$(pwd -P)"/parent-of-0000/dir \
     true
-command stat -c '%a' parent > dir-permissions
+command stat -f '%OLp' parent > dir-permissions
 assert_file_has_content dir-permissions '^755$'
-command stat -c '%a' parent-of-1777 > dir-permissions
+command stat -f '%OLp' parent-of-1777 > dir-permissions
 assert_file_has_content dir-permissions '^755$'
-command stat -c '%a' parent-of-0755 > dir-permissions
+command stat -f '%OLp' parent-of-0755 > dir-permissions
 assert_file_has_content dir-permissions '^755$'
-command stat -c '%a' parent-of-0644 > dir-permissions
+command stat -f '%OLp' parent-of-0644 > dir-permissions
 assert_file_has_content dir-permissions '^755$'
-command stat -c '%a' parent-of-0750 > dir-permissions
+command stat -f '%OLp' parent-of-0750 > dir-permissions
 assert_file_has_content dir-permissions '^750$'
-command stat -c '%a' parent-of-0710 > dir-permissions
+command stat -f '%OLp' parent-of-0710 > dir-permissions
 assert_file_has_content dir-permissions '^750$'
-command stat -c '%a' parent-of-0720 > dir-permissions
+command stat -f '%OLp' parent-of-0720 > dir-permissions
 assert_file_has_content dir-permissions '^750$'
-command stat -c '%a' parent-of-0640 > dir-permissions
+command stat -f '%OLp' parent-of-0640 > dir-permissions
 assert_file_has_content dir-permissions '^750$'
-command stat -c '%a' parent-of-0700 > dir-permissions
+command stat -f '%OLp' parent-of-0700 > dir-permissions
 assert_file_has_content dir-permissions '^700$'
-command stat -c '%a' parent-of-0600 > dir-permissions
+command stat -f '%OLp' parent-of-0600 > dir-permissions
 assert_file_has_content dir-permissions '^700$'
-command stat -c '%a' parent-of-0705 > dir-permissions
+command stat -f '%OLp' parent-of-0705 > dir-permissions
 assert_file_has_content dir-permissions '^705$'
-command stat -c '%a' parent-of-0604 > dir-permissions
+command stat -f '%OLp' parent-of-0604 > dir-permissions
 assert_file_has_content dir-permissions '^705$'
-command stat -c '%a' parent-of-0000 > dir-permissions
+command stat -f '%OLp' parent-of-0000 > dir-permissions
 assert_file_has_content dir-permissions '^700$'
 chmod -R 0700 parent*
 rm -fr parent*
@@ -442,11 +442,11 @@ $RUN \
     cat /proc/self/mountinfo >&2
 $RUN \
     --perms 01777 --tmpfs "$(pwd -P)" \
-    stat -c '%a' "$(pwd -P)" > dir-permissions
+    stat -f '%OMp%OLp' "$(pwd -P)" > dir-permissions
 assert_file_has_content dir-permissions '^1777$'
 $RUN \
     --tmpfs "$(pwd -P)" \
-    stat -c '%a' "$(pwd -P)" > dir-permissions
+    stat -f '%OLp' "$(pwd -P)" > dir-permissions
 assert_file_has_content dir-permissions '^755$'
 ok "tmpfs has expected permissions"
 
@@ -487,27 +487,27 @@ fi
 
 $RUN \
     --file 0 /tmp/file \
-    stat -c '%a' /tmp/file < /dev/null > file-permissions
+    stat -f '%OLp' /tmp/file < /dev/null > file-permissions
 assert_file_has_content file-permissions '^666$'
 $RUN \
     --perms 0640 --file 0 /tmp/file \
-    stat -c '%a' /tmp/file < /dev/null > file-permissions
+    stat -f '%OLp' /tmp/file < /dev/null > file-permissions
 assert_file_has_content file-permissions '^640$'
 $RUN \
     --bind-data 0 /tmp/file \
-    stat -c '%a' /tmp/file < /dev/null > file-permissions
+    stat -f '%OLp' /tmp/file < /dev/null > file-permissions
 assert_file_has_content file-permissions '^600$'
 $RUN \
     --perms 0640 --bind-data 0 /tmp/file \
-    stat -c '%a' /tmp/file < /dev/null > file-permissions
+    stat -f '%OLp' /tmp/file < /dev/null > file-permissions
 assert_file_has_content file-permissions '^640$'
 $RUN \
     --ro-bind-data 0 /tmp/file \
-    stat -c '%a' /tmp/file < /dev/null > file-permissions
+    stat -f '%OLp' /tmp/file < /dev/null > file-permissions
 assert_file_has_content file-permissions '^600$'
 $RUN \
     --perms 0640 --ro-bind-data 0 /tmp/file \
-    stat -c '%a' /tmp/file < /dev/null > file-permissions
+    stat -f '%OLp' /tmp/file < /dev/null > file-permissions
 assert_file_has_content file-permissions '^640$'
 ok "files have expected permissions"
 
