From 990a22f0392fffb226228e3fbcfc9ca500349d6b Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sun, 10 Apr 2022 19:29:49 +0200
Subject: [PATCH] add chimera logo variant

---
 neofetch | 28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/neofetch b/neofetch
index c116070..4ce39e1 100755
--- a/neofetch
+++ b/neofetch
@@ -777,7 +777,7 @@ image_source="auto"
 #       ArchStrike, XFerience, ArchMerge, Arch, Artix, Arya, Bedrock,
 #       Bitrig, BlackArch, BLAG, BlankOn, BlueLight, bonsai, BSD,
 #       BunsenLabs, Calculate, Carbs, CentOS, Chakra, ChaletOS,
-#       Chapeau, Chrom*, Cleanjaro, ClearOS, Clear_Linux, Clover,
+#       Chapeau, Chimera, Chrom*, Cleanjaro, ClearOS, Clear_Linux, Clover,
 #       Condres, Container_Linux, CRUX, Cucumber, Debian, Deepin,
 #       DesaOS, Devuan, DracOS, DarkOs, DragonFly, Drauger, Elementary,
 #       EndeavourOS, Endless, EuroLinux, Exherbo, Fedora, Feren, FreeBSD,
@@ -4920,7 +4920,7 @@ ASCII:
                                 ARCHlabs, ArchStrike, XFerience, ArchMerge, Arch, Artix, Arya,
                                 Bedrock, Bitrig, BlackArch, BLAG, BlankOn, BlueLight, bonsai, BSD,
                                 BunsenLabs, Calculate, Carbs, CentOS, Chakra, ChaletOS, Chapeau,
-                                Chrom, Cleanjaro, ClearOS, Clear_Linux, Clover, Condres,
+                                Chimera, Chrom, Cleanjaro, ClearOS, Clear_Linux, Clover, Condres,
                                 Container_Linux, CRUX, Cucumber, Debian, Deepin, DesaOS, Devuan,
                                 DracOS, DarkOs, DragonFly, Drauger, Elementary, EndeavourOS, Endless,
                                 EuroLinux, Exherbo, Fedora, Feren, FreeBSD, FreeMiNT, Frugalware,
@@ -10352,6 +10352,30 @@ ssssssssssssso/-`      `-/osssssssssssss
 EOF
         ;;
 
+        "Chimera"*)
+            set_colors 1 5 4 1
+            read -rd '' ascii_data <<'EOF'
+${c3}ddddddddddddddc  ${c1},cc:
+${c3}ddddddddddddddc  ${c1},cc:
+${c3}ddddddddddddddd  ${c1},cc:
+${c3}ddddddddddddl:'  ${c1},cc:
+${c3}dddddddddl'    ${c1}..;cc:
+${c3}dddddddo.   ${c1},:cccccc:
+${c3}ddddddl   ${c1},ccc:'''''
+${c3}dddddo.  ${c1};ccc.          ............
+        .ccc.           cccccccccccc
+${c2}......  ${c1}.ccc.          .ccc'''''''''
+${c2}OOOOOk.  ${c1};ccc.        .ccc;   ......
+${c2}OOOOOOd   ${c1}'ccc:,....,:ccc'   ${c4}coooooo
+${c2}OOOOOOOx.   ${c1}':cccccccc:'   ${c4}.looooooo
+${c2}OOOOOOOOOd,     ${c1}`'''`     ${c4}.coooooooo
+${c2}OOOOOOOOOOOOdc,.    ${c4}..,coooooooooooo
+${c2}OOOOOOOOOOOOOOOO'  ${c4}.oooooooooooooooo
+${c2}OOOOOOOOOOOOOOOO'  ${c4}.oooooooooooooooo
+${c2}OOOOOOOOOOOOOOOO'  ${c4}.oooooooooooooooo
+EOF
+        ;;
+
         *)
             case $kernel_name in
                 *"BSD")
-- 
2.35.1

