https://github.com/ClangBuiltLinux/linux/issues/1907#issuecomment-1670475621

diff --git a/arch/x86/kernel/vmlinux.lds.S b/arch/x86/kernel/vmlinux.lds.S
index fd03f5a..8d83621 100644
--- a/arch/x86/kernel/vmlinux.lds.S
+++ b/arch/x86/kernel/vmlinux.lds.S
@@ -139,14 +139,6 @@ SECTIONS
 
 		ENTRY_TEXT
 
-#ifdef CONFIG_CPU_SRSO
-		/*
-		 * See the comment above srso_untrain_ret_alias()'s
-		 * definition.
-		 */
-		. = srso_untrain_ret_alias | (1 << 2) | (1 << 8) | (1 << 14) | (1 << 20);
-		*(.text.__x86.rethunk_safe)
-#endif
 		ALIGN_ENTRY_TEXT_END
 		SOFTIRQENTRY_TEXT
 		STATIC_CALL_TEXT
@@ -517,9 +509,14 @@ INIT_PER_CPU(irq_stack_backing_store);
  * GNU ld cannot do XOR so do: (A | B) - (A & B) in order to compute the XOR
  * of the two function addresses:
  */
+#ifdef CONFIG_LD_IS_BFD
 . = ASSERT(((srso_untrain_ret_alias | srso_safe_ret_alias) -
 		(srso_untrain_ret_alias & srso_safe_ret_alias)) == ((1 << 2) | (1 << 8) | (1 << 14) | (1 << 20)),
 		"SRSO function pair won't alias");
+#else
+. = ASSERT(srso_untrain_ret_alias ^ srso_safe_ret_alias == ((1 << 2) | (1 << 8) | (1 << 14) | (1 << 20)),
+               "SRSO function pair won't alias");
+#endif
 #endif
 
 #endif /* CONFIG_X86_64 */
diff --git a/arch/x86/lib/retpoline.S b/arch/x86/lib/retpoline.S
index 30e76fa..453467e 100644
--- a/arch/x86/lib/retpoline.S
+++ b/arch/x86/lib/retpoline.S
@@ -100,8 +100,7 @@ SYM_START(srso_untrain_ret_alias, SYM_L_GLOBAL, SYM_A_NONE)
 	jmp __x86_return_thunk
 SYM_FUNC_END(srso_untrain_ret_alias)
 __EXPORT_THUNK(srso_untrain_ret_alias)
-
-	.section .text.__x86.rethunk_safe
+.fill (. - srso_untrain_ret_alias + 0x104104 - 0x14), 1, 0xcc
 #endif
 
 /* Needs a definition for the __x86_return_thunk alternative below. */
