From 2e7b817e1a4dbb623d922eb5eec94749002e5585 Mon Sep 17 00:00:00 2001
From: Arina Neshlyaeva <arina.neshlyaeva@intel.com>
Date: Mon, 10 Mar 2025 17:11:32 -0700
Subject: [PATCH] Fix getWithCaptureInfo for LLVM 20.1

---
 src/module.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/module.cpp b/src/module.cpp
index 112aaacee66..ec33261b815 100644
--- a/src/module.cpp
+++ b/src/module.cpp
@@ -1180,7 +1180,7 @@ void Module::AddFunctionDeclaration(const std::string &name, const FunctionType
 
             if (attrList->HasAttribute("noescape")) {
                 if (argType->IsPointerType() && argType->IsUniformType()) {
-#if ISPC_LLVM_VERSION > ISPC_LLVM_20_0
+#if ISPC_LLVM_VERSION >= ISPC_LLVM_21_0
                     function->addParamAttr(
                         i, llvm::Attribute::getWithCaptureInfo(function->getContext(), llvm::CaptureInfo::none()));
 #else
