diff --git a/crates/uv/Cargo.toml b/crates/uv/Cargo.toml
index 52be3cac4..581728e70 100644
--- a/crates/uv/Cargo.toml
+++ b/crates/uv/Cargo.toml
@@ -133,7 +133,7 @@ ignored = [
 ]
 
 [features]
-default = ["python", "python-managed", "pypi", "git", "performance", "crates-io", "test-ecosystem"]
+default = ["python", "python-managed", "pypi", "git", "crates-io", "test-ecosystem"]
 # Use better memory allocators, etc.
 performance = [
     "performance-memory-allocator",
