diff --git a/src/meson.build b/src/meson.build
index 4f7c3c2b..af70dd9c 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -40,8 +40,6 @@ else
   message('Building in release mode')
 endif
 
-cargo_env = [ 'CARGO_HOME=' + meson.project_build_root() / 'cargo-home' ]
-
 if build_env_only
   depends = []
 else
@@ -54,16 +52,11 @@ custom_target(
   build_always_stale: true,
   output: meson.project_name(),
   console: true,
-  install: true,
   install_dir: bindir,
   depends: depends,
   command: [
-    'env',
-    cargo_env,
     cargo, 'build',
     cargo_options,
-    '&&',
-    'cp', 'src' / rust_target / meson.project_name(), '@OUTPUT@',
   ]
 )
 
@@ -104,7 +97,7 @@ custom_target(
     output: 'doc',
     console: true,
     command: [
-      'env', cargo_env + doc_env,
+      'env', doc_env,
       cargo, 'doc',
       cargo_options + doc_deps + ['--no-deps', '-Zrustdoc-map'] ,
     ],
