--- a/Makefile
+++ b/Makefile
@@ -123,8 +123,8 @@
 	if [ -f .custom ]; then \
 		install -m 644 -t $(DESTDIR)/$(CRDA_KEY_PATH)/ $(shell cat .custom); \
 	fi
-	install -m 644 -t $(DESTDIR)/$(CRDA_KEY_PATH)/ $(REGDB_UPSTREAM_PUBKEY)
-	install -m 644 -t $(DESTDIR)/$(CRDA_PATH)/ regulatory.bin
-	install -m 644 -t $(DESTDIR)/$(FIRMWARE_PATH) regulatory.db regulatory.db.p7s
+	install -m 644 $(REGDB_UPSTREAM_PUBKEY) $(DESTDIR)/$(CRDA_KEY_PATH)/
+	install -m 644 regulatory.bin $(DESTDIR)/$(CRDA_PATH)/
+	install -m 644 regulatory.db regulatory.db.p7s $(DESTDIR)/$(FIRMWARE_PATH)/
 	install -m 755 -d $(DESTDIR)/$(MANDIR)/man5/
-	install -m 644 -t $(DESTDIR)/$(MANDIR)/man5/ regulatory.bin.5.gz regulatory.db.5.gz
+	install -m 644 regulatory.bin.5.gz regulatory.db.5.gz $(DESTDIR)/$(MANDIR)/man5/
