the release was 2024 and it's 2025 now

--- a/test/plugins/test_bucket.py
+++ b/test/plugins/test_bucket.py
@@ -51,7 +51,7 @@ class BucketPluginTest(BeetsTestCase):
         year."""
         self._setup_config(bucket_year=["1950", "1970"])
         assert self.plugin._tmpl_bucket("2014") == "1970"
-        assert self.plugin._tmpl_bucket("2025") == "2025"
+        assert self.plugin._tmpl_bucket("2026") == "2026"
 
     def test_year_two_years(self):
         """Buckets can be named with the 'from-to' syntax."""
