diff --git a/GNUmakefile b/GNUmakefile
index 1a22fb5..0da6645 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -44,7 +44,7 @@ ifeq ($(BSD_LIBS),)
   $(error You probably need to install "libbsd-dev" or "libbsd-devel" or something like that.)
 endif
 
-CURSES_LIBS:= $(shell $(PKG_CONFIG) --libs ncurses)
+CURSES_LIBS:= $(shell $(PKG_CONFIG) --libs ncursesw)
 ifeq ($(CURSES_LIBS),)
   $(error You probably need to install "libncurses5-dev" or "libncurses6-devel" or something like that.)
 endif
