requires zve32f extension and i cba to plug it in

diff --git a/third_party/llama.cpp/ggml/src/ggml-cpu/simd-mappings.h b/third_party/llama.cpp/ggml/src/ggml-cpu/simd-mappings.h
index 62e6872..eb7efd7 100644
--- a/third_party/llama.cpp/ggml/src/ggml-cpu/simd-mappings.h
+++ b/third_party/llama.cpp/ggml/src/ggml-cpu/simd-mappings.h
@@ -1266,8 +1266,6 @@ static inline void __lzs_f16cx4_store(ggml_fp16_t * x, float32x4_t v_y) {
 
 // compatible with vlen >= 128
 
-#define GGML_SIMD
-
 // F32
 
 #define GGML_F32_STEP 16
