commit 026a5feb719d69a44f984048cb9cdf296fb15801
Author: q66 <q66@chimera-linux.org>
Date:   Mon Aug 22 23:42:54 2022 +0200

    use integrated-as on ppc64

diff --git a/xpcom/reflect/xptcall/md/unix/moz.build b/xpcom/reflect/xptcall/md/unix/moz.build
index 1779c14..cd4595c 100644
--- a/xpcom/reflect/xptcall/md/unix/moz.build
+++ b/xpcom/reflect/xptcall/md/unix/moz.build
@@ -110,10 +110,6 @@ if CONFIG["OS_ARCH"] in ("Linux", "FreeBSD", "NetBSD", "OpenBSD"):
             "xptcstubs_asm_mips64.S",
             "xptcstubs_mips64.cpp",
         ]
-        if CONFIG["CC_TYPE"] == "clang":
-            ASFLAGS += [
-                "-fno-integrated-as",
-            ]
     if CONFIG["TARGET_CPU"] == "mips32":
         SOURCES += [
             "xptcinvoke_asm_mips.S",
