diff --git a/exp_chan.c b/exp_chan.c
index ba1fa80..be70135 100644
--- a/exp_chan.c
+++ b/exp_chan.c
@@ -15,7 +15,7 @@
 
 #include "expect_cf.h"
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/exp_clib.c b/exp_clib.c
index 53cfc90..50ee766 100644
--- a/exp_clib.c
+++ b/exp_clib.c
@@ -18,7 +18,7 @@ would appreciate credit if this program or parts of it are used.
 #ifdef HAVE_UNISTD_H
 # include <unistd.h>
 #endif
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/exp_event.c b/exp_event.c
index 8ebe4b6..7271871 100644
--- a/exp_event.c
+++ b/exp_event.c
@@ -12,7 +12,7 @@ NIST would appreciate credit if this program or parts of it are used.
 #include <errno.h>
 #include <sys/types.h>
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/exp_inter.c b/exp_inter.c
index 8802032..f17a400 100644
--- a/exp_inter.c
+++ b/exp_inter.c
@@ -29,7 +29,7 @@ would appreciate credit if this program or parts of it are used.
 # endif
 #endif
   
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/exp_main_sub.c b/exp_main_sub.c
index fcfaa6e..806b472 100644
--- a/exp_main_sub.c
+++ b/exp_main_sub.c
@@ -12,7 +12,7 @@
 # include <unistd.h>
 #endif
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/exp_noevent.c b/exp_noevent.c
index d145b49..0f5595a 100644
--- a/exp_noevent.c
+++ b/exp_noevent.c
@@ -24,7 +24,7 @@ it successfully, please mail back the changes to me.  - Don
 #include <sys/types.h>
 #include <sys/time.h>
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/exp_select.c b/exp_select.c
index de30853..da9386c 100644
--- a/exp_select.c
+++ b/exp_select.c
@@ -17,7 +17,7 @@ void exp_unused() {}
 #include <errno.h>
 #include <sys/types.h>
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/exp_simple.c b/exp_simple.c
index 780c05e..8f87e44 100644
--- a/exp_simple.c
+++ b/exp_simple.c
@@ -309,7 +309,7 @@ it successfully, please mail back the changes to me.  - Don
 #include <sys/types.h>
 #include <sys/time.h>
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/exp_trap.c b/exp_trap.c
index a55a74d..596b5ac 100644
--- a/exp_trap.c
+++ b/exp_trap.c
@@ -15,7 +15,7 @@ would appreciate credit if this program or parts of it are used.
 #include <sys/types.h>
 #include <string.h>
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 #ifdef HAVE_STRING_H
diff --git a/exp_tty.c b/exp_tty.c
index 3f22c90..f0268c9 100644
--- a/exp_tty.c
+++ b/exp_tty.c
@@ -22,7 +22,7 @@
 # include <unistd.h>
 #endif
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
diff --git a/expect.c b/expect.c
index 86b4abf..654666a 100644
--- a/expect.c
+++ b/expect.c
@@ -17,7 +17,7 @@ would appreciate credit if this program or parts of it are used.
 
 #include "expect_cf.h"
 
-#ifdef HAVE_SYS_WAIT_H
+#ifndef NO_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
 
