wrong dir
--
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -112,7 +112,7 @@
     DESTINATION ${ZSH_COMP_DIR}
 )
 install(FILES ansifilter.desktop DESTINATION ${DESKTOP_APPS_DIR} COMPONENT gui)
-install(FILES src/gui-qt/ansifilter.xpm DESTINATION ${DESKTOP_ICONS_DIR} COMPONENT gui)
+install(FILES src/qt-gui/ansifilter.xpm DESTINATION ${DESKTOP_ICONS_DIR} COMPONENT gui)
 
 # Installation of binaries
 install(TARGETS ansifilter DESTINATION ${BIN_DIR} )
